<HTML>

<HEAD>
  <TITLE>
  EMBOSS: antigenic
  </TITLE>
</HEAD>
<BODY BGCOLOR="#FFFFFF" text="#000000">

<table align=center border=0 cellspacing=0 cellpadding=0>
<tr><td valign=top>
<A HREF="/" ONMOUSEOVER="self.status='Go to the EMBOSS home page';return true"><img border=0 src="/images/emboss_icon.jpg" alt="" width=150 height=48></a>
</td>
<td align=left valign=middle>
<b><font size="+6">
antigenic
</font></b>
</td></tr>
</table>
<br>&nbsp;
<p>


<H2>
Wiki
</H2>

The master copies of EMBOSS documentation are available
at <a href="http://emboss.open-bio.org/wiki/Appdocs">
http://emboss.open-bio.org/wiki/Appdocs</a>
on the EMBOSS Wiki.

<p>
Please help by correcting and extending the Wiki pages.

<H2>
    Function
</H2>
Find antigenic sites in proteins




<H2>
    Description
</H2>


<p><B>Antigenic</B> predicts potentially <b>antigenic</b> regions of a
protein sequence, using the method of Kolaskar and Tongaonkar. This
method is based on a single parameter and thus very simple to use.</p>


<H2>
    Algorithm
</H2>

<p>Analysis of data from experimentally determined <b>antigenic</b>
sites on proteins has revealed that the hydrophobic residues Cys, Leu
and Val, if they occur on the surface of a protein, are more likely to
be a part of <b>antigenic</b> sites.  The method of Kolaskar and
Tongaonkar to predict <b>antigenic</b> determinants in proteins is
semi-empirical and makes use of physicochemical properties of amino
acid residues and their frequencies of occurrence in experimentally
known segmental epitopes.</p>



<H2>
    Usage
</H2>

Here is a sample session with <b>antigenic</b>
<p>

<p>
<table width="90%"><tr><td bgcolor="#CCFFFF"><pre>

% <b>antigenic </b>
Find antigenic sites in proteins
Input protein sequence(s): <b>tsw:actb1_takru</b>
Minimum length of antigenic region [6]: <b></b>
Output report [actb1_takru.antigenic]: <b></b>

</pre></td></tr></table><p>
<p>
<a href="#input.1">Go to the input files for this example</a><br><a href="#output.1">Go to the output files for this example</a><p><p>
<p>
<b>Example 2</b>
<p>
By using the '-rformat gff' qualifier, a GFF file of the predicted regions can be produced. 
<p>

<p>
<table width="90%"><tr><td bgcolor="#CCFFFF"><pre>

% <b>antigenic -rformat gff </b>
Find antigenic sites in proteins
Input protein sequence(s): <b>tsw:actb1_takru</b>
Minimum length of antigenic region [6]: <b></b>
Output report [actb1_takru.antigenic]: <b></b>

</pre></td></tr></table><p>
<p>
<a href="#output.2">Go to the output files for this example</a><p><p>



<H2>
    Command line arguments
</H2>



<!--
DON'T WRITE ANYTHING HERE.
IT IS DONE FOR YOU.
-->

<table CELLSPACING=0 CELLPADDING=3 BGCOLOR="#f5f5ff" ><tr><td>
<pre>
Find antigenic sites in proteins
Version: EMBOSS:6.6.0.0

   Standard (Mandatory) qualifiers:
  [-sequence]          seqall     Protein sequence(s) filename and optional
                                  format, or reference (input USA)
   -minlen             integer    [6] Minimum length of antigenic region
                                  (Integer from 1 to 50)
  [-outfile]           report     [*.antigenic] Output report file name
                                  (default -rformat motif)

   Additional (Optional) qualifiers: (none)
   Advanced (Unprompted) qualifiers: (none)
   Associated qualifiers:

   "-sequence" associated qualifiers
   -sbegin1            integer    Start of each sequence to be used
   -send1              integer    End of each sequence to be used
   -sreverse1          boolean    Reverse (if DNA)
   -sask1              boolean    Ask for begin/end/reverse
   -snucleotide1       boolean    Sequence is nucleotide
   -sprotein1          boolean    Sequence is protein
   -slower1            boolean    Make lower case
   -supper1            boolean    Make upper case
   -scircular1         boolean    Sequence is circular
   -squick1            boolean    Read id and sequence only
   -sformat1           string     Input sequence format
   -iquery1            string     Input query fields or ID list
   -ioffset1           integer    Input start position offset
   -sdbname1           string     Database name
   -sid1               string     Entryname
   -ufo1               string     UFO features
   -fformat1           string     Features format
   -fopenfile1         string     Features file name

   "-outfile" associated qualifiers
   -rformat2           string     Report format
   -rname2             string     Base file name
   -rextension2        string     File name extension
   -rdirectory2        string     Output directory
   -raccshow2          boolean    Show accession number in the report
   -rdesshow2          boolean    Show description in the report
   -rscoreshow2        boolean    Show the score in the report
   -rstrandshow2       boolean    Show the nucleotide strand in the report
   -rusashow2          boolean    Show the full USA in the report
   -rmaxall2           integer    Maximum total hits to report
   -rmaxseq2           integer    Maximum hits to report for one sequence

   General qualifiers:
   -auto               boolean    Turn off prompts
   -stdout             boolean    Write first file to standard output
   -filter             boolean    Read first file from standard input, write
                                  first file to standard output
   -options            boolean    Prompt for standard and additional values
   -debug              boolean    Write debug output to program.dbg
   -verbose            boolean    Report some/full command line options
   -help               boolean    Report command line options and exit. More
                                  information on associated and general
                                  qualifiers can be found with -help -verbose
   -warning            boolean    Report warnings
   -error              boolean    Report errors
   -fatal              boolean    Report fatal errors
   -die                boolean    Report dying program messages
   -version            boolean    Report version number and exit

</pre>
</td></tr></table>
<P>
<table border cellspacing=0 cellpadding=3 bgcolor="#ccccff">
<tr bgcolor="#FFFFCC">
<th align="left">Qualifier</th>
<th align="left">Type</th>
<th align="left">Description</th>
<th align="left">Allowed values</th>
<th align="left">Default</th>
</tr>

<tr bgcolor="#FFFFCC">
<th align="left" colspan=5>Standard (Mandatory) qualifiers</th>
</tr>

<tr bgcolor="#FFFFCC">
<td>[-sequence]<br>(Parameter 1)</td>
<td>seqall</td>
<td>Protein sequence(s) filename and optional format, or reference (input USA)</td>
<td>Readable sequence(s)</td>
<td><b>Required</b></td>
</tr>

<tr bgcolor="#FFFFCC">
<td>-minlen</td>
<td>integer</td>
<td>Minimum length of antigenic region</td>
<td>Integer from 1 to 50</td>
<td>6</td>
</tr>

<tr bgcolor="#FFFFCC">
<td>[-outfile]<br>(Parameter 2)</td>
<td>report</td>
<td>Output report file name</td>
<td>(default -rformat motif)</td>
<td><i>&lt;*&gt;</i>.antigenic</td>
</tr>

<tr bgcolor="#FFFFCC">
<th align="left" colspan=5>Additional (Optional) qualifiers</th>
</tr>

<tr>
<td colspan=5>(none)</td>
</tr>

<tr bgcolor="#FFFFCC">
<th align="left" colspan=5>Advanced (Unprompted) qualifiers</th>
</tr>

<tr>
<td colspan=5>(none)</td>
</tr>

<tr bgcolor="#FFFFCC">
<th align="left" colspan=5>Associated qualifiers</th>
</tr>

<tr bgcolor="#FFFFCC">
<td align="left" colspan=5>"-sequence" associated seqall qualifiers
</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -sbegin1<br>-sbegin_sequence</td>
<td>integer</td>
<td>Start of each sequence to be used</td>
<td>Any integer value</td>
<td>0</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -send1<br>-send_sequence</td>
<td>integer</td>
<td>End of each sequence to be used</td>
<td>Any integer value</td>
<td>0</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -sreverse1<br>-sreverse_sequence</td>
<td>boolean</td>
<td>Reverse (if DNA)</td>
<td>Boolean value Yes/No</td>
<td>N</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -sask1<br>-sask_sequence</td>
<td>boolean</td>
<td>Ask for begin/end/reverse</td>
<td>Boolean value Yes/No</td>
<td>N</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -snucleotide1<br>-snucleotide_sequence</td>
<td>boolean</td>
<td>Sequence is nucleotide</td>
<td>Boolean value Yes/No</td>
<td>N</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -sprotein1<br>-sprotein_sequence</td>
<td>boolean</td>
<td>Sequence is protein</td>
<td>Boolean value Yes/No</td>
<td>N</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -slower1<br>-slower_sequence</td>
<td>boolean</td>
<td>Make lower case</td>
<td>Boolean value Yes/No</td>
<td>N</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -supper1<br>-supper_sequence</td>
<td>boolean</td>
<td>Make upper case</td>
<td>Boolean value Yes/No</td>
<td>N</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -scircular1<br>-scircular_sequence</td>
<td>boolean</td>
<td>Sequence is circular</td>
<td>Boolean value Yes/No</td>
<td>N</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -squick1<br>-squick_sequence</td>
<td>boolean</td>
<td>Read id and sequence only</td>
<td>Boolean value Yes/No</td>
<td>N</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -sformat1<br>-sformat_sequence</td>
<td>string</td>
<td>Input sequence format</td>
<td>Any string</td>
<td>&nbsp;</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -iquery1<br>-iquery_sequence</td>
<td>string</td>
<td>Input query fields or ID list</td>
<td>Any string</td>
<td>&nbsp;</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -ioffset1<br>-ioffset_sequence</td>
<td>integer</td>
<td>Input start position offset</td>
<td>Any integer value</td>
<td>0</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -sdbname1<br>-sdbname_sequence</td>
<td>string</td>
<td>Database name</td>
<td>Any string</td>
<td>&nbsp;</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -sid1<br>-sid_sequence</td>
<td>string</td>
<td>Entryname</td>
<td>Any string</td>
<td>&nbsp;</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -ufo1<br>-ufo_sequence</td>
<td>string</td>
<td>UFO features</td>
<td>Any string</td>
<td>&nbsp;</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -fformat1<br>-fformat_sequence</td>
<td>string</td>
<td>Features format</td>
<td>Any string</td>
<td>&nbsp;</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -fopenfile1<br>-fopenfile_sequence</td>
<td>string</td>
<td>Features file name</td>
<td>Any string</td>
<td>&nbsp;</td>
</tr>

<tr bgcolor="#FFFFCC">
<td align="left" colspan=5>"-outfile" associated report qualifiers
</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -rformat2<br>-rformat_outfile</td>
<td>string</td>
<td>Report format</td>
<td>Any string</td>
<td>motif</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -rname2<br>-rname_outfile</td>
<td>string</td>
<td>Base file name</td>
<td>Any string</td>
<td>&nbsp;</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -rextension2<br>-rextension_outfile</td>
<td>string</td>
<td>File name extension</td>
<td>Any string</td>
<td>&nbsp;</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -rdirectory2<br>-rdirectory_outfile</td>
<td>string</td>
<td>Output directory</td>
<td>Any string</td>
<td>&nbsp;</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -raccshow2<br>-raccshow_outfile</td>
<td>boolean</td>
<td>Show accession number in the report</td>
<td>Boolean value Yes/No</td>
<td>N</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -rdesshow2<br>-rdesshow_outfile</td>
<td>boolean</td>
<td>Show description in the report</td>
<td>Boolean value Yes/No</td>
<td>N</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -rscoreshow2<br>-rscoreshow_outfile</td>
<td>boolean</td>
<td>Show the score in the report</td>
<td>Boolean value Yes/No</td>
<td>Y</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -rstrandshow2<br>-rstrandshow_outfile</td>
<td>boolean</td>
<td>Show the nucleotide strand in the report</td>
<td>Boolean value Yes/No</td>
<td>Y</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -rusashow2<br>-rusashow_outfile</td>
<td>boolean</td>
<td>Show the full USA in the report</td>
<td>Boolean value Yes/No</td>
<td>N</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -rmaxall2<br>-rmaxall_outfile</td>
<td>integer</td>
<td>Maximum total hits to report</td>
<td>Any integer value</td>
<td>0</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -rmaxseq2<br>-rmaxseq_outfile</td>
<td>integer</td>
<td>Maximum hits to report for one sequence</td>
<td>Any integer value</td>
<td>0</td>
</tr>

<tr bgcolor="#FFFFCC">
<th align="left" colspan=5>General qualifiers</th>
</tr>

<tr bgcolor="#FFFFCC">
<td> -auto</td>
<td>boolean</td>
<td>Turn off prompts</td>
<td>Boolean value Yes/No</td>
<td>N</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -stdout</td>
<td>boolean</td>
<td>Write first file to standard output</td>
<td>Boolean value Yes/No</td>
<td>N</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -filter</td>
<td>boolean</td>
<td>Read first file from standard input, write first file to standard output</td>
<td>Boolean value Yes/No</td>
<td>N</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -options</td>
<td>boolean</td>
<td>Prompt for standard and additional values</td>
<td>Boolean value Yes/No</td>
<td>N</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -debug</td>
<td>boolean</td>
<td>Write debug output to program.dbg</td>
<td>Boolean value Yes/No</td>
<td>N</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -verbose</td>
<td>boolean</td>
<td>Report some/full command line options</td>
<td>Boolean value Yes/No</td>
<td>Y</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -help</td>
<td>boolean</td>
<td>Report command line options and exit. More information on associated and general qualifiers can be found with -help -verbose</td>
<td>Boolean value Yes/No</td>
<td>N</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -warning</td>
<td>boolean</td>
<td>Report warnings</td>
<td>Boolean value Yes/No</td>
<td>Y</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -error</td>
<td>boolean</td>
<td>Report errors</td>
<td>Boolean value Yes/No</td>
<td>Y</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -fatal</td>
<td>boolean</td>
<td>Report fatal errors</td>
<td>Boolean value Yes/No</td>
<td>Y</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -die</td>
<td>boolean</td>
<td>Report dying program messages</td>
<td>Boolean value Yes/No</td>
<td>Y</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -version</td>
<td>boolean</td>
<td>Report version number and exit</td>
<td>Boolean value Yes/No</td>
<td>N</td>
</tr>

</table>


<H2>
    Input file format
</H2>


The input sequence can be one or more protein sequences.

<p>


<a name="input.1"></a>
<h3>Input files for usage example </h3>

'tsw:actb1_takru' is a sequence entry in the example protein database 'tsw'
<p>
<p><h3>Database entry: tsw:actb1_takru</h3>
<table width="90%"><tr><td bgcolor="#FFCCFF">
<pre>
ID   ACTB1_TAKRU             Reviewed;         375 AA.
AC   P68142; P53484;
DT   25-OCT-2004, integrated into UniProtKB/Swiss-Prot.
DT   25-OCT-2004, sequence version 1.
DT   16-MAY-2012, entry version 49.
DE   RecName: Full=Actin, cytoplasmic 1;
DE   AltName: Full=Beta-actin A;
DE   Contains:
DE     RecName: Full=Actin, cytoplasmic 1, N-terminally processed;
GN   Name=actba;
OS   Takifugu rubripes (Japanese pufferfish) (Fugu rubripes).
OC   Eukaryota; Metazoa; Chordata; Craniata; Vertebrata; Euteleostomi;
OC   Actinopterygii; Neopterygii; Teleostei; Euteleostei; Neoteleostei;
OC   Acanthomorpha; Acanthopterygii; Percomorpha; Tetraodontiformes;
OC   Tetradontoidea; Tetraodontidae; Takifugu.
OX   NCBI_TaxID=31033;
RN   [1]
RP   NUCLEOTIDE SEQUENCE [GENOMIC DNA], AND TISSUE SPECIFICITY.
RX   MEDLINE=96275651; PubMed=8683572; DOI=10.1006/jmbi.1996.0347;
RA   Venkatesh B., Tay B.H., Elgar G., Brenner S.;
RT   "Isolation, characterization and evolution of nine pufferfish (Fugu
RT   rubripes) actin genes.";
RL   J. Mol. Biol. 259:655-665(1996).
CC   -!- FUNCTION: Actins are highly conserved proteins that are involved
CC       in various types of cell motility and are ubiquitously expressed
CC       in all eukaryotic cells.
CC   -!- SUBUNIT: Polymerization of globular actin (G-actin) leads to a
CC       structural filament (F-actin) in the form of a two-stranded helix.
CC       Each actin can bind to 4 others.
CC   -!- SUBCELLULAR LOCATION: Cytoplasm, cytoskeleton.
CC   -!- TISSUE SPECIFICITY: Widely distributed. Not expressed in skeletal
CC       muscle.
CC   -!- PTM: Oxidation of Met-44 by MICALs (MICAL1, MICAL2 or MICAL3) to
CC       form methionine sulfoxide promotes actin filament
CC       depolymerization. Methionine sulfoxide is produced
CC       stereospecifically, but it is not known whether the (S)-S-oxide or
CC       the (R)-S-oxide is produced (By similarity).
CC   -!- MISCELLANEOUS: In vertebrates 3 main groups of actin isoforms,
CC       alpha, beta and gamma have been identified. The alpha actins are
CC       found in muscle tissues and are a major constituent of the
CC       contractile apparatus. The beta and gamma actins coexist in most
CC       cell types as components of the cytoskeleton and as mediators of
CC       internal cell motility.
CC   -!- MISCELLANEOUS: There are three different beta-cytoplasmic actins
CC       in Fugu rubripes.
CC   -!- SIMILARITY: Belongs to the actin family.
CC   -----------------------------------------------------------------------
CC   Copyrighted by the UniProt Consortium, see http://www.uniprot.org/terms
CC   Distributed under the Creative Commons Attribution-NoDerivs License
CC   -----------------------------------------------------------------------
DR   EMBL; U37499; AAC59889.1; -; Genomic_DNA.
DR   PIR; S71124; S71124.
DR   ProteinModelPortal; P68142; -.
DR   SMR; P68142; 2-375.
DR   Ensembl; ENSTRUT00000013141; ENSTRUP00000013080; ENSTRUG00000005447.
DR   eggNOG; COG5277; -.
DR   GeneTree; ENSGT00630000089629; -.
DR   InParanoid; P68142; -.
DR   OMA; IKNLMER; -.
DR   OrthoDB; EOG41JZC9; -.
DR   GO; GO:0005737; C:cytoplasm; IEA:UniProtKB-KW.
DR   GO; GO:0005856; C:cytoskeleton; IEA:UniProtKB-SubCell.
DR   GO; GO:0005524; F:ATP binding; IEA:UniProtKB-KW.
DR   InterPro; IPR004000; Actin-like.
DR   InterPro; IPR020902; Actin/actin-like_CS.
DR   InterPro; IPR004001; Actin_CS.
DR   PANTHER; PTHR11937; Actin_like; 1.
DR   Pfam; PF00022; Actin; 1.
DR   PRINTS; PR00190; ACTIN.
DR   SMART; SM00268; ACTIN; 1.
DR   PROSITE; PS00406; ACTINS_1; 1.
DR   PROSITE; PS00432; ACTINS_2; 1.
DR   PROSITE; PS01132; ACTINS_ACT_LIKE; 1.
PE   2: Evidence at transcript level;
KW   Acetylation; ATP-binding; Complete proteome; Cytoplasm; Cytoskeleton;
KW   Methylation; Nucleotide-binding; Oxidation; Reference proteome.
FT   CHAIN         1    375       Actin, cytoplasmic 1.
FT                                /FTId=PRO_0000367094.
FT   INIT_MET      1      1       Removed; alternate (By similarity).
FT   CHAIN         2    375       Actin, cytoplasmic 1, N-terminally
FT                                processed.
FT                                /FTId=PRO_0000000809.
FT   MOD_RES       1      1       N-acetylmethionine; in Actin, cytoplasmic
FT                                1; alternate (By similarity).
FT   MOD_RES       2      2       N-acetylglutamate; in Actin, cytoplasmic
FT                                1, N-terminally processed (By
FT                                similarity).
FT   MOD_RES      44     44       Methionine sulfoxide (By similarity).
FT   MOD_RES      73     73       Tele-methylhistidine (By similarity).
SQ   SEQUENCE   375 AA;  41767 MW;  9C505616D33E9495 CRC64;
     MEDEIAALVV DNGSGMCKAG FAGDDAPRAV FPSIVGRPRH QGVMVGMGQK DSYVGDEAQS
     KRGILTLKYP IEHGIVTNWD DMEKIWHHTF YNELRVAPEE HPVLLTEAPL NPKANREKMT
     QIMFETFNTP AMYVAIQAVL SLYASGRTTG IVMDSGDGVT HTVPIYEGYA LPHAILRLDL
     AGRDLTDYLM KILTERGYSF TTTAEREIVR DIKEKLCYVA LDFEQEMGTA ASSSSLEKSY
     ELPDGQVITI GNERFRCPEA LFQPSFLGME SCGIHETTYN SIMKCDVDIR KDLYANTVLS
     GGTTMYPGIA DRMQKEITAL APSTMKIKII APPERKYSVW IGGSILASLS TFQQMWISKQ
     EYDESGPSIV HRKCF
//
</pre>
</td></tr></table><p>

<H2>
    Output file format
</H2>

<p>

The output is a standard EMBOSS report file. 

<p>

The results can be output in one of several styles by using the
command-line qualifier <tt>-rformat xxx</tt>, where 'xxx' is replaced
by the name of the required format.  The available format names are:
embl, genbank, gff, pir, swiss, dasgff, debug, listfile, dbmotif,
diffseq, draw, restrict, excel, feattable, motif, nametable, regions,
seqtable, simple, srs, table, tagseq.

<p>

See:
<A href="http://emboss.sf.net/docs/themes/ReportFormats.html">
http://emboss.sf.net/docs/themes/ReportFormats.html</A>
for further information on report formats.

<p>


By default <b>antigenic</b> writes a 'motif' report file.

<p>


<a name="output.1"></a>
<h3>Output files for usage example </h3>
<p><h3>File: actb1_takru.antigenic</h3>
<table width="90%"><tr><td bgcolor="#CCFFCC">
<pre>
########################################
# Program: antigenic
# Rundate: Mon 15 Jul 2013 12:00:00
# Commandline: antigenic
#    -sequence tsw:actb1_takru
# Report_format: motif
# Report_file: actb1_takru.antigenic
########################################

#=======================================
#
# Sequence: ACTB1_TAKRU     from: 1   to: 375
# HitCount: 18
#=======================================

Max_score_pos at "*"

(1) Score 1.207 length 9 at residues 214-&gt;222
               *
 Sequence: EKLCYVALD
           |       |
         214       222
 Max_score_pos: 218

(2) Score 1.187 length 15 at residues 131-&gt;145
                 *
 Sequence: AMYVAIQAVLSLYAS
           |             |
         131             145
 Max_score_pos: 137

(3) Score 1.166 length 8 at residues 5-&gt;12
              *
 Sequence: IAALVVDN
           |      |
           5      12
 Max_score_pos: 8

(4) Score 1.164 length 12 at residues 27-&gt;38
                *
 Sequence: PRAVFPSIVGRP
           |          |
          27          38
 Max_score_pos: 32

(5) Score 1.136 length 24 at residues 160-&gt;183
                        *
 Sequence: THTVPIYEGYALPHAILRLDLAGR
           |                      |
         160                      183


<font color=red>  [Part of this file has been deleted for brevity]</font>

                        *
 Sequence: SSSSLEKSYELPDGQVITI
           |                 |
         232                 250
 Max_score_pos: 245

(13) Score 1.083 length 6 at residues 327-&gt;332
              *
 Sequence: IKIIAP
           |    |
         327    332
 Max_score_pos: 330

(14) Score 1.074 length 7 at residues 317-&gt;323
              *
 Sequence: ITALAPS
           |     |
         317     323
 Max_score_pos: 320

(15) Score 1.068 length 7 at residues 186-&gt;192
                *
 Sequence: TDYLMKI
           |     |
         186     192
 Max_score_pos: 191

(16) Score 1.066 length 7 at residues 40-&gt;46
              *
 Sequence: HQGVMVG
           |     |
          40     46
 Max_score_pos: 43

(17) Score 1.045 length 7 at residues 269-&gt;275
           *
 Sequence: MESCGIH
           |     |
         269     275
 Max_score_pos: 269

(18) Score 1.034 length 7 at residues 51-&gt;57
            *
 Sequence: DSYVGDE
           |     |
          51     57
 Max_score_pos: 52


#---------------------------------------
#---------------------------------------
</pre>
</td></tr></table><p>

<a name="output.2"></a>
<h3>Output files for usage example 2</h3>
<p><h3>File: actb1_takru.antigenic</h3>
<table width="90%"><tr><td bgcolor="#CCFFCC">
<pre>
##gff-version 3
##sequence-region ACTB1_TAKRU 1 375
#!Date 2013-07-15
#!Type Protein
#!Source-version EMBOSS 6.6.0.0
ACTB1_TAKRU	antigenic	epitope	214	222	1.207	.	.	ID=ACTB1_TAKRU.1;note=*pos 218
ACTB1_TAKRU	antigenic	epitope	131	145	1.187	.	.	ID=ACTB1_TAKRU.2;note=*pos 137
ACTB1_TAKRU	antigenic	epitope	5	12	1.166	.	.	ID=ACTB1_TAKRU.3;note=*pos 8
ACTB1_TAKRU	antigenic	epitope	27	38	1.164	.	.	ID=ACTB1_TAKRU.4;note=*pos 32
ACTB1_TAKRU	antigenic	epitope	160	183	1.136	.	.	ID=ACTB1_TAKRU.5;note=*pos 173
ACTB1_TAKRU	antigenic	epitope	367	372	1.135	.	.	ID=ACTB1_TAKRU.6;note=*pos 372
ACTB1_TAKRU	antigenic	epitope	93	108	1.116	.	.	ID=ACTB1_TAKRU.7;note=*pos 103
ACTB1_TAKRU	antigenic	epitope	295	301	1.113	.	.	ID=ACTB1_TAKRU.8;note=*pos 296
ACTB1_TAKRU	antigenic	epitope	256	266	1.11	.	.	ID=ACTB1_TAKRU.9;note=*pos 264
ACTB1_TAKRU	antigenic	epitope	336	352	1.107	.	.	ID=ACTB1_TAKRU.10;note=*pos 347
ACTB1_TAKRU	antigenic	epitope	62	76	1.102	.	.	ID=ACTB1_TAKRU.11;note=*pos 68
ACTB1_TAKRU	antigenic	epitope	232	250	1.086	.	.	ID=ACTB1_TAKRU.12;note=*pos 245
ACTB1_TAKRU	antigenic	epitope	327	332	1.083	.	.	ID=ACTB1_TAKRU.13;note=*pos 330
ACTB1_TAKRU	antigenic	epitope	317	323	1.074	.	.	ID=ACTB1_TAKRU.14;note=*pos 320
ACTB1_TAKRU	antigenic	epitope	186	192	1.068	.	.	ID=ACTB1_TAKRU.15;note=*pos 191
ACTB1_TAKRU	antigenic	epitope	40	46	1.066	.	.	ID=ACTB1_TAKRU.16;note=*pos 43
ACTB1_TAKRU	antigenic	epitope	269	275	1.045	.	.	ID=ACTB1_TAKRU.17;note=*pos 269
ACTB1_TAKRU	antigenic	epitope	51	57	1.034	.	.	ID=ACTB1_TAKRU.18;note=*pos 52
</pre>
</td></tr></table><p>


<H2>
    Data files
</H2>


Antigenic uses a data file called <em>Eantigenic.dat</em>.


<p>
EMBOSS data files are distributed with the application and stored
in the standard EMBOSS data directory, which is defined
by the EMBOSS environment variable EMBOSS_DATA.

<p>

To see the available EMBOSS data files, run:
<p>
<pre>
% embossdata -showall
</pre>
<p>
To fetch one of the data files (for example 'Exxx.dat') into your
current directory for you to inspect or modify, run:

<pre>

% embossdata -fetch -file Exxx.dat

</pre>
<p>

Users can provide their own data files in their own directories.
Project specific files can be put in the current directory, or for
tidier directory listings in a subdirectory called
".embossdata". Files for all EMBOSS runs can be put in the user's home
directory, or again in a subdirectory called ".embossdata".

<p>
The directories are searched in the following order:

<ul>
   <li> . (your current directory)
   <li> .embossdata (under your current directory)
   <li> ~/ (your home directory)
   <li> ~/.embossdata
</ul>
<p>

<p>
Here is the default <em>Eantigenic.dat</em> file:

<pre>
# Kolaskar AS and Tongaonkar PC (1990) FEBS Letters 276:172-174
# "A semi-emipirical method for prediction of antigenic determinants
# on protein antigens"
#
# TABLE 1: Occurrence of amino acids in epitopes, proteins and on the surface,
#         and their antigenic propensity, A(p), values
#
# 169 antigenic determinants experimentally determined. Selected those 156
# which have less than 20 amino acids per determinant (total 2066 residues).
# Calculated f(Ag) as frequency of occurrence of each residue in antigenic
# determinants [f(Ag) = Epitope_occurrence/2066].
#
# Used Hydrophilicity, Accessibility and Flexibility of Parker JMR, Guo D,
# Hodges, RS (1986) Biochemistry 25:5425-5432. In a given protein, calculated
# average for each 7-mer and assigned values to central residue of 7-mer.
# Residue considered to be on the surface if any of the 7-mer values was above
# the average for the protein. Used these results to get f(s) frequency of
# occurrence of amino acids at the surface.
#
# Original table covers the 20 naturally occurring amino acids.
# Values for B, Z, X use weighted averages from Edayhoff.dat
# and are ignored when calculating totals
#
# Antigenic propensity column A(p) = f(Ag)/f(s)
#
# f(s) values below were back-calculated from the table in the paper
#
# Prediction algorithm:
#
# 1. calculate average propensity <A(p)> for each overlapping 7-mer, assign to
#    central residue (i+3)
#
# 2. calculate average for whole protein A(p)av
#
# 3. (a) if average for whole protein >= 1.0 then all residues having
#        <A(p)> >= 1.0 are potentially antigenic.
#    (b) if average for whole protein <  1.0 then all residues having
#        <A(p)> > average for whole protein (??? paper has a mangled
#        formula here :-) are potentially antigenic.
#
# 4. Find 6-mers where all residues are selected by step 3 above
#
#                                               Antigenic  Surface  Antigenic
# Amino     -- Occurrence of amino acids in --   frequency frequency propensity
# Acid       Epitopes      Surface     Protein   f(Ag)    f(s)      A(p)
  A             135          328         524     0.065    0.061     1.064
  B             107          334         410     0.052    0.062     0.827
  C              53           97         186     0.026    0.018     1.412
  D             118          352         414     0.057    0.066     0.866
  E             132          401         499     0.064    0.075     0.851
  F              76          180         365     0.037    0.034     1.091
  G             116          343         487     0.056    0.064     0.874
  H              59          138         191     0.029    0.026     1.105
  I              86          193         437     0.042    0.036     1.152
  K             158          439         523     0.076    0.082     0.930
  L             149          308         684     0.072    0.058     1.250
  M              23           72         152     0.011    0.013     0.826
  N              94          313         407     0.045    0.058     0.776
  P             135          328         411     0.065    0.061     1.064
  Q              99          252         332     0.048    0.047     1.015
  R             106          314         394     0.051    0.058     0.873
  S             168          429         553     0.081    0.080     1.012
  T             141          401         522     0.068    0.075     0.909
  V             128          239         515     0.062    0.045     1.383
  W              19           55         103     0.009    0.010     0.893
  X             118          306         453     0.057    0.057     1.025
  Y              71          158         245     0.034    0.029     1.161
  Z             119          342         433     0.058    0.064     0.916
Total          2066         5340        7944
</pre>

<H2>
    Notes
</H2>
<p>Application of this method to a large number of proteins has shown that their method can predict <b>antigenic</b> determinants with about 75% accuracy which is better than most of the known methods.</p>


<H2>
    References
</H2>


<ol>
<li>
Kolaskar,AS and Tongaonkar,PC (1990).
A semi-empirical method for prediction of antigenic determinants
on protein antigens.
FEBS Letters 276: 172-174.

<li>
Parker,JMR, Guo,D and Hodges,RS (1986).
Biochemistry 25: 5425-5432.
</ol>

<H2>
    Warnings
</H2>


The program will warn you if the sequence is not a protein or has
ambiguity codes.


<H2>
    Diagnostic Error Messages
</H2>


<H2>
    Exit status
</H2>


It exits with status 0.

<H2>
    Known bugs
</H2>


None.

<h2><a name="See also">See also</a></h2>
<table border cellpadding=4 bgcolor="#FFFFF0">
<tr><th>Program name</th>
<th>Description</th></tr>
<tr>
<td><a href="epestfind.html">epestfind</a></td>
<td>Find PEST motifs as potential proteolytic cleavage sites</td>
</tr>

<tr>
<td><a href="fuzzpro.html">fuzzpro</a></td>
<td>Search for patterns in protein sequences</td>
</tr>

<tr>
<td><a href="fuzztran.html">fuzztran</a></td>
<td>Search for patterns in protein sequences (translated)</td>
</tr>

<tr>
<td><a href="patmatdb.html">patmatdb</a></td>
<td>Search protein sequences with a sequence motif</td>
</tr>

<tr>
<td><a href="patmatmotifs.html">patmatmotifs</a></td>
<td>Scan a protein sequence with motifs from the PROSITE database</td>
</tr>

<tr>
<td><a href="preg.html">preg</a></td>
<td>Regular expression search of protein sequence(s)</td>
</tr>

<tr>
<td><a href="pscan.html">pscan</a></td>
<td>Scan protein sequence(s) with fingerprints from the PRINTS database</td>
</tr>

<tr>
<td><a href="sigcleave.html">sigcleave</a></td>
<td>Report on signal cleavage sites in a protein sequence</td>
</tr>

</table>

<!--
        Add any comments about other associated programs (to prepare
        data files?) that seealso doesn't find.
   -->


<H2>
    Author(s)
</H2>

Alan Bleasby 
<br>
European Bioinformatics Institute, Wellcome Trust Genome Campus, Hinxton, Cambridge CB10 1SD, UK

<p>
Please report all bugs to the EMBOSS bug team (emboss-bug&nbsp;&copy;&nbsp;emboss.open-bio.org) not to the original author.


<p>
Original program "ANTIGENIC" by Peter Rice (EGCG 1991)

Peter Rice
<br>
European Bioinformatics Institute, Wellcome Trust Genome Campus, Hinxton, Cambridge CB10 1SD, UK

<p>
Please report all bugs to the EMBOSS bug team (emboss-bug&nbsp;&copy;&nbsp;emboss.open-bio.org) not to the original author.
<H2>
    History
</H2>


<p>
Completed 9th March 1999


<H2>
    Target users
</H2>

This program is intended to be used by everyone and everything, from naive users to embedded scripts.



<H2>
    Comments
</H2>
None



</BODY>
</HTML>

