<HTML>
<HEAD>
  <TITLE>
  EMBOSS: btwisted
  </TITLE>
</HEAD>
<BODY BGCOLOR="#FFFFFF" text="#000000">

<table align=center border=0 cellspacing=0 cellpadding=0>
<tr><td valign=top>
<A HREF="/" ONMOUSEOVER="self.status='Go to the EMBOSS home page';return true"><img border=0 src="/images/emboss_icon.jpg" alt="" width=150 height=48></a>
</td>
<td align=left valign=middle>
<b><font size="+6">
btwisted
</font></b>
</td></tr>
</table>
<br>&nbsp;
<p>


<H2>
Wiki
</H2>

The master copies of EMBOSS documentation are available
at <a href="http://emboss.open-bio.org/wiki/Appdocs">
http://emboss.open-bio.org/wiki/Appdocs</a>
on the EMBOSS Wiki.

<p>
Please help by correcting and extending the Wiki pages.

<H2>
    Function
</H2>
Calculate the twisting in a B-DNA sequence
<H2>
    Description
</H2>

<p><b>btwisted</b> reads a pure DNA sequence and calculates by simple arithmetic the probable overall twist of the sequence and the stacking energy.</p>


<H2>
    Usage
</H2>
Here is a sample session with <b>btwisted</b>
<p>

<p>
<table width="90%"><tr><td bgcolor="#CCFFFF"><pre>

% <b>btwisted -auto tembl:ab000095 -sbegin 100 -send 120 </b>

</pre></td></tr></table><p>
<p>
<a href="#input.1">Go to the input files for this example</a><br><a href="#output.1">Go to the output files for this example</a><p><p>

<H2>
    Command line arguments
</H2>

<table CELLSPACING=0 CELLPADDING=3 BGCOLOR="#f5f5ff" ><tr><td>
<pre>
Calculate the twisting in a B-DNA sequence
Version: EMBOSS:6.5.6.0

   Standard (Mandatory) qualifiers:
  [-sequence]          sequence   Nucleotide sequence filename and optional
                                  format, or reference (input USA)
  [-outfile]           outfile    [*.btwisted] Output file name

   Additional (Optional) qualifiers: (none)
   Advanced (Unprompted) qualifiers:
   -angledata          datafile   [Eangles.dat] DNA base pair twist angle data
                                  file
   -energydata         datafile   [Eenergy.dat] DNA base pair stacking
                                  energies data file

   Associated qualifiers:

   "-sequence" associated qualifiers
   -sbegin1            integer    Start of the sequence to be used
   -send1              integer    End of the sequence to be used
   -sreverse1          boolean    Reverse (if DNA)
   -sask1              boolean    Ask for begin/end/reverse
   -snucleotide1       boolean    Sequence is nucleotide
   -sprotein1          boolean    Sequence is protein
   -slower1            boolean    Make lower case
   -supper1            boolean    Make upper case
   -scircular1         boolean    Sequence is circular
   -sformat1           string     Input sequence format
   -iquery1            string     Input query fields or ID list
   -ioffset1           integer    Input start position offset
   -sdbname1           string     Database name
   -sid1               string     Entryname
   -ufo1               string     UFO features
   -fformat1           string     Features format
   -fopenfile1         string     Features file name

   "-outfile" associated qualifiers
   -odirectory2        string     Output directory

   General qualifiers:
   -auto               boolean    Turn off prompts
   -stdout             boolean    Write first file to standard output
   -filter             boolean    Read first file from standard input, write
                                  first file to standard output
   -options            boolean    Prompt for standard and additional values
   -debug              boolean    Write debug output to program.dbg
   -verbose            boolean    Report some/full command line options
   -help               boolean    Report command line options and exit. More
                                  information on associated and general
                                  qualifiers can be found with -help -verbose
   -warning            boolean    Report warnings
   -error              boolean    Report errors
   -fatal              boolean    Report fatal errors
   -die                boolean    Report dying program messages
   -version            boolean    Report version number and exit

</pre>
</td></tr></table>
<P>
<table border cellspacing=0 cellpadding=3 bgcolor="#ccccff">
<tr bgcolor="#FFFFCC">
<th align="left">Qualifier</th>
<th align="left">Type</th>
<th align="left">Description</th>
<th align="left">Allowed values</th>
<th align="left">Default</th>
</tr>

<tr bgcolor="#FFFFCC">
<th align="left" colspan=5>Standard (Mandatory) qualifiers</th>
</tr>

<tr bgcolor="#FFFFCC">
<td>[-sequence]<br>(Parameter 1)</td>
<td>sequence</td>
<td>Nucleotide sequence filename and optional format, or reference (input USA)</td>
<td>Readable sequence</td>
<td><b>Required</b></td>
</tr>

<tr bgcolor="#FFFFCC">
<td>[-outfile]<br>(Parameter 2)</td>
<td>outfile</td>
<td>Output file name</td>
<td>Output file</td>
<td><i>&lt;*&gt;</i>.btwisted</td>
</tr>

<tr bgcolor="#FFFFCC">
<th align="left" colspan=5>Additional (Optional) qualifiers</th>
</tr>

<tr>
<td colspan=5>(none)</td>
</tr>

<tr bgcolor="#FFFFCC">
<th align="left" colspan=5>Advanced (Unprompted) qualifiers</th>
</tr>

<tr bgcolor="#FFFFCC">
<td>-angledata</td>
<td>datafile</td>
<td>DNA base pair twist angle data file</td>
<td>Data file</td>
<td>Eangles.dat</td>
</tr>

<tr bgcolor="#FFFFCC">
<td>-energydata</td>
<td>datafile</td>
<td>DNA base pair stacking energies data file</td>
<td>Data file</td>
<td>Eenergy.dat</td>
</tr>

<tr bgcolor="#FFFFCC">
<th align="left" colspan=5>Associated qualifiers</th>
</tr>

<tr bgcolor="#FFFFCC">
<td align="left" colspan=5>"-sequence" associated sequence qualifiers
</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -sbegin1<br>-sbegin_sequence</td>
<td>integer</td>
<td>Start of the sequence to be used</td>
<td>Any integer value</td>
<td>0</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -send1<br>-send_sequence</td>
<td>integer</td>
<td>End of the sequence to be used</td>
<td>Any integer value</td>
<td>0</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -sreverse1<br>-sreverse_sequence</td>
<td>boolean</td>
<td>Reverse (if DNA)</td>
<td>Boolean value Yes/No</td>
<td>N</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -sask1<br>-sask_sequence</td>
<td>boolean</td>
<td>Ask for begin/end/reverse</td>
<td>Boolean value Yes/No</td>
<td>N</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -snucleotide1<br>-snucleotide_sequence</td>
<td>boolean</td>
<td>Sequence is nucleotide</td>
<td>Boolean value Yes/No</td>
<td>N</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -sprotein1<br>-sprotein_sequence</td>
<td>boolean</td>
<td>Sequence is protein</td>
<td>Boolean value Yes/No</td>
<td>N</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -slower1<br>-slower_sequence</td>
<td>boolean</td>
<td>Make lower case</td>
<td>Boolean value Yes/No</td>
<td>N</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -supper1<br>-supper_sequence</td>
<td>boolean</td>
<td>Make upper case</td>
<td>Boolean value Yes/No</td>
<td>N</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -scircular1<br>-scircular_sequence</td>
<td>boolean</td>
<td>Sequence is circular</td>
<td>Boolean value Yes/No</td>
<td>N</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -sformat1<br>-sformat_sequence</td>
<td>string</td>
<td>Input sequence format</td>
<td>Any string</td>
<td>&nbsp;</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -iquery1<br>-iquery_sequence</td>
<td>string</td>
<td>Input query fields or ID list</td>
<td>Any string</td>
<td>&nbsp;</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -ioffset1<br>-ioffset_sequence</td>
<td>integer</td>
<td>Input start position offset</td>
<td>Any integer value</td>
<td>0</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -sdbname1<br>-sdbname_sequence</td>
<td>string</td>
<td>Database name</td>
<td>Any string</td>
<td>&nbsp;</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -sid1<br>-sid_sequence</td>
<td>string</td>
<td>Entryname</td>
<td>Any string</td>
<td>&nbsp;</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -ufo1<br>-ufo_sequence</td>
<td>string</td>
<td>UFO features</td>
<td>Any string</td>
<td>&nbsp;</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -fformat1<br>-fformat_sequence</td>
<td>string</td>
<td>Features format</td>
<td>Any string</td>
<td>&nbsp;</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -fopenfile1<br>-fopenfile_sequence</td>
<td>string</td>
<td>Features file name</td>
<td>Any string</td>
<td>&nbsp;</td>
</tr>

<tr bgcolor="#FFFFCC">
<td align="left" colspan=5>"-outfile" associated outfile qualifiers
</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -odirectory2<br>-odirectory_outfile</td>
<td>string</td>
<td>Output directory</td>
<td>Any string</td>
<td>&nbsp;</td>
</tr>

<tr bgcolor="#FFFFCC">
<th align="left" colspan=5>General qualifiers</th>
</tr>

<tr bgcolor="#FFFFCC">
<td> -auto</td>
<td>boolean</td>
<td>Turn off prompts</td>
<td>Boolean value Yes/No</td>
<td>N</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -stdout</td>
<td>boolean</td>
<td>Write first file to standard output</td>
<td>Boolean value Yes/No</td>
<td>N</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -filter</td>
<td>boolean</td>
<td>Read first file from standard input, write first file to standard output</td>
<td>Boolean value Yes/No</td>
<td>N</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -options</td>
<td>boolean</td>
<td>Prompt for standard and additional values</td>
<td>Boolean value Yes/No</td>
<td>N</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -debug</td>
<td>boolean</td>
<td>Write debug output to program.dbg</td>
<td>Boolean value Yes/No</td>
<td>N</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -verbose</td>
<td>boolean</td>
<td>Report some/full command line options</td>
<td>Boolean value Yes/No</td>
<td>Y</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -help</td>
<td>boolean</td>
<td>Report command line options and exit. More information on associated and general qualifiers can be found with -help -verbose</td>
<td>Boolean value Yes/No</td>
<td>N</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -warning</td>
<td>boolean</td>
<td>Report warnings</td>
<td>Boolean value Yes/No</td>
<td>Y</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -error</td>
<td>boolean</td>
<td>Report errors</td>
<td>Boolean value Yes/No</td>
<td>Y</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -fatal</td>
<td>boolean</td>
<td>Report fatal errors</td>
<td>Boolean value Yes/No</td>
<td>Y</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -die</td>
<td>boolean</td>
<td>Report dying program messages</td>
<td>Boolean value Yes/No</td>
<td>Y</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -version</td>
<td>boolean</td>
<td>Report version number and exit</td>
<td>Boolean value Yes/No</td>
<td>N</td>
</tr>

</table>


<H2>
    Input file format
</H2>

<b>btwisted</b> reads a single nucleotide sequence.
<p>
<p>

The input is a standard EMBOSS sequence query (also known as a 'USA').

<p>
Major sequence database sources defined as standard in EMBOSS
installations include srs:embl, srs:uniprot and ensembl

<p>

Data can also be read from sequence output in any supported format
written by an EMBOSS or third-party application.

<p>
The input format can be specified by using the
command-line qualifier <tt>-sformat xxx</tt>, where 'xxx' is replaced
by the name of the required format.  The available format names are:
gff (gff3), gff2, embl (em), genbank (gb, refseq), ddbj, refseqp, pir
(nbrf), swissprot (swiss, sw), dasgff and debug.

<p>

See:
<A href="http://emboss.sf.net/docs/themes/SequenceFormats.html">
http://emboss.sf.net/docs/themes/SequenceFormats.html</A>
for further information on sequence formats.


<p>

<a name="input.1"></a>
<h3>Input files for usage example </h3>

'tembl:ab000095' is a sequence entry in the example nucleic acid database 'tembl'
<p>
<p><h3>Database entry: tembl:ab000095</h3>
<table width="90%"><tr><td bgcolor="#FFCCFF">
<pre>
ID   AB000095; SV 1; linear; mRNA; STD; HUM; 2399 BP.
XX
AC   AB000095;
XX
DT   10-MAR-1998 (Rel. 54, Created)
DT   07-OCT-2008 (Rel. 97, Last updated, Version 3)
XX
DE   Homo sapiens mRNA for hepatocyte growth factor activator inhibitor,
DE   complete cds.
XX
KW   hepatocyte growth factor activator inhibitor.
XX
OS   Homo sapiens (human)
OC   Eukaryota; Metazoa; Chordata; Craniata; Vertebrata; Euteleostomi; Mammalia;
OC   Eutheria; Euarchontoglires; Primates; Haplorrhini; Catarrhini; Hominidae;
OC   Homo.
XX
RN   [1]
RP   1-2399
RA   Denda K.;
RT   ;
RL   Submitted (24-DEC-1996) to the INSDC.
RL   Kimitoshi Denda, Tokyo Institute of Technology, Department of Life Science;
RL   4259 Nagatsuta, Midori-ku, Yokohama, Kanagawa 227, Japan
RL   (E-mail:kdenda@bio.titech.ac.jp, Tel:45-924-5702, Fax:45-924-5771)
XX
RN   [2]
RX   DOI; 10.1074/jbc.272.10.6370.
RX   PUBMED; 9045658.
RA   Shimomura T., Denda K., Kitamura A., Kawaguchi T., Kito M., Kondo J.,
RA   Kagaya S., Qin L., Takata H., Miyazawa K., Kitamura N.;
RT   "Hepatocyte growth factor activator inhibitor, a novel Kunitz-type serine
RT   protease inhibitor";
RL   J Biol Chem 272(10):6370-6376(1997).
XX
DR   Ensembl-Gn; ENSG00000166145; Homo_sapiens.
DR   Ensembl-Tr; ENST00000344051; Homo_sapiens.
DR   Ensembl-Tr; ENST00000431806; Homo_sapiens.
DR   Ensembl-Tr; ENST00000536281; Homo_sapiens.
DR   Ensembl-Tr; ENST00000562057; Homo_sapiens.
DR   Ensembl-Tr; ENST00000563656; Homo_sapiens.
DR   Ensembl-Tr; ENST00000564375; Homo_sapiens.
DR   Ensembl-Tr; ENST00000566928; Homo_sapiens.
DR   Ensembl-Tr; ENST00000568580; Homo_sapiens.
DR   Ensembl-Tr; ENST00000568823; Homo_sapiens.
DR   Ensembl-Tr; ENST00000569589; Homo_sapiens.
XX
FH   Key             Location/Qualifiers
FH
FT   source          1..2399


<font color=red>  [Part of this file has been deleted for brevity]</font>

FT                   QPDRGEDAIAACFLINCLYEQNFVCKFAPREGFINYLTREVYRSYRQLRTQGFGGSGIP
FT                   KAWAGIDLKVQPQEPLVLKDVENTDWRLLRGDTDVRVERKDPNQVELWGLKEGTYLFQL
FT                   TVTSSDHPEDTANVTVTVLSTKQTEDYCLASNKVGRCRGSFPRWYYDPTEQICKSFVYG
FT                   GCLGNKNNYLREEECILACRGVQGPSMERRHPVCSGTCQPTQFRCSNGCCIDSFLECDD
FT                   TPNCPDASDEAACEKYTSGFDELQRIHFPSDKGHCVDLPDTGLCKESIPRWYYNPFSEH
FT                   CARFTYGGCYGNKNNFEEEQQCLESCRGISKKDVFGLRREIPIPSTGSVEMAVAVFLVI
FT                   CIVVVVAILGYCFFKNQRKDFHGHHHHPPPTPASSTVSTTEDTEHLVYNHTTRPL"
FT   polyA_signal    2379..2384
XX
SQ   Sequence 2399 BP; 490 A; 777 C; 684 G; 448 T; 0 other;
     cggccgagcc cagctctccg agcaccgggt cggaagccgc gacccgagcc gcgcaggaag        60
     ctgggaccgg aacctcggcg gacccggccc cacccaactc acctgcgcag gtcaccagca       120
     ccctcggaac ccagaggccc gcgctctgaa ggtgaccccc ctggggagga aggcgatggc       180
     ccctgcgagg acgatggccc gcgcccgcct cgccccggcc ggcatccctg ccgtcgcctt       240
     gtggcttctg tgcacgctcg gcctccaggg cacccaggcc gggccaccgc ccgcgccccc       300
     tgggctgccc gcgggagccg actgcctgaa cagctttacc gccggggtgc ctggcttcgt       360
     gctggacacc aacgcctcgg tcagcaacgg agctaccttc ctggagtccc ccaccgtgcg       420
     ccggggctgg gactgcgtgc gcgcctgctg caccacccag aactgcaact tggcgctagt       480
     ggagctgcag cccgaccgcg gggaggacgc catcgccgcc tgcttcctca tcaactgcct       540
     ctacgagcag aacttcgtgt gcaagttcgc gcccagggag ggcttcatca actacctcac       600
     gagggaagtg taccgctcct accgccagct gcggacccag ggctttggag ggtctgggat       660
     ccccaaggcc tgggcaggca tagacttgaa ggtacaaccc caggaacccc tggtgctgaa       720
     ggatgtggaa aacacagatt ggcgcctact gcggggtgac acggatgtca gggtagagag       780
     gaaagaccca aaccaggtgg aactgtgggg actcaaggaa ggcacctacc tgttccagct       840
     gacagtgact agctcagacc acccagagga cacggccaac gtcacagtca ctgtgctgtc       900
     caccaagcag acagaagact actgcctcgc atccaacaag gtgggtcgct gccggggctc       960
     tttcccacgc tggtactatg accccacgga gcagatctgc aagagtttcg tttatggagg      1020
     ctgcttgggc aacaagaaca actaccttcg ggaagaagag tgcattctag cctgtcgggg      1080
     tgtgcaaggc ccctccatgg aaaggcgcca tccagtgtgc tctggcacct gtcagcccac      1140
     ccagttccgc tgcagcaatg gctgctgcat cgacagtttc ctggagtgtg acgacacccc      1200
     caactgcccc gacgcctccg acgaggctgc ctgtgaaaaa tacacgagtg gctttgacga      1260
     gctccagcgc atccatttcc ccagtgacaa agggcactgc gtggacctgc cagacacagg      1320
     actctgcaag gagagcatcc cgcgctggta ctacaacccc ttcagcgaac actgcgcccg      1380
     ctttacctat ggtggttgtt atggcaacaa gaacaacttt gaggaagagc agcagtgcct      1440
     cgagtcttgt cgcggcatct ccaagaagga tgtgtttggc ctgaggcggg aaatccccat      1500
     tcccagcaca ggctctgtgg agatggctgt cgcagtgttc ctggtcatct gcattgtggt      1560
     ggtggtagcc atcttgggtt actgcttctt caagaaccag agaaaggact tccacggaca      1620
     ccaccaccac ccaccaccca cccctgccag ctccactgtc tccactaccg aggacacgga      1680
     gcacctggtc tataaccaca ccacccggcc cctctgagcc tgggtctcac cggctctcac      1740
     ctggccctgc ttcctgcttg ccaaggcaga ggcctgggct gggaaaaact ttggaaccag      1800
     actcttgcct gtttcccagg cccactgtgc ctcagagacc agggctccag cccctcttgg      1860
     agaagtctca gctaagctca cgtcctgaga aagctcaaag gtttggaagg agcagaaaac      1920
     ccttgggcca gaagtaccag actagatgga cctgcctgca taggagtttg gaggaagttg      1980
     gagttttgtt tcctctgttc aaagctgcct gtccctaccc catggtgcta ggaagaggag      2040
     tggggtggtg tcagaccctg gaggccccaa ccctgtcctc ccgagctcct cttccatgct      2100
     gtgcgcccag ggctgggagg aaggacttcc ctgtgtagtt tgtgctgtaa agagttgctt      2160
     tttgtttatt taatgctgtg gcatgggtga agaggagggg aagaggcctg tttggcctct      2220
     ctgtcctctc ttcctcttcc cccaagattg agctctctgc ccttgatcag ccccaccctg      2280
     gcctagacca gcagacagag ccaggagagg ctcagctgca ttccgcagcc cccaccccca      2340
     aggttctcca acatcacagc ccagcccacc cactgggtaa taaaagtggt ttgtggaaa       2399
//
</pre>
</td></tr></table><p>

<H2>
    Output file format
</H2>

<a name="output.1"></a>
<h3>Output files for usage example </h3>
<p><h3>File: ab000095.btwisted</h3>
<table width="90%"><tr><td bgcolor="#CCFFCC">
<pre>
# Output from BTWISTED
# Twisting calculated from 100 to 120 of AB000095
Total twist (degrees): 681.1
Total turns : 1.89
Average bases per turn: 11.10
Total stacking energy : -179.34
Average stacking energy per dinucleotide: -8.97
</pre>
</td></tr></table><p>


<H2>
    Data files
</H2>
<p>The program requires two data files.  A file (<tt>Eangles.dat</tt> by default) containing base pair twist angles, and another (<tt>Eenergy.dat</tt>) containing base pair stacking energies. No check is made that all dinucleotides are listed in the files and so read for their energy and twist. Attempting to run <b>btwisted</b> with an incomplete data file will result a fatal error if the sequence contains a dinucleotide not in the data files.</p>


<P><tt>Eangles.dat</tt> consists of the data:</P>
<pre>
aa      35.6    0.1
ac      34.4    1.3
ag      27.7    1.5
at      31.5    1.1
ca      34.5    0.9
cc      33.7    0.1
cg      29.8    1.1
ct      27.7    1.5
ga      36.9    0.9
gc      40.0    1.2
gg      33.7    0.1
gt      34.4    1.3
ta      36.0    1.0
tc      36.9    0.9
tg      34.5    0.9
tt      35.6    0.1
</pre>


<p><tt>'Eenergy.dat'</tt> consists of the data:</P>
<pre>
#base pair stacking energy for B-DNA
aa      -5.37
ac      -10.51
ag      -6.78
at      -6.57
ca      -6.57
cc      -8.26
cg      -9.69
ct      -6.78
ga      -9.81
gc      -14.59
gg      -8.26
gt      -10.51
ta      -3.82
tc      -9.81
tg      -6.57
tt      -5.37
</pre>
<P>


 

<P>
<p>
EMBOSS data files are distributed with the application and stored
in the standard EMBOSS data directory, which is defined
by the EMBOSS environment variable EMBOSS_DATA.

<p>

To see the available EMBOSS data files, run:
<p>
<pre>
% embossdata -showall
</pre>
<p>
To fetch one of the data files (for example 'Exxx.dat') into your
current directory for you to inspect or modify, run:

<pre>

% embossdata -fetch -file Exxx.dat

</pre>
<p>

Users can provide their own data files in their own directories.
Project specific files can be put in the current directory, or for
tidier directory listings in a subdirectory called
".embossdata". Files for all EMBOSS runs can be put in the user's home
directory, or again in a subdirectory called ".embossdata".

<p>
The directories are searched in the following order:

<ul>
   <li> . (your current directory)
   <li> .embossdata (under your current directory)
   <li> ~/ (your home directory)
   <li> ~/.embossdata
</ul>
<p>
<p>

<H2>
    Notes
</H2>

None.

<H2>
    References
</H2>


None.

<H2>
    Warnings
</H2>

No check is made that all dinucleotides have been read for the energy
and twist. Attempting to run btwisted with an incomplete data file
will result a fatal error if the sequence contains a dinucleotide not
in the data files.

<H2>
    Diagnostic Error Messages
</H2>

"Incomplete table" 

<BR> 

You have supplied data files for either the angles or energies that do
not contain the fulll required set of possible dinucleotides. 



<H2>
    Exit status
</H2>

It always exits with status 0.

<H2>
    Known bugs
</H2>


No check is made that all dinucleotides have been read for the energy
and twist. Attempting to run btwisted with an incomplete data file
will result a fatal error if the sequence contains a dinucleotide not
in the data files.



<h2><a name="See also">See also</a></h2>
<table border cellpadding=4 bgcolor="#FFFFF0">
<tr><th>Program name</th>
<th>Description</th></tr>
<tr>
<td><a href="banana.html">banana</a></td>
<td>Plot bending and curvature data for B-DNA</td>
</tr>

<tr>
<td><a href="chaos.html">chaos</a></td>
<td>Draw a chaos game representation plot for a nucleotide sequence</td>
</tr>

<tr>
<td><a href="compseq.html">compseq</a></td>
<td>Calculate the composition of unique words in sequences</td>
</tr>

<tr>
<td><a href="dan.html">dan</a></td>
<td>Calculate nucleic acid melting temperature</td>
</tr>

<tr>
<td><a href="density.html">density</a></td>
<td>Draw a nucleic acid density plot</td>
</tr>

<tr>
<td><a href="einverted.html">einverted</a></td>
<td>Find inverted repeats in nucleotide sequences</td>
</tr>

<tr>
<td><a href="freak.html">freak</a></td>
<td>Generate residue/base frequency table or plot</td>
</tr>

<tr>
<td><a href="isochore.html">isochore</a></td>
<td>Plot isochores in DNA sequences</td>
</tr>

<tr>
<td><a href="sirna.html">sirna</a></td>
<td>Find siRNA duplexes in mRNA</td>
</tr>

<tr>
<td><a href="wordcount.html">wordcount</a></td>
<td>Count and extract unique words in molecular sequence(s)</td>
</tr>

</table>


<H2>
    Author(s)
</H2>
David Martin  

<p>
Please report all bugs to the EMBOSS bug team (emboss-bug&nbsp;&copy;&nbsp;emboss.open-bio.org) not to the original author.
and

Alan Bleasby 
<br>
European Bioinformatics Institute, Wellcome Trust Genome Campus, Hinxton, Cambridge CB10 1SD, UK

<p>
Please report all bugs to the EMBOSS bug team (emboss-bug&nbsp;&copy;&nbsp;emboss.open-bio.org) not to the original author.

<p>
This application was written during the first Scandinavian EMBOSS Workshop.


<H2>
    History
</H2>

Written (Sept 2000) - David Martin & Alan Bleasby.



<H2>
    Target users
</H2>
This program is intended to be used by everyone and everything, from naive users to embedded scripts.


<H2>
    Comments
</H2>
None


</BODY>
</HTML>

