<HTML>

<HEAD>
  <TITLE>
  EMBOSS: dbiflat
  </TITLE>
</HEAD>
<BODY BGCOLOR="#FFFFFF" text="#000000">

<table align=center border=0 cellspacing=0 cellpadding=0>
<tr><td valign=top>
<A HREF="/" ONMOUSEOVER="self.status='Go to the EMBOSS home page';return true"><img border=0 src="emboss_icon.jpg" alt="" width=150 height=48></a>
</td>
<td align=left valign=middle>
<b><font size="+6">
dbiflat
</font></b>
</td></tr>
</table>
<br>&nbsp;
<p>


<H2>
Wiki
</H2>

The master copies of EMBOSS documentation are available
at <a href="http://emboss.open-bio.org/wiki/Appdocs">
http://emboss.open-bio.org/wiki/Appdocs</a>
on the EMBOSS Wiki.

<p>
Please help by correcting and extending the Wiki pages.

<H2>
    Function
</H2>
Index a flat file database
<H2>
    Description
</H2>

<b>dbiflat</b> indexes a flat file database of one or more files, and builds
EMBL CD-ROM format index files.

<p>
The resulting index-file format is used by the software on
the EMBL database CD-ROM distribution and by the Staden package in
addition to EMBOSS, and appears to be the most generally used and
publicly available index file format for these databases.


<p>
A <b>flat file database</b> is a database in one of the native formats:
EMBL, Swiss-Prot or GenBank, as distributed by the EBI, NCBI, or DDBJ. 
(It is called a 'flat' file to distinguish it from the complex
relational database formats that this data is held in before being
written to files and distributed.)

                        


<H2>
    Usage
</H2>
Here is a sample session with <b>dbiflat</b>
<p>

<p>
<table width="90%"><tr><td bgcolor="#CCFFFF"><pre>

% <b>dbiflat </b>
Index a flat file database
Database name: <b>EMBL</b>
      EMBL : EMBL
     SWISS : Swiss-Prot, SpTrEMBL, TrEMBLnew
        GB : Genbank, DDBJ
    REFSEQ : Refseq
Entry format [SWISS]: <b>EMBL</b>
Database directory [.]: <b>embl</b>
Wildcard database filename [*.dat]: <b>rod.dat</b>
Release number [0.0]: <b></b>
Index date [00/00/00]: <b></b>
General log output file [outfile.dbiflat]: <b></b>

</pre></td></tr></table><p>
<p>
<a href="#output.1">Go to the output files for this example</a><p><p>

<H2>
    Command line arguments
</H2>
<table CELLSPACING=0 CELLPADDING=3 BGCOLOR="#f5f5ff" ><tr><td>
<pre>
Index a flat file database
Version: EMBOSS:6.4.0.0

   Standard (Mandatory) qualifiers:
  [-dbname]            string     Database name (Any string from 2 to 19
                                  characters, matching regular expression
                                  /[A-z][A-z0-9_]+/)
   -idformat           menu       [SWISS] Entry format (Values: EMBL (EMBL);
                                  SWISS (Swiss-Prot, SpTrEMBL, TrEMBLnew); GB
                                  (Genbank, DDBJ); REFSEQ (Refseq))
   -directory          directory  [.] Database directory
   -filenames          string     [*.dat] Wildcard database filename (Any
                                  string)
   -release            string     [0.0] Release number (Any string up to 9
                                  characters)
   -date               string     [00/00/00] Index date (Date string dd/mm/yy)
   -outfile            outfile    [*.dbiflat] General log output file

   Additional (Optional) qualifiers: (none)
   Advanced (Unprompted) qualifiers:
   -fields             menu       [acc] Index fields (Values: acc (acnum
                                  accession number index); sv (seqvn sequence
                                  version and gi number index); des (des
                                  description index); key (keyword keywords
                                  index); org (taxon taxonomy and organism
                                  index))
   -exclude            string     Wildcard filename(s) to exclude (Any string)
   -maxindex           integer    [0] Maximum index length (Integer 0 or more)
   -sortoptions        string     [-T . -k 1,1] Sort options, typically '-T .'
                                  to use current directory for work files and
                                  '-k 1,1' to force GNU sort to use the first
                                  field (Any string)
   -[no]systemsort     boolean    [Y] Use system sort utility
   -[no]cleanup        boolean    [Y] Clean up temporary files
   -indexoutdir        outdir     [.] Index file output directory

   Associated qualifiers:

   "-directory" associated qualifiers
   -extension          string     Default file extension

   "-outfile" associated qualifiers
   -odirectory         string     Output directory

   "-indexoutdir" associated qualifiers
   -extension          string     Default file extension

   General qualifiers:
   -auto               boolean    Turn off prompts
   -stdout             boolean    Write first file to standard output
   -filter             boolean    Read first file from standard input, write
                                  first file to standard output
   -options            boolean    Prompt for standard and additional values
   -debug              boolean    Write debug output to program.dbg
   -verbose            boolean    Report some/full command line options
   -help               boolean    Report command line options and exit. More
                                  information on associated and general
                                  qualifiers can be found with -help -verbose
   -warning            boolean    Report warnings
   -error              boolean    Report errors
   -fatal              boolean    Report fatal errors
   -die                boolean    Report dying program messages
   -version            boolean    Report version number and exit

</pre>
</td></tr></table>
<P>
<table border cellspacing=0 cellpadding=3 bgcolor="#ccccff">
<tr bgcolor="#FFFFCC">
<th align="left">Qualifier</th>
<th align="left">Type</th>
<th align="left">Description</th>
<th align="left">Allowed values</th>
<th align="left">Default</th>
</tr>

<tr bgcolor="#FFFFCC">
<th align="left" colspan=5>Standard (Mandatory) qualifiers</th>
</tr>

<tr bgcolor="#FFFFCC">
<td>[-dbname]<br>(Parameter 1)</td>
<td>string</td>
<td>Database name</td>
<td>Any string from 2 to 19 characters, matching regular expression /[A-z][A-z0-9_]+/</td>
<td><b>Required</b></td>
</tr>

<tr bgcolor="#FFFFCC">
<td>-idformat</td>
<td>list</td>
<td>Entry format</td>
<td><table><tr><td>EMBL</td> <td><i>(EMBL)</i></td></tr><tr><td>SWISS</td> <td><i>(Swiss-Prot, SpTrEMBL, TrEMBLnew)</i></td></tr><tr><td>GB</td> <td><i>(Genbank, DDBJ)</i></td></tr><tr><td>REFSEQ</td> <td><i>(Refseq)</i></td></tr></table></td>
<td>SWISS</td>
</tr>

<tr bgcolor="#FFFFCC">
<td>-directory</td>
<td>directory</td>
<td>Database directory</td>
<td>Directory</td>
<td>.</td>
</tr>

<tr bgcolor="#FFFFCC">
<td>-filenames</td>
<td>string</td>
<td>Wildcard database filename</td>
<td>Any string</td>
<td>*.dat</td>
</tr>

<tr bgcolor="#FFFFCC">
<td>-release</td>
<td>string</td>
<td>Release number</td>
<td>Any string up to 9 characters</td>
<td>0.0</td>
</tr>

<tr bgcolor="#FFFFCC">
<td>-date</td>
<td>string</td>
<td>Index date</td>
<td>Date string dd/mm/yy</td>
<td>00/00/00</td>
</tr>

<tr bgcolor="#FFFFCC">
<td>-outfile</td>
<td>outfile</td>
<td>General log output file</td>
<td>Output file</td>
<td><i>&lt;*&gt;</i>.dbiflat</td>
</tr>

<tr bgcolor="#FFFFCC">
<th align="left" colspan=5>Additional (Optional) qualifiers</th>
</tr>

<tr>
<td colspan=5>(none)</td>
</tr>

<tr bgcolor="#FFFFCC">
<th align="left" colspan=5>Advanced (Unprompted) qualifiers</th>
</tr>

<tr bgcolor="#FFFFCC">
<td>-fields</td>
<td>list</td>
<td>Index fields</td>
<td><table><tr><td>acc</td> <td><i>(acnum accession number index)</i></td></tr><tr><td>sv</td> <td><i>(seqvn sequence version and gi number index)</i></td></tr><tr><td>des</td> <td><i>(des description index)</i></td></tr><tr><td>key</td> <td><i>(keyword keywords index)</i></td></tr><tr><td>org</td> <td><i>(taxon taxonomy and organism index)</i></td></tr></table></td>
<td>acc</td>
</tr>

<tr bgcolor="#FFFFCC">
<td>-exclude</td>
<td>string</td>
<td>Wildcard filename(s) to exclude</td>
<td>Any string</td>
<td>&nbsp;</td>
</tr>

<tr bgcolor="#FFFFCC">
<td>-maxindex</td>
<td>integer</td>
<td>Maximum index length</td>
<td>Integer 0 or more</td>
<td>0</td>
</tr>

<tr bgcolor="#FFFFCC">
<td>-sortoptions</td>
<td>string</td>
<td>Sort options, typically '-T .' to use current directory for work files and '-k 1,1' to force GNU sort to use the first field</td>
<td>Any string</td>
<td>-T . -k 1,1</td>
</tr>

<tr bgcolor="#FFFFCC">
<td>-[no]systemsort</td>
<td>boolean</td>
<td>Use system sort utility</td>
<td>Boolean value Yes/No</td>
<td>Yes</td>
</tr>

<tr bgcolor="#FFFFCC">
<td>-[no]cleanup</td>
<td>boolean</td>
<td>Clean up temporary files</td>
<td>Boolean value Yes/No</td>
<td>Yes</td>
</tr>

<tr bgcolor="#FFFFCC">
<td>-indexoutdir</td>
<td>outdir</td>
<td>Index file output directory</td>
<td>Output directory</td>
<td>.</td>
</tr>

<tr bgcolor="#FFFFCC">
<th align="left" colspan=5>Associated qualifiers</th>
</tr>

<tr bgcolor="#FFFFCC">
<th align="left" colspan=5>"-directory" associated directory qualifiers
</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -extension</td>
<td>string</td>
<td>Default file extension</td>
<td>Any string</td>
<td>&nbsp;</td>
</tr>

<tr bgcolor="#FFFFCC">
<th align="left" colspan=5>"-outfile" associated outfile qualifiers
</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -odirectory</td>
<td>string</td>
<td>Output directory</td>
<td>Any string</td>
<td>&nbsp;</td>
</tr>

<tr bgcolor="#FFFFCC">
<th align="left" colspan=5>"-indexoutdir" associated outdir qualifiers
</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -extension</td>
<td>string</td>
<td>Default file extension</td>
<td>Any string</td>
<td>&nbsp;</td>
</tr>

<tr bgcolor="#FFFFCC">
<th align="left" colspan=5>General qualifiers</th>
</tr>

<tr bgcolor="#FFFFCC">
<td> -auto</td>
<td>boolean</td>
<td>Turn off prompts</td>
<td>Boolean value Yes/No</td>
<td>N</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -stdout</td>
<td>boolean</td>
<td>Write first file to standard output</td>
<td>Boolean value Yes/No</td>
<td>N</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -filter</td>
<td>boolean</td>
<td>Read first file from standard input, write first file to standard output</td>
<td>Boolean value Yes/No</td>
<td>N</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -options</td>
<td>boolean</td>
<td>Prompt for standard and additional values</td>
<td>Boolean value Yes/No</td>
<td>N</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -debug</td>
<td>boolean</td>
<td>Write debug output to program.dbg</td>
<td>Boolean value Yes/No</td>
<td>N</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -verbose</td>
<td>boolean</td>
<td>Report some/full command line options</td>
<td>Boolean value Yes/No</td>
<td>Y</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -help</td>
<td>boolean</td>
<td>Report command line options and exit. More information on associated and general qualifiers can be found with -help -verbose</td>
<td>Boolean value Yes/No</td>
<td>N</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -warning</td>
<td>boolean</td>
<td>Report warnings</td>
<td>Boolean value Yes/No</td>
<td>Y</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -error</td>
<td>boolean</td>
<td>Report errors</td>
<td>Boolean value Yes/No</td>
<td>Y</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -fatal</td>
<td>boolean</td>
<td>Report fatal errors</td>
<td>Boolean value Yes/No</td>
<td>Y</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -die</td>
<td>boolean</td>
<td>Report dying program messages</td>
<td>Boolean value Yes/No</td>
<td>Y</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -version</td>
<td>boolean</td>
<td>Report version number and exit</td>
<td>Boolean value Yes/No</td>
<td>N</td>
</tr>

</table>

<H2>
    Input file format
</H2>

'flat' files (native distribution format) of a protein or nucleic
sequence database. 

<p>

<H2>
    Output file format
</H2>


<a name="output.1"></a>
<h3>Output files for usage example </h3>
<p><h3>File: division.lkp</h3>
<p>This file contains non-printing characters and so cannot be displayed here.
<p><h3>File: entrynam.idx</h3>
<p>This file contains non-printing characters and so cannot be displayed here.
<p><h3>File: acnum.hit</h3>
<p>This file contains non-printing characters and so cannot be displayed here.
<p><h3>File: acnum.trg</h3>
<p>This file contains non-printing characters and so cannot be displayed here.
<p><h3>File: outfile.dbiflat</h3>
<table width="90%"><tr><td bgcolor="#CCFFCC">
<pre>
########################################
# Program: dbiflat
# Rundate: Fri 15 Jul 2011 12:00:00
# Dbname: EMBL
# Release: 0.0
# Date: 15/07/11
# CurrentDirectory: /homes/user/test/qa/dbiflat-ex-keep/
# IndexDirectory: ./
# IndexDirectoryPath: /homes/user/test/qa/dbiflat-ex-keep/
# Maxindex: 0
# Fields: 2
#   Field 1: id
#   Field 2: acc
# Directory: /homes/user/test/embl/
# DirectoryPath: /homes/user/test/embl/
# Filenames: rod.dat
# Exclude: 
# Files: 1
#   File 1: /homes/user/test/embl/rod.dat
########################################
# Commandline: dbiflat
#    -dbname EMBL
#    -idformat EMBL
#    -directory ../../embl
#    -filenames rod.dat
########################################

filename: 'rod.dat'
    id: 6
   acc: 9

Index acc: maxlen 6 items 7

Total 1 files 6 entries (0 duplicates)
</pre>
</td></tr></table><p>
<p>

<b>dbiflat</b> creates four index files.  All are binary but with a
simple format. 

<ul>

<li>division.lkp is the master index file, and has a 300 byte header
containing the database name and date plus information on the record
size. This header is followed by one record for each database file,
giving the full file name for the data file, and optionally a second
sequence file.

<li>entryname.idx is the entry name index. It has the same 300 byte
header, mainly used to store the record size which will depend on the
size of the longest entryname in the database. Each entry is stored in
sorted alphanumeric order so that a binary search can be used to
efficiently find any record. The record also holds the file number
from division.lkp and the offsets in the data and sequence files for
that entry.

<li>acnum.trg holds the accession number information. The file has the
usual 300 byte header, and a sorted list of record by accession number.
Each accession number record contains the first record number in acnum.hit
and the total number of records in acnum.hit so that secondary (duplicated)
accession numbers can be searched.

<li>acnum.hit is a very simple file. After the usual 300 byte header,
each record simply holds the record number in entryname.idx. An
accession number search will use acnum.trg to find a start position
and number of records to read in this file, and will then simply
read the entryname.idx records for each entry in turn.

</ul>

<H2>
    Data files
</H2>

None.

<H2>
    Notes
</H2>


The indexing method depends on each entry having a unique entry name.
No allowance is made for two entries with the same name so it is
not possible to index EMBL and EMBLNEW together.


<p>

Having created the EMBOSS indices for this file, a database can then be
defined in the file <b>emboss.defaults</b> as something like:

<p>

<pre>
DB emrod [
   type: N
   format: embl
   method: emblcd
   directory: /data/embl
]  
</pre>



<h3>Fields Indexed</H3>

By default, <b>dbiflat</b> will index the ID name and the accession
number.

<br>

If they are present in your database, you may also specify that
<b>dbiflat</b> should index the Sequence Version and GI number, the
Keywords and Taxonomy names and the words in the description by using
the '-fields' qualifier with the appropriate values. 


<H2>
    References
</H2>

None.


<H2>
    Warnings
</H2>

None.


<H2>
    Diagnostic Error Messages
</H2>

None.


<H2>
    Exit status
</H2>

It exits with a status of 0 if no problems.


<H2>
    Known bugs
</H2>

None.

<h2><a name="See also">See also</a></h2>
<table border cellpadding=4 bgcolor="#FFFFF0">
<tr><th>Program name</th>
<th>Description</th></tr>
<tr>
<td><a href="dbiblast.html">dbiblast</a></td>
<td>Index a BLAST database</td>
</tr>

<tr>
<td><a href="dbifasta.html">dbifasta</a></td>
<td>Index a fasta file database</td>
</tr>

<tr>
<td><a href="dbigcg.html">dbigcg</a></td>
<td>Index a GCG formatted database</td>
</tr>

<tr>
<td><a href="dbxcompress.html">dbxcompress</a></td>
<td>Compress an uncompressed dbx index</td>
</tr>

<tr>
<td><a href="dbxedam.html">dbxedam</a></td>
<td>Index the EDAM ontology using b+tree indices</td>
</tr>

<tr>
<td><a href="dbxfasta.html">dbxfasta</a></td>
<td>Index a fasta file database using b+tree indices</td>
</tr>

<tr>
<td><a href="dbxflat.html">dbxflat</a></td>
<td>Index a flat file database using b+tree indices</td>
</tr>

<tr>
<td><a href="dbxgcg.html">dbxgcg</a></td>
<td>Index a GCG formatted database using b+tree indices</td>
</tr>

<tr>
<td><a href="dbxobo.html">dbxobo</a></td>
<td>Index an obo ontology using b+tree indices</td>
</tr>

<tr>
<td><a href="dbxreport.html">dbxreport</a></td>
<td>Validate index and report internals for dbx databases</td>
</tr>

<tr>
<td><a href="dbxresource.html">dbxresource</a></td>
<td>Index a data resource catalogue using b+tree indices</td>
</tr>

<tr>
<td><a href="dbxstat.html">dbxstat</a></td>
<td>Dump statistics for dbx databases</td>
</tr>

<tr>
<td><a href="dbxtax.html">dbxtax</a></td>
<td>Index NCBI taxonomy using b+tree indices</td>
</tr>

<tr>
<td><a href="dbxuncompress.html">dbxuncompress</a></td>
<td>Uncompress a compressed dbx index</td>
</tr>

</table>

<H2>
    Author(s)
</H2>

Peter Rice
<br>
European Bioinformatics Institute, Wellcome Trust Genome Campus, Hinxton, Cambridge CB10 1SD, UK

<p>
Please report all bugs to the EMBOSS bug team (emboss-bug&nbsp;&copy;&nbsp;emboss.open-bio.org) not to the original author.
<H2>
    History
</H2>

Completed December 1999

<H2>
    Target users
</H2>
This program is intended to be used by administrators responsible
for software and database installation and maintenance. 

<H2>
    Comments
</H2>
None

</BODY>
</HTML>

