<HTML>
<HEAD>
  <TITLE>
  EMBOSS: drfindresource
  </TITLE>
</HEAD>
<BODY BGCOLOR="#FFFFFF" text="#000000">

<table align=center border=0 cellspacing=0 cellpadding=0>
<tr><td valign=top>
<A HREF="/" ONMOUSEOVER="self.status='Go to the EMBOSS home page';return true"><img border=0 src="/images/emboss_icon.jpg" alt="" width=150 height=48></a>
</td>
<td align=left valign=middle>
<b><font size="+6">
drfindresource
</font></b>
</td></tr>
</table>
<br>&nbsp;
<p>


<H2>
Wiki
</H2>

The master copies of EMBOSS documentation are available
at <a href="http://emboss.open-bio.org/wiki/Appdocs">
http://emboss.open-bio.org/wiki/Appdocs</a>
on the EMBOSS Wiki.

<p>
Please help by correcting and extending the Wiki pages.

<H2>
    Function
</H2>
Find public databases by resource
<!--
DON'T WRITE ANYTHING HERE.
IT IS DONE FOR YOU.
-->




<H2>
    Description
</H2>


<b>drfindresource</b> searches the Data Resource Catalogue to find entries
with EDAM resource terms matching a query string.




<H2>
    Algorithm
</H2>

The first search is of the EDAM ontology topic namespace, using the
term names and their synonynms. All child terms are automatically
included in the set of matches inless the <tt>-nosubclasses</tt>
qualifier is used.

<p>
The <tt>-sensitive</tt> qualifier also searches the definition strings.

<p>
The set of EDAM terms are then compared to entries in the Data
Resource Catalogue, searching the 'etpc' EDAM topic index.



<H2>
    Usage
</H2>

<!--  
	Example usage, as run from the command-line.
        Many examples illustrating different behaviours is good.
 -->

Here is a sample session with <b>drfindresource</b>
<p>

<p>
<table width="90%"><tr><td bgcolor="#CCFFFF"><pre>

% <b>drfindresource pathogen </b>
Find public databases by resource
Data resource output file [drfindresource.drcat]: <b></b>

</pre></td></tr></table><p>
<p>
<a href="#output.1">Go to the output files for this example</a><p><p>



<H2>
    Command line arguments
</H2>

<table CELLSPACING=0 CELLPADDING=3 BGCOLOR="#f5f5ff" ><tr><td>
<pre>
Find public databases by resource
Version: EMBOSS:6.5.0.0

   Standard (Mandatory) qualifiers:
  [-query]             string     List of EDAM data keywords (Any string)
  [-outfile]           outresource [*.drfindresource] Output data resource file
                                  name

   Additional (Optional) qualifiers: (none)
   Advanced (Unprompted) qualifiers:
   -sensitive          boolean    [N] By default, the query keywords are
                                  matched against the EDAM term names (and
                                  synonyms) only. This option also matches the
                                  keywords against the EDAM term definitions
                                  and will therefore (typically) report more
                                  matches.
   -[no]subclasses     boolean    [Y] Extend the query matches to include all
                                  terms which are specialisations (EDAM
                                  sub-classes) of the matched type.

   Associated qualifiers:

   "-outfile" associated qualifiers
   -odirectory2        string     Output directory
   -oformat2           string     Data resource output format

   General qualifiers:
   -auto               boolean    Turn off prompts
   -stdout             boolean    Write first file to standard output
   -filter             boolean    Read first file from standard input, write
                                  first file to standard output
   -options            boolean    Prompt for standard and additional values
   -debug              boolean    Write debug output to program.dbg
   -verbose            boolean    Report some/full command line options
   -help               boolean    Report command line options and exit. More
                                  information on associated and general
                                  qualifiers can be found with -help -verbose
   -warning            boolean    Report warnings
   -error              boolean    Report errors
   -fatal              boolean    Report fatal errors
   -die                boolean    Report dying program messages
   -version            boolean    Report version number and exit

</pre>
</td></tr></table>
<P>
<table border cellspacing=0 cellpadding=3 bgcolor="#ccccff">
<tr bgcolor="#FFFFCC">
<th align="left">Qualifier</th>
<th align="left">Type</th>
<th align="left">Description</th>
<th align="left">Allowed values</th>
<th align="left">Default</th>
</tr>

<tr bgcolor="#FFFFCC">
<th align="left" colspan=5>Standard (Mandatory) qualifiers</th>
</tr>

<tr bgcolor="#FFFFCC">
<td>[-query]<br>(Parameter 1)</td>
<td>string</td>
<td>List of EDAM data keywords</td>
<td>Any string</td>
<td>&nbsp;</td>
</tr>

<tr bgcolor="#FFFFCC">
<td>[-outfile]<br>(Parameter 2)</td>
<td>outresource</td>
<td>Output data resource file name</td>
<td>Data resource entry</td>
<td><i>&lt;*&gt;</i>.drfindresource</td>
</tr>

<tr bgcolor="#FFFFCC">
<th align="left" colspan=5>Additional (Optional) qualifiers</th>
</tr>

<tr>
<td colspan=5>(none)</td>
</tr>

<tr bgcolor="#FFFFCC">
<th align="left" colspan=5>Advanced (Unprompted) qualifiers</th>
</tr>

<tr bgcolor="#FFFFCC">
<td>-sensitive</td>
<td>boolean</td>
<td>By default, the query keywords are matched against the EDAM term names (and synonyms) only. This option also matches the keywords against the EDAM term definitions and will therefore (typically) report more matches.</td>
<td>Boolean value Yes/No</td>
<td>No</td>
</tr>

<tr bgcolor="#FFFFCC">
<td>-[no]subclasses</td>
<td>boolean</td>
<td>Extend the query matches to include all terms which are specialisations (EDAM sub-classes) of the matched type.</td>
<td>Boolean value Yes/No</td>
<td>Yes</td>
</tr>

<tr bgcolor="#FFFFCC">
<th align="left" colspan=5>Associated qualifiers</th>
</tr>

<tr bgcolor="#FFFFCC">
<td align="left" colspan=5>"-outfile" associated outresource qualifiers
</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -odirectory2<br>-odirectory_outfile</td>
<td>string</td>
<td>Output directory</td>
<td>Any string</td>
<td>&nbsp;</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -oformat2<br>-oformat_outfile</td>
<td>string</td>
<td>Data resource output format</td>
<td>Any string</td>
<td>&nbsp;</td>
</tr>

<tr bgcolor="#FFFFCC">
<th align="left" colspan=5>General qualifiers</th>
</tr>

<tr bgcolor="#FFFFCC">
<td> -auto</td>
<td>boolean</td>
<td>Turn off prompts</td>
<td>Boolean value Yes/No</td>
<td>N</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -stdout</td>
<td>boolean</td>
<td>Write first file to standard output</td>
<td>Boolean value Yes/No</td>
<td>N</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -filter</td>
<td>boolean</td>
<td>Read first file from standard input, write first file to standard output</td>
<td>Boolean value Yes/No</td>
<td>N</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -options</td>
<td>boolean</td>
<td>Prompt for standard and additional values</td>
<td>Boolean value Yes/No</td>
<td>N</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -debug</td>
<td>boolean</td>
<td>Write debug output to program.dbg</td>
<td>Boolean value Yes/No</td>
<td>N</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -verbose</td>
<td>boolean</td>
<td>Report some/full command line options</td>
<td>Boolean value Yes/No</td>
<td>Y</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -help</td>
<td>boolean</td>
<td>Report command line options and exit. More information on associated and general qualifiers can be found with -help -verbose</td>
<td>Boolean value Yes/No</td>
<td>N</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -warning</td>
<td>boolean</td>
<td>Report warnings</td>
<td>Boolean value Yes/No</td>
<td>Y</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -error</td>
<td>boolean</td>
<td>Report errors</td>
<td>Boolean value Yes/No</td>
<td>Y</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -fatal</td>
<td>boolean</td>
<td>Report fatal errors</td>
<td>Boolean value Yes/No</td>
<td>Y</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -die</td>
<td>boolean</td>
<td>Report dying program messages</td>
<td>Boolean value Yes/No</td>
<td>Y</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -version</td>
<td>boolean</td>
<td>Report version number and exit</td>
<td>Boolean value Yes/No</td>
<td>N</td>
</tr>

</table>

<!--
DON'T WRITE ANYTHING HERE.
IT IS DONE FOR YOU.
-->








<H2>
    Input file format
</H2>

None.

<p>






<H2>
    Output file format
</H2>

<p>

The output is a standard EMBOSS resource file. 

<p>

The results can be output in one of several styles by using the
command-line qualifier <tt>-oformat xxx</tt>, where 'xxx' is replaced by
the name of the required format.  The available format names are: 
drcat, basic, wsbasic, list.
<p>

See:
<A href="http://emboss.sf.net/docs/themes/ResourceFormats.html">
http://emboss.sf.net/docs/themes/ResourceFormats.html</A>
for further information on resource formats.

<p>

<p>


<a name="output.1"></a>
<h3>Output files for usage example </h3>
<p><h3>File: drfindresource.drcat</h3>
<table width="90%"><tr><td bgcolor="#CCFFCC">
<pre>
ID      ApiDB_TrichDB
IDalt   TrichDB
Name    Trichomonas genome resources (TrichDB)
Desc    Genomic-scale datasets associated with the eukaryotic Trichomonas.
URL     http://trichdb.org/trichdb/
URLrest http://eupathdb.org/eupathdb/serviceList.jsp
Cat     Not available
Taxon   5721 | Trichomonas
EDAMtpc 2821 | Unicellular eukaryotes
EDAMtpc 0783 | Pathogens
EDAMdat 0916 | Gene annotation
EDAMid  2295 | Gene ID
EDAMfmt 2331 | HTML
Xref    SP_FT | None
Query    Gene annotation | HTML | Gene ID | http://trichdb.org/gene/%s
Example Gene ID | TVAG_386080

ID      ApiDB_GiardiaDB
IDalt   GiardiaDB
Name    Giardia genome resources (GiardiaDB)
Desc    Genomic-scale datasets for the eukaryotic pathogen Giardia.
URL     http://giardiadb.org/giardiadb/
URLrest http://eupathdb.org/eupathdb/serviceList.jsp
Cat     Not available
Taxon   5740 | Giardia
EDAMtpc 2821 | Unicellular eukaryotes
EDAMtpc 0783 | Pathogens
EDAMdat 0916 | Gene annotation
EDAMid  2295 | Gene ID
EDAMfmt 2331 | HTML
Xref    SP_FT | None
Query    Gene annotation | HTML | Gene ID | http://giardiadb.org/gene/%s
Example Gene ID | GL50803_102438

ID      GeneDB
Name    GeneDB database from Sanger Institute Pathogen Sequencing Units
Desc    Sequence data and annotation/curation for the whole range of organisms sequenced by the PSU (Sanger Institute Pathogen Sequencing Units).
URL     http://www.genedb.org/
Taxon   1 | all
EDAMtpc 0783 | Pathogens
EDAMdat 0916 | Gene annotation
EDAMid  1026 | Gene symbol
EDAMfmt 2331 | HTML
Xref    SP_FT | None
Query    Gene annotation | HTML | Gene symbol | http://www.genedb.org/gene/%s
Example Gene symbol | ECA4014

ID      EuPathDB
IDalt   ApiDB
Acc     DB-0153


<font color=red>  [Part of this file has been deleted for brevity]</font>

Taxon   1 | all
EDAMtpc 0783 | Pathogens
EDAMdat 2399 | Gene annotation (transcript)
EDAMdat 0895 | Peptide annotation
EDAMdat 0916 | Gene annotation
EDAMid  2759 | Gene ID (VectorBase)
EDAMfmt 2331 | HTML
Xref    SP_explicit | Gene ID (VectorBase)
Query    Gene annotation {VectorBase entry} | HTML | Gene ID (VectorBase) | http://www.vectorbase.org/Genome/BRCGene/?feature=%s
Query    Peptide annotation {VectorBase peptide page} | HTML | Gene ID (VectorBase) | http://www.vectorbase.org/Genome/BRCGene/?feature=%s-PA
Query    Gene annotation (transcript) | HTML | Gene ID (VectorBase) | http://www.vectorbase.org/Genome/BRCGene/?feature=%s-RA
Example Gene ID (VectorBase) | AGAP005025

ID      ApiDB_TriTryPDB
IDalt   TritryPDB
Name    Kinetoplastid genome resources (TritryPDB)
Desc    Kinetoplastid genome resources.
URL     http://tritryPDB.org/tritryPDB/
URLrest http://eupathdb.org/eupathdb/serviceList.jsp
Cat     Not available
Taxon   5653 | Kinetoplastida
EDAMtpc 2821 | Unicellular eukaryotes
EDAMtpc 0783 | Pathogens
EDAMdat 0916 | Gene annotation
EDAMid  2295 | Gene ID
EDAMfmt 2331 | HTML
Xref    SP_FT | None
Query    Gene annotation | HTML | Gene ID | http://tritryPDB.org/gene/%s
Example Gene ID | Tb927.8.620

ID      NMPDR
Acc     DB-0125
Name    National microbial pathogen
Desc    Curated annotations in an environment for comparative analysis of genomes and biological subsystems, with an emphasis on the food-borne pathogens Campylobacter, Listeria, Staphylococcus, Streptococcus, and Vibrio as well as the STD pathogens Chlamydiaceae, Haemophilus, Mycoplasma, Neisseria, Treponema, and Ureaplasma.
URL     http://www.nmpdr.org
URLlink http://www.nmpdr.org/FIG/wiki/view.cgi/FIG/LinkingToTheGenomeViewer
Cat     Genome annotation databases
Taxon   2 | Bacteria
EDAMtpc 0783 | Pathogens
EDAMtpc 0797 | Comparative genomics
EDAMdat 0916 | Gene annotation
EDAMid  1179 | NCBI taxonomy ID
EDAMid  2737 | FIG ID
EDAMfmt 2331 | HTML
Xref    SP_explicit | FIG ID
Query    Gene annotation | HTML | NCBI taxonomy ID | http://www.nmpdr.org/linkin.cgi?genome=%s
Query    Gene annotation {Protein encoding gene or other feature} | HTML | FIG ID | http://www.nmpdr.org/linkin.cgi?id=%s
Example NCBI taxonomy ID | 83333.1
Example FIG ID | 83333.1.peg.123
Example FIG ID | 83333.1.rna.1

</pre>
</td></tr></table><p>





<H2>
    Data files
</H2>


The Data Resource Catalogue is included in EMBOSS as local
database <b>drcat</b>.


The EDAM Ontology is included in EMBOSS as local
database <b>edam</b>.




<H2>
    Notes
</H2>

<!-- 
        Restrictions.
        Interesting behaviour.
        Useful things you can do with this program.
   -->

None.







<H2>
    References
</H2>

<!-- 
        Bibliography for methods used.
<ol>

<li>

</ol>

   -->

None.








<H2>
    Warnings
</H2>

<!-- 
        Potentially stupid things the program will let you do.
   -->

None.







<H2>
    Diagnostic Error Messages
</H2>

<!-- 
        Error messages specific to this program, eg:
        "FATAL xxx" - means you have not set up the xxx data using program <b>prog</b>.<p>
   -->

None.







<H2>
    Exit status
</H2>

<!-- 
        Description of the exit status for various error conditions
   -->

It always exits with status 0.








<H2>
    Known bugs
</H2>


<!-- 
        Bugs noted but not yet fixed.
   -->

None.








<!--
<H2>
    See also
</H2>
-->
<h2><a name="See also">See also</a></h2>
<table border cellpadding=4 bgcolor="#FFFFF0">
<tr><th>Program name</th>
<th>Description</th></tr>
<tr>
<td><a href="drfinddata.html">drfinddata</a></td>
<td>Find public databases by data type</td>
</tr>

<tr>
<td><a href="drfindformat.html">drfindformat</a></td>
<td>Find public databases by format</td>
</tr>

<tr>
<td><a href="drfindid.html">drfindid</a></td>
<td>Find public databases by identifier</td>
</tr>

<tr>
<td><a href="drget.html">drget</a></td>
<td>Get data resource entries</td>
</tr>

<tr>
<td><a href="drtext.html">drtext</a></td>
<td>Get data resource entries complete text</td>
</tr>

<tr>
<td><a href="edamdef.html">edamdef</a></td>
<td>Find EDAM ontology terms by definition</td>
</tr>

<tr>
<td><a href="edamhasinput.html">edamhasinput</a></td>
<td>Find EDAM ontology terms by has_input relation</td>
</tr>

<tr>
<td><a href="edamhasoutput.html">edamhasoutput</a></td>
<td>Find EDAM ontology terms by has_output relation</td>
</tr>

<tr>
<td><a href="edamisformat.html">edamisformat</a></td>
<td>Find EDAM ontology terms by is_format_of relation</td>
</tr>

<tr>
<td><a href="edamisid.html">edamisid</a></td>
<td>Find EDAM ontology terms by is_identifier_of relation</td>
</tr>

<tr>
<td><a href="edamname.html">edamname</a></td>
<td>Find EDAM ontology terms by name</td>
</tr>

<tr>
<td><a href="wossdata.html">wossdata</a></td>
<td>Find programs by EDAM data</td>
</tr>

<tr>
<td><a href="wossinput.html">wossinput</a></td>
<td>Find programs by EDAM input data</td>
</tr>

<tr>
<td><a href="wossoperation.html">wossoperation</a></td>
<td>Find programs by EDAM operation</td>
</tr>

<tr>
<td><a href="wossoutput.html">wossoutput</a></td>
<td>Find programs by EDAM output data</td>
</tr>

<tr>
<td><a href="wossparam.html">wossparam</a></td>
<td>Find programs by EDAM parameter</td>
</tr>

<tr>
<td><a href="wosstopic.html">wosstopic</a></td>
<td>Find programs by EDAM topic</td>
</tr>

</table>
<!-- 
        Add any comments about other associated programs (to prepare
        data files?) that seealso doesn't find. 
   -->










<H2>
    Author(s)
</H2>

Peter Rice
<br>
European Bioinformatics Institute, Wellcome Trust Genome Campus, Hinxton, Cambridge CB10 1SD, UK

<p>
Please report all bugs to the EMBOSS bug team (emboss-bug&nbsp;&copy;&nbsp;emboss.open-bio.org) not to the original author.



<H2>
    History
</H2>
<!--
        Date written and what changes have been made go in this file.
   -->




<H2>
    Target users
</H2>
<!--
        For general users, use this text
   -->
This program is intended to be used by everyone and everything, from naive users to embedded scripts.

<H2>
    Comments
</H2>
<!--
        User/developer/other comments go in this file.
   -->
None


</BODY>
</HTML>

