<HTML>
<HEAD>
  <TITLE>
  EMBOSS: edamhasinput
  </TITLE>
</HEAD>
<BODY BGCOLOR="#FFFFFF" text="#000000">

<table align=center border=0 cellspacing=0 cellpadding=0>
<tr><td valign=top>
<A HREF="/" ONMOUSEOVER="self.status='Go to the EMBOSS home page';return true"><img border=0 src="/images/emboss_icon.jpg" alt="" width=150 height=48></a>
</td>
<td align=left valign=middle>
<b><font size="+6">
edamhasinput
</font></b>
</td></tr>
</table>
<br>&nbsp;
<p>


<H2>
Wiki
</H2>

The master copies of EMBOSS documentation are available
at <a href="http://emboss.open-bio.org/wiki/Appdocs">
http://emboss.open-bio.org/wiki/Appdocs</a>
on the EMBOSS Wiki.

<p>
Please help by correcting and extending the Wiki pages.

<H2>
    Function
</H2>
Find EDAM ontology terms by has_input relation
<!--
DON'T WRITE ANYTHING HERE.
IT IS DONE FOR YOU.
-->




<H2>
    Description
</H2>

<b>edamhasinput</b> searches the has_input relations of EDAM terms and
returns matching terms.  The input is read from the installed EDAM
databas. The ontology term output can be written to screen, to file,
or passed to another program. A wide range of standard ontology term
formats may be specified for input and output.

<p>
Optionally the search can be restricted to specified EDAM namespaces.



<H2>
    Usage
</H2>

<!--  
	Example usage, as run from the command-line.
        Many examples illustrating different behaviours is good.
 -->

Here is a sample session with <b>edamhasinput</b>
<p>

<p>
<table width="90%"><tr><td bgcolor="#CCFFFF"><pre>

% <b>edamhasinput sequence </b>
Find EDAM ontology terms by has_input relation
Obo output file [edamhasinput.obo]: <b></b>

</pre></td></tr></table><p>
<p>
<a href="#output.1">Go to the output files for this example</a><p><p>



<H2>
    Command line arguments
</H2>

<table CELLSPACING=0 CELLPADDING=3 BGCOLOR="#f5f5ff" ><tr><td>
<pre>
Find EDAM ontology terms by has_input relation
Version: EMBOSS:6.5.0.0

   Standard (Mandatory) qualifiers:
  [-identifier]        string     Identifier(s) to search for in ontology (Any
                                  string)
  [-outfile]           outobo     [*.edamhasinput] Output ontology term file
                                  name

   Additional (Optional) qualifiers:
   -namespace          menu       [*] By default all terms are returned.
                                  Searches can be limited to one or a few
                                  namespaces. (Values: data (Data entity);
                                  entity (Biological entity); format (Data
                                  format); identifier (Identifier); operation
                                  (Bioinformatics operation); resource (Data
                                  resource); topic (Field of bioinformatics
                                  study))

   Advanced (Unprompted) qualifiers:
   -sensitive          boolean    [N] By default, the query keywords are
                                  matched against the EDAM term names (and
                                  synonyms) only. This option also matches the
                                  keywords against the EDAM term definitions
                                  and will therefore (typically) report more
                                  matches.
   -subclasses         boolean    [N] Extend the query matches to include all
                                  terms which are specialisations (EDAM
                                  sub-classes) of the matched type.
   -obsolete           boolean    [N] The default behaviour is to not use or
                                  return obsolete terms. This option if set
                                  will include all terms.

   Associated qualifiers:

   "-outfile" associated qualifiers
   -odirectory2        string     Output directory
   -oformat2           string     Ontology term output format

   General qualifiers:
   -auto               boolean    Turn off prompts
   -stdout             boolean    Write first file to standard output
   -filter             boolean    Read first file from standard input, write
                                  first file to standard output
   -options            boolean    Prompt for standard and additional values
   -debug              boolean    Write debug output to program.dbg
   -verbose            boolean    Report some/full command line options
   -help               boolean    Report command line options and exit. More
                                  information on associated and general
                                  qualifiers can be found with -help -verbose
   -warning            boolean    Report warnings
   -error              boolean    Report errors
   -fatal              boolean    Report fatal errors
   -die                boolean    Report dying program messages
   -version            boolean    Report version number and exit

</pre>
</td></tr></table>
<P>
<table border cellspacing=0 cellpadding=3 bgcolor="#ccccff">
<tr bgcolor="#FFFFCC">
<th align="left">Qualifier</th>
<th align="left">Type</th>
<th align="left">Description</th>
<th align="left">Allowed values</th>
<th align="left">Default</th>
</tr>

<tr bgcolor="#FFFFCC">
<th align="left" colspan=5>Standard (Mandatory) qualifiers</th>
</tr>

<tr bgcolor="#FFFFCC">
<td>[-identifier]<br>(Parameter 1)</td>
<td>string</td>
<td>Identifier(s) to search for in ontology</td>
<td>Any string</td>
<td>&nbsp;</td>
</tr>

<tr bgcolor="#FFFFCC">
<td>[-outfile]<br>(Parameter 2)</td>
<td>outobo</td>
<td>Output ontology term file name</td>
<td>OBO ontology term(s)</td>
<td><i>&lt;*&gt;</i>.edamhasinput</td>
</tr>

<tr bgcolor="#FFFFCC">
<th align="left" colspan=5>Additional (Optional) qualifiers</th>
</tr>

<tr bgcolor="#FFFFCC">
<td>-namespace</td>
<td>list</td>
<td>By default all terms are returned. Searches can be limited to one or a few namespaces.</td>
<td><table><tr><td>data</td> <td><i>(Data entity)</i></td></tr><tr><td>entity</td> <td><i>(Biological entity)</i></td></tr><tr><td>format</td> <td><i>(Data format)</i></td></tr><tr><td>identifier</td> <td><i>(Identifier)</i></td></tr><tr><td>operation</td> <td><i>(Bioinformatics operation)</i></td></tr><tr><td>resource</td> <td><i>(Data resource)</i></td></tr><tr><td>topic</td> <td><i>(Field of bioinformatics study)</i></td></tr></table></td>
<td>*</td>
</tr>

<tr bgcolor="#FFFFCC">
<th align="left" colspan=5>Advanced (Unprompted) qualifiers</th>
</tr>

<tr bgcolor="#FFFFCC">
<td>-sensitive</td>
<td>boolean</td>
<td>By default, the query keywords are matched against the EDAM term names (and synonyms) only. This option also matches the keywords against the EDAM term definitions and will therefore (typically) report more matches.</td>
<td>Boolean value Yes/No</td>
<td>No</td>
</tr>

<tr bgcolor="#FFFFCC">
<td>-subclasses</td>
<td>boolean</td>
<td>Extend the query matches to include all terms which are specialisations (EDAM sub-classes) of the matched type.</td>
<td>Boolean value Yes/No</td>
<td>No</td>
</tr>

<tr bgcolor="#FFFFCC">
<td>-obsolete</td>
<td>boolean</td>
<td>The default behaviour is to not use or return obsolete terms. This option if set will include all terms.</td>
<td>Boolean value Yes/No</td>
<td>No</td>
</tr>

<tr bgcolor="#FFFFCC">
<th align="left" colspan=5>Associated qualifiers</th>
</tr>

<tr bgcolor="#FFFFCC">
<td align="left" colspan=5>"-outfile" associated outobo qualifiers
</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -odirectory2<br>-odirectory_outfile</td>
<td>string</td>
<td>Output directory</td>
<td>Any string</td>
<td>&nbsp;</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -oformat2<br>-oformat_outfile</td>
<td>string</td>
<td>Ontology term output format</td>
<td>Any string</td>
<td>&nbsp;</td>
</tr>

<tr bgcolor="#FFFFCC">
<th align="left" colspan=5>General qualifiers</th>
</tr>

<tr bgcolor="#FFFFCC">
<td> -auto</td>
<td>boolean</td>
<td>Turn off prompts</td>
<td>Boolean value Yes/No</td>
<td>N</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -stdout</td>
<td>boolean</td>
<td>Write first file to standard output</td>
<td>Boolean value Yes/No</td>
<td>N</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -filter</td>
<td>boolean</td>
<td>Read first file from standard input, write first file to standard output</td>
<td>Boolean value Yes/No</td>
<td>N</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -options</td>
<td>boolean</td>
<td>Prompt for standard and additional values</td>
<td>Boolean value Yes/No</td>
<td>N</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -debug</td>
<td>boolean</td>
<td>Write debug output to program.dbg</td>
<td>Boolean value Yes/No</td>
<td>N</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -verbose</td>
<td>boolean</td>
<td>Report some/full command line options</td>
<td>Boolean value Yes/No</td>
<td>Y</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -help</td>
<td>boolean</td>
<td>Report command line options and exit. More information on associated and general qualifiers can be found with -help -verbose</td>
<td>Boolean value Yes/No</td>
<td>N</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -warning</td>
<td>boolean</td>
<td>Report warnings</td>
<td>Boolean value Yes/No</td>
<td>Y</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -error</td>
<td>boolean</td>
<td>Report errors</td>
<td>Boolean value Yes/No</td>
<td>Y</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -fatal</td>
<td>boolean</td>
<td>Report fatal errors</td>
<td>Boolean value Yes/No</td>
<td>Y</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -die</td>
<td>boolean</td>
<td>Report dying program messages</td>
<td>Boolean value Yes/No</td>
<td>Y</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -version</td>
<td>boolean</td>
<td>Report version number and exit</td>
<td>Boolean value Yes/No</td>
<td>N</td>
</tr>

</table>

<!--
DON'T WRITE ANYTHING HERE.
IT IS DONE FOR YOU.
-->








<H2>
    Input file format
</H2>

<!-- 
        This includes example input file formats.
        This should be a detailed description and example - assume
        someone will want to parse this file and will want to know what
        happens in unusual cases - null input, etc. 
   -->

<b>edamhasinput</b> queries the EDAM ontology.

<p>






<H2>
    Output file format
</H2>

<!-- 
        This includes example output file formats.
        This should be a detailed description and example - assume
        someone will want to parse this file and will want to know what
        happens in unusual cases - null output, errors etc. 

        If you wish to include the standard description of the avalable
	report formats, use:
        #include file="inc/reportformats.ihtml"
   -->

<p>

The output is a standard EMBOSS ontology term file. 

<p>

The results can be output in one of several styles by using the
command-line qualifier <tt>-oformat xxx</tt>, where 'xxx' is replaced by
the name of the required format.  The available format names are: obo,
brief, list, html, xml, json, excel.
<p>

See:
<A href="http://emboss.sf.net/docs/themes/OntologyFormats.html">
http://emboss.sf.net/docs/themes/OntologyFormats.html</A>
for further information on ontology formats.

<p>

<p>


<a name="output.1"></a>
<h3>Output files for usage example </h3>
<p><h3>File: edamhasinput.obo</h3>
<table width="90%"><tr><td bgcolor="#CCFFCC">
<pre>
[Term]
id: EDAM_operation:2403
name: Sequence analysis
namespace: operation
def: Analyse one or more known molecular sequences.
subset: operations
synonym: "Sequence analysis (general)" EXACT [http://edamontology.org]
is_a: EDAM_operation:2446 ! Sequence processing
relationship: has_input EDAM_data:2044 {min_cardinality=1} ! Sequence
relationship: has_output EDAM_data:2955 {min_cardinality=1} ! Sequence report
relationship: has_topic EDAM_topic:0080 ! Sequence analysis

[Term]
id: EDAM_operation:0292
name: Sequence alignment construction
namespace: operation
def: Align (identify equivalent sites within) molecular sequences.
subset: operations
synonym: "Sequence alignment" EXACT [http://edamontology.org]
synonym: "Sequence alignment computation" EXACT [http://edamontology.org]
synonym: "Sequence alignment generation" EXACT [http://edamontology.org]
is_a: EDAM_operation:2463 ! Sequence alignment processing !(?)
is_a: EDAM_operation:2451 ! Sequence comparison
is_a: EDAM_operation:2928 ! Alignment construction
relationship: has_input EDAM_data:2044 {minCardinality=2, note="Applies to both pure/raw sequences and sequence records. Indirectly also to sequence sets or IDs (?)"} ! Sequence
relationship: has_output EDAM_data:0863 {min_cardinality=1} ! Sequence alignment
relationship: has_topic EDAM_topic:0182 ! Sequence alignment

[Term]
id: EDAM_operation:1780
name: Sequence submission
namespace: operation
def: Submit a molecular sequence to a database.
subset: operations
is_a: EDAM_operation:2446 ! Sequence processing
relationship: has_input EDAM_data:2044 {min_cardinality=1} ! Sequence
relationship: has_topic EDAM_topic:3077 ! Data acquisition and deposition

[Term]
id: EDAM_operation:2446
name: Sequence processing
namespace: operation
def: Process (read and / or write) one or more molecular sequences and associated annotation.
subset: operations
synonym: "Sequence processing (general)" EXACT [http://edamontology.org]
is_a: EDAM_operation:2420 ! Analysis and processing
relationship: has_input EDAM_data:2044 {min_cardinality=0} ! Sequence
relationship: has_output EDAM_data:2044 {min_cardinality=0} ! Sequence
relationship: has_topic EDAM_topic:0080 ! Sequence analysis



<font color=red>  [Part of this file has been deleted for brevity]</font>

synonym: "PCR primer prediction" EXACT [http://edamontology.org]
is_a: EDAM_operation:2419 ! Primer and probe design
relationship: has_input EDAM_data:2977 ! Nucleic acid sequence
relationship: has_output EDAM_data:1240 {min_cardinality=0} ! PCR primers
relationship: has_topic EDAM_topic:3032 ! Primer or probe design

[Term]
id: EDAM_operation:2506
name: Sequence alignment analysis (protein)
namespace: operation
def: Analyse a protein sequence alignment, typically to detect features or make predictions.
subset: operations
is_a: EDAM_operation:0258 ! Sequence alignment analysis
is_a: EDAM_operation:2502 ! Protein data processing
is_a: EDAM_operation:3023 ! Prediction, detection and recognition (protein)
relationship: has_input EDAM_data:1384 {min_cardinality=1} ! Sequence alignment (protein)
relationship: has_topic EDAM_topic:0741 ! Protein sequence alignment

[Term]
id: EDAM_operation:0361
name: Sequence annotation
namespace: operation
def: Annotate a molecular sequence record with terms from a controlled vocabulary.
subset: operations
is_a: EDAM_operation:0226 ! Annotation
relationship: has_input EDAM_data:0849 {min_cardinality=1} ! Sequence record
relationship: has_output EDAM_data:0849 {min_cardinality=1} ! Sequence record

[Term]
id: EDAM_operation:0368
name: Sequence masking
namespace: operation
def: Mask characters in a molecular sequence (replacing those characters with a mask character).
comment: For example, SNPs or repeats in a DNA sequence might be masked.
subset: operations
is_a: EDAM_operation:0231 ! Sequence editing
relationship: has_input EDAM_data:0851 {min_cardinality=1} ! Sequence mask character

[Term]
id: EDAM_operation:3180
name: Sequence assembly evaluation
namespace: operation
def: Evaluate a DNA sequence assembly, typically for purposes of quality control.
subset: operations
is_a: EDAM_operation:2448 ! Sequence processing (nucleic acid)
is_a: EDAM_operation:2478 ! Nucleic acid sequence analysis
is_a: EDAM_operation:2425 ! Optimisation and refinement
relationship: has_input EDAM_data:0925 ! Sequence assembly
relationship: has_output EDAM_data:3181 ! Sequence assembly report
relationship: has_topic EDAM_topic:0196 ! Sequence assembly

</pre>
</td></tr></table><p>





<H2>
    Data files
</H2>

The EDAM Ontology is included in EMBOSS as local
database <b>edam</b>.


<H2>
    Notes
</H2>

<!-- 
        Restrictions.
        Interesting behaviour.
        Useful things you can do with this program.
   -->

None.







<H2>
    References
</H2>

<!-- 
        Bibliography for methods used.
<ol>

<li>

</ol>

   -->

None.








<H2>
    Warnings
</H2>

<!-- 
        Potentially stupid things the program will let you do.
   -->

None.







<H2>
    Diagnostic Error Messages
</H2>

<!-- 
        Error messages specific to this program, eg:
        "FATAL xxx" - means you have not set up the xxx data using program <b>prog</b>.<p>
   -->

None.







<H2>
    Exit status
</H2>

<!-- 
        Description of the exit status for various error conditions
   -->

It always exits with status 0.








<H2>
    Known bugs
</H2>


<!-- 
        Bugs noted but not yet fixed.
   -->

None.








<!--
<H2>
    See also
</H2>
-->
<h2><a name="See also">See also</a></h2>
<table border cellpadding=4 bgcolor="#FFFFF0">
<tr><th>Program name</th>
<th>Description</th></tr>
<tr>
<td><a href="drfinddata.html">drfinddata</a></td>
<td>Find public databases by data type</td>
</tr>

<tr>
<td><a href="drfindformat.html">drfindformat</a></td>
<td>Find public databases by format</td>
</tr>

<tr>
<td><a href="drfindid.html">drfindid</a></td>
<td>Find public databases by identifier</td>
</tr>

<tr>
<td><a href="drfindresource.html">drfindresource</a></td>
<td>Find public databases by resource</td>
</tr>

<tr>
<td><a href="edamdef.html">edamdef</a></td>
<td>Find EDAM ontology terms by definition</td>
</tr>

<tr>
<td><a href="edamhasoutput.html">edamhasoutput</a></td>
<td>Find EDAM ontology terms by has_output relation</td>
</tr>

<tr>
<td><a href="edamisformat.html">edamisformat</a></td>
<td>Find EDAM ontology terms by is_format_of relation</td>
</tr>

<tr>
<td><a href="edamisid.html">edamisid</a></td>
<td>Find EDAM ontology terms by is_identifier_of relation</td>
</tr>

<tr>
<td><a href="edamname.html">edamname</a></td>
<td>Find EDAM ontology terms by name</td>
</tr>

<tr>
<td><a href="godef.html">godef</a></td>
<td>Find GO ontology terms by definition</td>
</tr>

<tr>
<td><a href="goname.html">goname</a></td>
<td>Find GO ontology terms by name</td>
</tr>

<tr>
<td><a href="ontoget.html">ontoget</a></td>
<td>Get ontology term(s)</td>
</tr>

<tr>
<td><a href="ontogetcommon.html">ontogetcommon</a></td>
<td>Get common ancestor for terms</td>
</tr>

<tr>
<td><a href="ontogetdown.html">ontogetdown</a></td>
<td>Get ontology term(s) by parent id</td>
</tr>

<tr>
<td><a href="ontogetobsolete.html">ontogetobsolete</a></td>
<td>Get ontology ontology terms</td>
</tr>

<tr>
<td><a href="ontogetroot.html">ontogetroot</a></td>
<td>Get ontology root terms by child identifier</td>
</tr>

<tr>
<td><a href="ontogetsibs.html">ontogetsibs</a></td>
<td>Get ontology term(s) by id with common parent</td>
</tr>

<tr>
<td><a href="ontogetup.html">ontogetup</a></td>
<td>Get ontology term(s) by id of child</td>
</tr>

<tr>
<td><a href="ontoisobsolete.html">ontoisobsolete</a></td>
<td>Report whether an ontology term id is obsolete</td>
</tr>

<tr>
<td><a href="ontotext.html">ontotext</a></td>
<td>Get ontology term(s) original full text</td>
</tr>

<tr>
<td><a href="wossdata.html">wossdata</a></td>
<td>Find programs by EDAM data</td>
</tr>

<tr>
<td><a href="wossinput.html">wossinput</a></td>
<td>Find programs by EDAM input data</td>
</tr>

<tr>
<td><a href="wossoperation.html">wossoperation</a></td>
<td>Find programs by EDAM operation</td>
</tr>

<tr>
<td><a href="wossoutput.html">wossoutput</a></td>
<td>Find programs by EDAM output data</td>
</tr>

<tr>
<td><a href="wossparam.html">wossparam</a></td>
<td>Find programs by EDAM parameter</td>
</tr>

<tr>
<td><a href="wosstopic.html">wosstopic</a></td>
<td>Find programs by EDAM topic</td>
</tr>

</table>
<!-- 
        Add any comments about other associated programs (to prepare
        data files?) that seealso doesn't find. 
   -->










<H2>
    Author(s)
</H2>

Peter Rice
<br>
European Bioinformatics Institute, Wellcome Trust Genome Campus, Hinxton, Cambridge CB10 1SD, UK

<p>
Please report all bugs to the EMBOSS bug team (emboss-bug&nbsp;&copy;&nbsp;emboss.open-bio.org) not to the original author.
<H2>
    History
</H2>
<!--
        Date written and what changes have been made go in this file.
   -->




<H2>
    Target users
</H2>
<!--
        For general users, use this text
   -->
This program is intended to be used by everyone and everything, from naive users to embedded scripts.

<H2>
    Comments
</H2>
<!--
        User/developer/other comments go in this file.
   -->
None


</BODY>
</HTML>

