<HTML>
<HEAD>
  <TITLE>
  EMBOSS: featcopy
  </TITLE>
</HEAD>
<BODY BGCOLOR="#FFFFFF" text="#000000">

<table align=center border=0 cellspacing=0 cellpadding=0>
<tr><td valign=top>
<A HREF="/" ONMOUSEOVER="self.status='Go to the EMBOSS home page';return true"><img border=0 src="emboss_icon.jpg" alt="" width=150 height=48></a>
</td>
<td align=left valign=middle>
<b><font size="+6">
featcopy
</font></b>
</td></tr>
</table>
<br>&nbsp;
<p>


<H2>
    Function
</H2>
Reads and writes a feature table
<!--
DON'T WRITE ANYTHING HERE.
IT IS DONE FOR YOU.
-->




<H2>
    Description
</H2>

<b>featcopy</b> reads a feature table and reformats it in any of the
supported output feature formats.













<H2>
    Usage
</H2>

<!--  
	Example usage, as run from the command-line.
        Many examples illustrating different behaviours is good.
 -->

<b>Here is a sample session with featcopy</b>
<p>

<p>
<table width="90%"><tr><td bgcolor="#CCFFFF"><pre>

% <b>featcopy </b>
Reads and writes a feature table
Input features: <b>paamir.gff3</b>
Features output [paamir.gff]: <b></b>

</pre></td></tr></table><p>
<p>
<a href="#input.1">Go to the input files for this example</a><br><a href="#output.1">Go to the output files for this example</a><p><p>



<H2>
    Command line arguments
</H2>

<table CELLSPACING=0 CELLPADDING=3 BGCOLOR="#f5f5ff" ><tr><td>
<pre>
   Standard (Mandatory) qualifiers:
  [-features]          features   (no help text) features value
  [-outfeat]           featout    [unknown.gff] Output features UFO

   Additional (Optional) qualifiers: (none)
   Advanced (Unprompted) qualifiers: (none)
   Associated qualifiers:

   "-features" associated qualifiers
   -fformat1           string     Features format
   -fopenfile1         string     Features file name
   -fask1              boolean    Prompt for begin/end/reverse
   -fbegin1            integer    Start of the features to be used
   -fend1              integer    End of the features to be used
   -freverse1          boolean    Reverse (if DNA)

   "-outfeat" associated qualifiers
   -offormat2          string     Output feature format
   -ofopenfile2        string     Features file name
   -ofextension2       string     File name extension
   -ofdirectory2       string     Output directory
   -ofname2            string     Base file name
   -ofsingle2          boolean    Separate file for each entry

   General qualifiers:
   -auto               boolean    Turn off prompts
   -stdout             boolean    Write first file to standard output
   -filter             boolean    Read first file from standard input, write
                                  first file to standard output
   -options            boolean    Prompt for standard and additional values
   -debug              boolean    Write debug output to program.dbg
   -verbose            boolean    Report some/full command line options
   -help               boolean    Report command line options. More
                                  information on associated and general
                                  qualifiers can be found with -help -verbose
   -warning            boolean    Report warnings
   -error              boolean    Report errors
   -fatal              boolean    Report fatal errors
   -die                boolean    Report dying program messages

</pre>
</td></tr></table>
<P>
<table border cellspacing=0 cellpadding=3 bgcolor="#ccccff">
<tr bgcolor="#FFFFCC">
<th align="left" colspan=2>Standard (Mandatory) qualifiers</th>
<th align="left">Allowed values</th>
<th align="left">Default</th>
</tr>

<tr>
<td>[-features]<br>(Parameter 1)</td>
<td>(no help text) features value</td>
<td>Readable feature table</td>
<td><b>Required</b></td>
</tr>

<tr>
<td>[-outfeat]<br>(Parameter 2)</td>
<td>Output features UFO</td>
<td>Writeable feature table</td>
<td><i>unknown.gff</i></td>
</tr>

<tr bgcolor="#FFFFCC">
<th align="left" colspan=2>Additional (Optional) qualifiers</th>
<th align="left">Allowed values</th>
<th align="left">Default</th>
</tr>

<tr>
<td colspan=4>(none)</td>
</tr>

<tr bgcolor="#FFFFCC">
<th align="left" colspan=2>Advanced (Unprompted) qualifiers</th>
<th align="left">Allowed values</th>
<th align="left">Default</th>
</tr>

<tr>
<td colspan=4>(none)</td>
</tr>

</table>

<!--
DON'T WRITE ANYTHING HERE.
IT IS DONE FOR YOU.
-->








<H2>
    Input file format
</H2>

<!-- 
        This includes example input file formats.
        This should be a detailed description and example - assume
        someone will want to parse this file and will want to know what
        happens in unusual cases - null input, etc. 
   -->

<b>featcopy</b> reads any normal features UFOs.

<p>


<a name="input.1"></a>
<h3>Input files for usage example </h3>
<p><h3>File: paamir.gff3</h3>
<table width="90%"><tr><td bgcolor="#FFCCFF">
<pre>
##gff-version 3
##sequence-region PAAMIR 1 2167
#!Date 2008-07-11
#!Type DNA
#!Source-version EMBOSS 6.0.0
PAAMIR	EMBL	databank_entry	1	2167	0.000	+	.	ID="PAAMIR.1";db_xref="taxon:287";organism="Pseudomonas aeruginosa";strain="PAC";isolate="PAC 1";map="38 min"
PAAMIR	EMBL	CDS	1289	1879	0.000	+	0	ID="PAAMIR.2";db_xref="SWISS-PROT:P10932";note="aliphatic amidase regulator, positive regulator of amiE";transl_table=11;gene="amiR";protein_id="CAA32023.1";translation="MSANSLLGSLRELQVLVLNPPGEVSDALVLQLIRIGCSVRQCWPPPEAFDVPVDVVFTSIFQNGHHDEIAALLAAGTPRTTLVALVEYESPAVLSQIIELECHGVITQPLDAHRVLPVLVSARRISEEMAKLKQKTEQLQDRIAGQARINQAKVLLMQRHGWDEREAHQHLSREAMKRREPILKIAQELLGNEPSA"
PAAMIR	EMBL	CDS	135	1292	0.000	+	0	ID="PAAMIR.3";db_xref="SWISS-PROT:P27017";note="negative regulator of amiR";transl_table=11;gene="amiC";protein_id="CAA32024.1";translation="MGSHQERPLIGLLFSETGVTADIERSHAYGALLAVEQLNREGGVGGRPIETLSQDPGGDPDRYRLCAEDFIRNRGVRFLVGCYMSHTRKAVMPVVERADALLCYPTPYEGFEYSPNIVYGGPAPNQNSAPLAAYLIRHYGERVVFIGSDYIYPRESNHVMRHLYRQHGGTVLEEIYIPLYPSDDDLQRAVERIYQARADVVFSTVVGTGTAELYRAIARRYGDGRRPPIASLTTSEAEVAKMESDVAEGQVVVAPYFSSIDTPASRAFVQACHGFFPENATITAWAEAAYWQTLLLGRAAQAAGNWRVEDVQRHLYDIDIDAPQGPVRVERQNNHSRLSSRIAEIDARGVFQVRWQSPEPIRPDPYVVVHNLDDWSASMGGGPLP"
PAAMIR	EMBL	promoter	8	24	0.000	+	.	ID="PAAMIR.4";note="proposed rpoN-dependent promoter"
PAAMIR	EMBL	promoter	65	81	0.000	+	.	ID="PAAMIR.5";note="proposed rpoN-dependent promoter"
PAAMIR	EMBL	ribosome_entry_site	121	126	0.000	+	.	ID="PAAMIR.6";note="proposed Shine-Dalgarno sequence"
PAAMIR	EMBL	sequence_variant	912	1167	0.000	+	.	ID="PAAMIR.7";note="ClaI fragment deleted in pSW36, constitutive phenotype";replace="";gene="amiC"
PAAMIR	EMBL	located_sequence_feature	1	1	0.000	+	.	ID="PAAMIR.8";featflags="0x40";note="last base of an XhoI site"
PAAMIR	EMBL	located_sequence_feature	648	653	0.000	+	.	ID="PAAMIR.9";note="end of 658bp XhoI fragment, deletion in pSW3 causes constitutive expression of amiE"
PAAMIR	EMBL	misc_difference	1281	1281	0.000	+	.	ID="PAAMIR.10";featflags="0x40";replace="g";citation=[3]
</pre>
</td></tr></table><p>





<H2>
    Output file format
</H2>

<!-- 
        This includes example output file formats.
        This should be a detailed description and example - assume
        someone will want to parse this file and will want to know what
        happens in unusual cases - null output, errors etc. 

        If you wish to include the standard description of the avalable
	report formats, use:
        #include file="inc/reportformats.ihtml"
   -->

<b>featcopy</b> 
outputs a feature file.


<p>


<a name="output.1"></a>
<h3>Output files for usage example </h3>
<p><h3>File: paamir.gff</h3>
<table width="90%"><tr><td bgcolor="#CCFFCC">
<pre>
##gff-version 3
##sequence-region PAAMIR 1 2167
#!Date 2008-07-15
#!Type DNA
#!Source-version EMBOSS 6.0.0
PAAMIR	EMBL	databank_entry	1	2167	0.000	+	.	ID="PAAMIR.1";db_xref="taxon:287";organism="Pseudomonas aeruginosa";strain="PAC";isolate="PAC 1";map="38 min"
PAAMIR	EMBL	CDS	1289	1879	0.000	+	0	ID="PAAMIR.2";db_xref="SWISS-PROT:P10932";note="aliphatic amidase regulator, positive regulator of amiE";transl_table=11;gene="amiR";protein_id="CAA32023.1";translation="MSANSLLGSLRELQVLVLNPPGEVSDALVLQLIRIGCSVRQCWPPPEAFDVPVDVVFTSIFQNGHHDEIAALLAAGTPRTTLVALVEYESPAVLSQIIELECHGVITQPLDAHRVLPVLVSARRISEEMAKLKQKTEQLQDRIAGQARINQAKVLLMQRHGWDEREAHQHLSREAMKRREPILKIAQELLGNEPSA"
PAAMIR	EMBL	CDS	135	1292	0.000	+	0	ID="PAAMIR.3";db_xref="SWISS-PROT:P27017";note="negative regulator of amiR";transl_table=11;gene="amiC";protein_id="CAA32024.1";translation="MGSHQERPLIGLLFSETGVTADIERSHAYGALLAVEQLNREGGVGGRPIETLSQDPGGDPDRYRLCAEDFIRNRGVRFLVGCYMSHTRKAVMPVVERADALLCYPTPYEGFEYSPNIVYGGPAPNQNSAPLAAYLIRHYGERVVFIGSDYIYPRESNHVMRHLYRQHGGTVLEEIYIPLYPSDDDLQRAVERIYQARADVVFSTVVGTGTAELYRAIARRYGDGRRPPIASLTTSEAEVAKMESDVAEGQVVVAPYFSSIDTPASRAFVQACHGFFPENATITAWAEAAYWQTLLLGRAAQAAGNWRVEDVQRHLYDIDIDAPQGPVRVERQNNHSRLSSRIAEIDARGVFQVRWQSPEPIRPDPYVVVHNLDDWSASMGGGPLP"
PAAMIR	EMBL	promoter	8	24	0.000	+	.	ID="PAAMIR.4";note="proposed rpoN-dependent promoter"
PAAMIR	EMBL	promoter	65	81	0.000	+	.	ID="PAAMIR.5";note="proposed rpoN-dependent promoter"
PAAMIR	EMBL	ribosome_entry_site	121	126	0.000	+	.	ID="PAAMIR.6";note="proposed Shine-Dalgarno sequence"
PAAMIR	EMBL	sequence_variant	912	1167	0.000	+	.	ID="PAAMIR.7";note="ClaI fragment deleted in pSW36, constitutive phenotype";replace="";gene="amiC"
PAAMIR	EMBL	located_sequence_feature	1	1	0.000	+	.	ID="PAAMIR.8";featflags="0x40";note="last base of an XhoI site"
PAAMIR	EMBL	located_sequence_feature	648	653	0.000	+	.	ID="PAAMIR.9";note="end of 658bp XhoI fragment, deletion in pSW3 causes constitutive expression of amiE"
PAAMIR	EMBL	sequence_difference	1281	1281	0.000	+	.	ID="PAAMIR.10";featflags="0x40";replace="g";citation=[3]
</pre>
</td></tr></table><p>





<H2>
    Data files
</H2>

None.



<H2>
    Notes
</H2>

<!-- 
        Restrictions.
        Interesting behaviour.
        Useful things you can do with this program.
   -->

None.







<H2>
    References
</H2>

<!-- 
        Bibliography for methods used.
<ol>

<li>

</ol>

   -->

None.








<H2>
    Warnings
</H2>

<!-- 
        Potentially stupid things the program will let you do.
   -->

None.







<H2>
    Diagnostic Error Messages
</H2>

<!-- 
        Error messages specific to this program, eg:
        "FATAL xxx" - means you have not set up the xxx data using program <b>prog</b>.<p>
   -->

None.







<H2>
    Exit status
</H2>

<!-- 
        Description of the exit status for various error conditions
   -->

It always exits with status 0.








<H2>
    Known bugs
</H2>


<!-- 
        Bugs noted but not yet fixed.
   -->

None.








<!--
<H2>
    See also
</H2>
-->
<h2><a name="See also">See also</a></h2>
<table border cellpadding=4 bgcolor="#FFFFF0">
<tr><th>Program name</th>
<th>Description</th></tr>
<tr>
<td><a href="aligncopy.html">aligncopy</a></td>
<td>Reads and writes alignments</td>
</tr>

<tr>
<td><a href="aligncopypair.html">aligncopypair</a></td>
<td>Reads and writes pairs from alignments</td>
</tr>

<tr>
<td><a href="biosed.html">biosed</a></td>
<td>Replace or delete sequence sections</td>
</tr>

<tr>
<td><a href="codcopy.html">codcopy</a></td>
<td>Copy and reformat a codon usage table</td>
</tr>

<tr>
<td><a href="cutseq.html">cutseq</a></td>
<td>Removes a section from a sequence</td>
</tr>

<tr>
<td><a href="degapseq.html">degapseq</a></td>
<td>Removes non-alphabetic (e.g. gap) characters from sequences</td>
</tr>

<tr>
<td><a href="descseq.html">descseq</a></td>
<td>Alter the name or description of a sequence</td>
</tr>

<tr>
<td><a href="entret.html">entret</a></td>
<td>Retrieves sequence entries from flatfile databases and files</td>
</tr>

<tr>
<td><a href="extractalign.html">extractalign</a></td>
<td>Extract regions from a sequence alignment</td>
</tr>

<tr>
<td><a href="extractfeat.html">extractfeat</a></td>
<td>Extract features from sequence(s)</td>
</tr>

<tr>
<td><a href="extractseq.html">extractseq</a></td>
<td>Extract regions from a sequence</td>
</tr>

<tr>
<td><a href="featreport.html">featreport</a></td>
<td>Reads and writes a feature table</td>
</tr>

<tr>
<td><a href="listor.html">listor</a></td>
<td>Write a list file of the logical OR of two sets of sequences</td>
</tr>

<tr>
<td><a href="makenucseq.html">makenucseq</a></td>
<td>Create random nucleotide sequences</td>
</tr>

<tr>
<td><a href="makeprotseq.html">makeprotseq</a></td>
<td>Create random protein sequences</td>
</tr>

<tr>
<td><a href="maskambignuc.html">maskambignuc</a></td>
<td>Masks all ambiguity characters in nucleotide sequences with N</td>
</tr>

<tr>
<td><a href="maskambigprot.html">maskambigprot</a></td>
<td>Masks all ambiguity characters in protein sequences with X</td>
</tr>

<tr>
<td><a href="maskfeat.html">maskfeat</a></td>
<td>Write a sequence with masked features</td>
</tr>

<tr>
<td><a href="maskseq.html">maskseq</a></td>
<td>Write a sequence with masked regions</td>
</tr>

<tr>
<td><a href="newseq.html">newseq</a></td>
<td>Create a sequence file from a typed-in sequence</td>
</tr>

<tr>
<td><a href="nohtml.html">nohtml</a></td>
<td>Remove mark-up (e.g. HTML tags) from an ASCII text file</td>
</tr>

<tr>
<td><a href="noreturn.html">noreturn</a></td>
<td>Remove carriage return from ASCII files</td>
</tr>

<tr>
<td><a href="nospace.html">nospace</a></td>
<td>Remove all whitespace from an ASCII text file</td>
</tr>

<tr>
<td><a href="notab.html">notab</a></td>
<td>Replace tabs with spaces in an ASCII text file</td>
</tr>

<tr>
<td><a href="notseq.html">notseq</a></td>
<td>Write to file a subset of an input stream of sequences</td>
</tr>

<tr>
<td><a href="nthseq.html">nthseq</a></td>
<td>Write to file a single sequence from an input stream of sequences</td>
</tr>

<tr>
<td><a href="pasteseq.html">pasteseq</a></td>
<td>Insert one sequence into another</td>
</tr>

<tr>
<td><a href="revseq.html">revseq</a></td>
<td>Reverse and complement a nucleotide sequence</td>
</tr>

<tr>
<td><a href="seqret.html">seqret</a></td>
<td>Reads and writes (returns) sequences</td>
</tr>

<tr>
<td><a href="seqretsplit.html">seqretsplit</a></td>
<td>Reads sequences and writes them to individual files</td>
</tr>

<tr>
<td><a href="sizeseq.html">sizeseq</a></td>
<td>Sort sequences by size</td>
</tr>

<tr>
<td><a href="skipredundant.html">skipredundant</a></td>
<td>Remove redundant sequences from an input set</td>
</tr>

<tr>
<td><a href="skipseq.html">skipseq</a></td>
<td>Reads and writes (returns) sequences, skipping first few</td>
</tr>

<tr>
<td><a href="splitter.html">splitter</a></td>
<td>Split sequence(s) into smaller sequences</td>
</tr>

<tr>
<td><a href="trimest.html">trimest</a></td>
<td>Remove poly-A tails from nucleotide sequences</td>
</tr>

<tr>
<td><a href="trimseq.html">trimseq</a></td>
<td>Remove unwanted characters from start and end of sequence(s)</td>
</tr>

<tr>
<td><a href="trimspace.html">trimspace</a></td>
<td>Remove extra whitespace from an ASCII text file</td>
</tr>

<tr>
<td><a href="union.html">union</a></td>
<td>Concatenate multiple sequences into a single sequence</td>
</tr>

<tr>
<td><a href="vectorstrip.html">vectorstrip</a></td>
<td>Removes vectors from the ends of nucleotide sequence(s)</td>
</tr>

<tr>
<td><a href="yank.html">yank</a></td>
<td>Add a sequence reference (a full USA) to a list file</td>
</tr>

</table>
<!-- 
        Add any comments about other associated programs (to prepare
        data files?) that seealso doesn't find. 
   -->










<H2>
    Author(s)
</H2>

Peter Rice (pmr&nbsp;&copy;&nbsp;ebi.ac.uk)
<br>
Informatics Division, European Bioinformatics Institute, Wellcome Trust Genome Campus, Hinxton, Cambridge CB10 1SD, UK



<H2>
    History
</H2>
<!--
        Date written and what changes have been made go in this file.
   -->




<H2>
    Target users
</H2>
<!--
        For general users, use this text
   -->
This program is intended to be used by everyone and everything, from naive users to embedded scripts.

<H2>
    Comments
</H2>
<!--
        User/developer/other comments go in this file.
   -->
None


</BODY>
</HTML>

