
<HTML>

<HEAD>
  <TITLE>
  EMBOSS: The Applications (programs)
  </TITLE>
</HEAD>
<BODY BGCOLOR="#FFFFFF" text="#000000">



<table align=center border=0 cellspacing=0 cellpadding=0>
<tr><td valign=top>
<A HREF="http://emboss.sourceforge.net/" ONMOUSEOVER="self.status='Go to the EMBOSS home page';return true">
<img border=0 src="emboss_icon.jpg" alt="" width=150 height=48></a>
</td>
<td align=left valign=middle>
<b><font size="+6"> 
The Applications (programs)
</font></b>
</td></tr>
</table>
<br>&nbsp;
<p>




The programs are listed in alphabetical order, Look at the individual
applications or go to the 
<a href="groups.html">GROUPS</a>
page to search by category. 
<p>

<a href="../../embassy/index.html">EMBASSY applications</a>
are described in separate documentation for each package.

<h3><A NAME="current">Applications</A> in the <a
href="ftp://emboss.open-bio.org/pub/EMBOSS/">current release</a></h3>

<table border cellpadding=4 bgcolor="#FFFFF0">

<tr>
<th>Program name</th>
<th>Description</th>
</tr>

<tr>
<td><a href="aaindexextract.html">aaindexextract</a></td>
<td>
Extract data from AAINDEX
</td>
</tr>
<tr>
<td><a href="abiview.html">abiview</a></td>
<td>
Reads ABI file and display the trace
</td>
</tr>
<tr>
<td><a href="acdc.html">acdc</a></td>
<td>
ACD compiler
</td>
</tr>
<tr>
<td><a href="acdpretty.html">acdpretty</a></td>
<td>
ACD pretty printing utility
</td>
</tr>
<tr>
<td><a href="acdtable.html">acdtable</a></td>
<td>
Creates an HTML table from an ACD file
</td>
</tr>
<tr>
<td><a href="acdtrace.html">acdtrace</a></td>
<td>
ACD compiler on-screen trace
</td>
</tr>
<tr>
<td><a href="acdvalid.html">acdvalid</a></td>
<td>
ACD file validation
</td>
</tr>
<tr>
<td><a href="antigenic.html">antigenic</a></td>
<td>
Finds antigenic sites in proteins
</td>
</tr>
<tr>
<td><a href="backtranambig.html">backtranambig</a></td>
<td>
Back translate a protein sequence to ambiguous codons
</td>
</tr>
<tr>
<td><a href="backtranseq.html">backtranseq</a></td>
<td>
Back translate a protein sequence
</td>
</tr>
<tr>
<td><a href="banana.html">banana</a></td>
<td>
Bending and curvature plot in B-DNA
</td>
</tr>
<tr>
<td><a href="biosed.html">biosed</a></td>
<td>
Replace or delete sequence sections
</td>
</tr>
<tr>
<td><a href="btwisted.html">btwisted</a></td>
<td>
Calculates the twisting in a B-DNA sequence
</td>
</tr>
<tr>
<td><a href="cai.html">cai</a></td>
<td>
CAI codon adaptation index
</td>
</tr>
<tr>
<td><a href="chaos.html">chaos</a></td>
<td>
Create a chaos game representation plot for a sequence
</td>
</tr>
<tr>
<td><a href="charge.html">charge</a></td>
<td>
Protein charge plot
</td>
</tr>
<tr>
<td><a href="checktrans.html">checktrans</a></td>
<td>
Reports STOP codons and ORF statistics of a protein
</td>
</tr>
<tr>
<td><a href="chips.html">chips</a></td>
<td>
Codon usage statistics
</td>
</tr>
<tr>
<td><a href="cirdna.html">cirdna</a></td>
<td>
Draws circular maps of DNA constructs
</td>
</tr>
<tr>
<td><a href="codcmp.html">codcmp</a></td>
<td>
Codon usage table comparison
</td>
</tr>
<tr>
<td><a href="codcopy.html">codcopy</a></td>
<td>
Reads and writes a codon usage table
</td>
</tr>
<tr>
<td><a href="coderet.html">coderet</a></td>
<td>
Extract CDS, mRNA and translations from feature tables
</td>
</tr>
<tr>
<td><a href="compseq.html">compseq</a></td>
<td>
Count composition of dimer/trimer/etc words in a sequence
</td>
</tr>
<tr>
<td><a href="cons.html">cons</a></td>
<td>
Creates a consensus from multiple alignments
</td>
</tr>
<tr>
<td><a href="cpgplot.html">cpgplot</a></td>
<td>
Plot CpG rich areas
</td>
</tr>
<tr>
<td><a href="cpgreport.html">cpgreport</a></td>
<td>
Reports all CpG rich regions
</td>
</tr>
<tr>
<td><a href="cusp.html">cusp</a></td>
<td>
Create a codon usage table
</td>
</tr>
<tr>
<td><a href="cutgextract.html">cutgextract</a></td>
<td>
Extract data from CUTG
</td>
</tr>
<tr>
<td><a href="cutseq.html">cutseq</a></td>
<td>
Removes a specified section from a sequence
</td>
</tr>
<tr>
<td><a href="dan.html">dan</a></td>
<td>
Calculates DNA RNA/DNA melting temperature
</td>
</tr>
<tr>
<td><a href="dbiblast.html">dbiblast</a></td>
<td>
Index a BLAST database
</td>
</tr>
<tr>
<td><a href="dbifasta.html">dbifasta</a></td>
<td>
Database indexing for fasta file databases
</td>
</tr>
<tr>
<td><a href="dbiflat.html">dbiflat</a></td>
<td>
Index a flat file database
</td>
</tr>
<tr>
<td><a href="dbigcg.html">dbigcg</a></td>
<td>
Index a GCG formatted database
</td>
</tr>
<tr>
<td><a href="dbxfasta.html">dbxfasta</a></td>
<td>
Database b+tree indexing for fasta file databases
</td>
</tr>
<tr>
<td><a href="dbxflat.html">dbxflat</a></td>
<td>
Database b+tree indexing for flat file databases
</td>
</tr>
<tr>
<td><a href="dbxgcg.html">dbxgcg</a></td>
<td>
Database b+tree indexing for GCG formatted databases
</td>
</tr>
<tr>
<td><a href="degapseq.html">degapseq</a></td>
<td>
Removes gap characters from sequences
</td>
</tr>
<tr>
<td><a href="descseq.html">descseq</a></td>
<td>
Alter the name or description of a sequence
</td>
</tr>
<tr>
<td><a href="diffseq.html">diffseq</a></td>
<td>
Find differences between nearly identical sequences
</td>
</tr>
<tr>
<td><a href="digest.html">digest</a></td>
<td>
Protein proteolytic enzyme or reagent cleavage digest
</td>
</tr>
<tr>
<td><a href="distmat.html">distmat</a></td>
<td>
Creates a distance matrix from multiple alignments
</td>
</tr>
<tr>
<td><a href="dotmatcher.html">dotmatcher</a></td>
<td>
Displays a thresholded dotplot of two sequences
</td>
</tr>
<tr>
<td><a href="dotpath.html">dotpath</a></td>
<td>
Non-overlapping wordmatch dotplot of two sequences
</td>
</tr>
<tr>
<td><a href="dottup.html">dottup</a></td>
<td>
Displays a wordmatch dotplot of two sequences
</td>
</tr>
<tr>
<td><a href="dreg.html">dreg</a></td>
<td>
Regular expression search of a nucleotide sequence
</td>
</tr>
<tr>
<td><a href="edialign.html">edialign</a></td>
<td>
Local multiple alignment of sequences
</td>
</tr>
<tr>
<td><a href="einverted.html">einverted</a></td>
<td>
Finds DNA inverted repeats
</td>
</tr>
<tr>
<td><a href="embossdata.html">embossdata</a></td>
<td>
Finds or fetches data files read by EMBOSS programs
</td>
</tr>
<tr>
<td><a href="embossversion.html">embossversion</a></td>
<td>
Writes the current EMBOSS version number
</td>
</tr>
<tr>
<td><a href="emma.html">emma</a></td>
<td>
Multiple alignment program - interface to ClustalW program
</td>
</tr>
<tr>
<td><a href="emowse.html">emowse</a></td>
<td>
Protein identification by mass spectrometry
</td>
</tr>
<tr>
<td><a href="entret.html">entret</a></td>
<td>
Reads and writes (returns) flatfile entries
</td>
</tr>
<tr>
<td><a href="epestfind.html">epestfind</a></td>
<td>
Finds PEST motifs as potential proteolytic cleavage sites
</td>
</tr>
<tr>
<td><a href="eprimer3.html">eprimer3</a></td>
<td>
Picks PCR primers and hybridization oligos
</td>
</tr>
<tr>
<td><a href="equicktandem.html">equicktandem</a></td>
<td>
Finds tandem repeats
</td>
</tr>
<tr>
<td><a href="est2genome.html">est2genome</a></td>
<td>
Align EST and genomic DNA sequences
</td>
</tr>
<tr>
<td><a href="etandem.html">etandem</a></td>
<td>
Looks for tandem repeats in a nucleotide sequence
</td>
</tr>
<tr>
<td><a href="extractalign.html">extractalign</a></td>
<td>
Extract regions from a sequence alignment
</td>
</tr>
<tr>
<td><a href="extractfeat.html">extractfeat</a></td>
<td>
Extract features from a sequence
</td>
</tr>
<tr>
<td><a href="extractseq.html">extractseq</a></td>
<td>
Extract regions from a sequence
</td>
</tr>
<tr>
<td><a href="findkm.html">findkm</a></td>
<td>
Find Km and Vmax for an enzyme reaction
</td>
</tr>
<tr>
<td><a href="freak.html">freak</a></td>
<td>
Residue/base frequency table or plot
</td>
</tr>
<tr>
<td><a href="fuzznuc.html">fuzznuc</a></td>
<td>
Nucleic acid pattern search
</td>
</tr>
<tr>
<td><a href="fuzzpro.html">fuzzpro</a></td>
<td>
Protein pattern search
</td>
</tr>
<tr>
<td><a href="fuzztran.html">fuzztran</a></td>
<td>
Protein pattern search after translation
</td>
</tr>
<tr>
<td><a href="garnier.html">garnier</a></td>
<td>
Predicts protein secondary structure
</td>
</tr>
<tr>
<td><a href="geecee.html">geecee</a></td>
<td>
Calculates fractional GC content of nucleic acid sequences
</td>
</tr>
<tr>
<td><a href="getorf.html">getorf</a></td>
<td>
Finds and extracts open reading frames (ORFs)
</td>
</tr>
<tr>
<td><a href="helixturnhelix.html">helixturnhelix</a></td>
<td>
Report nucleic acid binding motifs
</td>
</tr>
<tr>
<td><a href="hmoment.html">hmoment</a></td>
<td>
Hydrophobic moment calculation
</td>
</tr>
<tr>
<td><a href="iep.html">iep</a></td>
<td>
Calculates the isoelectric point of a protein
</td>
</tr>
<tr>
<td><a href="infoalign.html">infoalign</a></td>
<td>
Information on a multiple sequence alignment
</td>
</tr>
<tr>
<td><a href="infoseq.html">infoseq</a></td>
<td>
Displays some simple information about sequences
</td>
</tr>
<tr>
<td><a href="isochore.html">isochore</a></td>
<td>
Plots isochores in large DNA sequences
</td>
</tr>
<tr>
<td><a href="lindna.html">lindna</a></td>
<td>
Draws linear maps of DNA constructs
</td>
</tr>
<tr>
<td><a href="listor.html">listor</a></td>
<td>
Write a list file of the logical OR of two sets of sequences
</td>
</tr>
<tr>
<td><a href="makenucseq.html">makenucseq</a></td>
<td>
Creates random nucleotide sequences
</td>
</tr>
<tr>
<td><a href="makeprotseq.html">makeprotseq</a></td>
<td>
Creates random protein sequences
</td>
</tr>
<tr>
<td><a href="marscan.html">marscan</a></td>
<td>
Finds MAR/SAR sites in nucleic sequences
</td>
</tr>
<tr>
<td><a href="maskfeat.html">maskfeat</a></td>
<td>
Mask off features of a sequence
</td>
</tr>
<tr>
<td><a href="maskseq.html">maskseq</a></td>
<td>
Mask off regions of a sequence
</td>
</tr>
<tr>
<td><a href="matcher.html">matcher</a></td>
<td>
Finds the best local alignments between two sequences
</td>
</tr>
<tr>
<td><a href="megamerger.html">megamerger</a></td>
<td>
Merge two large overlapping nucleic acid sequences
</td>
</tr>
<tr>
<td><a href="merger.html">merger</a></td>
<td>
Merge two overlapping sequences
</td>
</tr>
<tr>
<td><a href="msbar.html">msbar</a></td>
<td>
Mutate sequence beyond all recognition
</td>
</tr>
<tr>
<td><a href="mwcontam.html">mwcontam</a></td>
<td>
Shows molwts that match across a set of files
</td>
</tr>
<tr>
<td><a href="mwfilter.html">mwfilter</a></td>
<td>
Filter noisy molwts from mass spec output
</td>
</tr>
<tr>
<td><a href="needle.html">needle</a></td>
<td>
Needleman-Wunsch global alignment
</td>
</tr>
<tr>
<td><a href="newcpgreport.html">newcpgreport</a></td>
<td>
Report CpG rich areas
</td>
</tr>
<tr>
<td><a href="newcpgseek.html">newcpgseek</a></td>
<td>
Reports CpG rich regions
</td>
</tr>
<tr>
<td><a href="newseq.html">newseq</a></td>
<td>
Type in a short new sequence
</td>
</tr>
<tr>
<td><a href="noreturn.html">noreturn</a></td>
<td>
Removes carriage return from ASCII files
</td>
</tr>
<tr>
<td><a href="notseq.html">notseq</a></td>
<td>
Exclude a set of sequences and write out the remaining ones
</td>
</tr>
<tr>
<td><a href="nthseq.html">nthseq</a></td>
<td>
Writes one sequence from a multiple set of sequences
</td>
</tr>
<tr>
<td><a href="octanol.html">octanol</a></td>
<td>
Displays protein hydropathy
</td>
</tr>
<tr>
<td><a href="oddcomp.html">oddcomp</a></td>
<td>
Find protein sequence regions with a biased composition
</td>
</tr>
<tr>
<td><a href="palindrome.html">palindrome</a></td>
<td>
Looks for inverted repeats in a nucleotide sequence
</td>
</tr>
<tr>
<td><a href="pasteseq.html">pasteseq</a></td>
<td>
Insert one sequence into another
</td>
</tr>
<tr>
<td><a href="patmatdb.html">patmatdb</a></td>
<td>
Search a protein sequence with a motif
</td>
</tr>
<tr>
<td><a href="patmatmotifs.html">patmatmotifs</a></td>
<td>
Search a PROSITE motif database with a protein sequence
</td>
</tr>
<tr>
<td><a href="pepcoil.html">pepcoil</a></td>
<td>
Predicts coiled coil regions
</td>
</tr>
<tr>
<td><a href="pepinfo.html">pepinfo</a></td>
<td>
Plots simple amino acid properties in parallel
</td>
</tr>
<tr>
<td><a href="pepnet.html">pepnet</a></td>
<td>
Displays proteins as a helical net
</td>
</tr>
<tr>
<td><a href="pepstats.html">pepstats</a></td>
<td>
Protein statistics
</td>
</tr>
<tr>
<td><a href="pepwheel.html">pepwheel</a></td>
<td>
Shows protein sequences as helices
</td>
</tr>
<tr>
<td><a href="pepwindow.html">pepwindow</a></td>
<td>
Displays protein hydropathy
</td>
</tr>
<tr>
<td><a href="pepwindowall.html">pepwindowall</a></td>
<td>
Displays protein hydropathy of a set of sequences
</td>
</tr>
<tr>
<td><a href="plotcon.html">plotcon</a></td>
<td>
Plot quality of conservation of a sequence alignment
</td>
</tr>
<tr>
<td><a href="plotorf.html">plotorf</a></td>
<td>
Plot potential open reading frames
</td>
</tr>
<tr>
<td><a href="polydot.html">polydot</a></td>
<td>
Displays all-against-all dotplots of a set of sequences
</td>
</tr>
<tr>
<td><a href="preg.html">preg</a></td>
<td>
Regular expression search of a protein sequence
</td>
</tr>
<tr>
<td><a href="prettyplot.html">prettyplot</a></td>
<td>
Displays aligned sequences, with colouring and boxing
</td>
</tr>
<tr>
<td><a href="prettyseq.html">prettyseq</a></td>
<td>
Output sequence with translated ranges
</td>
</tr>
<tr>
<td><a href="primersearch.html">primersearch</a></td>
<td>
Searches DNA sequences for matches with primer pairs
</td>
</tr>
<tr>
<td><a href="printsextract.html">printsextract</a></td>
<td>
Extract data from PRINTS
</td>
</tr>
<tr>
<td><a href="profit.html">profit</a></td>
<td>
Scan a sequence or database with a matrix or profile
</td>
</tr>
<tr>
<td><a href="prophecy.html">prophecy</a></td>
<td>
Creates matrices/profiles from multiple alignments
</td>
</tr>
<tr>
<td><a href="prophet.html">prophet</a></td>
<td>
Gapped alignment for profiles
</td>
</tr>
<tr>
<td><a href="prosextract.html">prosextract</a></td>
<td>
Build the PROSITE motif database for use by patmatmotifs
</td>
</tr>
<tr>
<td><a href="pscan.html">pscan</a></td>
<td>
Scans proteins using PRINTS
</td>
</tr>
<tr>
<td><a href="psiphi.html">psiphi</a></td>
<td>
Phi and psi torsion angles from protein coordinates
</td>
</tr>
<tr>
<td><a href="rebaseextract.html">rebaseextract</a></td>
<td>
Extract data from REBASE
</td>
</tr>
<tr>
<td><a href="recoder.html">recoder</a></td>
<td>
Remove restriction sites but maintain same translation
</td>
</tr>
<tr>
<td><a href="redata.html">redata</a></td>
<td>
Search REBASE for enzyme name, references, suppliers etc
</td>
</tr>
<tr>
<td><a href="remap.html">remap</a></td>
<td>
Display sequence with restriction sites, translation etc
</td>
</tr>
<tr>
<td><a href="restover.html">restover</a></td>
<td>
Find restriction enzymes producing specific overhang
</td>
</tr>
<tr>
<td><a href="restrict.html">restrict</a></td>
<td>
Finds restriction enzyme cleavage sites
</td>
</tr>
<tr>
<td><a href="revseq.html">revseq</a></td>
<td>
Reverse and complement a sequence
</td>
</tr>
<tr>
<td><a href="seealso.html">seealso</a></td>
<td>
Finds programs sharing group names
</td>
</tr>
<tr>
<td><a href="seqmatchall.html">seqmatchall</a></td>
<td>
All-against-all comparison of a set of sequences
</td>
</tr>
<tr>
<td><a href="seqret.html">seqret</a></td>
<td>
Reads and writes (returns) sequences
</td>
</tr>
<tr>
<td><a href="seqretsplit.html">seqretsplit</a></td>
<td>
Reads and writes (returns) sequences in individual files
</td>
</tr>
<tr>
<td><a href="showalign.html">showalign</a></td>
<td>
Displays a multiple sequence alignment
</td>
</tr>
<tr>
<td><a href="showdb.html">showdb</a></td>
<td>
Displays information on the currently available databases
</td>
</tr>
<tr>
<td><a href="showfeat.html">showfeat</a></td>
<td>
Show features of a sequence
</td>
</tr>
<tr>
<td><a href="showorf.html">showorf</a></td>
<td>
Pretty output of DNA translations
</td>
</tr>
<tr>
<td><a href="showseq.html">showseq</a></td>
<td>
Display a sequence with features, translation etc
</td>
</tr>
<tr>
<td><a href="shuffleseq.html">shuffleseq</a></td>
<td>
Shuffles a set of sequences maintaining composition
</td>
</tr>
<tr>
<td><a href="sigcleave.html">sigcleave</a></td>
<td>
Reports protein signal cleavage sites
</td>
</tr>
<tr>
<td><a href="silent.html">silent</a></td>
<td>
Silent mutation restriction enzyme scan
</td>
</tr>
<tr>
<td><a href="sirna.html">sirna</a></td>
<td>
Finds siRNA duplexes in mRNA
</td>
</tr>
<tr>
<td><a href="sixpack.html">sixpack</a></td>
<td>
Display a DNA sequence with 6-frame translation and ORFs
</td>
</tr>
<tr>
<td><a href="skipseq.html">skipseq</a></td>
<td>
Reads and writes (returns) sequences, skipping first few
</td>
</tr>
<tr>
<td><a href="splitter.html">splitter</a></td>
<td>
Split a sequence into (overlapping) smaller sequences
</td>
</tr>
<tr>
<td><a href="stretcher.html">stretcher</a></td>
<td>
Finds the best global alignment between two sequences
</td>
</tr>
<tr>
<td><a href="stssearch.html">stssearch</a></td>
<td>
Search a DNA database for matches with a set of STS primers
</td>
</tr>
<tr>
<td><a href="supermatcher.html">supermatcher</a></td>
<td>
Match large sequences against one or more other sequences
</td>
</tr>
<tr>
<td><a href="syco.html">syco</a></td>
<td>
Synonymous codon usage Gribskov statistic plot
</td>
</tr>
<tr>
<td><a href="tcode.html">tcode</a></td>
<td>
Fickett TESTCODE statistic to identify protein-coding DNA
</td>
</tr>
<tr>
<td><a href="textsearch.html">textsearch</a></td>
<td>
Search sequence documentation. Slow, use SRS and Entrez!
</td>
</tr>
<tr>
<td><a href="tfextract.html">tfextract</a></td>
<td>
Extract data from TRANSFAC
</td>
</tr>
<tr>
<td><a href="tfm.html">tfm</a></td>
<td>
Displays a program's help documentation manual
</td>
</tr>
<tr>
<td><a href="tfscan.html">tfscan</a></td>
<td>
Scans DNA sequences for transcription factors
</td>
</tr>
<tr>
<td><a href="tmap.html">tmap</a></td>
<td>
Displays membrane spanning regions
</td>
</tr>
<tr>
<td><a href="tranalign.html">tranalign</a></td>
<td>
Align nucleic coding regions given the aligned proteins
</td>
</tr>
<tr>
<td><a href="transeq.html">transeq</a></td>
<td>
Translate nucleic acid sequences
</td>
</tr>
<tr>
<td><a href="trimest.html">trimest</a></td>
<td>
Trim poly-A tails off EST sequences
</td>
</tr>
<tr>
<td><a href="trimseq.html">trimseq</a></td>
<td>
Trim ambiguous bits off the ends of sequences
</td>
</tr>
<tr>
<td><a href="twofeat.html">twofeat</a></td>
<td>
Finds neighbouring pairs of features in sequences
</td>
</tr>
<tr>
<td><a href="union.html">union</a></td>
<td>
Reads sequence fragments and builds one sequence
</td>
</tr>
<tr>
<td><a href="vectorstrip.html">vectorstrip</a></td>
<td>
Strips out DNA between a pair of vector sequences
</td>
</tr>
<tr>
<td><a href="water.html">water</a></td>
<td>
Smith-Waterman local alignment
</td>
</tr>
<tr>
<td><a href="whichdb.html">whichdb</a></td>
<td>
Search all databases for an entry
</td>
</tr>
<tr>
<td><a href="wobble.html">wobble</a></td>
<td>
Wobble base plot
</td>
</tr>
<tr>
<td><a href="wordcount.html">wordcount</a></td>
<td>
Counts words of a specified size in a DNA sequence
</td>
</tr>
<tr>
<td><a href="wordfinder.html">wordfinder</a></td>
<td>
Match large sequences against one or more other sequences
</td>
</tr>
<tr>
<td><a href="wordmatch.html">wordmatch</a></td>
<td>
Finds all exact matches of a given size between 2 sequences
</td>
</tr>
<tr>
<td><a href="wossname.html">wossname</a></td>
<td>
Finds programs by keywords in their one-line documentation
</td>
</tr>
<tr>
<td><a href="yank.html">yank</a></td>
<td>
Reads a sequence range, appends the full USA to a list file
</td>
</tr>


</table>




</BODY>
</HTML>
