
<HTML>

<HEAD>
  <TITLE>
  EMBOSS: The Applications (programs)
  </TITLE>
</HEAD>
<BODY BGCOLOR="#FFFFFF" text="#000000">



<table align=center border=0 cellspacing=0 cellpadding=0>
<tr><td valign=top>
<A HREF="http://emboss.sourceforge.net/" ONMOUSEOVER="self.status='Go to the EMBOSS home page';return true">
<img border=0 src="/emboss_icon.jpg" alt="" width=150 height=48></a>
</td>
<td align=left valign=middle>
<b><font size="+6"> 
The Applications (programs)
</font></b>
</td></tr>
</table>
<br>&nbsp;
<p>




The programs are listed in alphabetical order, Look at the individual
applications or go to the 
<a href="groups.html">GROUPS</a>
page to search by category. 
<p>

<a href="../../embassy/index.html">EMBASSY applications</a>
are described in separate documentation for each package.

<h3><A NAME="current">Applications</A> in the <a
href="ftp://emboss.open-bio.org/pub/EMBOSS/">current release</a></h3>

<table border cellpadding=4 bgcolor="#FFFFF0">

<tr>
<th>Program name</th>
<th>Description</th>
</tr>

<tr>
<td><a href="aaindexextract.html">aaindexextract</a></td>
<td>
Extract amino acid property data from AAINDEX
</td>
</tr>
<tr>
<td><a href="abiview.html">abiview</a></td>
<td>
Display the trace in an ABI sequencer file
</td>
</tr>
<tr>
<td><a href="acdc.html">acdc</a></td>
<td>
Test an application ACD file
</td>
</tr>
<tr>
<td><a href="acdpretty.html">acdpretty</a></td>
<td>
Correctly reformat an application ACD file
</td>
</tr>
<tr>
<td><a href="acdtable.html">acdtable</a></td>
<td>
Generate an HTML table of parameters from an application ACD file
</td>
</tr>
<tr>
<td><a href="acdtrace.html">acdtrace</a></td>
<td>
Trace processing of an application ACD file (for testing)
</td>
</tr>
<tr>
<td><a href="acdvalid.html">acdvalid</a></td>
<td>
Validate an application ACD file
</td>
</tr>
<tr>
<td><a href="aligncopy.html">aligncopy</a></td>
<td>
Reads and writes alignments
</td>
</tr>
<tr>
<td><a href="aligncopypair.html">aligncopypair</a></td>
<td>
Reads and writes pairs from alignments
</td>
</tr>
<tr>
<td><a href="antigenic.html">antigenic</a></td>
<td>
Finds antigenic sites in proteins
</td>
</tr>
<tr>
<td><a href="backtranambig.html">backtranambig</a></td>
<td>
Back-translate a protein sequence to ambiguous nucleotide sequence
</td>
</tr>
<tr>
<td><a href="backtranseq.html">backtranseq</a></td>
<td>
Back-translate a protein sequence to a nucleotide sequence
</td>
</tr>
<tr>
<td><a href="banana.html">banana</a></td>
<td>
Plot bending and curvature data for B-DNA
</td>
</tr>
<tr>
<td><a href="biosed.html">biosed</a></td>
<td>
Replace or delete sequence sections
</td>
</tr>
<tr>
<td><a href="btwisted.html">btwisted</a></td>
<td>
Calculate the twisting in a B-DNA sequence
</td>
</tr>
<tr>
<td><a href="cai.html">cai</a></td>
<td>
Calculate codon adaptation index
</td>
</tr>
<tr>
<td><a href="chaos.html">chaos</a></td>
<td>
Draw a chaos game representation plot for a nucleotide sequence
</td>
</tr>
<tr>
<td><a href="charge.html">charge</a></td>
<td>
Draw a protein charge plot
</td>
</tr>
<tr>
<td><a href="checktrans.html">checktrans</a></td>
<td>
Reports STOP codons and ORF statistics of a protein
</td>
</tr>
<tr>
<td><a href="chips.html">chips</a></td>
<td>
Calculates Nc codon usage statistic
</td>
</tr>
<tr>
<td><a href="cirdna.html">cirdna</a></td>
<td>
Draws circular maps of DNA constructs
</td>
</tr>
<tr>
<td><a href="codcmp.html">codcmp</a></td>
<td>
Codon usage table comparison
</td>
</tr>
<tr>
<td><a href="codcopy.html">codcopy</a></td>
<td>
Copy and reformat a codon usage table
</td>
</tr>
<tr>
<td><a href="coderet.html">coderet</a></td>
<td>
Extract CDS, mRNA and translations from feature tables
</td>
</tr>
<tr>
<td><a href="compseq.html">compseq</a></td>
<td>
Calculate the composition of unique words in sequences
</td>
</tr>
<tr>
<td><a href="cons.html">cons</a></td>
<td>
Create a consensus sequence from a multiple alignment
</td>
</tr>
<tr>
<td><a href="consambig.html">consambig</a></td>
<td>
Create an ambiguous consensus sequence from a multiple alignment
</td>
</tr>
<tr>
<td><a href="cpgplot.html">cpgplot</a></td>
<td>
Identify and plot CpG islands in nucleotide sequence(s)
</td>
</tr>
<tr>
<td><a href="cpgreport.html">cpgreport</a></td>
<td>
Identify and report CpG-rich regions in nucleotide sequence(s)
</td>
</tr>
<tr>
<td><a href="cusp.html">cusp</a></td>
<td>
Create a codon usage table from nucleotide sequence(s)
</td>
</tr>
<tr>
<td><a href="cutgextract.html">cutgextract</a></td>
<td>
Extract codon usage tables from CUTG database
</td>
</tr>
<tr>
<td><a href="cutseq.html">cutseq</a></td>
<td>
Removes a section from a sequence
</td>
</tr>
<tr>
<td><a href="dan.html">dan</a></td>
<td>
Calculates nucleic acid melting temperature
</td>
</tr>
<tr>
<td><a href="dbiblast.html">dbiblast</a></td>
<td>
Index a BLAST database
</td>
</tr>
<tr>
<td><a href="dbifasta.html">dbifasta</a></td>
<td>
Index a fasta file database
</td>
</tr>
<tr>
<td><a href="dbiflat.html">dbiflat</a></td>
<td>
Index a flat file database
</td>
</tr>
<tr>
<td><a href="dbigcg.html">dbigcg</a></td>
<td>
Index a GCG formatted database
</td>
</tr>
<tr>
<td><a href="dbxfasta.html">dbxfasta</a></td>
<td>
Index a fasta file database using b+tree indices
</td>
</tr>
<tr>
<td><a href="dbxflat.html">dbxflat</a></td>
<td>
Index a flat file database using b+tree indices
</td>
</tr>
<tr>
<td><a href="dbxgcg.html">dbxgcg</a></td>
<td>
Index a GCG formatted database using b+tree indices
</td>
</tr>
<tr>
<td><a href="degapseq.html">degapseq</a></td>
<td>
Removes non-alphabetic (e.g. gap) characters from sequences
</td>
</tr>
<tr>
<td><a href="density.html">density</a></td>
<td>
Draw a nucleic acid density plot
</td>
</tr>
<tr>
<td><a href="descseq.html">descseq</a></td>
<td>
Alter the name or description of a sequence
</td>
</tr>
<tr>
<td><a href="diffseq.html">diffseq</a></td>
<td>
Compare and report features of two similar sequences
</td>
</tr>
<tr>
<td><a href="digest.html">digest</a></td>
<td>
Reports on protein proteolytic enzyme or reagent cleavage sites
</td>
</tr>
<tr>
<td><a href="distmat.html">distmat</a></td>
<td>
Create a distance matrix from a multiple sequence alignment
</td>
</tr>
<tr>
<td><a href="dotmatcher.html">dotmatcher</a></td>
<td>
Draw a threshold dotplot of two sequences
</td>
</tr>
<tr>
<td><a href="dotpath.html">dotpath</a></td>
<td>
Draw a non-overlapping wordmatch dotplot of two sequences
</td>
</tr>
<tr>
<td><a href="dottup.html">dottup</a></td>
<td>
Displays a wordmatch dotplot of two sequences
</td>
</tr>
<tr>
<td><a href="dreg.html">dreg</a></td>
<td>
Regular expression search of nucleotide sequence(s)
</td>
</tr>
<tr>
<td><a href="edialign.html">edialign</a></td>
<td>
Local multiple alignment of sequences
</td>
</tr>
<tr>
<td><a href="einverted.html">einverted</a></td>
<td>
Finds inverted repeats in nucleotide sequences
</td>
</tr>
<tr>
<td><a href="embossdata.html">embossdata</a></td>
<td>
Find and retrieve EMBOSS data files
</td>
</tr>
<tr>
<td><a href="embossversion.html">embossversion</a></td>
<td>
Reports the current EMBOSS version number
</td>
</tr>
<tr>
<td><a href="emma.html">emma</a></td>
<td>
Multiple sequence alignment (ClustalW wrapper)
</td>
</tr>
<tr>
<td><a href="emowse.html">emowse</a></td>
<td>
Search protein sequences by digest fragment molecular weight
</td>
</tr>
<tr>
<td><a href="entret.html">entret</a></td>
<td>
Retrieves sequence entries from flatfile databases and files
</td>
</tr>
<tr>
<td><a href="epestfind.html">epestfind</a></td>
<td>
Finds PEST motifs as potential proteolytic cleavage sites
</td>
</tr>
<tr>
<td><a href="eprimer3.html">eprimer3</a></td>
<td>
Picks PCR primers and hybridization oligos
</td>
</tr>
<tr>
<td><a href="equicktandem.html">equicktandem</a></td>
<td>
Finds tandem repeats in nucleotide sequences
</td>
</tr>
<tr>
<td><a href="est2genome.html">est2genome</a></td>
<td>
Align EST sequences to genomic DNA sequence
</td>
</tr>
<tr>
<td><a href="etandem.html">etandem</a></td>
<td>
Finds tandem repeats in a nucleotide sequence
</td>
</tr>
<tr>
<td><a href="extractalign.html">extractalign</a></td>
<td>
Extract regions from a sequence alignment
</td>
</tr>
<tr>
<td><a href="extractfeat.html">extractfeat</a></td>
<td>
Extract features from sequence(s)
</td>
</tr>
<tr>
<td><a href="extractseq.html">extractseq</a></td>
<td>
Extract regions from a sequence
</td>
</tr>
<tr>
<td><a href="featcopy.html">featcopy</a></td>
<td>
Reads and writes a feature table
</td>
</tr>
<tr>
<td><a href="featreport.html">featreport</a></td>
<td>
Reads and writes a feature table
</td>
</tr>
<tr>
<td><a href="findkm.html">findkm</a></td>
<td>
Calculate and plot enzyme reaction data
</td>
</tr>
<tr>
<td><a href="freak.html">freak</a></td>
<td>
Generate residue/base frequency table or plot
</td>
</tr>
<tr>
<td><a href="fuzznuc.html">fuzznuc</a></td>
<td>
Search for patterns in nucleotide sequences
</td>
</tr>
<tr>
<td><a href="fuzzpro.html">fuzzpro</a></td>
<td>
Search for patterns in protein sequences
</td>
</tr>
<tr>
<td><a href="fuzztran.html">fuzztran</a></td>
<td>
Search for patterns in protein sequences (translated)
</td>
</tr>
<tr>
<td><a href="garnier.html">garnier</a></td>
<td>
Predicts protein secondary structure using GOR method
</td>
</tr>
<tr>
<td><a href="geecee.html">geecee</a></td>
<td>
Calculate fractional GC content of nucleic acid sequences
</td>
</tr>
<tr>
<td><a href="getorf.html">getorf</a></td>
<td>
Finds and extracts open reading frames (ORFs)
</td>
</tr>
<tr>
<td><a href="helixturnhelix.html">helixturnhelix</a></td>
<td>
Identify nucleic acid-binding motifs in protein sequences
</td>
</tr>
<tr>
<td><a href="hmoment.html">hmoment</a></td>
<td>
Calculate and plot hydrophobic moment for protein sequence(s)
</td>
</tr>
<tr>
<td><a href="iep.html">iep</a></td>
<td>
Calculate the isoelectric point of proteins
</td>
</tr>
<tr>
<td><a href="infoalign.html">infoalign</a></td>
<td>
Display basic information about a multiple sequence alignment
</td>
</tr>
<tr>
<td><a href="infobase.html">infobase</a></td>
<td>
Return information on a given nucleotide base
</td>
</tr>
<tr>
<td><a href="inforesidue.html">inforesidue</a></td>
<td>
Return information on a given amino acid residue
</td>
</tr>
<tr>
<td><a href="infoseq.html">infoseq</a></td>
<td>
Display basic information about sequences
</td>
</tr>
<tr>
<td><a href="isochore.html">isochore</a></td>
<td>
Plots isochores in DNA sequences
</td>
</tr>
<tr>
<td><a href="jaspextract.html">jaspextract</a></td>
<td>
Extract data from JASPAR
</td>
</tr>
<tr>
<td><a href="jaspscan.html">jaspscan</a></td>
<td>
Scans DNA sequences for transcription factors
</td>
</tr>
<tr>
<td><a href="lindna.html">lindna</a></td>
<td>
Draws linear maps of DNA constructs
</td>
</tr>
<tr>
<td><a href="listor.html">listor</a></td>
<td>
Write a list file of the logical OR of two sets of sequences
</td>
</tr>
<tr>
<td><a href="makenucseq.html">makenucseq</a></td>
<td>
Create random nucleotide sequences
</td>
</tr>
<tr>
<td><a href="makeprotseq.html">makeprotseq</a></td>
<td>
Create random protein sequences
</td>
</tr>
<tr>
<td><a href="marscan.html">marscan</a></td>
<td>
Finds matrix/scaffold recognition (MRS) signatures in DNA sequences
</td>
</tr>
<tr>
<td><a href="maskambignuc.html">maskambignuc</a></td>
<td>
Masks all ambiguity characters in nucleotide sequences with N
</td>
</tr>
<tr>
<td><a href="maskambigprot.html">maskambigprot</a></td>
<td>
Masks all ambiguity characters in protein sequences with X
</td>
</tr>
<tr>
<td><a href="maskfeat.html">maskfeat</a></td>
<td>
Write a sequence with masked features
</td>
</tr>
<tr>
<td><a href="maskseq.html">maskseq</a></td>
<td>
Write a sequence with masked regions
</td>
</tr>
<tr>
<td><a href="matcher.html">matcher</a></td>
<td>
Waterman-Eggert local alignment of two sequences
</td>
</tr>
<tr>
<td><a href="megamerger.html">megamerger</a></td>
<td>
Merge two large overlapping DNA sequences
</td>
</tr>
<tr>
<td><a href="merger.html">merger</a></td>
<td>
Merge two overlapping sequences
</td>
</tr>
<tr>
<td><a href="msbar.html">msbar</a></td>
<td>
Mutate a sequence
</td>
</tr>
<tr>
<td><a href="mwcontam.html">mwcontam</a></td>
<td>
Find weights common to multiple molecular weights files
</td>
</tr>
<tr>
<td><a href="mwfilter.html">mwfilter</a></td>
<td>
Filter noisy data from molecular weights file
</td>
</tr>
<tr>
<td><a href="needle.html">needle</a></td>
<td>
Needleman-Wunsch global alignment of two sequences
</td>
</tr>
<tr>
<td><a href="needleall.html">needleall</a></td>
<td>
Many-to-many pairwise alignments of two sequence sets
</td>
</tr>
<tr>
<td><a href="newcpgreport.html">newcpgreport</a></td>
<td>
Identify CpG islands in nucleotide sequence(s)
</td>
</tr>
<tr>
<td><a href="newcpgseek.html">newcpgseek</a></td>
<td>
Identify and report CpG-rich regions in nucleotide sequence(s)
</td>
</tr>
<tr>
<td><a href="newseq.html">newseq</a></td>
<td>
Create a sequence file from a typed-in sequence
</td>
</tr>
<tr>
<td><a href="nohtml.html">nohtml</a></td>
<td>
Remove mark-up (e.g. HTML tags) from an ASCII text file
</td>
</tr>
<tr>
<td><a href="noreturn.html">noreturn</a></td>
<td>
Remove carriage return from ASCII files
</td>
</tr>
<tr>
<td><a href="nospace.html">nospace</a></td>
<td>
Remove whitespace from an ASCII text file
</td>
</tr>
<tr>
<td><a href="notab.html">notab</a></td>
<td>
Replace tabs with spaces in an ASCII text file
</td>
</tr>
<tr>
<td><a href="notseq.html">notseq</a></td>
<td>
Write to file a subset of an input stream of sequences
</td>
</tr>
<tr>
<td><a href="nthseq.html">nthseq</a></td>
<td>
Write to file a single sequence from an input stream of sequences
</td>
</tr>
<tr>
<td><a href="nthseqset.html">nthseqset</a></td>
<td>
Reads and writes (returns) one set of sequences from many
</td>
</tr>
<tr>
<td><a href="octanol.html">octanol</a></td>
<td>
Draw a White-Wimley protein hydropathy plot
</td>
</tr>
<tr>
<td><a href="oddcomp.html">oddcomp</a></td>
<td>
Identify proteins with specified sequence word composition
</td>
</tr>
<tr>
<td><a href="palindrome.html">palindrome</a></td>
<td>
Finds inverted repeats in nucleotide sequence(s)
</td>
</tr>
<tr>
<td><a href="pasteseq.html">pasteseq</a></td>
<td>
Insert one sequence into another
</td>
</tr>
<tr>
<td><a href="patmatdb.html">patmatdb</a></td>
<td>
Searches protein sequences with a sequence motif
</td>
</tr>
<tr>
<td><a href="patmatmotifs.html">patmatmotifs</a></td>
<td>
Scan a protein sequence with motifs from the PROSITE database
</td>
</tr>
<tr>
<td><a href="pepcoil.html">pepcoil</a></td>
<td>
Predicts coiled coil regions in protein sequences
</td>
</tr>
<tr>
<td><a href="pepinfo.html">pepinfo</a></td>
<td>
Plot amino acid properties of a protein sequence in parallel
</td>
</tr>
<tr>
<td><a href="pepnet.html">pepnet</a></td>
<td>
Draw a helical net for a protein sequence
</td>
</tr>
<tr>
<td><a href="pepstats.html">pepstats</a></td>
<td>
Calculates statistics of protein properties
</td>
</tr>
<tr>
<td><a href="pepwheel.html">pepwheel</a></td>
<td>
Draw a helical wheel diagram for a protein sequence
</td>
</tr>
<tr>
<td><a href="pepwindow.html">pepwindow</a></td>
<td>
Draw a hydropathy plot for a protein sequence
</td>
</tr>
<tr>
<td><a href="pepwindowall.html">pepwindowall</a></td>
<td>
Draw Kyte-Doolittle hydropathy plot for a protein alignment
</td>
</tr>
<tr>
<td><a href="plotcon.html">plotcon</a></td>
<td>
Plot conservation of a sequence alignment
</td>
</tr>
<tr>
<td><a href="plotorf.html">plotorf</a></td>
<td>
Plot potential open reading frames in a nucleotide sequence
</td>
</tr>
<tr>
<td><a href="polydot.html">polydot</a></td>
<td>
Draw dotplots for all-against-all comparison of a sequence set
</td>
</tr>
<tr>
<td><a href="preg.html">preg</a></td>
<td>
Regular expression search of protein sequence(s)
</td>
</tr>
<tr>
<td><a href="prettyplot.html">prettyplot</a></td>
<td>
Draw a sequence alignment with pretty formatting
</td>
</tr>
<tr>
<td><a href="prettyseq.html">prettyseq</a></td>
<td>
Write a nucleotide sequence and its translation to file
</td>
</tr>
<tr>
<td><a href="primersearch.html">primersearch</a></td>
<td>
Search DNA sequences for matches with primer pairs
</td>
</tr>
<tr>
<td><a href="printsextract.html">printsextract</a></td>
<td>
Extract data from PRINTS database for use by pscan
</td>
</tr>
<tr>
<td><a href="profit.html">profit</a></td>
<td>
Scan one or more sequences with a simple frequency matrix
</td>
</tr>
<tr>
<td><a href="prophecy.html">prophecy</a></td>
<td>
Create frequency matrix or profile from a multiple alignment
</td>
</tr>
<tr>
<td><a href="prophet.html">prophet</a></td>
<td>
Scan one or more sequences with a Gribskov or Henikoff profile
</td>
</tr>
<tr>
<td><a href="prosextract.html">prosextract</a></td>
<td>
Processes the PROSITE motif database for use by patmatmotifs
</td>
</tr>
<tr>
<td><a href="pscan.html">pscan</a></td>
<td>
Scans protein sequence(s) with fingerprints from the PRINTS database
</td>
</tr>
<tr>
<td><a href="psiphi.html">psiphi</a></td>
<td>
Calculates phi and psi torsion angles from protein coordinates
</td>
</tr>
<tr>
<td><a href="rebaseextract.html">rebaseextract</a></td>
<td>
Process the REBASE database for use by restriction enzyme applications
</td>
</tr>
<tr>
<td><a href="recoder.html">recoder</a></td>
<td>
Find restriction sites to remove (mutate) with no translation change
</td>
</tr>
<tr>
<td><a href="redata.html">redata</a></td>
<td>
Retrieve information from REBASE restriction enzyme database
</td>
</tr>
<tr>
<td><a href="remap.html">remap</a></td>
<td>
Display restriction enzyme binding sites in a nucleotide sequence
</td>
</tr>
<tr>
<td><a href="restover.html">restover</a></td>
<td>
Find restriction enzymes producing a specific overhang
</td>
</tr>
<tr>
<td><a href="restrict.html">restrict</a></td>
<td>
Report restriction enzyme cleavage sites in a nucleotide sequence
</td>
</tr>
<tr>
<td><a href="revseq.html">revseq</a></td>
<td>
Reverse and complement a nucleotide sequence
</td>
</tr>
<tr>
<td><a href="seealso.html">seealso</a></td>
<td>
Finds programs with similar function to a specified program
</td>
</tr>
<tr>
<td><a href="seqmatchall.html">seqmatchall</a></td>
<td>
All-against-all word comparison of a sequence set
</td>
</tr>
<tr>
<td><a href="seqret.html">seqret</a></td>
<td>
Reads and writes (returns) sequences
</td>
</tr>
<tr>
<td><a href="seqretsetall.html">seqretsetall</a></td>
<td>
Reads and writes (returns) many sets of sequences
</td>
</tr>
<tr>
<td><a href="seqretsplit.html">seqretsplit</a></td>
<td>
Reads sequences and writes them to individual files
</td>
</tr>
<tr>
<td><a href="showalign.html">showalign</a></td>
<td>
Display a multiple sequence alignment in pretty format
</td>
</tr>
<tr>
<td><a href="showdb.html">showdb</a></td>
<td>
Displays information on configured databases
</td>
</tr>
<tr>
<td><a href="showfeat.html">showfeat</a></td>
<td>
Display features of a sequence in pretty format
</td>
</tr>
<tr>
<td><a href="showorf.html">showorf</a></td>
<td>
Display a nucleotide sequence and translation in pretty format
</td>
</tr>
<tr>
<td><a href="showpep.html">showpep</a></td>
<td>
Displays protein sequences with features in pretty format
</td>
</tr>
<tr>
<td><a href="showseq.html">showseq</a></td>
<td>
Displays sequences with features in pretty format
</td>
</tr>
<tr>
<td><a href="shuffleseq.html">shuffleseq</a></td>
<td>
Shuffles a set of sequences maintaining composition
</td>
</tr>
<tr>
<td><a href="sigcleave.html">sigcleave</a></td>
<td>
Reports on signal cleavage sites in a protein sequence
</td>
</tr>
<tr>
<td><a href="silent.html">silent</a></td>
<td>
Find restriction sites to insert (mutate) with no translation change
</td>
</tr>
<tr>
<td><a href="sirna.html">sirna</a></td>
<td>
Finds siRNA duplexes in mRNA
</td>
</tr>
<tr>
<td><a href="sixpack.html">sixpack</a></td>
<td>
Display a DNA sequence with 6-frame translation and ORFs
</td>
</tr>
<tr>
<td><a href="sizeseq.html">sizeseq</a></td>
<td>
Sort sequences by size
</td>
</tr>
<tr>
<td><a href="skipredundant.html">skipredundant</a></td>
<td>
Remove redundant sequences from an input set
</td>
</tr>
<tr>
<td><a href="skipseq.html">skipseq</a></td>
<td>
Reads and writes (returns) sequences, skipping first few
</td>
</tr>
<tr>
<td><a href="splitsource.html">splitsource</a></td>
<td>
Split sequence(s) into original source sequences
</td>
</tr>
<tr>
<td><a href="splitter.html">splitter</a></td>
<td>
Split sequence(s) into smaller sequences
</td>
</tr>
<tr>
<td><a href="stretcher.html">stretcher</a></td>
<td>
Needleman-Wunsch rapid global alignment of two sequences
</td>
</tr>
<tr>
<td><a href="stssearch.html">stssearch</a></td>
<td>
Search a DNA database for matches with a set of STS primers
</td>
</tr>
<tr>
<td><a href="supermatcher.html">supermatcher</a></td>
<td>
Calculate approximate local pair-wise alignments of larger sequences
</td>
</tr>
<tr>
<td><a href="syco.html">syco</a></td>
<td>
Draw synonymous codon usage statistic plot for a nucleotide sequence
</td>
</tr>
<tr>
<td><a href="tcode.html">tcode</a></td>
<td>
Identify protein-coding regions using Fickett TESTCODE statistic
</td>
</tr>
<tr>
<td><a href="textsearch.html">textsearch</a></td>
<td>
Search the textual description of sequence(s)
</td>
</tr>
<tr>
<td><a href="tfextract.html">tfextract</a></td>
<td>
Process TRANSFAC transcription factor database for use by tfscan
</td>
</tr>
<tr>
<td><a href="tfm.html">tfm</a></td>
<td>
Displays full documentation for an application
</td>
</tr>
<tr>
<td><a href="tfscan.html">tfscan</a></td>
<td>
Identify transcription factor binding sites in DNA sequences
</td>
</tr>
<tr>
<td><a href="tmap.html">tmap</a></td>
<td>
Predict and plot transmembrane segments in protein sequences
</td>
</tr>
<tr>
<td><a href="tranalign.html">tranalign</a></td>
<td>
Generate an alignment of nucleic coding regions from aligned proteins
</td>
</tr>
<tr>
<td><a href="transeq.html">transeq</a></td>
<td>
Translate nucleic acid sequences
</td>
</tr>
<tr>
<td><a href="trimest.html">trimest</a></td>
<td>
Remove poly-A tails from nucleotide sequences
</td>
</tr>
<tr>
<td><a href="trimseq.html">trimseq</a></td>
<td>
Remove unwanted characters from start and end of sequence(s)
</td>
</tr>
<tr>
<td><a href="trimspace.html">trimspace</a></td>
<td>
Remove extra whitespace from an ASCII text file
</td>
</tr>
<tr>
<td><a href="twofeat.html">twofeat</a></td>
<td>
Finds neighbouring pairs of features in sequence(s)
</td>
</tr>
<tr>
<td><a href="union.html">union</a></td>
<td>
Concatenate multiple sequences into a single sequence
</td>
</tr>
<tr>
<td><a href="vectorstrip.html">vectorstrip</a></td>
<td>
Removes vectors from the ends of nucleotide sequence(s)
</td>
</tr>
<tr>
<td><a href="water.html">water</a></td>
<td>
Smith-Waterman local alignment of sequences
</td>
</tr>
<tr>
<td><a href="whichdb.html">whichdb</a></td>
<td>
Search all sequence databases for an entry and retrieve it
</td>
</tr>
<tr>
<td><a href="wobble.html">wobble</a></td>
<td>
Plot third base position variability in a nucleotide sequence
</td>
</tr>
<tr>
<td><a href="wordcount.html">wordcount</a></td>
<td>
Count and extract unique words in molecular sequence(s)
</td>
</tr>
<tr>
<td><a href="wordfinder.html">wordfinder</a></td>
<td>
Match large sequences against one or more other sequences
</td>
</tr>
<tr>
<td><a href="wordmatch.html">wordmatch</a></td>
<td>
Finds regions of identity (exact matches) of two sequences
</td>
</tr>
<tr>
<td><a href="wossname.html">wossname</a></td>
<td>
Finds programs by keywords in their short description
</td>
</tr>
<tr>
<td><a href="yank.html">yank</a></td>
<td>
Add a sequence reference (a full USA) to a list file
</td>
</tr>


</table>




</BODY>
</HTML>
