<HTML>
<HEAD>
  <TITLE>
  EMBOSS: jaspscan
  </TITLE>
</HEAD>
<BODY BGCOLOR="#FFFFFF" text="#000000">

<table align=center border=0 cellspacing=0 cellpadding=0>
<tr><td valign=top>
<A HREF="/" ONMOUSEOVER="self.status='Go to the EMBOSS home page';return true"><img border=0 src="/images/emboss_icon.jpg" alt="" width=150 height=48></a>
</td>
<td align=left valign=middle>
<b><font size="+6">
jaspscan
</font></b>
</td></tr>
</table>
<br>&nbsp;
<p>


<H2>
Wiki
</H2>

The master copies of EMBOSS documentation are available
at <a href="http://emboss.open-bio.org/wiki/Appdocs">
http://emboss.open-bio.org/wiki/Appdocs</a>
on the EMBOSS Wiki.

<p>
Please help by correcting and extending the Wiki pages.

<H2>
    Function
</H2>
Scan DNA sequences for transcription factors
<!--
DON'T WRITE ANYTHING HERE.
IT IS DONE FOR YOU.
-->




<H2>
    Description
</H2>

<b>jaspscan</b> scans one or more DNA sequences for transcription factor binding sites from the JASPAR database. Matches are searched for using fast sequence word-matching, optionally allowing mismatches.  Because the binding sites are so small, there will be many spurious (false positive) matches.</p>
<p>An output file is written with information on the matches, including sequence ID and accession number, the start and end positions of the match in an input sequence and the sequence of the region where a match has been found. Binding factor information, where available, is given at the end of the matches for each matching entry.</p>

















<H2>
    Usage
</H2>

<!--  
	Example usage, as run from the command-line.
        Many examples illustrating different behaviours is good.
 -->

Here is a sample session with <b>jaspscan</b>
<p>

<p>
<table width="90%"><tr><td bgcolor="#CCFFFF"><pre>

% <b>jaspscan -both </b>
Scan DNA sequences for transcription factors
Input nucleotide sequence(s): <b>tembl:m23100</b>
Jaspar matrix set
         C : Core
         F : Fam
         P : Phylofacts
         N : CNE
         O : POLII
         S : SPLICE
         B : PBM
         L : PBM_HLH
         H : PBM_HOMEO
Matrix set [C]: <b></b>
Comma separated matrix list [all]: <b>ma0079.1</b>
Threshold percentage [80.0]: <b></b>
Output report [m23100.jaspscan]: <b></b>

</pre></td></tr></table><p>
<p>
<a href="#input.1">Go to the input files for this example</a><br><a href="#output.1">Go to the output files for this example</a><p><p>



<H2>
    Command line arguments
</H2>

<table CELLSPACING=0 CELLPADDING=3 BGCOLOR="#f5f5ff" ><tr><td>
<pre>
Scan DNA sequences for transcription factors
Version: EMBOSS:6.6.0.0

   Standard (Mandatory) qualifiers:
  [-sequence]          seqall     Nucleotide sequence(s) filename and optional
                                  format, or reference (input USA)
   -menu               menu       [C] Matrix set (Values: C (Core); F (Fam); P
                                  (Phylofacts); N (CNE); O (POLII); S
                                  (SPLICE); B (PBM); L (PBM_HLH); H
                                  (PBM_HOMEO))
   -matrices           string     [all] The name 'all' reads in all matrix
                                  files from the selected
                                  JASPAR matrix set. You can specify
                                  individual matrices by giving
                                  their names with commas between then, such
                                  as:
                                  'ma0001.1,ma0015*'.
                                  The case of the names is not important. You
                                  can specify a file of
                                  matrix names to read in by giving the name
                                  of the file holding the
                                  matrix names with a '@' character in front
                                  of it, for example,
                                  '@matrix.list'.
                                  Blank lines and lines starting with a hash
                                  character or '!' are ignored and all other
                                  lines are concatenated together with a comma
                                  character ',' and then treated as the list
                                  of enzymes to search
                                  for.
                                  An example of a file of matrix names is:
                                  ! my matrices
                                  ma0001.1, ma0002.1
                                  ! other matrices
                                  ma0010.1
                                  ma0032*
                                  ma0053.1 (Any string)
   -threshold          float      [80.0] If the matrix score is greater than
                                  or equal to this percentage
                                  then a hit will be reported (Any numeric
                                  value)
  [-outfile]           report     [*.jaspscan] Output report file name
                                  (default -rformat seqtable)

   Additional (Optional) qualifiers:
   -exclude            string     The names of any matrices to exclude from
                                  the 'matrices'
                                  list. Matrices are specified in the same way
                                  as for the
                                  selection list. (Any string)
   -both               boolean    [N] If set then both the forward and reverse
                                  strands are searched

   Advanced (Unprompted) qualifiers: (none)
   Associated qualifiers:

   "-sequence" associated qualifiers
   -sbegin1            integer    Start of each sequence to be used
   -send1              integer    End of each sequence to be used
   -sreverse1          boolean    Reverse (if DNA)
   -sask1              boolean    Ask for begin/end/reverse
   -snucleotide1       boolean    Sequence is nucleotide
   -sprotein1          boolean    Sequence is protein
   -slower1            boolean    Make lower case
   -supper1            boolean    Make upper case
   -scircular1         boolean    Sequence is circular
   -squick1            boolean    Read id and sequence only
   -sformat1           string     Input sequence format
   -iquery1            string     Input query fields or ID list
   -ioffset1           integer    Input start position offset
   -sdbname1           string     Database name
   -sid1               string     Entryname
   -ufo1               string     UFO features
   -fformat1           string     Features format
   -fopenfile1         string     Features file name

   "-outfile" associated qualifiers
   -rformat2           string     Report format
   -rname2             string     Base file name
   -rextension2        string     File name extension
   -rdirectory2        string     Output directory
   -raccshow2          boolean    Show accession number in the report
   -rdesshow2          boolean    Show description in the report
   -rscoreshow2        boolean    Show the score in the report
   -rstrandshow2       boolean    Show the nucleotide strand in the report
   -rusashow2          boolean    Show the full USA in the report
   -rmaxall2           integer    Maximum total hits to report
   -rmaxseq2           integer    Maximum hits to report for one sequence

   General qualifiers:
   -auto               boolean    Turn off prompts
   -stdout             boolean    Write first file to standard output
   -filter             boolean    Read first file from standard input, write
                                  first file to standard output
   -options            boolean    Prompt for standard and additional values
   -debug              boolean    Write debug output to program.dbg
   -verbose            boolean    Report some/full command line options
   -help               boolean    Report command line options and exit. More
                                  information on associated and general
                                  qualifiers can be found with -help -verbose
   -warning            boolean    Report warnings
   -error              boolean    Report errors
   -fatal              boolean    Report fatal errors
   -die                boolean    Report dying program messages
   -version            boolean    Report version number and exit

</pre>
</td></tr></table>
<P>
<table border cellspacing=0 cellpadding=3 bgcolor="#ccccff">
<tr bgcolor="#FFFFCC">
<th align="left">Qualifier</th>
<th align="left">Type</th>
<th align="left">Description</th>
<th align="left">Allowed values</th>
<th align="left">Default</th>
</tr>

<tr bgcolor="#FFFFCC">
<th align="left" colspan=5>Standard (Mandatory) qualifiers</th>
</tr>

<tr bgcolor="#FFFFCC">
<td>[-sequence]<br>(Parameter 1)</td>
<td>seqall</td>
<td>Nucleotide sequence(s) filename and optional format, or reference (input USA)</td>
<td>Readable sequence(s)</td>
<td><b>Required</b></td>
</tr>

<tr bgcolor="#FFFFCC">
<td>-menu</td>
<td>list</td>
<td>Matrix set</td>
<td><table><tr><td>C</td> <td><i>(Core)</i></td></tr><tr><td>F</td> <td><i>(Fam)</i></td></tr><tr><td>P</td> <td><i>(Phylofacts)</i></td></tr><tr><td>N</td> <td><i>(CNE)</i></td></tr><tr><td>O</td> <td><i>(POLII)</i></td></tr><tr><td>S</td> <td><i>(SPLICE)</i></td></tr><tr><td>B</td> <td><i>(PBM)</i></td></tr><tr><td>L</td> <td><i>(PBM_HLH)</i></td></tr><tr><td>H</td> <td><i>(PBM_HOMEO)</i></td></tr></table></td>
<td>C</td>
</tr>

<tr bgcolor="#FFFFCC">
<td>-matrices</td>
<td>string</td>
<td>The name 'all' reads in all matrix files from the selected
JASPAR matrix set. You can specify individual matrices by giving
their names with commas between then, such as:
'ma0001.1,ma0015*'.
The case of the names is not important. You can specify a file of
matrix names to read in by giving the name of the file holding the
matrix names with a '@' character in front of it, for example,
'@matrix.list'.
Blank lines and lines starting with a hash character or '!' are ignored and all other lines are concatenated together with a comma
character ',' and then treated as the list of enzymes to search
for.
An example of a file of matrix names is:
! my matrices
ma0001.1, ma0002.1
! other matrices
ma0010.1
ma0032*
ma0053.1</td>
<td>Any string</td>
<td>all</td>
</tr>

<tr bgcolor="#FFFFCC">
<td>-threshold</td>
<td>float</td>
<td>If the matrix score is greater than or equal to this percentage
then a hit will be reported</td>
<td>Any numeric value</td>
<td>80.0</td>
</tr>

<tr bgcolor="#FFFFCC">
<td>[-outfile]<br>(Parameter 2)</td>
<td>report</td>
<td>Output report file name</td>
<td>(default -rformat seqtable)</td>
<td><i>&lt;*&gt;</i>.jaspscan</td>
</tr>

<tr bgcolor="#FFFFCC">
<th align="left" colspan=5>Additional (Optional) qualifiers</th>
</tr>

<tr bgcolor="#FFFFCC">
<td>-exclude</td>
<td>string</td>
<td>The names of any matrices to exclude from the 'matrices'
list. Matrices are specified in the same way as for the
selection list.</td>
<td>Any string</td>
<td>&nbsp;</td>
</tr>

<tr bgcolor="#FFFFCC">
<td>-both</td>
<td>boolean</td>
<td>If set then both the forward and reverse strands are searched</td>
<td>Boolean value Yes/No</td>
<td>No</td>
</tr>

<tr bgcolor="#FFFFCC">
<th align="left" colspan=5>Advanced (Unprompted) qualifiers</th>
</tr>

<tr>
<td colspan=5>(none)</td>
</tr>

<tr bgcolor="#FFFFCC">
<th align="left" colspan=5>Associated qualifiers</th>
</tr>

<tr bgcolor="#FFFFCC">
<td align="left" colspan=5>"-sequence" associated seqall qualifiers
</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -sbegin1<br>-sbegin_sequence</td>
<td>integer</td>
<td>Start of each sequence to be used</td>
<td>Any integer value</td>
<td>0</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -send1<br>-send_sequence</td>
<td>integer</td>
<td>End of each sequence to be used</td>
<td>Any integer value</td>
<td>0</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -sreverse1<br>-sreverse_sequence</td>
<td>boolean</td>
<td>Reverse (if DNA)</td>
<td>Boolean value Yes/No</td>
<td>N</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -sask1<br>-sask_sequence</td>
<td>boolean</td>
<td>Ask for begin/end/reverse</td>
<td>Boolean value Yes/No</td>
<td>N</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -snucleotide1<br>-snucleotide_sequence</td>
<td>boolean</td>
<td>Sequence is nucleotide</td>
<td>Boolean value Yes/No</td>
<td>N</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -sprotein1<br>-sprotein_sequence</td>
<td>boolean</td>
<td>Sequence is protein</td>
<td>Boolean value Yes/No</td>
<td>N</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -slower1<br>-slower_sequence</td>
<td>boolean</td>
<td>Make lower case</td>
<td>Boolean value Yes/No</td>
<td>N</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -supper1<br>-supper_sequence</td>
<td>boolean</td>
<td>Make upper case</td>
<td>Boolean value Yes/No</td>
<td>N</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -scircular1<br>-scircular_sequence</td>
<td>boolean</td>
<td>Sequence is circular</td>
<td>Boolean value Yes/No</td>
<td>N</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -squick1<br>-squick_sequence</td>
<td>boolean</td>
<td>Read id and sequence only</td>
<td>Boolean value Yes/No</td>
<td>N</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -sformat1<br>-sformat_sequence</td>
<td>string</td>
<td>Input sequence format</td>
<td>Any string</td>
<td>&nbsp;</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -iquery1<br>-iquery_sequence</td>
<td>string</td>
<td>Input query fields or ID list</td>
<td>Any string</td>
<td>&nbsp;</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -ioffset1<br>-ioffset_sequence</td>
<td>integer</td>
<td>Input start position offset</td>
<td>Any integer value</td>
<td>0</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -sdbname1<br>-sdbname_sequence</td>
<td>string</td>
<td>Database name</td>
<td>Any string</td>
<td>&nbsp;</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -sid1<br>-sid_sequence</td>
<td>string</td>
<td>Entryname</td>
<td>Any string</td>
<td>&nbsp;</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -ufo1<br>-ufo_sequence</td>
<td>string</td>
<td>UFO features</td>
<td>Any string</td>
<td>&nbsp;</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -fformat1<br>-fformat_sequence</td>
<td>string</td>
<td>Features format</td>
<td>Any string</td>
<td>&nbsp;</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -fopenfile1<br>-fopenfile_sequence</td>
<td>string</td>
<td>Features file name</td>
<td>Any string</td>
<td>&nbsp;</td>
</tr>

<tr bgcolor="#FFFFCC">
<td align="left" colspan=5>"-outfile" associated report qualifiers
</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -rformat2<br>-rformat_outfile</td>
<td>string</td>
<td>Report format</td>
<td>Any string</td>
<td>seqtable</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -rname2<br>-rname_outfile</td>
<td>string</td>
<td>Base file name</td>
<td>Any string</td>
<td>&nbsp;</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -rextension2<br>-rextension_outfile</td>
<td>string</td>
<td>File name extension</td>
<td>Any string</td>
<td>&nbsp;</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -rdirectory2<br>-rdirectory_outfile</td>
<td>string</td>
<td>Output directory</td>
<td>Any string</td>
<td>&nbsp;</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -raccshow2<br>-raccshow_outfile</td>
<td>boolean</td>
<td>Show accession number in the report</td>
<td>Boolean value Yes/No</td>
<td>N</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -rdesshow2<br>-rdesshow_outfile</td>
<td>boolean</td>
<td>Show description in the report</td>
<td>Boolean value Yes/No</td>
<td>N</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -rscoreshow2<br>-rscoreshow_outfile</td>
<td>boolean</td>
<td>Show the score in the report</td>
<td>Boolean value Yes/No</td>
<td>Y</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -rstrandshow2<br>-rstrandshow_outfile</td>
<td>boolean</td>
<td>Show the nucleotide strand in the report</td>
<td>Boolean value Yes/No</td>
<td>Y</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -rusashow2<br>-rusashow_outfile</td>
<td>boolean</td>
<td>Show the full USA in the report</td>
<td>Boolean value Yes/No</td>
<td>N</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -rmaxall2<br>-rmaxall_outfile</td>
<td>integer</td>
<td>Maximum total hits to report</td>
<td>Any integer value</td>
<td>0</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -rmaxseq2<br>-rmaxseq_outfile</td>
<td>integer</td>
<td>Maximum hits to report for one sequence</td>
<td>Any integer value</td>
<td>0</td>
</tr>

<tr bgcolor="#FFFFCC">
<th align="left" colspan=5>General qualifiers</th>
</tr>

<tr bgcolor="#FFFFCC">
<td> -auto</td>
<td>boolean</td>
<td>Turn off prompts</td>
<td>Boolean value Yes/No</td>
<td>N</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -stdout</td>
<td>boolean</td>
<td>Write first file to standard output</td>
<td>Boolean value Yes/No</td>
<td>N</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -filter</td>
<td>boolean</td>
<td>Read first file from standard input, write first file to standard output</td>
<td>Boolean value Yes/No</td>
<td>N</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -options</td>
<td>boolean</td>
<td>Prompt for standard and additional values</td>
<td>Boolean value Yes/No</td>
<td>N</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -debug</td>
<td>boolean</td>
<td>Write debug output to program.dbg</td>
<td>Boolean value Yes/No</td>
<td>N</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -verbose</td>
<td>boolean</td>
<td>Report some/full command line options</td>
<td>Boolean value Yes/No</td>
<td>Y</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -help</td>
<td>boolean</td>
<td>Report command line options and exit. More information on associated and general qualifiers can be found with -help -verbose</td>
<td>Boolean value Yes/No</td>
<td>N</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -warning</td>
<td>boolean</td>
<td>Report warnings</td>
<td>Boolean value Yes/No</td>
<td>Y</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -error</td>
<td>boolean</td>
<td>Report errors</td>
<td>Boolean value Yes/No</td>
<td>Y</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -fatal</td>
<td>boolean</td>
<td>Report fatal errors</td>
<td>Boolean value Yes/No</td>
<td>Y</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -die</td>
<td>boolean</td>
<td>Report dying program messages</td>
<td>Boolean value Yes/No</td>
<td>Y</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -version</td>
<td>boolean</td>
<td>Report version number and exit</td>
<td>Boolean value Yes/No</td>
<td>N</td>
</tr>

</table>

<!--
DON'T WRITE ANYTHING HERE.
IT IS DONE FOR YOU.
-->








<H2>
    Input file format
</H2>


<b>jaspscan</b> reads one or more nucleotide sequences.


<p>
<p>

The input is a standard EMBOSS sequence query (also known as a 'USA').

<p>
Major sequence database sources defined as standard in EMBOSS
installations include srs:embl, srs:uniprot and ensembl

<p>

Data can also be read from sequence output in any supported format
written by an EMBOSS or third-party application.

<p>
The input format can be specified by using the
command-line qualifier <tt>-sformat xxx</tt>, where 'xxx' is replaced
by the name of the required format.  The available format names are:
gff (gff3), gff2, embl (em), genbank (gb, refseq), ddbj, refseqp, pir
(nbrf), swissprot (swiss, sw), dasgff and debug.

<p>

See:
<A href="http://emboss.sf.net/docs/themes/SequenceFormats.html">
http://emboss.sf.net/docs/themes/SequenceFormats.html</A>
for further information on sequence formats.

<p>

<a name="input.1"></a>
<h3>Input files for usage example </h3>

'tembl:m23100' is a sequence entry in the example nucleic acid database 'tembl'
<p>
<p><h3>Database entry: tembl:m23100</h3>
<table width="90%"><tr><td bgcolor="#FFCCFF">
<pre>
ID   M23100; SV 1; linear; genomic DNA; STD; HUM; 2085 BP.
XX
AC   M23100; M32822;
XX
DT   06-JUL-1989 (Rel. 20, Created)
DT   14-NOV-2006 (Rel. 89, Last updated, Version 10)
XX
DE   Human insulin receptor (INSR) gene, exon 1, clones lambda-hINSR-(1-13).
XX
KW   Alu repeat; insulin receptor.
XX
OS   Homo sapiens (human)
OC   Eukaryota; Metazoa; Chordata; Craniata; Vertebrata; Euteleostomi; Mammalia;
OC   Eutheria; Euarchontoglires; Primates; Haplorrhini; Catarrhini; Hominidae;
OC   Homo.
XX
RN   [1]
RP   1-1933
RX   DOI; 10.1073/pnas.86.1.114.
RX   PUBMED; 2911561.
RA   Seino S., Seino M., Nishi S., Bell G.I.;
RT   "Structure of the human insulin receptor gene and characterization of its
RT   promoter";
RL   Proc. Natl. Acad. Sci. U.S.A. 86(1):114-118(1989).
XX
RN   [2]
RP   1-2085
RX   PUBMED; 2210055.
RA   Seino S., Seino M., Bell G.I.;
RT   "Human insulin-receptor gene. Partial sequence and amplification of exons
RT   by polymerase chain reaction";
RL   Diabetes 39(1):123-128(1990).
XX
DR   Ensembl-Gn; ENSG00000171105; Homo_sapiens.
DR   Ensembl-Tr; ENST00000302850; Homo_sapiens.
XX
CC   Draft entry and computer-readable sequence for [1] kindly submitted
CC   by G.Bell, 14-MAR-1990.
XX
FH   Key             Location/Qualifiers
FH
FT   source          1..2085
FT                   /organism="Homo sapiens"
FT                   /map="19p13.3-p13.2"
FT                   /mol_type="genomic DNA"
FT                   /dev_stage="foetus"
FT                   /tissue_type="liver"
FT                   /db_xref="taxon:9606"
FT   repeat_region   &lt;1..76
FT                   /rpt_family="Alu"


<font color=red>  [Part of this file has been deleted for brevity]</font>

FT                   /number=1
FT                   /note="alternative exon 1; G00-119-352"
FT   exon            1548..1923
FT                   /gene="INSR"
FT                   /number=1
FT                   /note="alternative exon 1; G00-119-352"
FT   sig_peptide     1824..1904
FT                   /gene="INSR"
FT                   /note="G00-119-352"
FT   intron          1924..&gt;2085
FT                   /gene="INSR"
FT                   /number=1
FT                   /note="G00-119-352"
XX
SQ   Sequence 2085 BP; 417 A; 631 C; 702 G; 335 T; 0 other;
     agatctggcc attgcactcc agcctgggca acagagaaaa actccatcta aaaaaaaaaa        60
     aaaaaaaaaa aaaaaacaga gagagagaga gagagagaga gaaggaaacg gaactggggg       120
     gaggatttgc aaaaatatgg ttagggatgg cacttcagag atgaagccat cctggagtgt       180
     tacgggcaag ggaaatgctg gggcaaagcc ccagaggcag gaataggttt ggcctgttgc       240
     atgaacagtg ggtccagctc ctagcaaact gtttattgaa tgaaagaaga atgaatgcct       300
     tgggtctagg gttgtgctgg gcgctttctt aagttttctt tcccgggtac ctccccagaa       360
     ctggcatgca ggtattatta aacccattac acaagtgaaa ctggcccaga gacagaaaag       420
     tccctggtcc aagaccacac aggagtgagg ggtggaggaa ccctcctccc attgagttct       480
     ggctttccta tactgaaagc cccttcctct cctgcagtaa ggtaggtgga accgctgtcc       540
     cgccttgttg gtgaatgtcg ttgctagact tcagacacat acaggctggt ctgctgaaaa       600
     tcagagatgt ccacctgcgc cctattcgag gtctccggcg tcttctttgg cgtcgtcttt       660
     gccctttcag aagcgtctgc acatttttcc aggtgtcatt tctccaactt gaacacaggg       720
     agcgcactgg gcacgcgggc acgtggctgt ccccaggggc ctggcttggg tctcgcccct       780
     gggccggggc gcacgcgcgg gcgggacatc tgggggcgcc cacgcgctct gggacgagtg       840
     tcgctggcca ggcccggact gaggaaaggc gagtgagaca ctactcgcct ggggtgcaaa       900
     atttaaggga gtgaaaaaaa aaaaaaaaga aagaaaccaa aaccacctcg agtcaccaaa       960
     ataaacattt taatgcagta ttttttaaaa aatcaacagg aatcctccaa agcccactat      1020
     gaacaaaata gcaaaatggt agagaaagga tctgtgccgc tgcgtcgggc ctgtggggcg      1080
     cctccggggg tctgaaactg gaggagactc ggggctgtag ggcgcgcgga tctggggcgc      1140
     gccctcggtc ccggcgcgcc cagggcctcc cgcgcggggc ccggcacagg gaggcgggga      1200
     ggcgggcggg gcggggcggg accgggcggc acctccctcc cctgcaagct ttccctccct      1260
     ctcctgggcc tctcccgggc gcagagtccc ttcctaggcc agatccgcgc cgccttttcc      1320
     cgcggcccgc acggggccca gctgacgggc cgcgttgttt acgggccgga gcagccctct      1380
     ctcccgccgc ccgcccgcca cccgccagcc caggtgcccg cccgccagtc agctagtccg      1440
     tcggtccgcg cgtccctctg tcccggagcc cgcagatcgc gacccagagc gcgcggggcc      1500
     gagagccgag agacagtccc gggcgcagcg cggagctccg ggccccgaga tcctgggacg      1560
     gggcccgggc cgcagcggcc ggggggtcgg ggccaccacc gcaagggcct ccgctcagta      1620
     tttgtagctg gcgaagccgc gcgcgccctt cccggggctg cctctgggcc ctccccggca      1680
     ggggggctgc ggcccgcggg tcgcgggcgt ggaagagaag gacgcgcggc ccccagcgcc      1740
     tcttgggtgg ccgcctcgga gcatgacccc cgcgggccag cgccgcgcgc tctgatccga      1800
     ggagaccccg cgctcccgca gccatgggca ccgggggccg gcggggagcg gcggccgcgc      1860
     cgctgctggt ggcggtggcc gcgctgctac tgggcgccgc gggccacctg taccccggag      1920
     agggtgagtc tgggggcgcg ggcgtgggcg gggagcgccg cgatggggag aggaccccac      1980
     ccaagccaaa atcgatcccc cgcttgtgga ctgagaaccc tccccagggg cggggggcgg      2040
     tggccaggac ggtagctcct gcatcgcgta gggggagcgg gaagc                      2085
//
</pre>
</td></tr></table><p>





<H2>
    Output file format
</H2>


<p>

The output is a standard EMBOSS report file. 

<p>

The results can be output in one of several styles by using the
command-line qualifier <tt>-rformat xxx</tt>, where 'xxx' is replaced
by the name of the required format.  The available format names are:
embl, genbank, gff, pir, swiss, dasgff, debug, listfile, dbmotif,
diffseq, draw, restrict, excel, feattable, motif, nametable, regions,
seqtable, simple, srs, table, tagseq.

<p>

See:
<A href="http://emboss.sf.net/docs/themes/ReportFormats.html">
http://emboss.sf.net/docs/themes/ReportFormats.html</A>
for further information on report formats.

<p>

<p>
By default the report is in 'seqtable' format.

<p>


<a name="output.1"></a>
<h3>Output files for usage example </h3>
<p><h3>File: m23100.jaspscan</h3>
<table width="90%"><tr><td bgcolor="#CCFFCC">
<pre>
########################################
# Program: jaspscan
# Rundate: Mon 15 Jul 2013 12:00:00
# Commandline: jaspscan
#    -both
#    -sequence tembl:m23100
#    -matrices ma0079.1
# Report_format: seqtable
# Report_file: m23100.jaspscan
########################################

#=======================================
#
# Sequence: M23100     from: 1   to: 2085
# HitCount: 18
#
# Database scanned: JASPAR_CORE  Threshold: 80.000
#
#=======================================

  Start     End  Strand Score_Percent ID       Name   Species Class             Supergroup Family Pazar_TF_ID Protein_Seq Experiment Source_PMID Info_Content Closest_Jaspar Closest_Transfac MCS_Score Built_From Sequence
    117     126       +        82.143 MA0079.1 SP1    9606    Zinc-coordinating .          .      .           P08047      SELEX      2192357            9.719 .              .                        . .          gggggaggat
   1191    1200       +        89.286 MA0079.1 SP1    9606    Zinc-coordinating .          .      .           P08047      SELEX      2192357            9.719 .              .                        . .          gaggcgggga
   1203    1212       +        83.929 MA0079.1 SP1    9606    Zinc-coordinating .          .      .           P08047      SELEX      2192357            9.719 .              .                        . .          cgggcggggc
   1208    1217       +        89.286 MA0079.1 SP1    9606    Zinc-coordinating .          .      .           P08047      SELEX      2192357            9.719 .              .                        . .          ggggcggggc
   1213    1222       +        80.357 MA0079.1 SP1    9606    Zinc-coordinating .          .      .           P08047      SELEX      2192357            9.719 .              .                        . .          ggggcgggac
   1704    1713       +        82.143 MA0079.1 SP1    9606    Zinc-coordinating .          .      .           P08047      SELEX      2192357            9.719 .              .                        . .          cgggcgtgga
   1838    1847       +        82.143 MA0079.1 SP1    9606    Zinc-coordinating .          .      .           P08047      SELEX      2192357            9.719 .              .                        . .          ccggcgggga
   1933    1942       +        82.143 MA0079.1 SP1    9606    Zinc-coordinating .          .      .           P08047      SELEX      2192357            9.719 .              .                        . .          ggggcgcggg
   1945    1954       +        89.286 MA0079.1 SP1    9606    Zinc-coordinating .          .      .           P08047      SELEX      2192357            9.719 .              .                        . .          tgggcgggga
   2027    2036       +        89.286 MA0079.1 SP1    9606    Zinc-coordinating .          .      .           P08047      SELEX      2192357            9.719 .              .                        . .          ggggcggggg
   2032    2041       +        87.500 MA0079.1 SP1    9606    Zinc-coordinating .          .      .           P08047      SELEX      2192357            9.719 .              .                        . .          ggggggcggt
   2034    2043       +        82.143 MA0079.1 SP1    9606    Zinc-coordinating .          .      .           P08047      SELEX      2192357            9.719 .              .                        . .          ggggcggtgg
   1252    1261       -        80.357 MA0079.1 SP1    9606    Zinc-coordinating .          .      .           P08047      SELEX      2192357            9.719 .              .                        . .          gagggaggga
   1304    1313       -        80.357 MA0079.1 SP1    9606    Zinc-coordinating .          .      .           P08047      SELEX      2192357            9.719 .              .                        . .          gcggcgcgga
   1805    1814       -        80.357 MA0079.1 SP1    9606    Zinc-coordinating .          .      .           P08047      SELEX      2192357            9.719 .              .                        . .          agcgcggggt
   1974    1983       -        89.286 MA0079.1 SP1    9606    Zinc-coordinating .          .      .           P08047      SELEX      2192357            9.719 .              .                        . .          tgggtggggt
   2016    2025       -        80.357 MA0079.1 SP1    9606    Zinc-coordinating .          .      .           P08047      SELEX      2192357            9.719 .              .                        . .          ggggagggtt
   2017    2026       -        87.500 MA0079.1 SP1    9606    Zinc-coordinating .          .      .           P08047      SELEX      2192357            9.719 .              .                        . .          tggggagggt

#---------------------------------------
#---------------------------------------
</pre>
</td></tr></table><p>





<H2>
    Data files
</H2>

<p>
EMBOSS data files are distributed with the application and stored
in the standard EMBOSS data directory, which is defined
by the EMBOSS environment variable EMBOSS_DATA.

<p>

To see the available EMBOSS data files, run:
<p>
<pre>
% embossdata -showall
</pre>
<p>
To fetch one of the data files (for example 'Exxx.dat') into your
current directory for you to inspect or modify, run:

<pre>

% embossdata -fetch -file Exxx.dat

</pre>
<p>

Users can provide their own data files in their own directories.
Project specific files can be put in the current directory, or for
tidier directory listings in a subdirectory called
".embossdata". Files for all EMBOSS runs can be put in the user's home
directory, or again in a subdirectory called ".embossdata".

<p>
The directories are searched in the following order:

<ul>
   <li> . (your current directory)
   <li> .embossdata (under your current directory)
   <li> ~/ (your home directory)
   <li> ~/.embossdata
</ul>
<p>
<p>
The EMBOSS JASPAR transcription site files are stored in
directory 'data/JASPAR/*' under the EMBOSS installation directory.

<p>

These files must first be set up using the program <a
href="jaspextract.html">'<b>jaspextract</b>'</a>.  Running
'jaspextract' may be the job of your system manager.

<p>
      
The data files are stored in the JASPAR directory of the standard EMBOSS
data directory. The subdirectory names are:

<ul>
<li> JASPAR_CORE      Core files
<li> JASPAR_FAM       Family files
<li> JASPAR_PHYLOFACT Phylogeny files
</ul>
	 
The column information is described on the JASPAR home page <a
href="http://jaspar.genereg.net/">http://jaspar.genereg.net/</a>
	 






<H2>
    Notes
</H2>

<!-- 
        Restrictions.
        Interesting behaviour.
        Useful things you can do with this program.
   -->

None.







<H2>
    References
</H2>

<OL>

<LI>DNA binding sites: representation and discovery Bioinformatics. 2000
      Jan;16(1):16-23

<LI>Applied bioinformatics for the identification of regulatory elements
      Nat Rev Genet. 2004 Apr;5(4):276-87 

</OL>



<H2>
    Warnings
</H2>

<!-- 
        Potentially stupid things the program will let you do.
   -->

None.







<H2>
    Diagnostic Error Messages
</H2>

<!-- 
        Error messages specific to this program, eg:
        "FATAL xxx" - means you have not set up the xxx data using program <b>prog</b>.<p>
   -->

None.







<H2>
    Exit status
</H2>

<!-- 
        Description of the exit status for various error conditions
   -->

It always exits with status 0.








<H2>
    Known bugs
</H2>


<!-- 
        Bugs noted but not yet fixed.
   -->

None.








<!--
<H2>
    See also
</H2>
-->
<h2><a name="See also">See also</a></h2>
<table border cellpadding=4 bgcolor="#FFFFF0">
<tr><th>Program name</th>
<th>Description</th></tr>
<tr>
<td><a href="marscan.html">marscan</a></td>
<td>Find matrix/scaffold recognition (MRS) signatures in DNA sequences</td>
</tr>

<tr>
<td><a href="tfscan.html">tfscan</a></td>
<td>Identify transcription factor binding sites in DNA sequences</td>
</tr>

</table>
<!-- 
        Add any comments about other associated programs (to prepare
        data files?) that seealso doesn't find. 
   -->










<H2>
    Author(s)
</H2>

Alan Bleasby 
<br>
European Bioinformatics Institute, Wellcome Trust Genome Campus, Hinxton, Cambridge CB10 1SD, UK

<p>
Please report all bugs to the EMBOSS bug team (emboss-bug&nbsp;&copy;&nbsp;emboss.open-bio.org) not to the original author.

<H2>
    History
</H2>
<!--
        Date written and what changes have been made go in this file.
   -->
   Completed 23rd July 2007



<H2>
    Target users
</H2>
<!--
        For general users, use this text
   -->
This program is intended to be used by everyone and everything, from naive users to embedded scripts.

<H2>
    Comments
</H2>
<!--
        User/developer/other comments go in this file.
   -->
None


</BODY>
</HTML>

