<HTML>
<HEAD>
  <TITLE>
  EMBOSS: makenucseq
  </TITLE>
</HEAD>
<BODY BGCOLOR="#FFFFFF" text="#000000">

<table align=center border=0 cellspacing=0 cellpadding=0>
<tr><td valign=top>
<A HREF="/" ONMOUSEOVER="self.status='Go to the EMBOSS home page';return true"><img border=0 src="emboss_icon.jpg" alt="" width=150 height=48></a>
</td>
<td align=left valign=middle>
<b><font size="+6">
makenucseq
</font></b>
</td></tr>
</table>
<br>&nbsp;
<p>


<H2>
Wiki
</H2>

The master copies of EMBOSS documentation are available
at <a href="http://emboss.open-bio.org/wiki/Appdocs">
http://emboss.open-bio.org/wiki/Appdocs</a>
on the EMBOSS Wiki.

<p>
Please help by correcting and extending the Wiki pages.

<H2>
    Function
</H2>
Create random nucleotide sequences
<!--
DON'T WRITE ANYTHING HERE.
IT IS DONE FOR YOU.
-->




<H2>
    Description
</H2>



<p><b>makenucseq</b> writes an output file with a set of random
nucleotide sequences. The sequence composition is defined from reading
a codon usage file: the sequences are created with triplet frequencies
matching those given in the file, with the end trimmed to be in the
correct reading frame.  The number of sequences to create and length
of each sequence is specified.  Optionally, a user-defined string may
be inserted into each output sequence at a specified position</p>



<H2>
    Algorithm
</H2>

<!-- 
        Algorithms
	  Pseudocode/Vague description/Handwaving/web links to useful pages
  -->











<H2>
    Usage
</H2>

<!--  
	Example usage, as run from the command-line.
        Many examples illustrating different behaviours is good.
 -->

<b>Here is a sample session with makenucseq</b>
<p>

<p>
<table width="90%"><tr><td bgcolor="#CCFFFF"><pre>

% <b>makenucseq </b>
Create random nucleotide sequences
Codon usage file (optional): <b></b>
Number of sequences created [100]: <b></b>
Length of each sequence [100]: <b></b>
nucleotide output sequence(s) [makeseq.fasta]: <b></b>

</pre></td></tr></table><p>
<p>
<a href="#output.1">Go to the output files for this example</a><p><p>
<p>
<b>Example 2</b>
<p>
Providing a codon usage file specifies the sequence composition. This Pseudomonas aeruginosa file specifies a high GC content. 
<p>

<p>
<table width="90%"><tr><td bgcolor="#CCFFFF"><pre>

% <b>makenucseq </b>
Create random nucleotide sequences
Codon usage file (optional): <b>Epseae.cut</b>
Number of sequences created [100]: <b></b>
Length of each sequence [100]: <b></b>
nucleotide output sequence(s) [makeseq.fasta]: <b></b>

</pre></td></tr></table><p>
<p>
<a href="#output.2">Go to the output files for this example</a><p><p>



<H2>
    Command line arguments
</H2>

<table CELLSPACING=0 CELLPADDING=3 BGCOLOR="#f5f5ff" ><tr><td>
<pre>
   Standard (Mandatory) qualifiers (* if not always prompted):
   -codonfile          codon      Optional codon usage file. Nucleotide
                                  sequences will be created as triplets
                                  matching the frequencies in the file, with
                                  the end trimmed to be in the correct reading
                                  frame.
   -amount             integer    [100] Number of sequences created (Integer 1
                                  or more)
   -length             integer    [100] Length of each sequence (Integer 1 or
                                  more)
*  -insert             string     String that is inserted into sequence (Any
                                  string is accepted)
*  -start              integer    [1] Start point of inserted sequence
                                  (Integer 1 or more)
  [-outseq]            seqoutall  [<sequence>.<format>] Nucleotide sequence
                                  set(s) filename and optional format (output
                                  USA)

   Additional (Optional) qualifiers:
   -useinsert          toggle     [N] Do you want to make an insert

   Advanced (Unprompted) qualifiers: (none)
   Associated qualifiers:

   "-codonfile" associated qualifiers
   -format             string     Data format

   "-outseq" associated qualifiers
   -osformat1          string     Output seq format
   -osextension1       string     File name extension
   -osname1            string     Base file name
   -osdirectory1       string     Output directory
   -osdbname1          string     Database name to add
   -ossingle1          boolean    Separate file for each entry
   -oufo1              string     UFO features
   -offormat1          string     Features format
   -ofname1            string     Features file name
   -ofdirectory1       string     Output directory

   General qualifiers:
   -auto               boolean    Turn off prompts
   -stdout             boolean    Write first file to standard output
   -filter             boolean    Read first file from standard input, write
                                  first file to standard output
   -options            boolean    Prompt for standard and additional values
   -debug              boolean    Write debug output to program.dbg
   -verbose            boolean    Report some/full command line options
   -help               boolean    Report command line options. More
                                  information on associated and general
                                  qualifiers can be found with -help -verbose
   -warning            boolean    Report warnings
   -error              boolean    Report errors
   -fatal              boolean    Report fatal errors
   -die                boolean    Report dying program messages

</pre>
</td></tr></table>
<P>
<table border cellspacing=0 cellpadding=3 bgcolor="#ccccff">
<tr bgcolor="#FFFFCC">
<th align="left" colspan=2>Standard (Mandatory) qualifiers</th>
<th align="left">Allowed values</th>
<th align="left">Default</th>
</tr>

<tr>
<td>-codonfile</td>
<td>Optional codon usage file. Nucleotide sequences will be created as triplets matching the frequencies in the file, with the end trimmed to be in the correct reading frame.</td>
<td>Codon usage file in EMBOSS data path</td>
<td>&nbsp;</td>
</tr>

<tr>
<td>-amount</td>
<td>Number of sequences created</td>
<td>Integer 1 or more</td>
<td>100</td>
</tr>

<tr>
<td>-length</td>
<td>Length of each sequence</td>
<td>Integer 1 or more</td>
<td>100</td>
</tr>

<tr>
<td>-insert</td>
<td>String that is inserted into sequence</td>
<td>Any string is accepted</td>
<td><i>An empty string is accepted</i></td>
</tr>

<tr>
<td>-start</td>
<td>Start point of inserted sequence</td>
<td>Integer 1 or more</td>
<td>1</td>
</tr>

<tr>
<td>[-outseq]<br>(Parameter 1)</td>
<td>Nucleotide sequence set(s) filename and optional format (output USA)</td>
<td>Writeable sequence(s)</td>
<td><i>&lt;*&gt;</i>.<i>format</i></td>
</tr>

<tr bgcolor="#FFFFCC">
<th align="left" colspan=2>Additional (Optional) qualifiers</th>
<th align="left">Allowed values</th>
<th align="left">Default</th>
</tr>

<tr>
<td>-useinsert</td>
<td>Do you want to make an insert</td>
<td>Toggle value Yes/No</td>
<td>No</td>
</tr>

<tr bgcolor="#FFFFCC">
<th align="left" colspan=2>Advanced (Unprompted) qualifiers</th>
<th align="left">Allowed values</th>
<th align="left">Default</th>
</tr>

<tr>
<td colspan=4>(none)</td>
</tr>

</table>

<!--
DON'T WRITE ANYTHING HERE.
IT IS DONE FOR YOU.
-->








<H2>
    Input file format
</H2>

<!-- 
        This includes example input file formats.
        This should be a detailed description and example - assume
        someone will want to parse this file and will want to know what
        happens in unusual cases - null input, etc. 
   -->

<b>makenucseq</b> reads any normal sequence USAs.

<p>






<H2>
    Output file format
</H2>

<!-- 
        This includes example output file formats.
        This should be a detailed description and example - assume
        someone will want to parse this file and will want to know what
        happens in unusual cases - null output, errors etc. 

        If you wish to include the standard description of the avalable
	report formats, use:
        #include file="inc/reportformats.ihtml"
   -->

<b>makenucseq</b> 
writes a sequence file with the number of sequences requested.


<p>


<a name="output.1"></a>
<h3>Output files for usage example </h3>
<p><h3>File: makeseq.fasta</h3>
<table width="90%"><tr><td bgcolor="#CCFFCC">
<pre>
&gt;EMBOSS_001
ggtccgaggggtagcttgatcgcctcttttgggaacgcaagcgtggccggtatgataaaa
taaaatgcgctccgctctggtaagacggacggtcgcccta
&gt;EMBOSS_002
tttcaatccattaggagatccttccgggcttactcttttttggtaggaatacagacgaat
gtgttgttgacactacaagtacgaactgtgatcgcaccct
&gt;EMBOSS_003
aacggctgttagaccgcatcattttggcaggaaccttgggtcggttcatctcttgggtat
gacagcacagggaggtttgagagctgcctcccggattttg
&gt;EMBOSS_004
atcgaagagacgcattcctatagtatcaatccttgacgtcggcatggttcggatcttacg
cgaagccctacgactccctaccggtattgtatcgttctag
&gt;EMBOSS_005
agagttcgtctacacatgggcacccttactacttgagtgcttaccaaaagtacgatccac
gaaacgtcgagcactggcatgcacacgctctccgacgtat
&gt;EMBOSS_006
caatcctcattgcttacacggccacaacaggaagtgcctcaagctgtagagcataggcat
gttctcaagatcgcgttaacagccattgctggagaatggc
&gt;EMBOSS_007
actcggagttcaaccgccgctgtgggctgagctatccctaatgacccgcgcgtagagtgt
taaaatctttcgacagtcacctgcacatcgttgtcttctg
&gt;EMBOSS_008
cgagggtcattctaaatttgagggtaatgctcgcgccatgcgacttgtcgggaaagcggc
cctttatgcgttgccggtcggcattgcccaaccctagtga
&gt;EMBOSS_009
gttatcacgtccaattaggtgggccacaatacgtcggcagaatcaagtgcataacggaga
ggtcactaggacgttactgtcattccctgccccgctatgt
&gt;EMBOSS_010
tatgtttctcgtgctcagcttagaccggagagctccacccaataggccgcacacaagggg
tctttaggatacccccccccttattaccccacagcaacag
&gt;EMBOSS_011
agacgcgccgctcgtagtatgttacctcccttacctgaggacctgcgtggatggcggcaa
cgtgcctcaaagcccgcagcgaatggaaagcaggttggtg
&gt;EMBOSS_012
accctcgcgctgtcgtgtatatttctagcgatgagctagttcgtgcggccatggttctgg
atgcatggctatggtttgttcagacggaatagtccgggac
&gt;EMBOSS_013
accctgacagctcttaccctctatggacggacagggcatgaccgtgcatgacctggcaga
gaacggcttcatattagggagcagcagaatcgtgtatcgt
&gt;EMBOSS_014
tcctgccgggaattagagacaaaaagattgtgaccttcacagcctcctactttcttttgt
tagtatggggcttgacactatacactgggacaatttagaa
&gt;EMBOSS_015
ccgtagcaaggccacagttacgaaggtcatgttcccacccacagcgttgttcccttttac
cgacccacgcctagaatttctcgaagaacagaaaccaccc
&gt;EMBOSS_016
cttctcccgtaagaaccttcgtccaacatccaatcaatcacggtgggtacgaggtatacc
tatctgtcacgaaatcctaagacatttctgcaaacagggg
&gt;EMBOSS_017
cagtagttgatgcacgtaacatttactcattggacaacagcacagtagctcccatcacat


<font color=red>  [Part of this file has been deleted for brevity]</font>

&gt;EMBOSS_084
ttaccatttaaacatgtccagatacggacaggtcgcactgtccttgggccacttttctgt
tacgcgcgtgagcttatgatgacgaatcgcgggaagggat
&gt;EMBOSS_085
cctcactctctcagaccccgaatcctaagtgatcaatacctctacgtcttggttttgccg
ctgatatcatgatatacatgagtctagcacggtctcctac
&gt;EMBOSS_086
aggcagggtatagtgctctagagtacaccccttgcgtgaaatcggcgtcctggtggtcga
ttaccctcaagcacttttcctcagttcggggggtcacgat
&gt;EMBOSS_087
ataagtctaccaatgaaacgatcgcgaatcggacgccactcgccagggtattcgcgcttc
acattatatatagtccggaagcacgactctccatttggtt
&gt;EMBOSS_088
tctatccgtgcctcccacgggtcagatatatgctctggacgtagtggtgccgcatctgtg
tacatgactaaagctacgactttttcgcgtagccaatcca
&gt;EMBOSS_089
ccgccgtaactttcccgggtagggggctaccgacgataatctagcgtttgatagcgtctg
gcgcatgtgtcatagggtgctggtgtccttgggcagtaca
&gt;EMBOSS_090
acgaatccccctttccacacgacaaaaacgagaccggtagctttgggcgagacgtacgct
gactaagcaatcgaagcccttcaagtacgctgccagtcaa
&gt;EMBOSS_091
gtgacctgagacttgagctactaatgcacttgtacgagggcttacaaaaagaacggatca
aagcaagccctcgggatgtgtgacccaggcgtttgccgtg
&gt;EMBOSS_092
accgttcgagagatctcacaactagcacatatacgccagaggtagtcacacagattaaga
tggtgcgacagctcgcgcaaccacacggggatgacacgta
&gt;EMBOSS_093
aaccctaaccgagggtttgtggctaacatcagccccactgtgcccgcaacgtcccaaatt
gactctatctcgttccaacactttaaatgcttatgtatat
&gt;EMBOSS_094
gtctggtacgttcagcgcgattagccgggcgctccggaccggatgtcctaccaggtacac
ctatccagtaccacggcgcggtgtagccgtacctcaaagc
&gt;EMBOSS_095
taggttggatagacgacggtctgtcaaggctcaccaagggttaatctgcctctgataagt
gttggactatgggggcaacgcctcgaccagttgaacctgt
&gt;EMBOSS_096
gaccgtggtgcattctagtacgtatgagcatccaaacatgcgccacatgtaccagggtga
tgctggagtcgctgtggaacaagagcgttctaccccctcg
&gt;EMBOSS_097
atcataggcccaagtcggatcatgtttatccgagtttaacaagttctgcagtggcagaaa
agtgtctttttagcagctcccctattgccacggctgctac
&gt;EMBOSS_098
taactagcagtttcgttatttaagagacttatgcgttaataaaatcttgttatggggcca
ccagtccttattaatccgtaaggagacgagccgcctggag
&gt;EMBOSS_099
tcgggatgggaactgtagtaagtaaatacagcagaacaagaccagctgtaggaccgcaag
taattcgcatggggccttgctctctacaatcacggcagcc
&gt;EMBOSS_100
taaatcggcatggtggagttatctttaagattgcaccatatgtggccgagccggaaaagc
ttatggggtgccaggtttagcctcttgttacatagtgacg
</pre>
</td></tr></table><p>

<a name="output.2"></a>
<h3>Output files for usage example 2</h3>
<p><h3>File: makeseq.fasta</h3>
<table width="90%"><tr><td bgcolor="#CCFFCC">
<pre>
&gt;EMBOSS_001
gggcgtctgggcgccgtcacggtcggccgctacctcgacccggcgttctccgtcaacttc
gcgcggcagggcttcatctgactgctgttcggcgtgcgcg
&gt;EMBOSS_002
gacgccggcgcgaccgagcggcaccgcttcgtccgaggaggcggcggcttcaccctcgtg
gccttcgacgacagcaccctggaagagaaggactacccgg
&gt;EMBOSS_003
cgcgccttcgccgccggtggcctgcactggggggttctgaccgagggcaggatcggccgc
aggggcccggtgctgatcgtcgcccaccacctgaccctgc
&gt;EMBOSS_004
ctgcagaaggacttaatgcacaccctgttcaagcggggtaaacccgagctgggcatcctg
ttcggccagcgcgtgccggccttgtcgaccggctacggcc
&gt;EMBOSS_005
ttcctctcctacctgcgctacctgaacgtgccagaagagtccagcggcgaacgcagcgcc
ggcaccgacctgcgctcccccttcctgcggctgtgaggtg
&gt;EMBOSS_006
gccgacgcgtgcgaccataccgaccgcctggacatccgcagcaccatgttccccctgcgc
gagctcatccgcgcggaggcgatcatctacgacgagggcg
&gt;EMBOSS_007
ccgcacctgcgcctgctgaccgtgaacgccattccggccaattggatcacctcgtgcctg
ctggtgggcatcaagggcgggaccgaggccggcttcctgg
&gt;EMBOSS_008
ggcggcctggcgcctcgctatcgtgccaactgcatcttcgcgctgctgcggcggccgctg
gacctgccggatgcggaggtcgccgagagtgacccggttg
&gt;EMBOSS_009
aagggcccgctgctctcagggaacgtcacgcgcgcgtacgtcggcgcggaagccgccgcg
gttgtggccctgtcgctgctgcgggactatgccgtagaga
&gt;EMBOSS_010
gtcgacggtaaggccgtggcgacctcgttccaggccctggaacttgagggctccacctcg
ggaaaagagctgatgatccgttacgtaagcggccgccgtc
&gt;EMBOSS_011
cggcgcgcggcactcccggtctcgttcgccccgcggaactggcacctgtggacgatccgc
caagtcgagaagtaaggccacgccctcacgggccgcaaga
&gt;EMBOSS_012
tatcaggcccagctcgagcaacaccatccgtgcagctgactcggccgtaattggcagcat
ctttcgatcttggccccgaagggtaagcgcttgctggcgg
&gt;EMBOSS_013
ctgatctacacccatgcccagaccctagtgggcccgcacaaggcgcaccaacgttgaact
ggctgagaccagtgggaaccgaaacgctacgggggcctcc
&gt;EMBOSS_014
gccatggccaccaacgaagatggctgcgccatccgcagcctccagcaggacgcgggcgag
acggaccaaccgctgcacggcgacgtgatgaacattttcg
&gt;EMBOSS_015
cgccagagctcgcgcggcgaggcgacccagggcgcgctgcagtggggacaggtcaacgcg
ccgctggacttcgcgagcaacctgcagatgctcattgagc
&gt;EMBOSS_016
ctgggcgcgtcgctggagggcgaccacggcgtggcgcagaaagctaatgccagtgatggt
gtcgatacccgactgcctgcgatgtgggccgacaagccga
&gt;EMBOSS_017
ctcggcttcagccccacccctgcgagcctggacgtgggcgctaagttcggcctctcgggc


<font color=red>  [Part of this file has been deleted for brevity]</font>

&gt;EMBOSS_084
ctgcgccacctgcgcggggaaaacggcatcgccacgatgtcggagatcggtgagtacgcg
agcaagcggggcatggaagccgaggccttcctgagcacac
&gt;EMBOSS_085
cactactacatccgcctgctgctgcaggccaccatcttcctggtcagccgcacgcgcacc
ggcgtattgattatctgtcgctttgagccggtgtttaaga
&gt;EMBOSS_086
actgccgagatcaagcagtacggtgcgccgccggcaccgggtgacgccctcggcgaccgc
gactgccgcggcgccgtgggtctgctgagccgccaatccg
&gt;EMBOSS_087
cggaggaacaccagcgacatctatgaatggttcctgcagctggccaccgtgtcgctgatc
gccgaggagctcatcaacttccgtaccggccagaatgccc
&gt;EMBOSS_088
gggatcctggacgaagcggccgccttctcgaaggagaacagaagctactatccggacggc
ctgtacacccacgtgtacaagcgcaacgagcaatggcgga
&gt;EMBOSS_089
gcgctgctgcgcgcccgctcgaaatgggatcgctacgagaacatcgccaagtacctgatg
aaggaggaggcagacgccgaccaggaacaggtcaccctcc
&gt;EMBOSS_090
ccgatcgacggctacggtaccaacctgtacgccccgatgccgcatctgggcaggagcgac
cccccgcgcgccctcgggcccccaattcaacagatcgctc
&gt;EMBOSS_091
gcgaaccactcgaacgttctggtgatggcgcataccaattacaaccgcgacgccggccgt
atcccctaccaaccgagtcgacacaccggcgaggccggcg
&gt;EMBOSS_092
tccatcgagaaggccatcccgccggtccgtatcaaggggctggccggccgcccgtgcctc
ctaggccgggcgagcctagcgaccatcgcctatgacgcca
&gt;EMBOSS_093
attctgcccgacgaagacctctacgagatccgcggcttcatgcacaaggccgggcatgcg
gacccgggcgacgaacatatcaccgtgccgctcaggcacc
&gt;EMBOSS_094
cgcgtctacccgcgtcaggacgcgctgaccgccgtcctggtgaaccgccttgacaatttc
ctgaccctgggcaaccagatgctggagctcaacgtccaat
&gt;EMBOSS_095
gacctgatggagcatttcagctcgtacctgtccctggccggcggcctcttaggcataaag
cgcaccgccccgctatcccgtatcgggacggccctggata
&gt;EMBOSS_096
tgactggtcagctacggccgcttcctgctcatcgccccgctgggtctgcggacggagttc
gaacagctgtccctgggcggcgagccgggcccggccgacc
&gt;EMBOSS_097
gcccatcggggtcgctgcgtgaatccgtgcgttggagcgcggcgcgcctgcgaaaacctg
ccgctgtcgccggcggtgctgctggaggtggtccgcagcg
&gt;EMBOSS_098
cgcgcccctatggaacgcctgttcgcggcgttcgcgatcgccatcctggggatcatggtg
ttcgccatgctggacatcagcatcgaagagcgtttcccgc
&gt;EMBOSS_099
gcgaagtgccgcaccttcgtgggcagctacacgattcagagtttggacacccgggagacg
aagcagaccgtcatcgatgaggagaccaagaccaaattga
&gt;EMBOSS_100
ctcggccggcaactgcggcgcgaaccgatgccgccctacgcggccctgcgttagatggac
ctcgcccccgcccgcgaccgagtcgcgctcgacacggtcg
</pre>
</td></tr></table><p>





<H2>
    Data files
</H2>

<!-- 
        Any data files used (e.g. translation table file)
        This includes example data file formats if they are
        not obvious.

	If you wish to include the standard description of what data
	files are and how to use embossdata to inspect and retrieve
	them, use:

        #include file="inc/localfiles.ihtml"
   -->













<H2>
    Notes
</H2>

<!-- 
        Restrictions.
        Interesting behaviour.
        Useful things you can do with this program.
   -->

None.







<H2>
    References
</H2>

<!-- 
        Bibliography for methods used.
<ol>

<li>

</ol>

   -->

None.








<H2>
    Warnings
</H2>

<!-- 
        Potentially stupid things the program will let you do.
   -->

None.







<H2>
    Diagnostic Error Messages
</H2>

<!-- 
        Error messages specific to this program, eg:
        "FATAL xxx" - means you have not set up the xxx data using program <b>prog</b>.<p>
   -->

None.







<H2>
    Exit status
</H2>

<!-- 
        Description of the exit status for various error conditions
   -->

It always exits with status 0.








<H2>
    Known bugs
</H2>


<!-- 
        Bugs noted but not yet fixed.
   -->

None.








<!--
<H2>
    See also
</H2>
-->
<h2><a name="See also">See also</a></h2>
<table border cellpadding=4 bgcolor="#FFFFF0">
<tr><th>Program name</th>
<th>Description</th></tr>
<tr>
<td><a href="aligncopy.html">aligncopy</a></td>
<td>Reads and writes alignments</td>
</tr>

<tr>
<td><a href="aligncopypair.html">aligncopypair</a></td>
<td>Reads and writes pairs from alignments</td>
</tr>

<tr>
<td><a href="biosed.html">biosed</a></td>
<td>Replace or delete sequence sections</td>
</tr>

<tr>
<td><a href="codcopy.html">codcopy</a></td>
<td>Copy and reformat a codon usage table</td>
</tr>

<tr>
<td><a href="cutseq.html">cutseq</a></td>
<td>Removes a section from a sequence</td>
</tr>

<tr>
<td><a href="degapseq.html">degapseq</a></td>
<td>Removes non-alphabetic (e.g. gap) characters from sequences</td>
</tr>

<tr>
<td><a href="descseq.html">descseq</a></td>
<td>Alter the name or description of a sequence</td>
</tr>

<tr>
<td><a href="entret.html">entret</a></td>
<td>Retrieves sequence entries from flatfile databases and files</td>
</tr>

<tr>
<td><a href="extractalign.html">extractalign</a></td>
<td>Extract regions from a sequence alignment</td>
</tr>

<tr>
<td><a href="extractfeat.html">extractfeat</a></td>
<td>Extract features from sequence(s)</td>
</tr>

<tr>
<td><a href="extractseq.html">extractseq</a></td>
<td>Extract regions from a sequence</td>
</tr>

<tr>
<td><a href="featcopy.html">featcopy</a></td>
<td>Reads and writes a feature table</td>
</tr>

<tr>
<td><a href="featreport.html">featreport</a></td>
<td>Reads and writes a feature table</td>
</tr>

<tr>
<td><a href="listor.html">listor</a></td>
<td>Write a list file of the logical OR of two sets of sequences</td>
</tr>

<tr>
<td><a href="makeprotseq.html">makeprotseq</a></td>
<td>Create random protein sequences</td>
</tr>

<tr>
<td><a href="maskambignuc.html">maskambignuc</a></td>
<td>Masks all ambiguity characters in nucleotide sequences with N</td>
</tr>

<tr>
<td><a href="maskambigprot.html">maskambigprot</a></td>
<td>Masks all ambiguity characters in protein sequences with X</td>
</tr>

<tr>
<td><a href="maskfeat.html">maskfeat</a></td>
<td>Write a sequence with masked features</td>
</tr>

<tr>
<td><a href="maskseq.html">maskseq</a></td>
<td>Write a sequence with masked regions</td>
</tr>

<tr>
<td><a href="newseq.html">newseq</a></td>
<td>Create a sequence file from a typed-in sequence</td>
</tr>

<tr>
<td><a href="nohtml.html">nohtml</a></td>
<td>Remove mark-up (e.g. HTML tags) from an ASCII text file</td>
</tr>

<tr>
<td><a href="noreturn.html">noreturn</a></td>
<td>Remove carriage return from ASCII files</td>
</tr>

<tr>
<td><a href="nospace.html">nospace</a></td>
<td>Remove all whitespace from an ASCII text file</td>
</tr>

<tr>
<td><a href="notab.html">notab</a></td>
<td>Replace tabs with spaces in an ASCII text file</td>
</tr>

<tr>
<td><a href="notseq.html">notseq</a></td>
<td>Write to file a subset of an input stream of sequences</td>
</tr>

<tr>
<td><a href="nthseq.html">nthseq</a></td>
<td>Write to file a single sequence from an input stream of sequences</td>
</tr>

<tr>
<td><a href="nthseqset.html">nthseqset</a></td>
<td>Reads and writes (returns) one set of sequences from many</td>
</tr>

<tr>
<td><a href="pasteseq.html">pasteseq</a></td>
<td>Insert one sequence into another</td>
</tr>

<tr>
<td><a href="revseq.html">revseq</a></td>
<td>Reverse and complement a nucleotide sequence</td>
</tr>

<tr>
<td><a href="seqret.html">seqret</a></td>
<td>Reads and writes (returns) sequences</td>
</tr>

<tr>
<td><a href="seqretsetall.html">seqretsetall</a></td>
<td>Reads and writes (returns) many sets of sequences</td>
</tr>

<tr>
<td><a href="seqretsplit.html">seqretsplit</a></td>
<td>Reads sequences and writes them to individual files</td>
</tr>

<tr>
<td><a href="sizeseq.html">sizeseq</a></td>
<td>Sort sequences by size</td>
</tr>

<tr>
<td><a href="skipredundant.html">skipredundant</a></td>
<td>Remove redundant sequences from an input set</td>
</tr>

<tr>
<td><a href="skipseq.html">skipseq</a></td>
<td>Reads and writes (returns) sequences, skipping first few</td>
</tr>

<tr>
<td><a href="splitsource.html">splitsource</a></td>
<td>Split sequence(s) into original source sequences</td>
</tr>

<tr>
<td><a href="splitter.html">splitter</a></td>
<td>Split sequence(s) into smaller sequences</td>
</tr>

<tr>
<td><a href="trimest.html">trimest</a></td>
<td>Remove poly-A tails from nucleotide sequences</td>
</tr>

<tr>
<td><a href="trimseq.html">trimseq</a></td>
<td>Remove unwanted characters from start and end of sequence(s)</td>
</tr>

<tr>
<td><a href="trimspace.html">trimspace</a></td>
<td>Remove extra whitespace from an ASCII text file</td>
</tr>

<tr>
<td><a href="union.html">union</a></td>
<td>Concatenate multiple sequences into a single sequence</td>
</tr>

<tr>
<td><a href="vectorstrip.html">vectorstrip</a></td>
<td>Removes vectors from the ends of nucleotide sequence(s)</td>
</tr>

<tr>
<td><a href="yank.html">yank</a></td>
<td>Add a sequence reference (a full USA) to a list file</td>
</tr>

</table>
<!-- 
        Add any comments about other associated programs (to prepare
        data files?) that seealso doesn't find. 
   -->










<H2>
    Author(s)
</H2>
This application was contributed by 
Henrikki Almusa, Medicel, Helsinki, Finland






<H2>
    History
</H2>
<!--
        Date written and what changes have been made go in this file.
   -->




<H2>
    Target users
</H2>
<!--
        For general users, use this text
   -->
This program is intended to be used by everyone and everything, from naive users to embedded scripts.

<H2>
    Comments
</H2>
<!--
        User/developer/other comments go in this file.
   -->
None


</BODY>
</HTML>

