<HTML>

<HEAD>
  <TITLE>
  EMBOSS: maskfeat
  </TITLE>
</HEAD>
<BODY BGCOLOR="#FFFFFF" text="#000000">

<table align=center border=0 cellspacing=0 cellpadding=0>
<tr><td valign=top>
<A HREF="/" ONMOUSEOVER="self.status='Go to the EMBOSS home page';return true"><img border=0 src="/images/emboss_icon.jpg" alt="" width=150 height=48></a>
</td>
<td align=left valign=middle>
<b><font size="+6">
maskfeat
</font></b>
</td></tr>
</table>
<br>&nbsp;
<p>



<H2>
Wiki
</H2>

The master copies of EMBOSS documentation are available
at <a href="http://emboss.open-bio.org/wiki/Appdocs">
http://emboss.open-bio.org/wiki/Appdocs</a>
on the EMBOSS Wiki.

<p>
Please help by correcting and extending the Wiki pages.

<H2>
    Function
</H2>
Write a sequence with masked features

<H2>
    Description
</H2>

<p><b>maskfeat</b> reads a sequence with associated features and
writes the same information to file but with features of the specified
type omitted (masked).  Sequence regions corresponding to the
masked-out features are masked such that characters in those region
are (optionally) converted to lower case and / or (optionally)
replaced with the specified mask character.</p>

<H2>
    Usage
</H2>
Here is a sample session with <b>maskfeat</b>
<p>
Mask out a feature whose type is "repeat_region" from position 2331 to 2356: 
<p>

<p>
<table width="90%"><tr><td bgcolor="#CCFFFF"><pre>

% <b>maskfeat tembl:ab000360 </b>
Write a sequence with masked features
output sequence [ab000360.fasta]: <b></b>

</pre></td></tr></table><p>
<p>
<a href="#input.1">Go to the input files for this example</a><br><a href="#output.1">Go to the output files for this example</a><p><p>
<p>
<b>Example 2</b>
<p>
Change to lower-case a feature whose type is "repeat_region" from position 2331 to 2356. Note that '-supper' is used to make the whole sequence upper-case before the lower-case masking: 
<p>

<p>
<table width="90%"><tr><td bgcolor="#CCFFFF"><pre>

% <b>maskfeat tembl:ab000360 -tolower -supper </b>
Write a sequence with masked features
output sequence [ab000360.fasta]: <b></b>

</pre></td></tr></table><p>
<p>
<a href="#output.2">Go to the output files for this example</a><p><p>

<H2>
    Command line arguments
</H2>
<table CELLSPACING=0 CELLPADDING=3 BGCOLOR="#f5f5ff" ><tr><td>
<pre>
Write a sequence with masked features
Version: EMBOSS:6.5.6.0

   Standard (Mandatory) qualifiers:
  [-sequence]          seqall     Sequence(s) filename and optional format, or
                                  reference (input USA)
  [-outseq]            seqout     [<sequence>.<format>] Sequence filename and
                                  optional format (output USA)

   Additional (Optional) qualifiers (* if not always prompted):
   -type               string     [repeat*] By default any feature in the
                                  feature table with a type starting 'repeat'
                                  is masked. You can set this to be any
                                  feature type you wish to mask.
                                  See http://www.ebi.ac.uk/embl/WebFeat/ for a
                                  list of the EMBL feature types and see
                                  Appendix A of the Swissprot user manual in
                                  http://www.expasy.org/sprot/userman.html for
                                  a list of the Swissprot feature types.
                                  The type may be wildcarded by using '*'.
                                  If you wish to mask more than one type,
                                  separate their names with spaces or commas,
                                  eg:
                                  *UTR repeat* (Any string)
   -tolower            toggle     [N] The region can be 'masked' by converting
                                  the sequence characters to lower-case, some
                                  non-EMBOSS programs e.g. fasta can
                                  interpret this as a masked region. The
                                  sequence is unchanged apart from the case
                                  change. You might like to ensure that the
                                  whole sequence is in upper-case before
                                  masking the specified regions to lower-case
                                  by using the '-supper' flag.
*  -maskchar           string     ['X' for protein, 'N' for nucleic] Character
                                  to use when masking.
                                  Default is 'X' for protein sequences, 'N'
                                  for nucleic sequences.
                                  If the mask character is set to be the SPACE
                                  character or a null character, then the
                                  sequence is 'masked' by changing it to
                                  lower-case, just as with the '-lowercase'
                                  flag. (Any string up to 1 characters)

   Advanced (Unprompted) qualifiers: (none)
   Associated qualifiers:

   "-sequence" associated qualifiers
   -sbegin1            integer    Start of each sequence to be used
   -send1              integer    End of each sequence to be used
   -sreverse1          boolean    Reverse (if DNA)
   -sask1              boolean    Ask for begin/end/reverse
   -snucleotide1       boolean    Sequence is nucleotide
   -sprotein1          boolean    Sequence is protein
   -slower1            boolean    Make lower case
   -supper1            boolean    Make upper case
   -scircular1         boolean    Sequence is circular
   -sformat1           string     Input sequence format
   -iquery1            string     Input query fields or ID list
   -ioffset1           integer    Input start position offset
   -sdbname1           string     Database name
   -sid1               string     Entryname
   -ufo1               string     UFO features
   -fformat1           string     Features format
   -fopenfile1         string     Features file name

   "-outseq" associated qualifiers
   -osformat2          string     Output seq format
   -osextension2       string     File name extension
   -osname2            string     Base file name
   -osdirectory2       string     Output directory
   -osdbname2          string     Database name to add
   -ossingle2          boolean    Separate file for each entry
   -oufo2              string     UFO features
   -offormat2          string     Features format
   -ofname2            string     Features file name
   -ofdirectory2       string     Output directory

   General qualifiers:
   -auto               boolean    Turn off prompts
   -stdout             boolean    Write first file to standard output
   -filter             boolean    Read first file from standard input, write
                                  first file to standard output
   -options            boolean    Prompt for standard and additional values
   -debug              boolean    Write debug output to program.dbg
   -verbose            boolean    Report some/full command line options
   -help               boolean    Report command line options and exit. More
                                  information on associated and general
                                  qualifiers can be found with -help -verbose
   -warning            boolean    Report warnings
   -error              boolean    Report errors
   -fatal              boolean    Report fatal errors
   -die                boolean    Report dying program messages
   -version            boolean    Report version number and exit

</pre>
</td></tr></table>
<P>

<table border cellspacing=0 cellpadding=3 bgcolor="#ccccff">
<tr bgcolor="#FFFFCC">
<th align="left">Qualifier</th>
<th align="left">Type</th>
<th align="left">Description</th>
<th align="left">Allowed values</th>
<th align="left">Default</th>
</tr>

<tr bgcolor="#FFFFCC">
<th align="left" colspan=5>Standard (Mandatory) qualifiers</th>
</tr>

<tr bgcolor="#FFFFCC">
<td>[-sequence]<br>(Parameter 1)</td>
<td>seqall</td>
<td>Sequence(s) filename and optional format, or reference (input USA)</td>
<td>Readable sequence(s)</td>
<td><b>Required</b></td>
</tr>

<tr bgcolor="#FFFFCC">
<td>[-outseq]<br>(Parameter 2)</td>
<td>seqout</td>
<td>Sequence filename and optional format (output USA)</td>
<td>Writeable sequence</td>
<td><i>&lt;*&gt;</i>.<i>format</i></td>
</tr>

<tr bgcolor="#FFFFCC">
<th align="left" colspan=5>Additional (Optional) qualifiers</th>
</tr>

<tr bgcolor="#FFFFCC">
<td>-type</td>
<td>string</td>
<td>By default any feature in the feature table with a type starting 'repeat' is masked. You can set this to be any feature type you wish to mask.
See http://www.ebi.ac.uk/embl/WebFeat/ for a list of the EMBL feature types and see Appendix A of the Swissprot user manual in http://www.expasy.org/sprot/userman.html for a list of the Swissprot feature types.
The type may be wildcarded by using '*'.
If you wish to mask more than one type, separate their names with spaces or commas, eg:
*UTR repeat*</td>
<td>Any string</td>
<td>repeat*</td>
</tr>

<tr bgcolor="#FFFFCC">
<td>-tolower</td>
<td>toggle</td>
<td>The region can be 'masked' by converting the sequence characters to lower-case, some non-EMBOSS programs e.g. fasta can interpret this as a masked region. The sequence is unchanged apart from the case change. You might like to ensure that the whole sequence is in upper-case before masking the specified regions to lower-case by using the '-supper' flag.</td>
<td>Toggle value Yes/No</td>
<td>No</td>
</tr>

<tr bgcolor="#FFFFCC">
<td>-maskchar</td>
<td>string</td>
<td>Character to use when masking.
Default is 'X' for protein sequences, 'N' for nucleic sequences.
If the mask character is set to be the SPACE character or a null character, then the sequence is 'masked' by changing it to lower-case, just as with the '-lowercase' flag.</td>
<td>Any string up to 1 characters</td>
<td>'X' for protein, 'N' for nucleic</td>
</tr>

<tr bgcolor="#FFFFCC">
<th align="left" colspan=5>Advanced (Unprompted) qualifiers</th>
</tr>

<tr>
<td colspan=5>(none)</td>
</tr>

<tr bgcolor="#FFFFCC">
<th align="left" colspan=5>Associated qualifiers</th>
</tr>

<tr bgcolor="#FFFFCC">
<td align="left" colspan=5>"-sequence" associated seqall qualifiers
</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -sbegin1<br>-sbegin_sequence</td>
<td>integer</td>
<td>Start of each sequence to be used</td>
<td>Any integer value</td>
<td>0</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -send1<br>-send_sequence</td>
<td>integer</td>
<td>End of each sequence to be used</td>
<td>Any integer value</td>
<td>0</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -sreverse1<br>-sreverse_sequence</td>
<td>boolean</td>
<td>Reverse (if DNA)</td>
<td>Boolean value Yes/No</td>
<td>N</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -sask1<br>-sask_sequence</td>
<td>boolean</td>
<td>Ask for begin/end/reverse</td>
<td>Boolean value Yes/No</td>
<td>N</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -snucleotide1<br>-snucleotide_sequence</td>
<td>boolean</td>
<td>Sequence is nucleotide</td>
<td>Boolean value Yes/No</td>
<td>N</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -sprotein1<br>-sprotein_sequence</td>
<td>boolean</td>
<td>Sequence is protein</td>
<td>Boolean value Yes/No</td>
<td>N</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -slower1<br>-slower_sequence</td>
<td>boolean</td>
<td>Make lower case</td>
<td>Boolean value Yes/No</td>
<td>N</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -supper1<br>-supper_sequence</td>
<td>boolean</td>
<td>Make upper case</td>
<td>Boolean value Yes/No</td>
<td>N</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -scircular1<br>-scircular_sequence</td>
<td>boolean</td>
<td>Sequence is circular</td>
<td>Boolean value Yes/No</td>
<td>N</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -sformat1<br>-sformat_sequence</td>
<td>string</td>
<td>Input sequence format</td>
<td>Any string</td>
<td>&nbsp;</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -iquery1<br>-iquery_sequence</td>
<td>string</td>
<td>Input query fields or ID list</td>
<td>Any string</td>
<td>&nbsp;</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -ioffset1<br>-ioffset_sequence</td>
<td>integer</td>
<td>Input start position offset</td>
<td>Any integer value</td>
<td>0</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -sdbname1<br>-sdbname_sequence</td>
<td>string</td>
<td>Database name</td>
<td>Any string</td>
<td>&nbsp;</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -sid1<br>-sid_sequence</td>
<td>string</td>
<td>Entryname</td>
<td>Any string</td>
<td>&nbsp;</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -ufo1<br>-ufo_sequence</td>
<td>string</td>
<td>UFO features</td>
<td>Any string</td>
<td>&nbsp;</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -fformat1<br>-fformat_sequence</td>
<td>string</td>
<td>Features format</td>
<td>Any string</td>
<td>&nbsp;</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -fopenfile1<br>-fopenfile_sequence</td>
<td>string</td>
<td>Features file name</td>
<td>Any string</td>
<td>&nbsp;</td>
</tr>

<tr bgcolor="#FFFFCC">
<td align="left" colspan=5>"-outseq" associated seqout qualifiers
</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -osformat2<br>-osformat_outseq</td>
<td>string</td>
<td>Output seq format</td>
<td>Any string</td>
<td>&nbsp;</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -osextension2<br>-osextension_outseq</td>
<td>string</td>
<td>File name extension</td>
<td>Any string</td>
<td>&nbsp;</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -osname2<br>-osname_outseq</td>
<td>string</td>
<td>Base file name</td>
<td>Any string</td>
<td>&nbsp;</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -osdirectory2<br>-osdirectory_outseq</td>
<td>string</td>
<td>Output directory</td>
<td>Any string</td>
<td>&nbsp;</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -osdbname2<br>-osdbname_outseq</td>
<td>string</td>
<td>Database name to add</td>
<td>Any string</td>
<td>&nbsp;</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -ossingle2<br>-ossingle_outseq</td>
<td>boolean</td>
<td>Separate file for each entry</td>
<td>Boolean value Yes/No</td>
<td>N</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -oufo2<br>-oufo_outseq</td>
<td>string</td>
<td>UFO features</td>
<td>Any string</td>
<td>&nbsp;</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -offormat2<br>-offormat_outseq</td>
<td>string</td>
<td>Features format</td>
<td>Any string</td>
<td>&nbsp;</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -ofname2<br>-ofname_outseq</td>
<td>string</td>
<td>Features file name</td>
<td>Any string</td>
<td>&nbsp;</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -ofdirectory2<br>-ofdirectory_outseq</td>
<td>string</td>
<td>Output directory</td>
<td>Any string</td>
<td>&nbsp;</td>
</tr>

<tr bgcolor="#FFFFCC">
<th align="left" colspan=5>General qualifiers</th>
</tr>

<tr bgcolor="#FFFFCC">
<td> -auto</td>
<td>boolean</td>
<td>Turn off prompts</td>
<td>Boolean value Yes/No</td>
<td>N</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -stdout</td>
<td>boolean</td>
<td>Write first file to standard output</td>
<td>Boolean value Yes/No</td>
<td>N</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -filter</td>
<td>boolean</td>
<td>Read first file from standard input, write first file to standard output</td>
<td>Boolean value Yes/No</td>
<td>N</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -options</td>
<td>boolean</td>
<td>Prompt for standard and additional values</td>
<td>Boolean value Yes/No</td>
<td>N</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -debug</td>
<td>boolean</td>
<td>Write debug output to program.dbg</td>
<td>Boolean value Yes/No</td>
<td>N</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -verbose</td>
<td>boolean</td>
<td>Report some/full command line options</td>
<td>Boolean value Yes/No</td>
<td>Y</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -help</td>
<td>boolean</td>
<td>Report command line options and exit. More information on associated and general qualifiers can be found with -help -verbose</td>
<td>Boolean value Yes/No</td>
<td>N</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -warning</td>
<td>boolean</td>
<td>Report warnings</td>
<td>Boolean value Yes/No</td>
<td>Y</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -error</td>
<td>boolean</td>
<td>Report errors</td>
<td>Boolean value Yes/No</td>
<td>Y</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -fatal</td>
<td>boolean</td>
<td>Report fatal errors</td>
<td>Boolean value Yes/No</td>
<td>Y</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -die</td>
<td>boolean</td>
<td>Report dying program messages</td>
<td>Boolean value Yes/No</td>
<td>Y</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -version</td>
<td>boolean</td>
<td>Report version number and exit</td>
<td>Boolean value Yes/No</td>
<td>N</td>
</tr>

</table>


<H2>
    Input file format
</H2>

<b>maskfeat</b> reads one or more nucleotide or protein sequences with features.


<p>
<p>

The input is a standard EMBOSS sequence query (also known as a 'USA')
with associated feature information.

<p>
Major sequence database sources defined as standard in EMBOSS
installations include srs:embl, srs:uniprot and ensembl

<p>

Data can also be read from sequence output in any supported format
written by an EMBOSS or third-party application.

<p>
The input format can be specified by using the command-line qualifier
<tt>-sformat xxx</tt>, where 'xxx' is replaced by the name of the
required format.  The available format names are: text, html, xml (uniprotxml),
obo, embl (swissprot)

<p> Where the sequence format has no feature information, a second
file can be read to load the feature data. The file is specified with
the qualifier <tt>-ufo xxx</tt> and the feature format is specified with
the qualifier <tt>-fformat xxx</tt>

<p>
See:
<A href="http://emboss.sf.net/docs/themes/SequenceFormats.html">
http://emboss.sf.net/docs/themes/SequenceFormats.html</A>
for further information on sequence formats.

<p>
See:
<A href="http://emboss.sf.net/docs/themes/FeatureFormats.html">
http://emboss.sf.net/docs/themes/FeatureFormats.html</A>
for further information on feature formats.
<p>


<a name="input.1"></a>
<h3>Input files for usage example </h3>

'tembl:ab000360' is a sequence entry in the example nucleic acid database 'tembl'
<p>
<p><h3>Database entry: tembl:ab000360</h3>
<table width="90%"><tr><td bgcolor="#FFCCFF">
<pre>
ID   AB000360; SV 1; linear; genomic DNA; STD; HUM; 2582 BP.
XX
AC   AB000360;
XX
DT   27-OCT-1997 (Rel. 53, Created)
DT   28-SEP-2008 (Rel. 97, Last updated, Version 5)
XX
DE   Homo sapiens PIGC gene, complete cds.
XX
KW   .
XX
OS   Homo sapiens (human)
OC   Eukaryota; Metazoa; Chordata; Craniata; Vertebrata; Euteleostomi; Mammalia;
OC   Eutheria; Euarchontoglires; Primates; Haplorrhini; Catarrhini; Hominidae;
OC   Homo.
XX
RN   [1]
RP   1-2582
RA   Hong Y., Ohishi K., Inoue N., Endo Y., Fujita T., Takeda J., Kinoshita T.;
RT   ;
RL   Submitted (08-JAN-1997) to the INSDC.
RL   Contact:Yeongjin Hong Research Institute for Microbial Diseases,
RL   Immunoregulation; 3-1 Yamada-oka, Suita, Osaka 565, Japan
XX
RN   [2]
RX   DOI; 10.1006/geno.1997.4893.
RX   PUBMED; 9325057.
RA   Hong Y., Ohishi K., Inoue N., Endo Y., Fujita T., Takeda J., Kinoshita T.;
RT   "Structures and chromosomal localizations of the
RT   glycosylphosphatidylinositol synthesis gene PIGC and its pseudogene
RT   PIGCP1";
RL   Genomics 44(3):347-349(1997).
XX
DR   Ensembl-Gn; ENSG00000135845; Homo_sapiens.
DR   Ensembl-Tr; ENST00000258324; Homo_sapiens.
DR   Ensembl-Tr; ENST00000344529; Homo_sapiens.
DR   Ensembl-Tr; ENST00000367728; Homo_sapiens.
XX
FH   Key             Location/Qualifiers
FH
FT   source          1..2582
FT                   /organism="Homo sapiens"
FT                   /chromosome="1"
FT                   /map="1q23-q25"
FT                   /mol_type="genomic DNA"
FT                   /db_xref="taxon:9606"
FT   exon            808..2266
FT   CDS             1101..1994
FT                   /codon_start=1
FT                   /transl_table=1


<font color=red>  [Part of this file has been deleted for brevity]</font>

FT   variation       2259
FT                   /replace="t"
FT   repeat_region   2331..2356
FT                   /rpt_unit_seq="gt"
XX
SQ   Sequence 2582 BP; 694 A; 494 C; 581 G; 813 T; 0 other;
     ggatccctgc tgcagagggg gtaacggtgt ctggcttgcc aagcaatatt tgttgtggtc        60
     tatcatggaa gaaataaagt cgggcaatat gaattttttt tttctcaaat ttgccggatg       120
     gctgtggtgt ttctgactct tagttttctc attgtgaaaa aggaatgatt atcttcttcg       180
     atcctctcaa gagtttcctt gttttgagta gattgatagc tctttaaagg atgctaagct       240
     cagctaatgg aagaagagtc tagtttcttt gaggctttga ttttggttaa actatagagc       300
     tcataccttt ctgtatggtg cagcttacta ttgtctttgg attggtaact taaaaaatac       360
     aaataacatg cctttgagaa ccaataaaaa ctatggatat tatccctata aatttacaca       420
     aatccagata taagcatgca atgtgatata cctaagggat atgtgaacca ctgagttaag       480
     aactgcttta gagggagata caatgtgaga cacaggcttt gggataagac tttggtttga       540
     atcctggctc tgctctgtta ccttagggca aagttactta agcatcttga atctcagctt       600
     ttttaccaaa gcaggactaa tactaactta caaggtggtg aggattaagt gaaagaagat       660
     acataaggca cttagcacat agtaggtact caataagcga tagctaacag atgtctatta       720
     ttattcaagg aattataatt ttcaaatctg aaatgcagtt ttaatgtccc ataaggtgac       780
     taccacatac atttttctca gacttttagt aaactgagtt gatttgactt tatctcagta       840
     ctactcttga cctttcacaa ctttcgtagg ttcacagtct ctctttttct aggaacttgg       900
     ctgtgttgtc ctgcctcaga gacaaattca tctattgtag gcctagcccc tgcctttgaa       960
     aacaaggaaa ggttggtaga acatcaacac agcatggaat ttccagggag gtctcatttc      1020
     aaaacttcat aaagaacaag aaccacctgg acttctgtga gggcgatgat taaactggcc      1080
     tgagtttgaa tgaaaggata atgtatgctc aacctgtgac taacaccaag gaggtcaagt      1140
     ggcagaaggt cttgtatgag cgacagccct ttcctgataa ctatgtggac cggcgattcc      1200
     tggaagagct ccggaaaaac atccatgctc ggaaatacca atattgggct gtggtatttg      1260
     agtccagtgt ggtgatccag cagctgtgca gtgtttgtgt ttttgtggtt atctggtggt      1320
     atatggatga gggtcttctg gccccccatt ggcttttagg gactggcctg gcttcttcac      1380
     tgattgggta tgttttgttt gatctcattg atggaggtga agggcggaag aagagtgggc      1440
     agacccggtg ggctgacctg aagagtgccc tagtcttcat tactttcact tatgggtttt      1500
     caccagtgct gaagaccctt acagagtctg tcagcactga caccatctat gccatgtcag      1560
     tcttcatgct gttaggccat ctcatctttt ttgactatgg tgccaatgct gccattgtat      1620
     ccagcacact atccttgaac atggccatct ttgcttctgt atgcttggca tcacgtcttc      1680
     cccggtccct gcatgccttc atcatggtga catttgccat tcagattttt gccctgtggc      1740
     ccatgttgca gaagaaacta aaggcatgta ctccccggag ctatgtgggg gtcacactgc      1800
     tttttgcatt ttcagccgtg ggaggcctac tgtccattag tgctgtggga gccgtactct      1860
     ttgcccttct gctgatgtct atctcatgtc tgtgttcatt ctacctcatt cgcttgcagc      1920
     tttttaaaga aaacattcat gggccttggg atgaagctga aatcaaggaa gacttgtcca      1980
     ggttcctcag ttaaattagg acatccatta cattattaaa gcaagctgat agattagcct      2040
     cctaactagt atagaactta aagacagagt tccattctgg aagcagcatg tcattgtggt      2100
     aagagaatag agatcaaaac caaaaaaaat gaaccaaagg cttgggtggt gagggtgctt      2160
     atcctttctg ttattttgta gatgaaaaaa ctttctgggg acctcttgaa ttacatgctg      2220
     taacatatga agtgatgtgg tttctattaa aaaaataaca catccatcaa gttgtctcat      2280
     gatttttcca taaacaggag gcagacagag gggcatgaag agtgaagtaa gtgtgtgtgt      2340
     gtgtgtgtgt gtgtgtaaag tcacttcttt ctaccctttt caatgtgcta atgctctttt      2400
     atttatctag ggctcaaatc ttagaacaca gggtgctatg ctcagttttg ttgcccaaga      2460
     tcacagaatt ggttacttaa ccttgactca gagtttctac cttgttctta gggaagcata      2520
     tcacaactaa ttgcaaagca gagtgtgatg tgtcacaata agcagaatgc tagggggaat      2580
     tc                                                                     2582
//
</pre>
</td></tr></table><p>

<H2>
    Output file format
</H2>


<p>

The output is a standard EMBOSS sequence file. 

<p>

The results can be output in one of several styles by using the
command-line qualifier <tt>-osformat xxx</tt>, where 'xxx' is replaced by
the name of the required format.  The available format names are: embl,
genbank, gff, pir, swiss, dasgff, debug, listfile, dbmotif, diffseq, excel,
feattable, motif, nametable, regions, seqtable, simple, srs, table, tagseq.

<p>

See:
<A href="http://emboss.sf.net/docs/themes/SequenceFormats.html">
http://emboss.sf.net/docs/themes/SequenceFormats.html</A>
for further information on sequence formats.

<p>
<p>
The sequence file has features masked with the specified character.
<p>


<a name="output.1"></a>
<h3>Output files for usage example </h3>
<p><h3>File: ab000360.fasta</h3>
<table width="90%"><tr><td bgcolor="#CCFFCC">
<pre>
&gt;AB000360 AB000360.1 Homo sapiens PIGC gene, complete cds.
ggatccctgctgcagagggggtaacggtgtctggcttgccaagcaatatttgttgtggtc
tatcatggaagaaataaagtcgggcaatatgaattttttttttctcaaatttgccggatg
gctgtggtgtttctgactcttagttttctcattgtgaaaaaggaatgattatcttcttcg
atcctctcaagagtttccttgttttgagtagattgatagctctttaaaggatgctaagct
cagctaatggaagaagagtctagtttctttgaggctttgattttggttaaactatagagc
tcatacctttctgtatggtgcagcttactattgtctttggattggtaacttaaaaaatac
aaataacatgcctttgagaaccaataaaaactatggatattatccctataaatttacaca
aatccagatataagcatgcaatgtgatatacctaagggatatgtgaaccactgagttaag
aactgctttagagggagatacaatgtgagacacaggctttgggataagactttggtttga
atcctggctctgctctgttaccttagggcaaagttacttaagcatcttgaatctcagctt
ttttaccaaagcaggactaatactaacttacaaggtggtgaggattaagtgaaagaagat
acataaggcacttagcacatagtaggtactcaataagcgatagctaacagatgtctatta
ttattcaaggaattataattttcaaatctgaaatgcagttttaatgtcccataaggtgac
taccacatacatttttctcagacttttagtaaactgagttgatttgactttatctcagta
ctactcttgacctttcacaactttcgtaggttcacagtctctctttttctaggaacttgg
ctgtgttgtcctgcctcagagacaaattcatctattgtaggcctagcccctgcctttgaa
aacaaggaaaggttggtagaacatcaacacagcatggaatttccagggaggtctcatttc
aaaacttcataaagaacaagaaccacctggacttctgtgagggcgatgattaaactggcc
tgagtttgaatgaaaggataatgtatgctcaacctgtgactaacaccaaggaggtcaagt
ggcagaaggtcttgtatgagcgacagccctttcctgataactatgtggaccggcgattcc
tggaagagctccggaaaaacatccatgctcggaaataccaatattgggctgtggtatttg
agtccagtgtggtgatccagcagctgtgcagtgtttgtgtttttgtggttatctggtggt
atatggatgagggtcttctggccccccattggcttttagggactggcctggcttcttcac
tgattgggtatgttttgtttgatctcattgatggaggtgaagggcggaagaagagtgggc
agacccggtgggctgacctgaagagtgccctagtcttcattactttcacttatgggtttt
caccagtgctgaagacccttacagagtctgtcagcactgacaccatctatgccatgtcag
tcttcatgctgttaggccatctcatcttttttgactatggtgccaatgctgccattgtat
ccagcacactatccttgaacatggccatctttgcttctgtatgcttggcatcacgtcttc
cccggtccctgcatgccttcatcatggtgacatttgccattcagatttttgccctgtggc
ccatgttgcagaagaaactaaaggcatgtactccccggagctatgtgggggtcacactgc
tttttgcattttcagccgtgggaggcctactgtccattagtgctgtgggagccgtactct
ttgcccttctgctgatgtctatctcatgtctgtgttcattctacctcattcgcttgcagc
tttttaaagaaaacattcatgggccttgggatgaagctgaaatcaaggaagacttgtcca
ggttcctcagttaaattaggacatccattacattattaaagcaagctgatagattagcct
cctaactagtatagaacttaaagacagagttccattctggaagcagcatgtcattgtggt
aagagaatagagatcaaaaccaaaaaaaatgaaccaaaggcttgggtggtgagggtgctt
atcctttctgttattttgtagatgaaaaaactttctggggacctcttgaattacatgctg
taacatatgaagtgatgtggtttctattaaaaaaataacacatccatcaagttgtctcat
gatttttccataaacaggaggcagacagaggggcatgaagagtgaagtaaNNNNNNNNNN
NNNNNNNNNNNNNNNNaaagtcacttctttctacccttttcaatgtgctaatgctctttt
atttatctagggctcaaatcttagaacacagggtgctatgctcagttttgttgcccaaga
tcacagaattggttacttaaccttgactcagagtttctaccttgttcttagggaagcata
tcacaactaattgcaaagcagagtgtgatgtgtcacaataagcagaatgctagggggaat
tc
</pre>
</td></tr></table><p>

<a name="output.2"></a>
<h3>Output files for usage example 2</h3>
<p><h3>File: ab000360.fasta</h3>
<table width="90%"><tr><td bgcolor="#CCFFCC">
<pre>
&gt;AB000360 AB000360.1 Homo sapiens PIGC gene, complete cds.
GGATCCCTGCTGCAGAGGGGGTAACGGTGTCTGGCTTGCCAAGCAATATTTGTTGTGGTC
TATCATGGAAGAAATAAAGTCGGGCAATATGAATTTTTTTTTTCTCAAATTTGCCGGATG
GCTGTGGTGTTTCTGACTCTTAGTTTTCTCATTGTGAAAAAGGAATGATTATCTTCTTCG
ATCCTCTCAAGAGTTTCCTTGTTTTGAGTAGATTGATAGCTCTTTAAAGGATGCTAAGCT
CAGCTAATGGAAGAAGAGTCTAGTTTCTTTGAGGCTTTGATTTTGGTTAAACTATAGAGC
TCATACCTTTCTGTATGGTGCAGCTTACTATTGTCTTTGGATTGGTAACTTAAAAAATAC
AAATAACATGCCTTTGAGAACCAATAAAAACTATGGATATTATCCCTATAAATTTACACA
AATCCAGATATAAGCATGCAATGTGATATACCTAAGGGATATGTGAACCACTGAGTTAAG
AACTGCTTTAGAGGGAGATACAATGTGAGACACAGGCTTTGGGATAAGACTTTGGTTTGA
ATCCTGGCTCTGCTCTGTTACCTTAGGGCAAAGTTACTTAAGCATCTTGAATCTCAGCTT
TTTTACCAAAGCAGGACTAATACTAACTTACAAGGTGGTGAGGATTAAGTGAAAGAAGAT
ACATAAGGCACTTAGCACATAGTAGGTACTCAATAAGCGATAGCTAACAGATGTCTATTA
TTATTCAAGGAATTATAATTTTCAAATCTGAAATGCAGTTTTAATGTCCCATAAGGTGAC
TACCACATACATTTTTCTCAGACTTTTAGTAAACTGAGTTGATTTGACTTTATCTCAGTA
CTACTCTTGACCTTTCACAACTTTCGTAGGTTCACAGTCTCTCTTTTTCTAGGAACTTGG
CTGTGTTGTCCTGCCTCAGAGACAAATTCATCTATTGTAGGCCTAGCCCCTGCCTTTGAA
AACAAGGAAAGGTTGGTAGAACATCAACACAGCATGGAATTTCCAGGGAGGTCTCATTTC
AAAACTTCATAAAGAACAAGAACCACCTGGACTTCTGTGAGGGCGATGATTAAACTGGCC
TGAGTTTGAATGAAAGGATAATGTATGCTCAACCTGTGACTAACACCAAGGAGGTCAAGT
GGCAGAAGGTCTTGTATGAGCGACAGCCCTTTCCTGATAACTATGTGGACCGGCGATTCC
TGGAAGAGCTCCGGAAAAACATCCATGCTCGGAAATACCAATATTGGGCTGTGGTATTTG
AGTCCAGTGTGGTGATCCAGCAGCTGTGCAGTGTTTGTGTTTTTGTGGTTATCTGGTGGT
ATATGGATGAGGGTCTTCTGGCCCCCCATTGGCTTTTAGGGACTGGCCTGGCTTCTTCAC
TGATTGGGTATGTTTTGTTTGATCTCATTGATGGAGGTGAAGGGCGGAAGAAGAGTGGGC
AGACCCGGTGGGCTGACCTGAAGAGTGCCCTAGTCTTCATTACTTTCACTTATGGGTTTT
CACCAGTGCTGAAGACCCTTACAGAGTCTGTCAGCACTGACACCATCTATGCCATGTCAG
TCTTCATGCTGTTAGGCCATCTCATCTTTTTTGACTATGGTGCCAATGCTGCCATTGTAT
CCAGCACACTATCCTTGAACATGGCCATCTTTGCTTCTGTATGCTTGGCATCACGTCTTC
CCCGGTCCCTGCATGCCTTCATCATGGTGACATTTGCCATTCAGATTTTTGCCCTGTGGC
CCATGTTGCAGAAGAAACTAAAGGCATGTACTCCCCGGAGCTATGTGGGGGTCACACTGC
TTTTTGCATTTTCAGCCGTGGGAGGCCTACTGTCCATTAGTGCTGTGGGAGCCGTACTCT
TTGCCCTTCTGCTGATGTCTATCTCATGTCTGTGTTCATTCTACCTCATTCGCTTGCAGC
TTTTTAAAGAAAACATTCATGGGCCTTGGGATGAAGCTGAAATCAAGGAAGACTTGTCCA
GGTTCCTCAGTTAAATTAGGACATCCATTACATTATTAAAGCAAGCTGATAGATTAGCCT
CCTAACTAGTATAGAACTTAAAGACAGAGTTCCATTCTGGAAGCAGCATGTCATTGTGGT
AAGAGAATAGAGATCAAAACCAAAAAAAATGAACCAAAGGCTTGGGTGGTGAGGGTGCTT
ATCCTTTCTGTTATTTTGTAGATGAAAAAACTTTCTGGGGACCTCTTGAATTACATGCTG
TAACATATGAAGTGATGTGGTTTCTATTAAAAAAATAACACATCCATCAAGTTGTCTCAT
GATTTTTCCATAAACAGGAGGCAGACAGAGGGGCATGAAGAGTGAAGTAAgtgtgtgtgt
gtgtgtgtgtgtgtgtAAAGTCACTTCTTTCTACCCTTTTCAATGTGCTAATGCTCTTTT
ATTTATCTAGGGCTCAAATCTTAGAACACAGGGTGCTATGCTCAGTTTTGTTGCCCAAGA
TCACAGAATTGGTTACTTAACCTTGACTCAGAGTTTCTACCTTGTTCTTAGGGAAGCATA
TCACAACTAATTGCAAAGCAGAGTGTGATGTGTCACAATAAGCAGAATGCTAGGGGGAAT
TC
</pre>
</td></tr></table><p>



<H2>
    Data files
</H2>

None.


<H2>
    Notes
</H2>

<p>Formats such as EMBL or SWISSPROT include features in the sequence
annotation.  Otherwise, features they may be supplied in a separate
file and specified with the qualifier <tt>-ufo</tt>
or <tt>-fopenfile</tt> / <tt>-fformat</tt>.</p>

<p>The feature type to mask is by default <tt>repeat*</tt>, i.e. any
type whose name starts with <tt>repeat</tt> will be omitted from the
output file. Any other type of feature(s) may be specified with
the <tt>-type</tt> qualifier.  To specify more than one type of
feature, separate their names with spaces or commas. The names may be
wild-carded with asterisks <tt>*</tt> to find gruops of feature types
sharing a common part of their names.</p>

<p>If you are unsure of the names of feature types in use, please
consult <a href="http://www.ebi.ac.uk/Services/WebFeat/">
http://www.ebi.ac.uk/Services/WebFeat/</a> (EMBL feature types) and
see Appendix A of the Swissprot user manual
in <a href="http://www.uniprot.org/manual/sequence_annotation">
http://www.uniprot.org/manual/sequence_annotation</a> (Swissprot
feature types).</p>


<H2>
    References
</H2>

None.

<H2>
    Warnings
</H2>

None.

<H2>
    Diagnostic Error Messages
</H2>

None.

<H2>
    Exit status
</H2>

It always exits with a status of 0.

<H2>
    Known bugs
</H2>

None.

<h2><a name="See also">See also</a></h2>
<table border cellpadding=4 bgcolor="#FFFFF0">
<tr><th>Program name</th>
<th>Description</th></tr>
<tr>
<td><a href="aligncopy.html">aligncopy</a></td>
<td>Read and write alignments</td>
</tr>

<tr>
<td><a href="aligncopypair.html">aligncopypair</a></td>
<td>Read and write pairs from alignments</td>
</tr>

<tr>
<td><a href="biosed.html">biosed</a></td>
<td>Replace or delete sequence sections</td>
</tr>

<tr>
<td><a href="codcopy.html">codcopy</a></td>
<td>Copy and reformat a codon usage table</td>
</tr>

<tr>
<td><a href="cutseq.html">cutseq</a></td>
<td>Remove a section from a sequence</td>
</tr>

<tr>
<td><a href="degapseq.html">degapseq</a></td>
<td>Remove non-alphabetic (e.g. gap) characters from sequences</td>
</tr>

<tr>
<td><a href="descseq.html">descseq</a></td>
<td>Alter the name or description of a sequence</td>
</tr>

<tr>
<td><a href="entret.html">entret</a></td>
<td>Retrieve sequence entries from flatfile databases and files</td>
</tr>

<tr>
<td><a href="extractalign.html">extractalign</a></td>
<td>Extract regions from a sequence alignment</td>
</tr>

<tr>
<td><a href="extractfeat.html">extractfeat</a></td>
<td>Extract features from sequence(s)</td>
</tr>

<tr>
<td><a href="extractseq.html">extractseq</a></td>
<td>Extract regions from a sequence</td>
</tr>

<tr>
<td><a href="featcopy.html">featcopy</a></td>
<td>Read and write a feature table</td>
</tr>

<tr>
<td><a href="featmerge.html">featmerge</a></td>
<td>Merge two overlapping feature tables</td>
</tr>

<tr>
<td><a href="featreport.html">featreport</a></td>
<td>Read and write a feature table</td>
</tr>

<tr>
<td><a href="feattext.html">feattext</a></td>
<td>Return a feature table original text</td>
</tr>

<tr>
<td><a href="listor.html">listor</a></td>
<td>Write a list file of the logical OR of two sets of sequences</td>
</tr>

<tr>
<td><a href="makenucseq.html">makenucseq</a></td>
<td>Create random nucleotide sequences</td>
</tr>

<tr>
<td><a href="makeprotseq.html">makeprotseq</a></td>
<td>Create random protein sequences</td>
</tr>

<tr>
<td><a href="maskambignuc.html">maskambignuc</a></td>
<td>Mask all ambiguity characters in nucleotide sequences with N</td>
</tr>

<tr>
<td><a href="maskambigprot.html">maskambigprot</a></td>
<td>Mask all ambiguity characters in protein sequences with X</td>
</tr>

<tr>
<td><a href="maskseq.html">maskseq</a></td>
<td>Write a sequence with masked regions</td>
</tr>

<tr>
<td><a href="newseq.html">newseq</a></td>
<td>Create a sequence file from a typed-in sequence</td>
</tr>

<tr>
<td><a href="nohtml.html">nohtml</a></td>
<td>Remove mark-up (e.g. HTML tags) from an ASCII text file</td>
</tr>

<tr>
<td><a href="noreturn.html">noreturn</a></td>
<td>Remove carriage return from ASCII files</td>
</tr>

<tr>
<td><a href="nospace.html">nospace</a></td>
<td>Remove whitespace from an ASCII text file</td>
</tr>

<tr>
<td><a href="notab.html">notab</a></td>
<td>Replace tabs with spaces in an ASCII text file</td>
</tr>

<tr>
<td><a href="notseq.html">notseq</a></td>
<td>Write to file a subset of an input stream of sequences</td>
</tr>

<tr>
<td><a href="nthseq.html">nthseq</a></td>
<td>Write to file a single sequence from an input stream of sequences</td>
</tr>

<tr>
<td><a href="nthseqset.html">nthseqset</a></td>
<td>Read and write (return) one set of sequences from many</td>
</tr>

<tr>
<td><a href="pasteseq.html">pasteseq</a></td>
<td>Insert one sequence into another</td>
</tr>

<tr>
<td><a href="revseq.html">revseq</a></td>
<td>Reverse and complement a nucleotide sequence</td>
</tr>

<tr>
<td><a href="seqcount.html">seqcount</a></td>
<td>Read and count sequences</td>
</tr>

<tr>
<td><a href="seqret.html">seqret</a></td>
<td>Read and write (return) sequences</td>
</tr>

<tr>
<td><a href="seqretsetall.html">seqretsetall</a></td>
<td>Read and write (return) many sets of sequences</td>
</tr>

<tr>
<td><a href="seqretsplit.html">seqretsplit</a></td>
<td>Read sequences and write them to individual files</td>
</tr>

<tr>
<td><a href="showfeat.html">showfeat</a></td>
<td>Display features of a sequence in pretty format</td>
</tr>

<tr>
<td><a href="sizeseq.html">sizeseq</a></td>
<td>Sort sequences by size</td>
</tr>

<tr>
<td><a href="skipredundant.html">skipredundant</a></td>
<td>Remove redundant sequences from an input set</td>
</tr>

<tr>
<td><a href="skipseq.html">skipseq</a></td>
<td>Read and write (return) sequences, skipping first few</td>
</tr>

<tr>
<td><a href="splitsource.html">splitsource</a></td>
<td>Split sequence(s) into original source sequences</td>
</tr>

<tr>
<td><a href="splitter.html">splitter</a></td>
<td>Split sequence(s) into smaller sequences</td>
</tr>

<tr>
<td><a href="trimest.html">trimest</a></td>
<td>Remove poly-A tails from nucleotide sequences</td>
</tr>

<tr>
<td><a href="trimseq.html">trimseq</a></td>
<td>Remove unwanted characters from start and end of sequence(s)</td>
</tr>

<tr>
<td><a href="trimspace.html">trimspace</a></td>
<td>Remove extra whitespace from an ASCII text file</td>
</tr>

<tr>
<td><a href="twofeat.html">twofeat</a></td>
<td>Find neighbouring pairs of features in sequence(s)</td>
</tr>

<tr>
<td><a href="union.html">union</a></td>
<td>Concatenate multiple sequences into a single sequence</td>
</tr>

<tr>
<td><a href="vectorstrip.html">vectorstrip</a></td>
<td>Remove vectors from the ends of nucleotide sequence(s)</td>
</tr>

<tr>
<td><a href="yank.html">yank</a></td>
<td>Add a sequence reference (a full USA) to a list file</td>
</tr>

</table>

<H2>
    Author(s)
</H2>


Gary Williams formerly at:
<br>
MRC Rosalind Franklin Centre for Genomics Research
Wellcome Trust Genome Campus, Hinxton, Cambridge, CB10 1SB, UK

<p>
Please report all bugs to the EMBOSS bug team (emboss-bug&nbsp;&copy;&nbsp;emboss.open-bio.org) not to the original author.


<H2>
    History
</H2>

Written (2000) - Gary Williams

<H2>
    Target users
</H2>
This program is intended to be used by everyone and everything, from naive users to embedded scripts.



<H2>
    Comments
</H2>
None

</BODY>
</HTML>

