<HTML>
<HEAD>
  <TITLE>
  EMBOSS: mwcontam
  </TITLE>
</HEAD>
<BODY BGCOLOR="#FFFFFF" text="#000000">

<table align=center border=0 cellspacing=0 cellpadding=0>
<tr><td valign=top>
<A HREF="/" ONMOUSEOVER="self.status='Go to the EMBOSS home page';return true"><img border=0 src="/images/emboss_icon.jpg" alt="" width=150 height=48></a>
</td>
<td align=left valign=middle>
<b><font size="+6">
mwcontam
</font></b>
</td></tr>
</table>
<br>&nbsp;
<p>


<H2>
Wiki
</H2>

The master copies of EMBOSS documentation are available
at <a href="http://emboss.open-bio.org/wiki/Appdocs">
http://emboss.open-bio.org/wiki/Appdocs</a>
on the EMBOSS Wiki.

<p>
Please help by correcting and extending the Wiki pages.

<H2>
    Function
</H2>
Find weights common to multiple molecular weights files
<H2>
    Description
</H2>

<p><p><b>mwcontam</b> finds molecular weights that are common between a set of mass spectrometry result files at a specified ppm tolerance. Such molecular weights are usually a form of contamination resulting from autolysis of a protease, degradation of the matrix or presence of keratin from the sample preparer. The input is a comma-separated list of files each containing a list of experimental molecular weights.  The output is a file with a list of the molecular weights common to all files.</p></p>



<H2>
    Usage
</H2>
Here is a sample session with <b>mwcontam</b>
<p>

<p>
<table width="90%"><tr><td bgcolor="#CCFFFF"><pre>

% <b>mwcontam </b>
Find weights common to multiple molecular weights files
Molecular weights file list: <b>../data/mw2.dat,mw3.dat</b>
Ppm tolerance [50.0]: <b></b>
Molecular weights output file [outfile.mwcontam]: <b></b>

</pre></td></tr></table><p>
<p>
<a href="#input.1">Go to the input files for this example</a><br><a href="#output.1">Go to the output files for this example</a><p><p>

<H2>
    Command line arguments
</H2>
<table CELLSPACING=0 CELLPADDING=3 BGCOLOR="#f5f5ff" ><tr><td>
<pre>
Find weights common to multiple molecular weights files
Version: EMBOSS:6.5.6.0

   Standard (Mandatory) qualifiers:
  [-files]             filelist   Molecular weights file list
   -tolerance          float      [50.0] Ppm tolerance (Any numeric value)
  [-outfile]           outfile    [*.mwcontam] Molecular weights output file

   Additional (Optional) qualifiers: (none)
   Advanced (Unprompted) qualifiers: (none)
   Associated qualifiers:

   "-outfile" associated qualifiers
   -odirectory2        string     Output directory

   General qualifiers:
   -auto               boolean    Turn off prompts
   -stdout             boolean    Write first file to standard output
   -filter             boolean    Read first file from standard input, write
                                  first file to standard output
   -options            boolean    Prompt for standard and additional values
   -debug              boolean    Write debug output to program.dbg
   -verbose            boolean    Report some/full command line options
   -help               boolean    Report command line options and exit. More
                                  information on associated and general
                                  qualifiers can be found with -help -verbose
   -warning            boolean    Report warnings
   -error              boolean    Report errors
   -fatal              boolean    Report fatal errors
   -die                boolean    Report dying program messages
   -version            boolean    Report version number and exit

</pre>
</td></tr></table>
<P>
<table border cellspacing=0 cellpadding=3 bgcolor="#ccccff">
<tr bgcolor="#FFFFCC">
<th align="left">Qualifier</th>
<th align="left">Type</th>
<th align="left">Description</th>
<th align="left">Allowed values</th>
<th align="left">Default</th>
</tr>

<tr bgcolor="#FFFFCC">
<th align="left" colspan=5>Standard (Mandatory) qualifiers</th>
</tr>

<tr bgcolor="#FFFFCC">
<td>[-files]<br>(Parameter 1)</td>
<td>filelist</td>
<td>Molecular weights file list</td>
<td>Comma-separated file list</td>
<td><i>comma-separated file list</i></td>
</tr>

<tr bgcolor="#FFFFCC">
<td>-tolerance</td>
<td>float</td>
<td>Ppm tolerance</td>
<td>Any numeric value</td>
<td>50.0</td>
</tr>

<tr bgcolor="#FFFFCC">
<td>[-outfile]<br>(Parameter 2)</td>
<td>outfile</td>
<td>Molecular weights output file</td>
<td>Output file</td>
<td><i>&lt;*&gt;</i>.mwcontam</td>
</tr>

<tr bgcolor="#FFFFCC">
<th align="left" colspan=5>Additional (Optional) qualifiers</th>
</tr>

<tr>
<td colspan=5>(none)</td>
</tr>

<tr bgcolor="#FFFFCC">
<th align="left" colspan=5>Advanced (Unprompted) qualifiers</th>
</tr>

<tr>
<td colspan=5>(none)</td>
</tr>

<tr bgcolor="#FFFFCC">
<th align="left" colspan=5>Associated qualifiers</th>
</tr>

<tr bgcolor="#FFFFCC">
<td align="left" colspan=5>"-outfile" associated outfile qualifiers
</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -odirectory2<br>-odirectory_outfile</td>
<td>string</td>
<td>Output directory</td>
<td>Any string</td>
<td>&nbsp;</td>
</tr>

<tr bgcolor="#FFFFCC">
<th align="left" colspan=5>General qualifiers</th>
</tr>

<tr bgcolor="#FFFFCC">
<td> -auto</td>
<td>boolean</td>
<td>Turn off prompts</td>
<td>Boolean value Yes/No</td>
<td>N</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -stdout</td>
<td>boolean</td>
<td>Write first file to standard output</td>
<td>Boolean value Yes/No</td>
<td>N</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -filter</td>
<td>boolean</td>
<td>Read first file from standard input, write first file to standard output</td>
<td>Boolean value Yes/No</td>
<td>N</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -options</td>
<td>boolean</td>
<td>Prompt for standard and additional values</td>
<td>Boolean value Yes/No</td>
<td>N</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -debug</td>
<td>boolean</td>
<td>Write debug output to program.dbg</td>
<td>Boolean value Yes/No</td>
<td>N</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -verbose</td>
<td>boolean</td>
<td>Report some/full command line options</td>
<td>Boolean value Yes/No</td>
<td>Y</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -help</td>
<td>boolean</td>
<td>Report command line options and exit. More information on associated and general qualifiers can be found with -help -verbose</td>
<td>Boolean value Yes/No</td>
<td>N</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -warning</td>
<td>boolean</td>
<td>Report warnings</td>
<td>Boolean value Yes/No</td>
<td>Y</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -error</td>
<td>boolean</td>
<td>Report errors</td>
<td>Boolean value Yes/No</td>
<td>Y</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -fatal</td>
<td>boolean</td>
<td>Report fatal errors</td>
<td>Boolean value Yes/No</td>
<td>Y</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -die</td>
<td>boolean</td>
<td>Report dying program messages</td>
<td>Boolean value Yes/No</td>
<td>Y</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -version</td>
<td>boolean</td>
<td>Report version number and exit</td>
<td>Boolean value Yes/No</td>
<td>N</td>
</tr>

</table>


<H2>
    Input file format
</H2>


<a name="input.1"></a>
<h3>Input files for usage example </h3>
<p><h3>File: mw1.dat</h3>
<table width="90%"><tr><td bgcolor="#FFCCFF">
<pre>
20.0
33.0
67.0
128.0
415.0
999.0
</pre>
</td></tr></table><p>
<p><h3>File: mw2.dat</h3>
<table width="90%"><tr><td bgcolor="#FFCCFF">
<pre>
13.0
41.0
67.0
415.0
846.0
781.0
</pre>
</td></tr></table><p>
<p><h3>File: mw3.dat</h3>
<table width="90%"><tr><td bgcolor="#FFCCFF">
<pre>
3.0
18.0
67.0
124.0
197.0
236.0
415.0
587.0
632.0
</pre>
</td></tr></table><p>

<p>

The input is a comma-separated list of files containing simple list of
experimental molecular weights. 

<p>

There should be one weight per line.
   
<p>

Comments in the data file start with a '#' character in the first
column. 

<p>

Blank lines are ignored.
   


<H2>
    Output file format
</H2>

The output is a list of the molecular weights common to all files.
   
<p>


<a name="output.1"></a>
<h3>Output files for usage example </h3>
<p><h3>File: outfile.mwcontam</h3>
<table width="90%"><tr><td bgcolor="#CCFFCC">
<pre>
67.000
415.000
</pre>
</td></tr></table><p>




<H2>
    Data files
</H2>

<b>mwcontam</b> uses no data files.

<p>

The output of <b>mwcontam</b>, with minimal editing, can be added to the
data file for the <a href="mwfilter.html"><b>mwfilter</b></a> program
('Emwfilter.dat').  For example you might like to fetch 'Emwfilter.dat'
using <b>embossdata</b> (as described in the <b>mwfilter</b>
documentation) and edit it to include the lines:

<p>

<hr>
<pre>
# data produced from the program mwcontam
mwcontam	67.000
mwcontam	415.000 
</pre>

<H2>
    Notes
</H2>

<p>The output of <b>mwcontam</b>, with minimal editing, can be added to the data file for the <b>mwfilter</b> program (<tt>Emwfilter.dat</tt>), which will then remove these weights from its input file of molecular weights. For example you might like to fetch <tt>Emwfilter.dat</tt> using <b>embossdata</b> (as described in the <b>mwfilter</b> documentation) and edit it to include the lines:

# data produced from the program mwcontam
mwcontam	67.000
mwcontam	415.000 
</p>


<H2>
    References
</H2>

None.

<H2>
    Warnings
</H2>

None.

<H2>
    Diagnostic Error Messages
</H2>

None.

<H2>
    Exit status
</H2>

It always exits with status 0.

<H2>
    Known bugs
</H2>

None.

<h2><a name="See also">See also</a></h2>
<table border cellpadding=4 bgcolor="#FFFFF0">
<tr><th>Program name</th>
<th>Description</th></tr>
<tr>
<td><a href="backtranambig.html">backtranambig</a></td>
<td>Back-translate a protein sequence to ambiguous nucleotide sequence</td>
</tr>

<tr>
<td><a href="backtranseq.html">backtranseq</a></td>
<td>Back-translate a protein sequence to a nucleotide sequence</td>
</tr>

<tr>
<td><a href="compseq.html">compseq</a></td>
<td>Calculate the composition of unique words in sequences</td>
</tr>

<tr>
<td><a href="emowse.html">emowse</a></td>
<td>Search protein sequences by digest fragment molecular weight</td>
</tr>

<tr>
<td><a href="freak.html">freak</a></td>
<td>Generate residue/base frequency table or plot</td>
</tr>

<tr>
<td><a href="mwfilter.html">mwfilter</a></td>
<td>Filter noisy data from molecular weights file</td>
</tr>

<tr>
<td><a href="oddcomp.html">oddcomp</a></td>
<td>Identify proteins with specified sequence word composition</td>
</tr>

<tr>
<td><a href="pepdigest.html">pepdigest</a></td>
<td>Report on protein proteolytic enzyme or reagent cleavage sites</td>
</tr>

<tr>
<td><a href="pepinfo.html">pepinfo</a></td>
<td>Plot amino acid properties of a protein sequence in parallel</td>
</tr>

<tr>
<td><a href="pepstats.html">pepstats</a></td>
<td>Calculate statistics of protein properties</td>
</tr>

<tr>
<td><a href="wordcount.html">wordcount</a></td>
<td>Count and extract unique words in molecular sequence(s)</td>
</tr>

</table>


<H2>
    Author(s)
</H2>
Alan Bleasby 
<br>
European Bioinformatics Institute, Wellcome Trust Genome Campus, Hinxton, Cambridge CB10 1SD, UK

<p>
Please report all bugs to the EMBOSS bug team (emboss-bug&nbsp;&copy;&nbsp;emboss.open-bio.org) not to the original author.


<H2>
    History
</H2>


Written (July 2002) - Alan Bleasby.

<H2>
    Target users
</H2>
This program is intended to be used by everyone and everything, from naive users to embedded scripts.

<H2>
    Comments
</H2>
None

</BODY>
</HTML>

