<HTML>

<HEAD>
  <TITLE>
  EMBOSS: prettyplot
  </TITLE>
</HEAD>
<BODY BGCOLOR="#FFFFFF" text="#000000">

<table align=center border=0 cellspacing=0 cellpadding=0>
<tr><td valign=top>
<A HREF="/" ONMOUSEOVER="self.status='Go to the EMBOSS home page';return true"><img border=0 src="/images/emboss_icon.jpg" alt="" width=150 height=48></a>
</td>
<td align=left valign=middle>
<b><font size="+6">
prettyplot
</font></b>
</td></tr>
</table>
<br>&nbsp;
<p>


<H2>
Wiki
</H2>

The master copies of EMBOSS documentation are available
at <a href="http://emboss.open-bio.org/wiki/Appdocs">
http://emboss.open-bio.org/wiki/Appdocs</a>
on the EMBOSS Wiki.

<p>
Please help by correcting and extending the Wiki pages.

<H2>
    Function
</H2>
Draw a sequence alignment with pretty formatting

<H2>
    Description
</H2>


<p><b>prettyplot</b> draws a plot of the input sequence alignment.
The sequences are rendered in pretty formatting on the specified
graphics device.  Drawing options control the appearance of the image,
such as boxes, colour and shading for highlighting conserved
regions.</p>

<H2>
    Usage
</H2>
Here is a sample session with <b>prettyplot</b>
<p>

<p>
<table width="90%"><tr><td bgcolor="#CCFFFF"><pre>

% <b>prettyplot -blocksperline=5 -boxcol -consensus -ratio=0.59 </b>
Draw a sequence alignment with pretty formatting
Input (aligned) sequence set: <b>globins.msf</b>
Graph type [x11]: <b>cps</b>

Created prettyplot.ps

</pre></td></tr></table><p>
<p>
<a href="#input.1">Go to the input files for this example</a><br><a href="#output.1">Go to the output files for this example</a><p><p>
<p>
<b>Example 2</b>
<p>

<p>
<table width="90%"><tr><td bgcolor="#CCFFFF"><pre>

% <b>prettyplot globins.msf -ratio=0.59 -docolour </b>
Draw a sequence alignment with pretty formatting
Graph type [x11]: <b>cps</b>

Created prettyplot.ps

</pre></td></tr></table><p>
<p>
<a href="#output.2">Go to the output files for this example</a><p><p>


<H2>
    Command line arguments
</H2>
<table CELLSPACING=0 CELLPADDING=3 BGCOLOR="#f5f5ff" ><tr><td>
<pre>
Draw a sequence alignment with pretty formatting
Version: EMBOSS:6.6.0.0

   Standard (Mandatory) qualifiers:
  [-sequences]         seqset     (Aligned) sequence set filename and optional
                                  format, or reference (input USA)
   -graph              graph      [$EMBOSS_GRAPHICS value, or x11] Graph type
                                  (ps, hpgl, hp7470, hp7580, meta, cps, x11,
                                  tek, tekt, none, data, xterm, png, gif, pdf,
                                  svg)

   Additional (Optional) qualifiers:
   -matrixfile         matrix     [EBLOSUM62 for protein, EDNAFULL for DNA]
                                  This is the scoring matrix file used when
                                  comparing sequences. By default it is the
                                  file 'EBLOSUM62' (for proteins) or the file
                                  'EDNAFULL' (for nucleic sequences). These
                                  files are found in the 'data' directory of
                                  the EMBOSS installation.
   -residuesperline    integer    [50] The number of residues to be displayed
                                  on each line (Any integer value)
   -blocksperline      integer    [1] Blocks of residues on each line (Integer
                                  1 or more)
   -[no]ccolours       boolean    [Y] Colour residues by their consensus
                                  value.
   -cidentity          string     [RED] Colour to display identical residues
                                  (RED) (Any string)
   -csimilarity        string     [GREEN] Colour to display similar residues
                                  (GREEN) (Any string)
   -cother             string     [BLACK] Colour to display other residues
                                  (BLACK) (Any string)
   -docolour           boolean    [N] Colour residues by table oily, amide
                                  etc.
   -shade              string     Set to BPLW for normal shading
                                  (black, pale, light, white)
                                  so for pair = 1.5,1.0,0.5 and shade = BPLW
                                  Residues score Colour
                                  1.5 or over... BLACK (B)
                                  1.0 to 1.5 ... BROWN (P)
                                  0.5 to 1.0 ... WHEAT (L)
                                  under 0.5 .... WHITE (W)
                                  The only four letters allowed are BPLW, in
                                  any order. (Any string up to 4 characters,
                                  matching regular expression
                                  /^([BPLW]{4})?$/)
   -pair               array      [1.5,1.0,0.5] Values to represent identical
                                  similar related
   -identity           integer    [0] Only match those which are identical in
                                  all sequences. (Integer 0 or more)
   -[no]doboxes        boolean    [Y] Display prettyboxes
   -boxcol             boolean    [N] Colour the background in the boxes
   -boxuse             string     [GREY] Colour to be used for background.
                                  (GREY) (Any string)
   -[no]name           boolean    [Y] Display the sequence names
   -maxnamelen         integer    [10] Margin size for the sequence name. (Any
                                  integer value)
   -[no]number         boolean    [Y] Display the residue number
   -[no]listoptions    boolean    [Y] Display the date and options used
   -ratio              float      [0.5] Plurality ratio for a consensus match
                                  (Number from 0.000 to 1.000)
   -consensus          boolean    [N] Display the consensus
   -[no]collision      boolean    [Y] Allow collisions in calculating
                                  consensus
   -alternative        menu       [0] Values are 0:Normal collision check.
                                  (default)
                                  1:Compares identical scores with the max
                                  score found. So if any other residue matches
                                  the identical score then a collision has
                                  occurred.
                                  2:If another residue has a greater than or
                                  equal to matching score and these do not
                                  match then a collision has occurred.
                                  3:Checks all those not in the current
                                  consensus.If any of these give a top score
                                  for matching or identical scores then a
                                  collision has occured. (Values: 0 (Normal
                                  collision check. (default)); 1 (Compares
                                  identical scores with the max score found.
                                  So if any other residue matches the
                                  identical score then a collision has
                                  occurred.); 2 (If another residue has a
                                  greater than or equal to matching score and
                                  these do not match then a collision has
                                  occurred.); 3 (Checks all those not in the
                                  current consensus.If any of these give a top
                                  score for matching or identical scores then
                                  a collision has occured.))
   -showscore          integer    [-1] Print residue scores (Any integer
                                  value)
   -portrait           boolean    [N] Set page to Portrait

   Advanced (Unprompted) qualifiers: (none)
   Associated qualifiers:

   "-sequences" associated qualifiers
   -sbegin1            integer    Start of each sequence to be used
   -send1              integer    End of each sequence to be used
   -sreverse1          boolean    Reverse (if DNA)
   -sask1              boolean    Ask for begin/end/reverse
   -snucleotide1       boolean    Sequence is nucleotide
   -sprotein1          boolean    Sequence is protein
   -slower1            boolean    Make lower case
   -supper1            boolean    Make upper case
   -scircular1         boolean    Sequence is circular
   -squick1            boolean    Read id and sequence only
   -sformat1           string     Input sequence format
   -iquery1            string     Input query fields or ID list
   -ioffset1           integer    Input start position offset
   -sdbname1           string     Database name
   -sid1               string     Entryname
   -ufo1               string     UFO features
   -fformat1           string     Features format
   -fopenfile1         string     Features file name

   "-graph" associated qualifiers
   -gprompt            boolean    Graph prompting
   -gdesc              string     Graph description
   -gtitle             string     Graph title
   -gsubtitle          string     Graph subtitle
   -gxtitle            string     Graph x axis title
   -gytitle            string     Graph y axis title
   -goutfile           string     Output file for non interactive displays
   -gdirectory         string     Output directory

   General qualifiers:
   -auto               boolean    Turn off prompts
   -stdout             boolean    Write first file to standard output
   -filter             boolean    Read first file from standard input, write
                                  first file to standard output
   -options            boolean    Prompt for standard and additional values
   -debug              boolean    Write debug output to program.dbg
   -verbose            boolean    Report some/full command line options
   -help               boolean    Report command line options and exit. More
                                  information on associated and general
                                  qualifiers can be found with -help -verbose
   -warning            boolean    Report warnings
   -error              boolean    Report errors
   -fatal              boolean    Report fatal errors
   -die                boolean    Report dying program messages
   -version            boolean    Report version number and exit

</pre>
</td></tr></table>
<P>

<table border cellspacing=0 cellpadding=3 bgcolor="#ccccff">
<tr bgcolor="#FFFFCC">
<th align="left">Qualifier</th>
<th align="left">Type</th>
<th align="left">Description</th>
<th align="left">Allowed values</th>
<th align="left">Default</th>
</tr>

<tr bgcolor="#FFFFCC">
<th align="left" colspan=5>Standard (Mandatory) qualifiers</th>
</tr>

<tr bgcolor="#FFFFCC">
<td>[-sequences]<br>(Parameter 1)</td>
<td>seqset</td>
<td>(Aligned) sequence set filename and optional format, or reference (input USA)</td>
<td>Readable set of sequences</td>
<td><b>Required</b></td>
</tr>

<tr bgcolor="#FFFFCC">
<td>-graph</td>
<td>graph</td>
<td>Graph type</td>
<td>EMBOSS has a list of known devices, including ps, hpgl, hp7470, hp7580, meta, cps, x11, tek, tekt, none, data, xterm, png, gif, pdf, svg</td>
<td><i>EMBOSS_GRAPHICS</i> value, or x11</td>
</tr>

<tr bgcolor="#FFFFCC">
<th align="left" colspan=5>Additional (Optional) qualifiers</th>
</tr>

<tr bgcolor="#FFFFCC">
<td>-matrixfile</td>
<td>matrix</td>
<td>This is the scoring matrix file used when comparing sequences. By default it is the file 'EBLOSUM62' (for proteins) or the file 'EDNAFULL' (for nucleic sequences). These files are found in the 'data' directory of the EMBOSS installation.</td>
<td>Comparison matrix file in EMBOSS data path</td>
<td>EBLOSUM62 for protein<br>EDNAFULL for DNA</td>
</tr>

<tr bgcolor="#FFFFCC">
<td>-residuesperline</td>
<td>integer</td>
<td>The number of residues to be displayed on each line</td>
<td>Any integer value</td>
<td>50</td>
</tr>

<tr bgcolor="#FFFFCC">
<td>-blocksperline</td>
<td>integer</td>
<td>Blocks of residues on each line</td>
<td>Integer 1 or more</td>
<td>1</td>
</tr>

<tr bgcolor="#FFFFCC">
<td>-[no]ccolours</td>
<td>boolean</td>
<td>Colour residues by their consensus value.</td>
<td>Boolean value Yes/No</td>
<td>Yes</td>
</tr>

<tr bgcolor="#FFFFCC">
<td>-cidentity</td>
<td>string</td>
<td>Colour to display identical residues (RED)</td>
<td>Any string</td>
<td>RED</td>
</tr>

<tr bgcolor="#FFFFCC">
<td>-csimilarity</td>
<td>string</td>
<td>Colour to display similar residues (GREEN)</td>
<td>Any string</td>
<td>GREEN</td>
</tr>

<tr bgcolor="#FFFFCC">
<td>-cother</td>
<td>string</td>
<td>Colour to display other residues (BLACK)</td>
<td>Any string</td>
<td>BLACK</td>
</tr>

<tr bgcolor="#FFFFCC">
<td>-docolour</td>
<td>boolean</td>
<td>Colour residues by table oily, amide etc.</td>
<td>Boolean value Yes/No</td>
<td>No</td>
</tr>

<tr bgcolor="#FFFFCC">
<td>-shade</td>
<td>string</td>
<td>Set to BPLW for normal shading
(black, pale, light, white)
so for pair = 1.5,1.0,0.5 and shade = BPLW
Residues score Colour
1.5 or over... BLACK (B)
1.0 to 1.5 ... BROWN (P)
0.5 to 1.0 ... WHEAT (L)
under 0.5 .... WHITE (W)
The only four letters allowed are BPLW, in any order.</td>
<td>Any string up to 4 characters, matching regular expression /^([BPLW]{4})?$/</td>
<td>&nbsp;</td>
</tr>

<tr bgcolor="#FFFFCC">
<td>-pair</td>
<td>array</td>
<td>Values to represent identical similar related</td>
<td>List of floating point numbers</td>
<td>1.5,1.0,0.5</td>
</tr>

<tr bgcolor="#FFFFCC">
<td>-identity</td>
<td>integer</td>
<td>Only match those which are identical in all sequences.</td>
<td>Integer 0 or more</td>
<td>0</td>
</tr>

<tr bgcolor="#FFFFCC">
<td>-[no]doboxes</td>
<td>boolean</td>
<td>Display prettyboxes</td>
<td>Boolean value Yes/No</td>
<td>Yes</td>
</tr>

<tr bgcolor="#FFFFCC">
<td>-boxcol</td>
<td>boolean</td>
<td>Colour the background in the boxes</td>
<td>Boolean value Yes/No</td>
<td>No</td>
</tr>

<tr bgcolor="#FFFFCC">
<td>-boxuse</td>
<td>string</td>
<td>Colour to be used for background. (GREY)</td>
<td>Any string</td>
<td>GREY</td>
</tr>

<tr bgcolor="#FFFFCC">
<td>-[no]name</td>
<td>boolean</td>
<td>Display the sequence names</td>
<td>Boolean value Yes/No</td>
<td>Yes</td>
</tr>

<tr bgcolor="#FFFFCC">
<td>-maxnamelen</td>
<td>integer</td>
<td>Margin size for the sequence name.</td>
<td>Any integer value</td>
<td>10</td>
</tr>

<tr bgcolor="#FFFFCC">
<td>-[no]number</td>
<td>boolean</td>
<td>Display the residue number</td>
<td>Boolean value Yes/No</td>
<td>Yes</td>
</tr>

<tr bgcolor="#FFFFCC">
<td>-[no]listoptions</td>
<td>boolean</td>
<td>Display the date and options used</td>
<td>Boolean value Yes/No</td>
<td>Yes</td>
</tr>

<tr bgcolor="#FFFFCC">
<td>-ratio</td>
<td>float</td>
<td>Plurality ratio for a consensus match</td>
<td>Number from 0.000 to 1.000</td>
<td>0.5</td>
</tr>

<tr bgcolor="#FFFFCC">
<td>-consensus</td>
<td>boolean</td>
<td>Display the consensus</td>
<td>Boolean value Yes/No</td>
<td>No</td>
</tr>

<tr bgcolor="#FFFFCC">
<td>-[no]collision</td>
<td>boolean</td>
<td>Allow collisions in calculating consensus</td>
<td>Boolean value Yes/No</td>
<td>Yes</td>
</tr>

<tr bgcolor="#FFFFCC">
<td>-alternative</td>
<td>list</td>
<td>Values are 0:Normal collision check. (default)
1:Compares identical scores with the max score found. So if any other residue matches the identical score then a collision has occurred.
2:If another residue has a greater than or equal to matching score and these do not match then a collision has occurred.
3:Checks all those not in the current consensus.If any of these give a top score for matching or identical scores then a collision has occured.</td>
<td><table><tr><td>0</td> <td><i>(Normal collision check. (default))</i></td></tr><tr><td>1</td> <td><i>(Compares identical scores with the max score found. So if any other residue matches the identical score then a collision has occurred.)</i></td></tr><tr><td>2</td> <td><i>(If another residue has a greater than or equal to matching score and these do not match then a collision has occurred.)</i></td></tr><tr><td>3</td> <td><i>(Checks all those not in the current consensus.If any of these give a top score for matching or identical scores then a collision has occured.)</i></td></tr></table></td>
<td>0</td>
</tr>

<tr bgcolor="#FFFFCC">
<td>-showscore</td>
<td>integer</td>
<td>Print residue scores</td>
<td>Any integer value</td>
<td>-1</td>
</tr>

<tr bgcolor="#FFFFCC">
<td>-portrait</td>
<td>boolean</td>
<td>Set page to Portrait</td>
<td>Boolean value Yes/No</td>
<td>No</td>
</tr>

<tr bgcolor="#FFFFCC">
<th align="left" colspan=5>Advanced (Unprompted) qualifiers</th>
</tr>

<tr>
<td colspan=5>(none)</td>
</tr>

<tr bgcolor="#FFFFCC">
<th align="left" colspan=5>Associated qualifiers</th>
</tr>

<tr bgcolor="#FFFFCC">
<td align="left" colspan=5>"-sequences" associated seqset qualifiers
</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -sbegin1<br>-sbegin_sequences</td>
<td>integer</td>
<td>Start of each sequence to be used</td>
<td>Any integer value</td>
<td>0</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -send1<br>-send_sequences</td>
<td>integer</td>
<td>End of each sequence to be used</td>
<td>Any integer value</td>
<td>0</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -sreverse1<br>-sreverse_sequences</td>
<td>boolean</td>
<td>Reverse (if DNA)</td>
<td>Boolean value Yes/No</td>
<td>N</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -sask1<br>-sask_sequences</td>
<td>boolean</td>
<td>Ask for begin/end/reverse</td>
<td>Boolean value Yes/No</td>
<td>N</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -snucleotide1<br>-snucleotide_sequences</td>
<td>boolean</td>
<td>Sequence is nucleotide</td>
<td>Boolean value Yes/No</td>
<td>N</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -sprotein1<br>-sprotein_sequences</td>
<td>boolean</td>
<td>Sequence is protein</td>
<td>Boolean value Yes/No</td>
<td>N</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -slower1<br>-slower_sequences</td>
<td>boolean</td>
<td>Make lower case</td>
<td>Boolean value Yes/No</td>
<td>N</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -supper1<br>-supper_sequences</td>
<td>boolean</td>
<td>Make upper case</td>
<td>Boolean value Yes/No</td>
<td>N</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -scircular1<br>-scircular_sequences</td>
<td>boolean</td>
<td>Sequence is circular</td>
<td>Boolean value Yes/No</td>
<td>N</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -squick1<br>-squick_sequences</td>
<td>boolean</td>
<td>Read id and sequence only</td>
<td>Boolean value Yes/No</td>
<td>N</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -sformat1<br>-sformat_sequences</td>
<td>string</td>
<td>Input sequence format</td>
<td>Any string</td>
<td>&nbsp;</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -iquery1<br>-iquery_sequences</td>
<td>string</td>
<td>Input query fields or ID list</td>
<td>Any string</td>
<td>&nbsp;</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -ioffset1<br>-ioffset_sequences</td>
<td>integer</td>
<td>Input start position offset</td>
<td>Any integer value</td>
<td>0</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -sdbname1<br>-sdbname_sequences</td>
<td>string</td>
<td>Database name</td>
<td>Any string</td>
<td>&nbsp;</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -sid1<br>-sid_sequences</td>
<td>string</td>
<td>Entryname</td>
<td>Any string</td>
<td>&nbsp;</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -ufo1<br>-ufo_sequences</td>
<td>string</td>
<td>UFO features</td>
<td>Any string</td>
<td>&nbsp;</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -fformat1<br>-fformat_sequences</td>
<td>string</td>
<td>Features format</td>
<td>Any string</td>
<td>&nbsp;</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -fopenfile1<br>-fopenfile_sequences</td>
<td>string</td>
<td>Features file name</td>
<td>Any string</td>
<td>&nbsp;</td>
</tr>

<tr bgcolor="#FFFFCC">
<td align="left" colspan=5>"-graph" associated graph qualifiers
</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -gprompt</td>
<td>boolean</td>
<td>Graph prompting</td>
<td>Boolean value Yes/No</td>
<td>N</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -gdesc</td>
<td>string</td>
<td>Graph description</td>
<td>Any string</td>
<td>Pretty plot</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -gtitle</td>
<td>string</td>
<td>Graph title</td>
<td>Any string</td>
<td>&nbsp;</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -gsubtitle</td>
<td>string</td>
<td>Graph subtitle</td>
<td>Any string</td>
<td>&nbsp;</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -gxtitle</td>
<td>string</td>
<td>Graph x axis title</td>
<td>Any string</td>
<td>&nbsp;</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -gytitle</td>
<td>string</td>
<td>Graph y axis title</td>
<td>Any string</td>
<td>&nbsp;</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -goutfile</td>
<td>string</td>
<td>Output file for non interactive displays</td>
<td>Any string</td>
<td>&nbsp;</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -gdirectory</td>
<td>string</td>
<td>Output directory</td>
<td>Any string</td>
<td>&nbsp;</td>
</tr>

<tr bgcolor="#FFFFCC">
<th align="left" colspan=5>General qualifiers</th>
</tr>

<tr bgcolor="#FFFFCC">
<td> -auto</td>
<td>boolean</td>
<td>Turn off prompts</td>
<td>Boolean value Yes/No</td>
<td>N</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -stdout</td>
<td>boolean</td>
<td>Write first file to standard output</td>
<td>Boolean value Yes/No</td>
<td>N</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -filter</td>
<td>boolean</td>
<td>Read first file from standard input, write first file to standard output</td>
<td>Boolean value Yes/No</td>
<td>N</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -options</td>
<td>boolean</td>
<td>Prompt for standard and additional values</td>
<td>Boolean value Yes/No</td>
<td>N</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -debug</td>
<td>boolean</td>
<td>Write debug output to program.dbg</td>
<td>Boolean value Yes/No</td>
<td>N</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -verbose</td>
<td>boolean</td>
<td>Report some/full command line options</td>
<td>Boolean value Yes/No</td>
<td>Y</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -help</td>
<td>boolean</td>
<td>Report command line options and exit. More information on associated and general qualifiers can be found with -help -verbose</td>
<td>Boolean value Yes/No</td>
<td>N</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -warning</td>
<td>boolean</td>
<td>Report warnings</td>
<td>Boolean value Yes/No</td>
<td>Y</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -error</td>
<td>boolean</td>
<td>Report errors</td>
<td>Boolean value Yes/No</td>
<td>Y</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -fatal</td>
<td>boolean</td>
<td>Report fatal errors</td>
<td>Boolean value Yes/No</td>
<td>Y</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -die</td>
<td>boolean</td>
<td>Report dying program messages</td>
<td>Boolean value Yes/No</td>
<td>Y</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -version</td>
<td>boolean</td>
<td>Report version number and exit</td>
<td>Boolean value Yes/No</td>
<td>N</td>
</tr>

</table>


<H2>
    Input file format
</H2>

<b>prettyplot</b> reads aligned protein sequences.
<p>
<p>

The input is a standard EMBOSS sequence query (also known as a 'USA').

<p>
Major sequence database sources defined as standard in EMBOSS
installations include srs:embl, srs:uniprot and ensembl

<p>

Data can also be read from sequence output in any supported format
written by an EMBOSS or third-party application.

<p>
The input format can be specified by using the
command-line qualifier <tt>-sformat xxx</tt>, where 'xxx' is replaced
by the name of the required format.  The available format names are:
gff (gff3), gff2, embl (em), genbank (gb, refseq), ddbj, refseqp, pir
(nbrf), swissprot (swiss, sw), dasgff and debug.

<p>

See:
<A href="http://emboss.sf.net/docs/themes/SequenceFormats.html">
http://emboss.sf.net/docs/themes/SequenceFormats.html</A>
for further information on sequence formats.

<p>


<a name="input.1"></a>
<h3>Input files for usage example </h3>
<p><h3>File: globins.msf</h3>
<table width="90%"><tr><td bgcolor="#FFCCFF">
<pre>
!!AA_MULTIPLE_ALIGNMENT 1.0

  ../data/globins.msf MSF:  164 Type: P 25/06/01 CompCheck: 4278 ..

  Name: HBB_HUMAN Len: 164  Check: 6914 Weight: 0.61
  Name: HBB_HORSE Len: 164  Check: 6007 Weight: 0.65
  Name: HBA_HUMAN Len: 164  Check: 3921 Weight: 0.65
  Name: HBA_HORSE Len: 164  Check: 4770 Weight: 0.83
  Name: MYG_PHYCA Len: 164  Check: 7930 Weight: 1.00
  Name: GLB5_PETMA Len: 164  Check: 1857 Weight: 0.91
  Name: LGB2_LUPLU Len: 164  Check: 2879 Weight: 0.43

//

           1                                               50
HBB_HUMAN  ~~~~~~~~VHLTPEEKSAVTALWGKVN.VDEVGGEALGR.LLVVYPWTQR
HBB_HORSE  ~~~~~~~~VQLSGEEKAAVLALWDKVN.EEEVGGEALGR.LLVVYPWTQR
HBA_HUMAN  ~~~~~~~~~~~~~~VLSPADKTNVKAA.WGKVGAHAGEYGAEALERMFLS
HBA_HORSE  ~~~~~~~~~~~~~~VLSAADKTNVKAA.WSKVGGHAGEYGAEALERMFLG
MYG_PHYCA  ~~~~~~~VLSEGEWQLVLHVWAKVEAD.VAGHGQDILIR.LFKSHPETLE
GLB5_PETMA PIVDTGSVAPLSAAEKTKIRSAWAPVYSTYETSGVDILVKFFTSTPAAQE
LGB2_LUPLU ~~~~~~~~GALTESQAALVKSSWEEFNANIPKHTHRFFILVLEIAPAAKD

           51                                             100
HBB_HUMAN  FFESFGDLSTPDAVMGNPKVKAHGKKVLGAFSDGLAHLDNLKGTFATLSE
HBB_HORSE  FFDSFGDLSNPGAVMGNPKVKAHGKKVLHSFGEGVHHLDNLKGTFAALSE
HBA_HUMAN  FPTTKTYFPHFDLSHGSAQVKGHGKKVADALTNAVAHVDDMPNALSALSD
HBA_HORSE  FPTTKTYFPHFDLSHGSAQVKAHGKKVGDALTLAVGHLDDLPGALSNLSD
MYG_PHYCA  KFDRFKHLKTEAEMKASEDLKKHGVTVLTALGAILKKKGHHEAELKPLAQ
GLB5_PETMA FFPKFKGLTTADQLKKSADVRWHAERIINAVNDAVASMDDTEKMSMKLRD
LGB2_LUPLU LFSFLKGTSEVPQNNPELQAHAGKVFKLVYEAAIQLQVTGVVVTDATLKN

           101                                            150
HBB_HUMAN  LHCDKLH..VDPENFRLLGNVLVCVLAHHFGKEFTPPVQAAYQKVVAGVA
HBB_HORSE  LHCDKLH..VDPENFRLLGNVLVVVLARHFGKDFTPELQASYQKVVAGVA
HBA_HUMAN  LHAHKLR..VDPVNFKLLSHCLLVTLAAHLPAEFTPAVHASLDKFLASVS
HBA_HORSE  LHAHKLR..VDPVNFKLLSHCLLSTLAVHLPNDFTPAVHASLDKFLSSVS
MYG_PHYCA  SHATKHK..IPIKYLEFISEAIIHVLHSRHPGDFGADAQGAMNKALELFR
GLB5_PETMA LSGKHAK..SFQVDPQYFKVLAAVIADTVAAGDAGFEKLMSMICILLRSA
LGB2_LUPLU LGSVHVSKGVADAHFPVVKEAILKTIKEVVGAKWSEELNSAWTIAYDELA

           151        164
HBB_HUMAN  NALAHKYH~~~~~~
HBB_HORSE  NALAHKYH~~~~~~
HBA_HUMAN  TVLTSKYR~~~~~~
HBA_HORSE  TVLTSKYR~~~~~~
MYG_PHYCA  KDIAAKYKELGYQG
GLB5_PETMA Y~~~~~~~~~~~~~
LGB2_LUPLU IVIKKEMNDAA~~~

</pre>
</td></tr></table><p>

<H2>
    Output file format
</H2>

<p>

The output is to the specified graphics device.

<p>

The results can be output in one of several formats by using the
command-line qualifier <b>-graph xxx</b>, where 'xxx' is replaced by
the name of the required device. Support depends on the availability
of third-party software packages.

<p>
The device names that output to a file are:
ps (postscript), cps (colourps), png, gif, pdf, svg, hpgl, hp7470,
hp7580, das, data.

<p> The other available device names are: meta, x11 (xwindows), tek
(tek4107t), tekt (tektronix), xterm, text.

<p>
Output can be turned off by specifying none (null).

<p>

See:
<A href="http://emboss.sf.net/docs/themes/GraphicsDevices.html">
http://emboss.sf.net/docs/themes/GraphicsDevices.html</A>
for further information on supported devices.

<p>

<p>


<a name="output.1"></a>
<h3>Output files for usage example </h3>
<p><h3>Graphics File: prettyplot.ps</h3>
<p><img src="prettyplot.1.prettyplot.gif" alt="[prettyplot results]">

<a name="output.2"></a>
<h3>Output files for usage example 2</h3>
<p><h3>Graphics File: prettyplot.ps</h3>
<p><img src="prettyplot.2.prettyplot.gif" alt="[prettyplot results]">

<H2>
    Data files
</H2>


Prettyplot uses a comparison matrix file to calculate similarity to
the consensus.

<p>
For protein sequences EBLOSUM62 is used for the substitution matrix.
For nucleotide sequence, EDNAFULL is used.

<p>
EMBOSS data files are distributed with the application and stored
in the standard EMBOSS data directory, which is defined
by the EMBOSS environment variable EMBOSS_DATA.

<p>

To see the available EMBOSS data files, run:
<p>
<pre>
% embossdata -showall
</pre>
<p>
To fetch one of the data files (for example 'Exxx.dat') into your
current directory for you to inspect or modify, run:

<pre>

% embossdata -fetch -file Exxx.dat

</pre>
<p>

Users can provide their own data files in their own directories.
Project specific files can be put in the current directory, or for
tidier directory listings in a subdirectory called
".embossdata". Files for all EMBOSS runs can be put in the user's home
directory, or again in a subdirectory called ".embossdata".

<p>
The directories are searched in the following order:

<ul>
   <li> . (your current directory)
   <li> .embossdata (under your current directory)
   <li> ~/ (your home directory)
   <li> ~/.embossdata
</ul>
<p>

<H2>
    Notes
</H2>

<p>A consesnsus sequence is calculated for the alignment and
individual sequences compared to the consensus using the specified
comparison matrix file. The default matrix for protein sequences
is <tt>EBLOSUM62</tt> and for nucleotide sequences
is <tt>EDNAFULL</tt>. The drawing options render conserved sites and
regions identified from the comparisons.  For example, residues in a
sequence are classed as "identical", "similar" or "other" to the
consensus depending on user-specified thresholds of sequence
similarity (<tt>-pair</tt> option).  Residues in each class are
rendered red, green and black by default (this can be changed).</p>

<p>There are other more general drawing options, for example,
controlling the number of residues displayed per line, background
shading and whether to display sequence names or not.</p>

<H2>
    References
</H2>

None.

<H2>
    Warnings
</H2>

None.

<H2>
    Diagnostic Error Messages
</H2>

None.

<H2>
    Exit status
</H2>

It exits with status 0 unless an error is reported.

<H2>
    Known bugs
</H2>


Portrait mode does not cover the whole page! This is a "feature" in
plplot.

<h2><a name="See also">See also</a></h2>
<table border cellpadding=4 bgcolor="#FFFFF0">
<tr><th>Program name</th>
<th>Description</th></tr>
<tr>
<td><a href="abiview.html">abiview</a></td>
<td>Display the trace in an ABI sequencer file</td>
</tr>

<tr>
<td><a href="cirdna.html">cirdna</a></td>
<td>Draw circular map of DNA constructs</td>
</tr>

<tr>
<td><a href="edialign.html">edialign</a></td>
<td>Local multiple alignment of sequences</td>
</tr>

<tr>
<td><a href="emma.html">emma</a></td>
<td>Multiple sequence alignment (ClustalW wrapper)</td>
</tr>

<tr>
<td><a href="iep.html">iep</a></td>
<td>Calculate the isoelectric point of proteins</td>
</tr>

<tr>
<td><a href="infoalign.html">infoalign</a></td>
<td>Display basic information about a multiple sequence alignment</td>
</tr>

<tr>
<td><a href="lindna.html">lindna</a></td>
<td>Draw linear maps of DNA constructs</td>
</tr>

<tr>
<td><a href="pepinfo.html">pepinfo</a></td>
<td>Plot amino acid properties of a protein sequence in parallel</td>
</tr>

<tr>
<td><a href="pepnet.html">pepnet</a></td>
<td>Draw a helical net for a protein sequence</td>
</tr>

<tr>
<td><a href="pepwheel.html">pepwheel</a></td>
<td>Draw a helical wheel diagram for a protein sequence</td>
</tr>

<tr>
<td><a href="plotcon.html">plotcon</a></td>
<td>Plot conservation of a sequence alignment</td>
</tr>

<tr>
<td><a href="plotorf.html">plotorf</a></td>
<td>Plot potential open reading frames in a nucleotide sequence</td>
</tr>

<tr>
<td><a href="prettyseq.html">prettyseq</a></td>
<td>Write a nucleotide sequence and its translation to file</td>
</tr>

<tr>
<td><a href="remap.html">remap</a></td>
<td>Display restriction enzyme binding sites in a nucleotide sequence</td>
</tr>

<tr>
<td><a href="showalign.html">showalign</a></td>
<td>Display a multiple sequence alignment in pretty format</td>
</tr>

<tr>
<td><a href="showfeat.html">showfeat</a></td>
<td>Display features of a sequence in pretty format</td>
</tr>

<tr>
<td><a href="showpep.html">showpep</a></td>
<td>Display protein sequences with features in pretty format</td>
</tr>

<tr>
<td><a href="sixpack.html">sixpack</a></td>
<td>Display a DNA sequence with 6-frame translation and ORFs</td>
</tr>

<tr>
<td><a href="tranalign.html">tranalign</a></td>
<td>Generate an alignment of nucleic coding regions from aligned proteins</td>
</tr>

</table>

<H2>
    Author(s)
</H2>
Ian Longden formerly at:
<br>
Sanger Institute, Wellcome Trust Genome Campus, Hinxton,
Cambridge, CB10 1SA, UK.                      

<p>
Please report all bugs to the EMBOSS bug team (emboss-bug&nbsp;&copy;&nbsp;emboss.open-bio.org) not to the original author.


<p>

Many features were first implemented in the EGCG program "prettyplot" by
Peter Rice
<br>
European Bioinformatics Institute, Wellcome Trust Genome Campus, Hinxton, Cambridge CB10 1SD, UK

<p>
Please report all bugs to the EMBOSS bug team (emboss-bug&nbsp;&copy;&nbsp;emboss.open-bio.org) not to the original author.

<p> 

The original suggestions for the PrettyPlot program were from Denis
Duboule and Sigfried Labeit at EMBL. Gert Vriend added the star
marking. Rita Grandori suggested the -NOCOLLISION option. 

<H2>
    History
</H2>

Completed 5th May 1999.

<H2>
    Target users
</H2>
This program is intended to be used by everyone and everything, from naive users to embedded scripts.

<H2>
    Comments
</H2>
None

</BODY>
</HTML>

