<HTML>

<HEAD>
  <TITLE>
  EMBOSS: prophet
  </TITLE>
</HEAD>
<BODY BGCOLOR="#FFFFFF" text="#000000">

<table align=center border=0 cellspacing=0 cellpadding=0>
<tr><td valign=top>
<A HREF="/" ONMOUSEOVER="self.status='Go to the EMBOSS home page';return true"><img border=0 src="/images/emboss_icon.jpg" alt="" width=150 height=48></a>
</td>
<td align=left valign=middle>
<b><font size="+6">
prophet
</font></b>
</td></tr>
</table>
<br>&nbsp;
<p>


<H2>
Wiki
</H2>

The master copies of EMBOSS documentation are available
at <a href="http://emboss.open-bio.org/wiki/Appdocs">
http://emboss.open-bio.org/wiki/Appdocs</a>
on the EMBOSS Wiki.

<p>
Please help by correcting and extending the Wiki pages.

<H2>
    Function
</H2>
Scan one or more sequences with a Gribskov or Henikoff profile

<H2>
    Description
</H2>

<p>prophet scans one or more sequences with the supplied Gribskov or Henikoff profile (produced by prophecy).  It writes the the highest-scoring (gapped) alignment for each sequence to file, which includes the score for the match.  Gap opening coefficient and gap extension coefficient must be specified.</p>



<H2>
Algorithm
</H2>
<p>prophet does NOT use the 'simple frequency matrices' produced by prophecy. If you have a 'simple frequency matrix'you should use the program profit to scan sequences.</p>



<H2>
    Usage
</H2>
Here is a sample session with <b>prophet</b>
<p>

<p>
<table width="90%"><tr><td bgcolor="#CCFFFF"><pre>

% <b>prophet </b>
Scan one or more sequences with a Gribskov or Henikoff profile
Input sequence(s): <b>tsw:h*</b>
Profile or weight matrix file: <b>globins.gribskov</b>
Gap opening coefficient [1.0]: <b></b>
Gap extension coefficient [1.0]: <b></b>
Output alignment [hba_human.prophet]: <b></b>

</pre></td></tr></table><p>
<p>
<a href="#input.1">Go to the input files for this example</a><br><a href="#output.1">Go to the output files for this example</a><p><p>


<H2>
    Command line arguments
</H2>
<table CELLSPACING=0 CELLPADDING=3 BGCOLOR="#f5f5ff" ><tr><td>
<pre>
Scan one or more sequences with a Gribskov or Henikoff profile
Version: EMBOSS:6.6.0.0

   Standard (Mandatory) qualifiers:
  [-sequence]          seqall     Sequence(s) filename and optional format, or
                                  reference (input USA)
  [-infile]            infile     Profile or weight matrix file
   -gapopen            float      [1.0] Gap opening coefficient (Number from
                                  0.000 to 100.000)
   -gapextend          float      [1.0] Gap extension coefficient (Number from
                                  0.000 to 100.000)
  [-outfile]           align      [*.prophet] Output alignment file name
                                  (default -aformat simple)

   Additional (Optional) qualifiers: (none)
   Advanced (Unprompted) qualifiers: (none)
   Associated qualifiers:

   "-sequence" associated qualifiers
   -sbegin1            integer    Start of each sequence to be used
   -send1              integer    End of each sequence to be used
   -sreverse1          boolean    Reverse (if DNA)
   -sask1              boolean    Ask for begin/end/reverse
   -snucleotide1       boolean    Sequence is nucleotide
   -sprotein1          boolean    Sequence is protein
   -slower1            boolean    Make lower case
   -supper1            boolean    Make upper case
   -scircular1         boolean    Sequence is circular
   -squick1            boolean    Read id and sequence only
   -sformat1           string     Input sequence format
   -iquery1            string     Input query fields or ID list
   -ioffset1           integer    Input start position offset
   -sdbname1           string     Database name
   -sid1               string     Entryname
   -ufo1               string     UFO features
   -fformat1           string     Features format
   -fopenfile1         string     Features file name

   "-outfile" associated qualifiers
   -aformat3           string     Alignment format
   -aextension3        string     File name extension
   -adirectory3        string     Output directory
   -aname3             string     Base file name
   -awidth3            integer    Alignment width
   -aaccshow3          boolean    Show accession number in the header
   -adesshow3          boolean    Show description in the header
   -ausashow3          boolean    Show the full USA in the alignment
   -aglobal3           boolean    Show the full sequence in alignment

   General qualifiers:
   -auto               boolean    Turn off prompts
   -stdout             boolean    Write first file to standard output
   -filter             boolean    Read first file from standard input, write
                                  first file to standard output
   -options            boolean    Prompt for standard and additional values
   -debug              boolean    Write debug output to program.dbg
   -verbose            boolean    Report some/full command line options
   -help               boolean    Report command line options and exit. More
                                  information on associated and general
                                  qualifiers can be found with -help -verbose
   -warning            boolean    Report warnings
   -error              boolean    Report errors
   -fatal              boolean    Report fatal errors
   -die                boolean    Report dying program messages
   -version            boolean    Report version number and exit

</pre>
</td></tr></table>
<P>

<table border cellspacing=0 cellpadding=3 bgcolor="#ccccff">
<tr bgcolor="#FFFFCC">
<th align="left">Qualifier</th>
<th align="left">Type</th>
<th align="left">Description</th>
<th align="left">Allowed values</th>
<th align="left">Default</th>
</tr>

<tr bgcolor="#FFFFCC">
<th align="left" colspan=5>Standard (Mandatory) qualifiers</th>
</tr>

<tr bgcolor="#FFFFCC">
<td>[-sequence]<br>(Parameter 1)</td>
<td>seqall</td>
<td>Sequence(s) filename and optional format, or reference (input USA)</td>
<td>Readable sequence(s)</td>
<td><b>Required</b></td>
</tr>

<tr bgcolor="#FFFFCC">
<td>[-infile]<br>(Parameter 2)</td>
<td>infile</td>
<td>Profile or weight matrix file</td>
<td>Input file</td>
<td><b>Required</b></td>
</tr>

<tr bgcolor="#FFFFCC">
<td>-gapopen</td>
<td>float</td>
<td>Gap opening coefficient</td>
<td>Number from 0.000 to 100.000</td>
<td>1.0</td>
</tr>

<tr bgcolor="#FFFFCC">
<td>-gapextend</td>
<td>float</td>
<td>Gap extension coefficient</td>
<td>Number from 0.000 to 100.000</td>
<td>1.0</td>
</tr>

<tr bgcolor="#FFFFCC">
<td>[-outfile]<br>(Parameter 3)</td>
<td>align</td>
<td>Output alignment file name</td>
<td>(default -aformat simple)</td>
<td><i>&lt;*&gt;</i>.prophet</td>
</tr>

<tr bgcolor="#FFFFCC">
<th align="left" colspan=5>Additional (Optional) qualifiers</th>
</tr>

<tr>
<td colspan=5>(none)</td>
</tr>

<tr bgcolor="#FFFFCC">
<th align="left" colspan=5>Advanced (Unprompted) qualifiers</th>
</tr>

<tr>
<td colspan=5>(none)</td>
</tr>

<tr bgcolor="#FFFFCC">
<th align="left" colspan=5>Associated qualifiers</th>
</tr>

<tr bgcolor="#FFFFCC">
<td align="left" colspan=5>"-sequence" associated seqall qualifiers
</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -sbegin1<br>-sbegin_sequence</td>
<td>integer</td>
<td>Start of each sequence to be used</td>
<td>Any integer value</td>
<td>0</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -send1<br>-send_sequence</td>
<td>integer</td>
<td>End of each sequence to be used</td>
<td>Any integer value</td>
<td>0</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -sreverse1<br>-sreverse_sequence</td>
<td>boolean</td>
<td>Reverse (if DNA)</td>
<td>Boolean value Yes/No</td>
<td>N</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -sask1<br>-sask_sequence</td>
<td>boolean</td>
<td>Ask for begin/end/reverse</td>
<td>Boolean value Yes/No</td>
<td>N</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -snucleotide1<br>-snucleotide_sequence</td>
<td>boolean</td>
<td>Sequence is nucleotide</td>
<td>Boolean value Yes/No</td>
<td>N</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -sprotein1<br>-sprotein_sequence</td>
<td>boolean</td>
<td>Sequence is protein</td>
<td>Boolean value Yes/No</td>
<td>N</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -slower1<br>-slower_sequence</td>
<td>boolean</td>
<td>Make lower case</td>
<td>Boolean value Yes/No</td>
<td>N</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -supper1<br>-supper_sequence</td>
<td>boolean</td>
<td>Make upper case</td>
<td>Boolean value Yes/No</td>
<td>N</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -scircular1<br>-scircular_sequence</td>
<td>boolean</td>
<td>Sequence is circular</td>
<td>Boolean value Yes/No</td>
<td>N</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -squick1<br>-squick_sequence</td>
<td>boolean</td>
<td>Read id and sequence only</td>
<td>Boolean value Yes/No</td>
<td>N</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -sformat1<br>-sformat_sequence</td>
<td>string</td>
<td>Input sequence format</td>
<td>Any string</td>
<td>&nbsp;</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -iquery1<br>-iquery_sequence</td>
<td>string</td>
<td>Input query fields or ID list</td>
<td>Any string</td>
<td>&nbsp;</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -ioffset1<br>-ioffset_sequence</td>
<td>integer</td>
<td>Input start position offset</td>
<td>Any integer value</td>
<td>0</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -sdbname1<br>-sdbname_sequence</td>
<td>string</td>
<td>Database name</td>
<td>Any string</td>
<td>&nbsp;</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -sid1<br>-sid_sequence</td>
<td>string</td>
<td>Entryname</td>
<td>Any string</td>
<td>&nbsp;</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -ufo1<br>-ufo_sequence</td>
<td>string</td>
<td>UFO features</td>
<td>Any string</td>
<td>&nbsp;</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -fformat1<br>-fformat_sequence</td>
<td>string</td>
<td>Features format</td>
<td>Any string</td>
<td>&nbsp;</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -fopenfile1<br>-fopenfile_sequence</td>
<td>string</td>
<td>Features file name</td>
<td>Any string</td>
<td>&nbsp;</td>
</tr>

<tr bgcolor="#FFFFCC">
<td align="left" colspan=5>"-outfile" associated align qualifiers
</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -aformat3<br>-aformat_outfile</td>
<td>string</td>
<td>Alignment format</td>
<td>Any string</td>
<td>simple</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -aextension3<br>-aextension_outfile</td>
<td>string</td>
<td>File name extension</td>
<td>Any string</td>
<td>&nbsp;</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -adirectory3<br>-adirectory_outfile</td>
<td>string</td>
<td>Output directory</td>
<td>Any string</td>
<td>&nbsp;</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -aname3<br>-aname_outfile</td>
<td>string</td>
<td>Base file name</td>
<td>Any string</td>
<td>&nbsp;</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -awidth3<br>-awidth_outfile</td>
<td>integer</td>
<td>Alignment width</td>
<td>Any integer value</td>
<td>0</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -aaccshow3<br>-aaccshow_outfile</td>
<td>boolean</td>
<td>Show accession number in the header</td>
<td>Boolean value Yes/No</td>
<td>N</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -adesshow3<br>-adesshow_outfile</td>
<td>boolean</td>
<td>Show description in the header</td>
<td>Boolean value Yes/No</td>
<td>N</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -ausashow3<br>-ausashow_outfile</td>
<td>boolean</td>
<td>Show the full USA in the alignment</td>
<td>Boolean value Yes/No</td>
<td>N</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -aglobal3<br>-aglobal_outfile</td>
<td>boolean</td>
<td>Show the full sequence in alignment</td>
<td>Boolean value Yes/No</td>
<td>N</td>
</tr>

<tr bgcolor="#FFFFCC">
<th align="left" colspan=5>General qualifiers</th>
</tr>

<tr bgcolor="#FFFFCC">
<td> -auto</td>
<td>boolean</td>
<td>Turn off prompts</td>
<td>Boolean value Yes/No</td>
<td>N</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -stdout</td>
<td>boolean</td>
<td>Write first file to standard output</td>
<td>Boolean value Yes/No</td>
<td>N</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -filter</td>
<td>boolean</td>
<td>Read first file from standard input, write first file to standard output</td>
<td>Boolean value Yes/No</td>
<td>N</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -options</td>
<td>boolean</td>
<td>Prompt for standard and additional values</td>
<td>Boolean value Yes/No</td>
<td>N</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -debug</td>
<td>boolean</td>
<td>Write debug output to program.dbg</td>
<td>Boolean value Yes/No</td>
<td>N</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -verbose</td>
<td>boolean</td>
<td>Report some/full command line options</td>
<td>Boolean value Yes/No</td>
<td>Y</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -help</td>
<td>boolean</td>
<td>Report command line options and exit. More information on associated and general qualifiers can be found with -help -verbose</td>
<td>Boolean value Yes/No</td>
<td>N</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -warning</td>
<td>boolean</td>
<td>Report warnings</td>
<td>Boolean value Yes/No</td>
<td>Y</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -error</td>
<td>boolean</td>
<td>Report errors</td>
<td>Boolean value Yes/No</td>
<td>Y</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -fatal</td>
<td>boolean</td>
<td>Report fatal errors</td>
<td>Boolean value Yes/No</td>
<td>Y</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -die</td>
<td>boolean</td>
<td>Report dying program messages</td>
<td>Boolean value Yes/No</td>
<td>Y</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -version</td>
<td>boolean</td>
<td>Report version number and exit</td>
<td>Boolean value Yes/No</td>
<td>N</td>
</tr>

</table>


<H2>
    Input file format
</H2>

<b>prophet</b> reads one or more nucleotide or protein sequences.
<p>
<p>

The input is a standard EMBOSS sequence query (also known as a 'USA').

<p>
Major sequence database sources defined as standard in EMBOSS
installations include srs:embl, srs:uniprot and ensembl

<p>

Data can also be read from sequence output in any supported format
written by an EMBOSS or third-party application.

<p>
The input format can be specified by using the
command-line qualifier <tt>-sformat xxx</tt>, where 'xxx' is replaced
by the name of the required format.  The available format names are:
gff (gff3), gff2, embl (em), genbank (gb, refseq), ddbj, refseqp, pir
(nbrf), swissprot (swiss, sw), dasgff and debug.

<p>

See:
<A href="http://emboss.sf.net/docs/themes/SequenceFormats.html">
http://emboss.sf.net/docs/themes/SequenceFormats.html</A>
for further information on sequence formats.


<b>prophet</b> also reads a GRIBSKOV or a HENIKOFF profile produced by
<b>prophecy</b>. 

<p>


<a name="input.1"></a>
<h3>Input files for usage example </h3>

'tsw:h*' is a sequence entry in the example protein database 'tsw'
<p>
<p><h3>File: globins.gribskov</h3>
<table width="90%"><tr><td bgcolor="#FFCCFF">
<pre>
# Gribskov Protein Profile
# Columns are amino acids A-&gt;Z
# Last column is indel penalty
# Rows are alignment positions 1-&gt;n
Gribskov
Name		globins
Matrix		pprofile
Length		164
Max_score	645.45
Threshold	75
Gap_open	3.00
Gap_extend	0.30
Consensus	PIVDTGSVVSLSEEELSAVDKAWVKANSVAEVGGHALERGLFASEPMTLEFFDTFKYLSTFDLSKGSADVKAHGKKVLDALGDAVAHLDDLEGTLAALSDLHAHKLKKGVDPVNFKLLSHCLLVVLASHLPGDFTPEVQASMDKFLASVATVLASKYRELGYQG
0.15 0.00 0.03 0.03 0.03 -0.18 0.09 0.06 -0.06 0.00 0.03 -0.09 -0.06 0.00 0.00 0.45 0.09 0.09 0.12 0.09 0.00 0.03 -0.24 0.00 -0.24 0.00 0.87 0.87
0.00 0.00 0.06 -0.06 -0.06 0.18 -0.09 -0.09 0.45 0.00 -0.06 0.24 0.18 -0.09 0.00 -0.06 -0.09 -0.09 -0.03 0.06 0.00 0.33 -0.15 0.00 0.03 0.00 0.87 0.87
0.06 0.00 0.06 -0.06 -0.06 0.06 0.06 -0.09 0.33 0.00 -0.06 0.24 0.18 -0.09 0.00 0.03 -0.06 -0.09 -0.03 0.06 0.00 0.45 -0.24 0.00 -0.03 0.00 0.87 0.87
0.09 0.00 -0.15 0.45 0.30 -0.30 0.18 0.12 -0.06 0.00 0.09 -0.15 -0.12 0.18 0.00 0.03 0.18 0.00 0.06 0.06 0.00 -0.06 -0.33 0.00 -0.15 0.00 0.87 0.87
0.12 0.00 0.06 0.06 0.06 -0.09 0.12 -0.03 0.06 0.00 0.06 -0.03 0.00 0.06 0.00 0.09 -0.03 -0.03 0.09 0.45 0.00 0.06 -0.18 0.00 -0.09 0.00 0.87 0.87
0.18 0.00 0.06 0.18 0.15 -0.18 0.45 -0.06 -0.09 0.00 -0.03 -0.15 -0.09 0.12 0.00 0.09 0.06 -0.09 0.18 0.12 0.00 0.06 -0.30 0.00 -0.18 0.00 0.87 0.87
0.12 0.00 0.18 0.06 0.06 -0.09 0.18 -0.06 -0.03 0.00 0.06 -0.12 -0.09 0.09 0.00 0.12 -0.03 0.03 0.45 0.09 0.00 -0.03 0.09 0.00 -0.12 0.00 0.87 0.87
0.13 0.00 0.13 -0.13 -0.13 0.13 0.13 -0.19 0.69 0.00 -0.13 0.50 0.38 -0.19 0.00 0.06 -0.13 -0.19 -0.06 0.13 0.00 0.94 -0.50 0.00 -0.06 0.00 0.87 0.87
0.59 0.00 -0.06 -0.07 -0.02 0.24 0.31 -0.25 0.68 0.00 -0.20 0.72 0.60 -0.14 0.00 0.14 -0.03 -0.39 0.03 0.23 0.00 0.97 -0.55 0.00 -0.12 0.00 0.87 0.87
0.52 0.00 0.12 0.35 0.35 -0.54 0.38 0.41 -0.22 0.00 0.20 -0.30 -0.22 0.31 0.00 0.76 0.53 0.27 0.61 0.20 0.00 -0.08 -0.38 0.00 -0.48 0.00 0.87 0.87
0.01 0.00 -0.88 -0.10 0.24 0.83 -0.26 -0.04 0.62 0.00 -0.16 1.19 0.96 -0.18 0.00 -0.22 0.11 -0.34 -0.28 -0.02 0.00 0.62 0.07 0.00 0.09 0.00 0.87 0.87
0.54 0.00 0.44 0.37 0.34 -0.45 0.94 -0.20 -0.08 0.00 0.14 -0.40 -0.25 0.35 0.00 0.41 -0.02 -0.08 1.07 0.80 0.00 0.08 -0.38 0.00 -0.51 0.00 0.87 0.87
0.82 0.00 -0.13 0.71 0.92 -0.68 0.80 0.16 -0.20 0.00 0.14 -0.34 -0.20 0.38 0.00 0.56 0.45 -0.09 0.42 0.36 0.00 0.03 -1.13 0.00 -0.61 0.00 0.87 0.87
0.37 0.00 -0.47 0.17 0.38 -0.05 0.14 0.07 -0.26 0.00 0.19 -0.05 -0.22 0.21 0.00 -0.01 0.13 0.35 0.52 0.05 0.00 -0.30 -0.16 0.00 0.01 0.00 0.87 0.87
0.41 0.00 -0.61 0.90 1.26 -0.71 0.55 0.42 0.25 0.00 0.31 0.13 0.15 0.40 0.00 0.26 1.04 0.04 0.05 0.19 0.00 0.49 -1.41 0.00 -0.69 0.00 4.55 4.55
0.13 0.00 -1.04 -0.15 0.01 0.48 -0.39 -0.11 0.51 0.00 0.83 0.99 1.12 -0.01 0.00 -0.10 0.23 0.20 -0.13 0.12 0.00 0.54 0.36 0.00 -0.23 0.00 4.55 4.55
1.00 0.00 0.64 0.24 0.24 -0.41 0.81 -0.30 0.35 0.00 0.13 -0.08 -0.01 0.24 0.00 0.58 -0.09 -0.17 1.23 0.86 0.00 0.56 -0.52 0.00 -0.50 0.00 4.55 4.55
1.09 0.00 -0.33 0.08 0.18 -0.09 0.21 -0.09 0.27 0.00 0.33 0.48 0.58 0.07 0.00 0.55 0.27 -0.09 0.23 0.35 0.00 0.48 -0.45 0.00 -0.42 0.00 4.55 4.55
0.81 0.00 0.28 0.11 0.11 0.02 0.25 0.19 0.96 0.00 -0.14 0.57 0.42 0.00 0.00 0.30 0.09 -0.24 0.04 0.33 0.00 1.16 -1.02 0.00 -0.07 0.00 4.55 4.55
0.18 0.00 -0.48 0.64 0.44 -0.46 0.23 0.20 0.36 0.00 0.51 0.16 0.35 0.23 0.00 0.18 0.36 0.36 0.10 0.44 0.00 0.49 -0.41 0.00 -0.54 0.00 4.55 4.55
0.54 0.00 -0.30 0.00 0.00 -0.24 0.14 -0.11 -0.31 0.00 0.85 -0.20 -0.13 0.31 0.00 0.17 0.07 0.74 1.03 0.20 0.00 -0.32 0.34 0.00 -0.23 0.00 4.55 4.55
1.15 0.00 0.04 0.11 0.19 -0.01 0.45 -0.22 0.41 0.00 0.00 0.45 0.45 0.10 0.00 0.39 0.02 -0.39 0.45 0.98 0.00 0.54 -0.54 0.00 -0.27 0.00 4.55 4.55
-0.59 0.00 -1.37 -0.55 -0.60 0.59 -0.70 0.19 -0.64 0.00 0.77 0.14 -0.34 0.60 0.00 -0.65 -0.10 1.43 0.47 -0.35 0.00 -0.90 1.17 0.00 0.70 0.00 4.55 4.55
0.84 0.00 0.10 0.51 0.46 -0.41 0.84 -0.17 0.76 0.00 -0.08 0.37 0.31 0.10 0.00 0.33 0.15 -0.39 0.23 0.43 0.00 1.25 -1.48 0.00 -0.47 0.00 4.55 4.55
0.29 0.00 -0.79 0.77 1.01 -1.00 0.24 0.34 -0.33 0.00 1.52 -0.50 0.03 0.60 0.00 0.59 0.73 0.81 0.39 0.36 0.00 -0.24 -0.67 0.00 -1.02 0.00 4.55 4.55
1.29 0.00 0.37 -0.04 0.02 -0.05 0.55 -0.31 0.87 0.00 -0.23 0.66 0.50 -0.12 0.00 0.39 -0.09 -0.53 0.21 0.43 0.00 1.26 -1.05 0.00 -0.13 0.00 4.55 4.55
0.85 0.00 0.11 0.82 0.60 -0.46 0.53 0.45 -0.20 0.00 0.14 -0.35 -0.33 1.10 0.00 0.04 0.34 -0.27 0.31 0.28 0.00 -0.17 -0.59 0.00 0.08 0.00 4.55 4.55
0.33 0.00 0.22 0.10 0.10 -0.16 0.27 -0.07 -0.03 0.00 0.06 -0.13 -0.09 0.12 0.00 0.19 0.00 -0.01 0.51 0.15 0.00 0.00 -0.02 0.00 -0.16 0.00 3.49 3.49
-0.07 0.00 -0.59 -0.28 -0.19 0.40 -0.10 -0.08 0.31 0.00 0.12 0.51 0.09 0.08 0.00 -0.22 -0.19 0.46 0.27 0.34 0.00 0.38 -0.15 0.00 0.27 0.00 4.55 4.55
0.76 0.00 0.40 0.62 0.60 -0.23 0.69 0.08 0.07 0.00 -0.05 -0.21 -0.21 0.39 0.00 0.11 0.11 -0.36 0.62 0.32 0.00 0.13 -0.59 0.00 -0.08 0.00 4.55 4.55
0.48 0.00 -0.64 1.07 1.40 -1.00 0.84 0.30 -0.37 0.00 0.92 -0.57 -0.17 0.68 0.00 0.43 0.73 0.33 0.49 0.41 0.00 -0.16 -1.18 0.00 -0.96 0.00 4.55 4.55
0.27 0.00 0.12 0.05 0.05 -0.03 0.22 0.21 0.92 0.00 0.13 0.58 0.47 0.01 0.00 0.26 0.04 -0.02 -0.04 0.63 0.00 1.28 -0.92 0.00 -0.17 0.00 4.55 4.55
0.84 0.00 0.41 0.85 0.73 -0.84 1.99 -0.09 -0.44 0.00 -0.05 -0.76 -0.50 0.65 0.00 0.52 0.30 -0.27 1.16 0.57 0.00 0.17 -1.15 0.00 -0.81 0.00 4.55 4.55
1.04 0.00 0.09 0.88 0.78 -1.00 1.73 -0.04 -0.37 0.00 0.06 -0.56 -0.30 0.60 0.00 0.54 0.72 -0.24 0.69 0.66 0.00 0.20 -1.41 0.00 -0.90 0.00 4.55 4.55
0.22 0.00 -0.42 1.10 1.14 -0.58 0.34 1.15 -0.01 0.00 0.23 -0.17 -0.22 0.63 0.00 0.23 0.76 0.22 -0.01 0.15 0.00 0.11 -1.12 0.00 -0.21 0.00 4.55 4.55
1.40 0.00 0.14 0.65 0.50 -0.62 0.58 0.00 0.39 0.00 0.14 -0.03 0.11 0.28 0.00 0.46 0.32 -0.15 0.40 0.47 0.00 0.44 -1.03 0.00 -0.47 0.00 4.55 4.55
0.15 0.00 -0.45 -0.28 -0.13 0.99 0.18 -0.35 0.98 0.00 -0.42 1.28 1.00 -0.26 0.00 -0.22 -0.18 -0.60 -0.08 0.14 0.00 1.05 -0.09 0.00 0.15 0.00 4.55 4.55


<font color=red>  [Part of this file has been deleted for brevity]</font>

0.16 0.00 -0.11 0.88 0.64 -0.34 0.31 0.88 -0.34 0.00 0.27 -0.44 -0.47 1.57 0.00 -0.20 0.43 -0.04 0.17 0.13 0.00 -0.41 -0.25 0.00 0.30 0.00 4.55 4.55
-0.40 0.00 -0.34 -1.18 -0.69 1.78 -0.70 -0.11 0.83 0.00 -0.69 1.65 0.86 -0.65 0.00 -0.27 -0.78 -0.56 -0.32 -0.26 0.00 0.50 1.18 0.00 1.21 0.00 4.55 4.55
0.11 0.00 -0.78 0.67 0.83 -1.02 0.09 0.60 -0.41 0.00 1.29 -0.48 0.09 0.52 0.00 0.51 1.05 1.17 0.23 0.13 0.00 -0.33 -0.04 0.00 -1.00 0.00 4.55 4.55
-0.32 0.00 -0.42 -0.96 -0.65 1.99 -0.80 -0.17 1.10 0.00 -0.68 1.95 1.30 -0.60 0.00 -0.69 -0.56 -0.75 -0.59 -0.25 0.00 0.97 1.06 0.00 1.13 0.00 4.55 4.55
-0.21 0.00 -0.66 -0.84 -0.54 1.76 -0.70 -0.35 1.55 0.00 -0.54 2.09 1.51 -0.65 0.00 -0.50 -0.46 -0.65 -0.50 -0.09 0.00 1.36 0.53 0.00 0.68 0.00 4.55 4.55
0.57 0.00 0.31 0.54 0.50 -0.76 1.07 -0.20 -0.29 0.00 0.79 -0.67 -0.28 0.59 0.00 0.49 0.18 0.31 1.56 0.50 0.00 -0.09 -0.13 0.00 -0.84 0.00 4.55 4.55
0.24 0.00 -0.40 0.85 1.05 -0.48 0.36 1.03 -0.03 0.00 0.30 -0.16 -0.18 1.01 0.00 0.17 0.68 0.19 0.09 0.19 0.00 0.09 -0.93 0.00 -0.16 0.00 4.55 4.55
0.91 0.00 0.71 -0.33 -0.32 0.16 0.31 -0.28 0.79 0.00 -0.46 0.35 0.32 -0.30 0.00 0.24 -0.31 -0.53 0.32 0.34 0.00 1.05 -1.14 0.00 0.39 0.00 4.55 4.55
0.36 0.00 -0.54 -0.45 -0.27 1.21 -0.41 -0.35 1.43 0.00 -0.36 1.70 1.36 -0.44 0.00 -0.21 -0.17 -0.59 -0.29 0.12 0.00 1.30 -0.03 0.00 0.23 0.00 4.55 4.55
0.47 0.00 -0.26 -0.37 -0.25 0.88 -0.15 -0.38 1.45 0.00 -0.34 1.51 1.20 -0.41 0.00 -0.06 -0.18 -0.56 -0.21 0.21 0.00 1.55 -0.42 0.00 0.09 0.00 4.55 4.55
0.28 0.00 0.49 -0.02 -0.04 -0.07 0.27 0.21 0.69 0.00 0.04 0.21 0.17 0.02 0.00 0.28 -0.04 0.03 0.42 0.26 0.00 0.98 -0.76 0.00 0.03 0.00 4.55 4.55
0.40 0.00 0.33 -0.08 -0.08 0.14 0.31 -0.38 1.39 0.00 -0.08 0.77 0.63 -0.19 0.00 0.20 -0.30 -0.38 0.08 1.15 0.00 1.57 -1.12 0.00 -0.23 0.00 4.55 4.55
0.33 0.00 -0.86 -0.55 -0.31 1.41 -0.48 -0.32 1.20 0.00 -0.40 1.93 1.56 -0.47 0.00 -0.25 -0.11 -0.62 -0.39 0.03 0.00 1.20 0.30 0.00 0.29 0.00 4.55 4.55
1.41 0.00 0.00 0.89 0.74 -0.87 0.64 0.54 -0.19 0.00 0.34 -0.35 -0.19 0.58 0.00 0.56 0.61 0.01 0.38 0.42 0.00 -0.01 -1.07 0.00 -0.41 0.00 4.55 4.55
0.59 0.00 0.21 0.36 0.43 -0.45 0.47 0.27 0.17 0.00 0.31 -0.13 0.02 0.31 0.00 0.47 0.22 0.28 0.65 0.68 0.00 0.32 -0.37 0.00 -0.45 0.00 4.55 4.55
-0.10 0.00 -0.10 0.27 0.27 -0.17 -0.19 1.38 0.12 0.00 0.26 0.04 0.06 0.35 0.00 0.32 0.58 0.81 -0.19 -0.03 0.00 0.30 -0.02 0.00 0.03 0.00 4.55 4.55
0.19 0.00 -0.34 -0.46 -0.20 1.05 -0.35 0.28 0.70 0.00 -0.39 1.24 0.78 -0.22 0.00 -0.16 -0.15 -0.37 -0.28 -0.06 0.00 0.65 0.35 0.00 0.68 0.00 4.55 4.55
1.19 0.00 0.28 0.51 0.45 -0.97 1.26 0.02 -0.33 0.00 0.03 -0.55 -0.33 0.28 0.00 1.54 0.42 -0.01 0.78 0.59 0.00 0.25 -1.45 0.00 -1.08 0.00 4.55 4.55
0.97 0.00 -0.10 0.77 0.68 -0.94 1.22 0.02 -0.35 0.00 0.67 -0.58 -0.19 0.90 0.00 0.41 0.47 0.06 0.68 0.53 0.00 0.03 -0.95 0.00 -0.76 0.00 4.55 4.55
0.46 0.00 -0.89 2.13 1.78 -1.45 0.86 0.63 -0.33 0.00 0.67 -0.72 -0.50 0.93 0.00 0.17 0.97 0.11 0.33 0.33 0.00 -0.33 -1.67 0.00 -0.85 0.00 4.55 4.55
-0.28 0.00 -0.20 -1.30 -0.80 1.86 -0.70 -0.17 0.67 0.00 -0.72 1.52 0.57 -0.60 0.00 -0.70 -0.99 -0.52 -0.21 -0.33 0.00 0.19 1.45 0.00 1.66 0.00 4.55 4.55
0.79 0.00 0.39 0.59 0.52 -0.69 1.39 -0.24 -0.02 0.00 0.15 -0.46 -0.23 0.47 0.00 0.52 0.02 -0.26 0.86 1.64 0.00 0.29 -1.13 0.00 -0.70 0.00 4.55 4.55
0.84 0.00 0.07 0.03 0.22 -0.34 0.36 0.17 -0.03 0.00 -0.05 0.01 -0.06 -0.01 0.00 1.35 0.18 0.02 0.43 0.34 0.00 0.19 -0.78 0.00 -0.50 0.00 4.55 4.55
1.12 0.00 -0.39 1.32 1.48 -1.09 0.88 0.39 -0.24 0.00 0.32 -0.47 -0.30 0.62 0.00 0.64 0.75 -0.09 0.47 0.45 0.00 -0.08 -1.63 0.00 -0.80 0.00 4.55 4.55
0.59 0.00 -0.23 -0.13 -0.06 0.22 0.13 -0.28 0.98 0.00 0.21 0.96 0.90 -0.16 0.00 0.16 0.01 -0.21 -0.02 0.29 0.00 1.32 -0.60 0.00 -0.24 0.00 4.55 4.55
0.10 0.00 -0.78 0.58 0.62 -0.35 -0.04 1.19 -0.17 0.00 0.31 0.22 0.17 0.63 0.00 0.23 1.43 0.43 -0.25 -0.12 0.00 -0.10 -0.31 0.00 -0.22 0.00 4.55 4.55
1.60 0.00 0.24 0.38 0.40 -0.54 1.03 -0.27 0.07 0.00 0.06 0.05 0.31 0.26 0.00 0.55 0.23 -0.29 0.68 0.53 0.00 0.41 -1.10 0.00 -0.55 0.00 4.55 4.55
1.41 0.00 0.80 0.40 0.40 -0.64 1.00 -0.27 -0.10 0.00 0.20 -0.47 -0.30 0.43 0.00 0.74 0.03 -0.10 1.77 0.57 0.00 0.03 -0.24 0.00 -0.60 0.00 4.55 4.55
-0.29 0.00 -0.52 -0.86 -0.64 1.61 -0.82 -0.18 0.74 0.00 -0.25 1.68 1.44 -0.47 0.00 -0.72 -0.37 -0.13 -0.51 -0.26 0.00 0.61 0.72 0.00 0.86 0.00 4.55 4.55
0.35 0.00 -0.50 1.14 0.87 -0.84 0.47 0.50 0.16 0.00 0.41 -0.19 -0.11 0.89 0.00 0.16 0.94 0.09 0.17 0.40 0.00 0.08 -1.08 0.00 -0.54 0.00 4.55 4.55
0.09 0.00 -0.26 0.19 0.16 -0.68 -0.11 0.05 0.03 0.00 1.63 -0.49 0.15 0.36 0.00 0.12 0.27 0.85 0.41 0.33 0.00 -0.03 -0.31 0.00 -0.42 0.00 4.55 4.55
0.55 0.00 0.24 -0.49 -0.29 0.76 -0.02 -0.31 1.20 0.00 -0.43 1.11 0.67 -0.36 0.00 -0.07 -0.47 -0.60 -0.03 0.19 0.00 1.14 -0.28 0.00 0.48 0.00 4.55 4.55
-0.07 0.00 -0.67 -0.71 -0.49 1.61 -0.56 -0.30 1.36 0.00 -0.50 2.05 1.57 -0.58 0.00 -0.41 -0.28 -0.66 -0.54 -0.07 0.00 1.50 0.38 0.00 0.51 0.00 4.55 4.55
1.16 0.00 -0.16 0.64 0.79 -0.38 0.64 0.01 0.12 0.00 0.11 0.11 0.16 0.34 0.00 0.38 0.35 -0.28 0.63 0.40 0.00 0.24 -0.79 0.00 -0.44 0.00 4.55 4.55
0.36 0.00 -0.06 0.32 0.42 -0.24 0.73 -0.04 -0.03 0.00 0.24 -0.07 0.16 0.28 0.00 0.32 0.21 0.24 0.90 0.28 0.00 0.18 0.13 0.00 -0.60 0.00 4.55 4.55
0.12 0.00 0.21 -0.52 -0.36 0.75 0.09 -0.39 1.27 0.00 -0.36 1.21 0.79 -0.40 0.00 -0.03 -0.49 -0.46 0.20 0.16 0.00 1.50 -0.16 0.00 0.26 0.00 4.55 4.55
1.38 0.00 0.45 0.35 0.35 -0.74 0.71 -0.02 -0.15 0.00 0.36 -0.41 -0.08 0.35 0.00 0.72 0.25 0.28 1.10 0.46 0.00 0.02 -0.11 0.00 -0.65 0.00 4.55 4.55
0.19 0.00 0.10 0.27 0.22 -0.07 0.10 0.24 0.15 0.00 0.55 -0.11 0.00 0.78 0.00 -0.09 0.03 0.03 0.20 0.82 0.00 0.03 -0.12 0.00 0.08 0.00 4.55 4.55
0.85 0.00 0.09 0.49 0.33 -0.41 0.57 -0.10 0.63 0.00 -0.03 0.30 0.25 0.09 0.00 0.30 0.15 -0.31 0.17 0.36 0.00 0.96 -1.20 0.00 -0.35 0.00 0.92 0.92
-0.09 0.00 -0.63 -0.54 -0.36 1.36 -0.59 -0.32 1.43 0.00 -0.36 1.73 1.36 -0.50 0.00 -0.36 -0.23 -0.50 -0.41 0.00 0.00 1.24 0.21 0.00 0.32 0.00 0.92 0.92
1.31 0.00 0.23 0.36 0.36 -0.60 0.63 -0.11 0.07 0.00 0.31 -0.17 0.03 0.30 0.00 0.53 0.16 -0.16 0.47 1.05 0.00 0.22 -0.87 0.00 -0.45 0.00 0.92 0.92
0.65 0.00 0.26 0.40 0.40 -0.44 0.39 0.51 -0.20 0.00 0.35 -0.35 -0.24 0.48 0.00 0.46 0.32 0.27 0.81 0.26 0.00 -0.14 -0.14 0.00 -0.25 0.00 0.92 0.92
0.04 0.00 -0.82 0.51 0.58 -0.82 -0.05 0.18 -0.27 0.00 1.89 -0.41 0.22 0.56 0.00 0.14 0.58 0.98 0.27 0.27 0.00 -0.27 -0.03 0.00 -0.81 0.00 0.92 0.92
-0.37 0.00 1.14 -0.67 -0.64 1.67 -0.78 0.33 0.21 0.00 -0.71 0.54 0.09 -0.17 0.00 -1.01 -0.74 -0.71 -0.53 -0.37 0.00 -0.04 1.31 0.00 1.83 0.00 0.92 0.92
-0.16 0.00 -0.43 0.35 0.34 -0.55 -0.20 0.97 -0.38 0.00 0.98 -0.43 0.00 0.60 0.00 0.26 0.63 1.21 0.07 0.00 0.00 -0.38 0.58 0.00 -0.38 0.00 0.92 0.92
0.14 0.00 -0.27 0.54 0.64 -0.34 0.25 0.19 -0.09 0.00 0.14 -0.17 -0.12 0.25 0.00 0.05 0.28 0.00 0.09 0.09 0.00 -0.09 -0.52 0.00 -0.24 0.00 0.92 0.92
0.18 0.00 -0.22 -0.12 -0.06 0.32 -0.08 -0.08 0.26 0.00 -0.10 0.48 0.39 -0.10 0.00 -0.03 0.00 -0.17 -0.07 0.02 0.00 0.29 0.05 0.00 0.06 0.00 0.92 0.92
0.41 0.00 0.11 0.24 0.21 -0.27 0.58 -0.08 -0.10 0.00 -0.03 -0.18 -0.10 0.16 0.00 0.17 0.09 -0.14 0.25 0.19 0.00 0.09 -0.44 0.00 -0.24 0.00 0.92 0.92
-0.10 0.00 0.33 -0.16 -0.16 0.43 -0.20 0.10 0.03 0.00 -0.20 0.10 -0.03 -0.03 0.00 -0.26 -0.20 -0.20 -0.13 -0.10 0.00 -0.03 0.36 0.00 0.49 0.00 0.92 0.92
0.07 0.00 -0.20 0.20 0.20 -0.26 0.07 0.20 -0.10 0.00 0.13 -0.03 0.00 0.13 0.00 0.10 0.49 0.13 -0.03 -0.03 0.00 -0.07 -0.16 0.00 -0.20 0.00 0.92 0.92
0.20 0.00 0.07 0.20 0.16 -0.20 0.49 -0.07 -0.10 0.00 -0.03 -0.16 -0.10 0.13 0.00 0.10 0.07 -0.10 0.20 0.13 0.00 0.07 -0.33 0.00 -0.20 0.00 0.92 0.92
</pre>
</td></tr></table><p>


<H2>
    Output file format
</H2>

<p>

The output is a standard EMBOSS alignment file. 

<p>

The results can be output in one of several styles by using the
command-line qualifier <tt>-aformat xxx</tt>, where 'xxx' is replaced by
the name of the required format.  Some of the alignment formats can cope
with an unlimited number of sequences, while others are only for pairs
of sequences.  

<p>

The available multiple alignment format names are: multiple, simple,
fasta, msf, clustal, mega, meganon, nexus,, nexusnon, phylip,
phylipnon, selex, treecon, tcoffee, debug, srs.

<p>

The available pairwise alignment format names are: pair, markx0, markx1,
markx2, markx3, markx10, match, sam, bam, score, srspair

<p>

See:
<A href="http://emboss.sf.net/docs/themes/AlignFormats.html">
http://emboss.sf.net/docs/themes/AlignFormats.html</A>
for further information on alignment formats.

<p>

<p>

By default the output is in 'simple' format.

<p>

<a name="output.1"></a>
<h3>Output files for usage example </h3>
<p><h3>File: hba_human.prophet</h3>
<table width="90%"><tr><td bgcolor="#CCFFCC">
<pre>
########################################
# Program: prophet
# Rundate: Mon 15 Jul 2013 12:00:00
# Commandline: prophet
#    -sequence "tsw:h*"
#    -infile ../../data/globins.gribskov
# Align_format: simple
# Report_file: hba_human.prophet
########################################

#=======================================
#
# Aligned_sequences: 2
# 1: globins
# 2: HBA_HUMAN
# Matrix: EBLOSUM62
#
# Length: 144
# Identity:      94/144 (65.3%)
# Similarity:   104/144 (72.2%)
# Gaps:           3/144 ( 2.1%)
# Score: 161.14
# 
#
#=======================================

globins           15 ELSAVDKAWVKANSVAEVGGHALERGLFASEPMTLEFFDTFKYLSTFDLS     64
                     .||..||..|||.. .:||.||.|.|..|.|.|.|.|..|..|...||||
HBA_HUMAN          2 VLSPADKTNVKAAW-GKVGAHAGEYGAEALERMFLSFPTTKTYFPHFDLS     50

globins           65 KGSADVKAHGKKVLDALGDAVAHLDDLEGTLAALSDLHAHKLKKGVDPVN    114
                     .|||.||.|||||.|||.:||||:||:...|:||||||||||:  |||||
HBA_HUMAN         51 HGSAQVKGHGKKVADALTNAVAHVDDMPNALSALSDLHAHKLR--VDPVN     98

globins          115 FKLLSHCLLVVLASHLPGDFTPEVQASMDKFLASVATVLASKYR    158
                     ||||||||||.||:|||.:|||.|.||:|||||||:|||.||||
HBA_HUMAN         99 FKLLSHCLLVTLAAHLPAEFTPAVHASLDKFLASVSTVLTSKYR    142


#=======================================
#
# Aligned_sequences: 2
# 1: globins
# 2: HBA_PANPA
# Matrix: EBLOSUM62
#
# Length: 144
# Identity:      94/144 (65.3%)
# Similarity:   104/144 (72.2%)
# Gaps:           3/144 ( 2.1%)


<font color=red>  [Part of this file has been deleted for brevity]</font>

                     :....| ..|| .|..:.........:.:......:....|...|....:
HD_TAKRU        2289 LQQPCVWNKLSTPEYNTHTCSLIYCLHHIILAVAVSPGDQLLHPERKKTK   2338

globins           51 FFDTFKYLSTFDLSKGSADVKAHGKKVLDALGDAVAHLDDLEGT-LAALS     99
                     ...........|....:..::....:::..|.:.:..:..|... ..|..
HD_TAKRU        2339 ALRHSDDEDQVDSVHDNHTLEWQACEIMAELVEGLQSVLSLGHHRNTAFP   2388

globins          100 DLHAHKLKKGVDPVNFKLLSHCLLVVLASHLPGDFTPEVQASMDKFLASV    149
                     ......|:..:..::...|.:....|........::|:........|..:
HD_TAKRU        2389 AFLTPTLRNIIISLSRLPLVNSHTRVPPLVWKLGWSPQPGGEFGTTLPEI   2438

globins          150 ATV-LASK--YRELGYQ    163
                     ... |..|  :||..|:
HD_TAKRU        2439 PVDFLQEKDVFREFLYR   2455


#=======================================
#
# Aligned_sequences: 2
# 1: globins
# 2: HIRA_TAKRU
# Matrix: EBLOSUM62
#
# Length: 161
# Identity:       8/161 ( 5.0%)
# Similarity:    40/161 (24.8%)
# Gaps:           2/161 ( 1.2%)
# Score: 64.35
# 
#
#=======================================

globins            1 PIVDTGSVVSLSEE-ELSAVDKAWVKANSVAEVGGHALERGLFASEPMTL     49
                     ......|...|:.. .............|..:....:.::....::.:..
HIRA_TAKRU       574 ATAAIASSTGLTPSERPKESTPMQKDVKSKEDTSSDSEDKMATINKNLAF    623

globins           50 EFFDTFKYLSTFDLSKGSADVKAHGKKVLDALGDAVAHLDDLEGTLAALS     99
                     ........:...::.:.....:....|:..::...:...........:.:
HIRA_TAKRU       624 NKRKPELLMDGAEVVEKRKKGRPRKDKMAASIAQPLTQTTSPAEREPSRA    673

globins          100 DLHAHKLKKGVDPVNFKLLSHCLLVVLASHLPGDFTPEVQASMDKFLASV    149
                     ................||.:..:.......:..|.:..::...:..:.:.
HIRA_TAKRU       674 AAAGAGAAAPTAAAALKLPTPSIKKAFTLQVSMDPSVVLEVENEVSVVAG    723

globins          150 ATVLASKY-RE    159
                     :.:...:. |:
HIRA_TAKRU       724 SRLSQLRCSRD    734


#---------------------------------------
#---------------------------------------
</pre>
</td></tr></table><p>

<H2>
    Data files
</H2>

None.

<H2>
    Notes
</H2>

None.

<H2>
    References
</H2>

None.

<H2>
    Warnings
</H2>

<b>prophet</b> does <b>NOT</b> use the 'simple frequency matrices'
produced by <b>prophecy</b>.  If you have a 'simple frequency matrix'you
should use the program <b>profit</b> to scan sequences. 


<H2>
    Diagnostic Error Messages
</H2>

If you see something like:

<p>

<pre>
   EMBOSS An error in prophet.c at line 110:
Unrecognised profile/matrix file format
</pre>

<p>

then you have probably given <b>prophet</b> a 'simple frequency matrix'
and not a a GRIBSKOV or HENIKOFF profile produced by <b>prophecy</b>.

<p>

If you see something like:

<p>

<pre>
Uncaught exception
Allocation Failed
raised at prophet.c:147
</pre>

<p>

then you have run out of memory.  This will occur if your sequence
and/or your profile are large. 

<p>

<b>prophet</b> allocates space based on the length of the sequence and
the size of the profile (8 x seqlen x profilelen)



<H2>
    Exit status
</H2>

It exits with a status of 0.

<H2>
    Known bugs
</H2>

None.

<h2><a name="See also">See also</a></h2>
<table border cellpadding=4 bgcolor="#FFFFF0">
<tr><th>Program name</th>
<th>Description</th></tr>
<tr>
<td><a href="profit.html">profit</a></td>
<td>Scan one or more sequences with a simple frequency matrix</td>
</tr>

<tr>
<td><a href="prophecy.html">prophecy</a></td>
<td>Create frequency matrix or profile from a multiple alignment</td>
</tr>

</table>


<H2>
    Author(s)
</H2>
Alan Bleasby 
<br>
European Bioinformatics Institute, Wellcome Trust Genome Campus, Hinxton, Cambridge CB10 1SD, UK

<p>
Please report all bugs to the EMBOSS bug team (emboss-bug&nbsp;&copy;&nbsp;emboss.open-bio.org) not to the original author.

<H2>
    History
</H2>

Written (1999) - Alan Bleasby

<H2>
    Target users
</H2>
This program is intended to be used by everyone and everything, from naive users to embedded scripts.


<H2>
    Comments
</H2>
None

</BODY>
</HTML>

