<HTML>

<HEAD>
  <TITLE>
  EMBOSS: showfeat
  </TITLE>
</HEAD>
<BODY BGCOLOR="#FFFFFF" text="#000000">

<table align=center border=0 cellspacing=0 cellpadding=0>
<tr><td valign=top>
<A HREF="/" ONMOUSEOVER="self.status='Go to the EMBOSS home page';return true"><img border=0 src="emboss_icon.jpg" alt="" width=150 height=48></a>
</td>
<td align=left valign=middle>
<b><font size="+6">
showfeat
</font></b>
</td></tr>
</table>
<br>&nbsp;
<p>

<H2>
    Function
</H2>
Display features of a sequence in pretty format

<H2>
    Description
</H2>

<p><b>showfeat</b> reads one or more protein or nucleic sequences with associated feature tables, and writes the features to file (or screen) in a style suitable for publication.  There are many options for controlling the format of the output.</p>


<H2>
    Usage
</H2>
<b>Here is a sample session with showfeat</b>
<p>

<p>
<table width="90%"><tr><td bgcolor="#CCFFFF"><pre>

% <b>showfeat </b>
Display features of a sequence in pretty format
Input sequence(s): <b>tembl:x65921</b>
Output file [x65921.showfeat]: <b></b>

</pre></td></tr></table><p>
<p>
<a href="#input.1">Go to the input files for this example</a><br><a href="#output.1">Go to the output files for this example</a><p><p>
<p>
<b>Example 2</b>
<p>
Display 'joined' features on one line with positions: 
<p>

<p>
<table width="90%"><tr><td bgcolor="#CCFFFF"><pre>

% <b>showfeat -sort join -pos </b>
Display features of a sequence in pretty format
Input sequence(s): <b>tembl:x65921</b>
Output file [x65921.showfeat]: <b></b>

</pre></td></tr></table><p>
<p>
<a href="#output.2">Go to the output files for this example</a><p><p>
<p>
<b>Example 3</b>
<p>
Display just positions and names of CDS features - this can be used as a regions file in showseq: 
<p>

<p>
<table width="90%"><tr><td bgcolor="#CCFFFF"><pre>

% <b>showfeat -matchtype CDS -width 0 -noid -nodesc -noscale -pos  </b>
Display features of a sequence in pretty format
Input sequence(s): <b>tembl:x65921</b>
Output file [x65921.showfeat]: <b></b>

</pre></td></tr></table><p>
<p>
<a href="#output.3">Go to the output files for this example</a><p><p>


<H2>
    Command line arguments
</H2>
<table CELLSPACING=0 CELLPADDING=3 BGCOLOR="#f5f5ff" ><tr><td>
<pre>
   Standard (Mandatory) qualifiers:
  [-sequence]          seqall     Sequence(s) filename and optional format, or
                                  reference (input USA)
  [-outfile]           outfile    [*.showfeat] Output file name

   Additional (Optional) qualifiers:
   -matchsource        string     [*] By default any feature source in the
                                  feature table is shown. You can set this to
                                  match any feature source you wish to show.
                                  The source name is usually either the name
                                  of the program that detected the feature or
                                  it is the feature table (eg: EMBL) that the
                                  feature came from.
                                  The source may be wildcarded by using '*'.
                                  If you wish to show more than one source,
                                  separate their names with the character '|',
                                  eg:
                                  gene* | embl (Any string is accepted)
   -matchtype          string     [*] By default any feature type in the
                                  feature table is shown. You can set this to
                                  match any feature type you wish to show.
                                  See http://www3.ebi.ac.uk/Services/WebFeat/
                                  for a list of the EMBL feature types and see
                                  Appendix A of the Swissprot user manual in
                                  http://www.expasy.ch/txt/userman.txt for a
                                  list of the Swissprot feature types.
                                  The type may be wildcarded by using '*'.
                                  If you wish to show more than one type,
                                  separate their names with the character '|',
                                  eg:
                                  *UTR | intron (Any string is accepted)
   -matchtag           string     [*] Tags are the types of extra values that
                                  a feature may have. For example in the EMBL
                                  feature table, a 'CDS' type of feature may
                                  have the tags '/codon', '/codon_start',
                                  '/db_xref', '/EC_number', '/evidence',
                                  '/exception', '/function', '/gene',
                                  '/label', '/map', '/note', '/number',
                                  '/partial', '/product', '/protein_id',
                                  '/pseudo', '/standard_name', '/translation',
                                  '/transl_except', '/transl_table', or
                                  '/usedin'. Some of these tags also have
                                  values, for example '/gene' can have the
                                  value of the gene name.
                                  By default any feature tag in the feature
                                  table is shown. You can set this to match
                                  any feature tag you wish to show.
                                  The tag may be wildcarded by using '*'.
                                  If you wish to show more than one tag,
                                  separate their names with the character '|',
                                  eg:
                                  gene | label (Any string is accepted)
   -matchvalue         string     [*] Tag values are the values associated
                                  with a feature tag. Tags are the types of
                                  extra values that a feature may have. For
                                  example in the EMBL feature table, a 'CDS'
                                  type of feature may have the tags '/codon',
                                  '/codon_start', '/db_xref', '/EC_number',
                                  '/evidence', '/exception', '/function',
                                  '/gene', '/label', '/map', '/note',
                                  '/number', '/partial', '/product',
                                  '/protein_id', '/pseudo', '/standard_name',
                                  '/translation', '/transl_except',
                                  '/transl_table', or '/usedin'. Only some of
                                  these tags can have values, for example
                                  '/gene' can have the value of the gene name.
                                  By default any feature tag value in the
                                  feature table is shown. You can set this to
                                  match any feature tag valueyou wish to show.
                                  The tag value may be wildcarded by using
                                  '*'.
                                  If you wish to show more than one tag value,
                                  separate their names with the character
                                  '|', eg:
                                  pax* | 10 (Any string is accepted)
   -sort               menu       [start] Sort features by Type, Start or
                                  Source, Nosort (don't sort - use input
                                  order) or join coding regions together and
                                  leave other features in the input order
                                  (Values: source (Sort by Source); start
                                  (Sort by Start position); type (Sort by
                                  Type); nosort (No sorting done); join (Join
                                  coding regions together))
   -annotation         range      [If this is left blank, then no annotation
                                  is added.] Regions to annotate by marking.
                                  If this is left blank, then no annotation is
                                  added.
                                  A set of regions is specified by a set of
                                  pairs of positions followed by optional
                                  text.
                                  The positions are integers.
                                  They are followed by any text (but not
                                  digits when on the command-line).
                                  Examples of region specifications are:
                                  24-45 new domain 56-78 match to Mouse
                                  1-100 First part 120-156 oligo
                                  A file of ranges to annotate (one range per
                                  line) can be specified as '@filename'.

   Advanced (Unprompted) qualifiers:
   -html               boolean    [N] Use HTML formatting
   -[no]id             boolean    [Y] Set this to be false if you do not wish
                                  to display the ID name of the sequence.
   -[no]description    boolean    [Y] Set this to be false if you do not wish
                                  to display the description of the sequence.
   -[no]scale          boolean    [Y] Set this to be false if you do not wish
                                  to display the scale line.
   -width              integer    [60] You can expand (or contract) the width
                                  of the ASCII-character graphics display of
                                  the positions of the features using this
                                  value.
                                  For example, a width of 80 characters would
                                  cover a standard page width and a width a 10
                                  characters would be nearly unreadable.
                                  If the width is set to less than 4, the
                                  graphics lines and the scale line will not
                                  be displayed. (Integer 0 or more)
   -collapse           boolean    [N] If this is set, then features from the
                                  same source and of the same type and sense
                                  are all printed on the same line. For
                                  instance if there are several features from
                                  the EMBL feature table (ie. the same source)
                                  which are all of type 'exon' in the same
                                  sense, then they will all be displayed on
                                  the same line. This makes it hard to
                                  distinguish overlapping features.
                                  If this is set to false then each feature is
                                  displayed on a separate line making it
                                  easier to distinguish where features start
                                  and end.
   -[no]forward        boolean    [Y] Set this to be false if you do not wish
                                  to display forward sense features.
   -[no]reverse        boolean    [Y] Set this to be false if you do not wish
                                  to display reverse sense features.
   -[no]unknown        boolean    [Y] Set this to be false if you do not wish
                                  to display unknown sense features. (ie.
                                  features with no directionality - all
                                  protein features are of this type and some
                                  nucleic features (for example, CG-rich
                                  regions)).
   -strand             boolean    [N] Set this if you wish to display the
                                  strand of the features. Protein features are
                                  always directionless (indicated by '0'),
                                  forward is indicated by '+' and reverse is
                                  '-'.
   -source             boolean    [N] Set this if you wish to display the
                                  source of the features.
                                  The source name is usually either the name
                                  of the program that detected the feature or
                                  it is the name of the feature table (eg:
                                  EMBL) that the feature came from.
   -position           boolean    [N] Set this if you wish to display the
                                  start and end position of the features. If
                                  several features are being displayed on the
                                  same line, then the start and end positions
                                  will be joined by a comma, for example:
                                  '189-189,225-225'.
   -[no]type           boolean    [Y] Set this to be false if you do not wish
                                  to display the type of the features.
   -tags               boolean    [N] Set this to be false if you do not wish
                                  to display the tags and values of the
                                  features.
   -[no]values         boolean    [Y] Set this to be false if you do not wish
                                  to display the tag values of the features.
                                  If this is set to be false, only the tag
                                  names will be displayed. If the tags are not
                                  displayed, then the values will not be
                                  displayed. The value of the 'translation'
                                  tag is never displayed as it is often
                                  extremely long.
   -stricttags         boolean    [N] By default if any tag/value pair in a
                                  feature matches the specified tag and value,
                                  then all the tags/value pairs of that
                                  feature will be displayed. If this is set to
                                  be true, then only those tag/value pairs in
                                  a feature that match the specified tag and
                                  value will be displayed.

   Associated qualifiers:

   "-sequence" associated qualifiers
   -sbegin1            integer    Start of each sequence to be used
   -send1              integer    End of each sequence to be used
   -sreverse1          boolean    Reverse (if DNA)
   -sask1              boolean    Ask for begin/end/reverse
   -snucleotide1       boolean    Sequence is nucleotide
   -sprotein1          boolean    Sequence is protein
   -slower1            boolean    Make lower case
   -supper1            boolean    Make upper case
   -sformat1           string     Input sequence format
   -sdbname1           string     Database name
   -sid1               string     Entryname
   -ufo1               string     UFO features
   -fformat1           string     Features format
   -fopenfile1         string     Features file name

   "-outfile" associated qualifiers
   -odirectory2        string     Output directory

   General qualifiers:
   -auto               boolean    Turn off prompts
   -stdout             boolean    Write first file to standard output
   -filter             boolean    Read first file from standard input, write
                                  first file to standard output
   -options            boolean    Prompt for standard and additional values
   -debug              boolean    Write debug output to program.dbg
   -verbose            boolean    Report some/full command line options
   -help               boolean    Report command line options. More
                                  information on associated and general
                                  qualifiers can be found with -help -verbose
   -warning            boolean    Report warnings
   -error              boolean    Report errors
   -fatal              boolean    Report fatal errors
   -die                boolean    Report dying program messages

</pre>
</td></tr></table>
<P>
<table border cellspacing=0 cellpadding=3 bgcolor="#ccccff">
<tr bgcolor="#FFFFCC">
<th align="left" colspan=2>Standard (Mandatory) qualifiers</th>
<th align="left">Allowed values</th>
<th align="left">Default</th>
</tr>

<tr>
<td>[-sequence]<br>(Parameter 1)</td>
<td>Sequence(s) filename and optional format, or reference (input USA)</td>
<td>Readable sequence(s)</td>
<td><b>Required</b></td>
</tr>

<tr>
<td>[-outfile]<br>(Parameter 2)</td>
<td>Output file name</td>
<td>Output file</td>
<td><i>&lt;*&gt;</i>.showfeat</td>
</tr>

<tr bgcolor="#FFFFCC">
<th align="left" colspan=2>Additional (Optional) qualifiers</th>
<th align="left">Allowed values</th>
<th align="left">Default</th>
</tr>

<tr>
<td>-matchsource</td>
<td>By default any feature source in the feature table is shown. You can set this to match any feature source you wish to show.
The source name is usually either the name of the program that detected the feature or it is the feature table (eg: EMBL) that the feature came from.
The source may be wildcarded by using '*'.
If you wish to show more than one source, separate their names with the character '|', eg:
gene* | embl</td>
<td>Any string is accepted</td>
<td>*</td>
</tr>

<tr>
<td>-matchtype</td>
<td>By default any feature type in the feature table is shown. You can set this to match any feature type you wish to show.
See http://www3.ebi.ac.uk/Services/WebFeat/ for a list of the EMBL feature types and see Appendix A of the Swissprot user manual in http://www.expasy.ch/txt/userman.txt for a list of the Swissprot feature types.
The type may be wildcarded by using '*'.
If you wish to show more than one type, separate their names with the character '|', eg:
*UTR | intron</td>
<td>Any string is accepted</td>
<td>*</td>
</tr>

<tr>
<td>-matchtag</td>
<td>Tags are the types of extra values that a feature may have. For example in the EMBL feature table, a 'CDS' type of feature may have the tags '/codon', '/codon_start', '/db_xref', '/EC_number', '/evidence', '/exception', '/function', '/gene', '/label', '/map', '/note', '/number', '/partial', '/product', '/protein_id', '/pseudo', '/standard_name', '/translation', '/transl_except', '/transl_table', or '/usedin'. Some of these tags also have values, for example '/gene' can have the value of the gene name.
By default any feature tag in the feature table is shown. You can set this to match any feature tag you wish to show.
The tag may be wildcarded by using '*'.
If you wish to show more than one tag, separate their names with the character '|', eg:
gene | label</td>
<td>Any string is accepted</td>
<td>*</td>
</tr>

<tr>
<td>-matchvalue</td>
<td>Tag values are the values associated with a feature tag. Tags are the types of extra values that a feature may have. For example in the EMBL feature table, a 'CDS' type of feature may have the tags '/codon', '/codon_start', '/db_xref', '/EC_number', '/evidence', '/exception', '/function', '/gene', '/label', '/map', '/note', '/number', '/partial', '/product', '/protein_id', '/pseudo', '/standard_name', '/translation', '/transl_except', '/transl_table', or '/usedin'. Only some of these tags can have values, for example '/gene' can have the value of the gene name. By default any feature tag value in the feature table is shown. You can set this to match any feature tag valueyou wish to show.
The tag value may be wildcarded by using '*'.
If you wish to show more than one tag value, separate their names with the character '|', eg:
pax* | 10</td>
<td>Any string is accepted</td>
<td>*</td>
</tr>

<tr>
<td>-sort</td>
<td>Sort features by Type, Start or Source, Nosort (don't sort - use input order) or join coding regions together and leave other features in the input order</td>
<td><table><tr><td>source</td> <td><i>(Sort by Source)</i></td></tr><tr><td>start</td> <td><i>(Sort by Start position)</i></td></tr><tr><td>type</td> <td><i>(Sort by Type)</i></td></tr><tr><td>nosort</td> <td><i>(No sorting done)</i></td></tr><tr><td>join</td> <td><i>(Join coding regions together)</i></td></tr></table></td>
<td>start</td>
</tr>

<tr>
<td>-annotation</td>
<td>Regions to annotate by marking.
If this is left blank, then no annotation is added.
A set of regions is specified by a set of pairs of positions followed by optional text.
The positions are integers.
They are followed by any text (but not digits when on the command-line).
Examples of region specifications are:
24-45 new domain 56-78 match to Mouse
1-100 First part 120-156 oligo
A file of ranges to annotate (one range per line) can be specified as '@filename'.</td>
<td>Sequence range</td>
<td>If this is left blank, then no annotation is added.</td>
</tr>

<tr bgcolor="#FFFFCC">
<th align="left" colspan=2>Advanced (Unprompted) qualifiers</th>
<th align="left">Allowed values</th>
<th align="left">Default</th>
</tr>

<tr>
<td>-html</td>
<td>Use HTML formatting</td>
<td>Boolean value Yes/No</td>
<td>No</td>
</tr>

<tr>
<td>-[no]id</td>
<td>Set this to be false if you do not wish to display the ID name of the sequence.</td>
<td>Boolean value Yes/No</td>
<td>Yes</td>
</tr>

<tr>
<td>-[no]description</td>
<td>Set this to be false if you do not wish to display the description of the sequence.</td>
<td>Boolean value Yes/No</td>
<td>Yes</td>
</tr>

<tr>
<td>-[no]scale</td>
<td>Set this to be false if you do not wish to display the scale line.</td>
<td>Boolean value Yes/No</td>
<td>Yes</td>
</tr>

<tr>
<td>-width</td>
<td>You can expand (or contract) the width of the ASCII-character graphics display of the positions of the features using this value.
For example, a width of 80 characters would cover a standard page width and a width a 10 characters would be nearly unreadable.
If the width is set to less than 4, the graphics lines and the scale line will not be displayed.</td>
<td>Integer 0 or more</td>
<td>60</td>
</tr>

<tr>
<td>-collapse</td>
<td>If this is set, then features from the same source and of the same type and sense are all printed on the same line. For instance if there are several features from the EMBL feature table (ie. the same source) which are all of type 'exon' in the same sense, then they will all be displayed on the same line. This makes it hard to distinguish overlapping features.
If this is set to false then each feature is displayed on a separate line making it easier to distinguish where features start and end.</td>
<td>Boolean value Yes/No</td>
<td>No</td>
</tr>

<tr>
<td>-[no]forward</td>
<td>Set this to be false if you do not wish to display forward sense features.</td>
<td>Boolean value Yes/No</td>
<td>Yes</td>
</tr>

<tr>
<td>-[no]reverse</td>
<td>Set this to be false if you do not wish to display reverse sense features.</td>
<td>Boolean value Yes/No</td>
<td>Yes</td>
</tr>

<tr>
<td>-[no]unknown</td>
<td>Set this to be false if you do not wish to display unknown sense features. (ie. features with no directionality - all protein features are of this type and some nucleic features (for example, CG-rich regions)).</td>
<td>Boolean value Yes/No</td>
<td>Yes</td>
</tr>

<tr>
<td>-strand</td>
<td>Set this if you wish to display the strand of the features. Protein features are always directionless (indicated by '0'), forward is indicated by '+' and reverse is '-'.</td>
<td>Boolean value Yes/No</td>
<td>No</td>
</tr>

<tr>
<td>-source</td>
<td>Set this if you wish to display the source of the features.
The source name is usually either the name of the program that detected the feature or it is the name of the feature table (eg: EMBL) that the feature came from.</td>
<td>Boolean value Yes/No</td>
<td>No</td>
</tr>

<tr>
<td>-position</td>
<td>Set this if you wish to display the start and end position of the features. If several features are being displayed on the same line, then the start and end positions will be joined by a comma, for example: '189-189,225-225'.</td>
<td>Boolean value Yes/No</td>
<td>No</td>
</tr>

<tr>
<td>-[no]type</td>
<td>Set this to be false if you do not wish to display the type of the features.</td>
<td>Boolean value Yes/No</td>
<td>Yes</td>
</tr>

<tr>
<td>-tags</td>
<td>Set this to be false if you do not wish to display the tags and values of the features.</td>
<td>Boolean value Yes/No</td>
<td>No</td>
</tr>

<tr>
<td>-[no]values</td>
<td>Set this to be false if you do not wish to display the tag values of the features. If this is set to be false, only the tag names will be displayed. If the tags are not displayed, then the values will not be displayed. The value of the 'translation' tag is never displayed as it is often extremely long.</td>
<td>Boolean value Yes/No</td>
<td>Yes</td>
</tr>

<tr>
<td>-stricttags</td>
<td>By default if any tag/value pair in a feature matches the specified tag and value, then all the tags/value pairs of that feature will be displayed. If this is set to be true, then only those tag/value pairs in a feature that match the specified tag and value will be displayed.</td>
<td>Boolean value Yes/No</td>
<td>No</td>
</tr>

</table>


<H2>
    Input file format
</H2>


The feature input is the feature table lines from a normal sequence USA.

<p>


<a name="input.1"></a>
<h3>Input files for usage example </h3>

'tembl:x65921' is a sequence entry in the example nucleic acid database 'tembl'
<p>
<p><h3>Database entry: tembl:x65921</h3>
<table width="90%"><tr><td bgcolor="#FFCCFF">
<pre>
ID   X65921; SV 1; linear; genomic DNA; STD; HUM; 2016 BP.
XX
AC   X65921; S45242;
XX
DT   13-MAY-1992 (Rel. 31, Created)
DT   14-NOV-2006 (Rel. 89, Last updated, Version 7)
XX
DE   H.sapiens fau 1 gene
XX
KW   fau 1 gene.
XX
OS   Homo sapiens (human)
OC   Eukaryota; Metazoa; Chordata; Craniata; Vertebrata; Euteleostomi; Mammalia;
OC   Eutheria; Euarchontoglires; Primates; Haplorrhini; Catarrhini; Hominidae;
OC   Homo.
XX
RN   [1]
RP   1-2016
RA   Kas K.;
RT   ;
RL   Submitted (29-APR-1992) to the EMBL/GenBank/DDBJ databases.
RL   K. Kas, University of Antwerp, Dept of Biochemistry T3.22,
RL   Universiteitsplein 1, 2610 Wilrijk, BELGIUM
XX
RN   [2]
RP   1-2016
RX   DOI; 10.1016/0006-291X(92)91286-Y.
RX   PUBMED; 1326960.
RA   Kas K., Michiels L., Merregaert J.;
RT   "Genomic structure and expression of the human fau gene: encoding the
RT   ribosomal protein S30 fused to a ubiquitin-like protein.";
RL   Biochem. Biophys. Res. Commun. 187(2):927-933(1992).
XX
DR   GDB; 191789.
DR   GDB; 191790.
DR   GDB; 354872.
DR   GDB; 4590236.
XX
FH   Key             Location/Qualifiers
FH
FT   source          1..2016
FT                   /organism="Homo sapiens"
FT                   /mol_type="genomic DNA"
FT                   /clone_lib="CML cosmid"
FT                   /clone="15.1"
FT                   /db_xref="taxon:9606"
FT   mRNA            join(408..504,774..856,951..1095,1557..1612,1787..&gt;1912)
FT                   /gene="fau 1"
FT   exon            408..504
FT                   /number=1


<font color=red>  [Part of this file has been deleted for brevity]</font>

FT                   RAKRRMQYNRRFVNVVPTFGKKKGPNANS"
FT   intron          857..950
FT                   /number=2
FT   exon            951..1095
FT                   /number=3
FT   intron          1096..1556
FT                   /number=3
FT   exon            1557..1612
FT                   /number=4
FT   intron          1613..1786
FT                   /number=4
FT   exon            1787..&gt;1912
FT                   /number=5
FT   polyA_signal    1938..1943
XX
SQ   Sequence 2016 BP; 421 A; 562 C; 538 G; 495 T; 0 other;
     ctaccatttt ccctctcgat tctatatgta cactcgggac aagttctcct gatcgaaaac        60
     ggcaaaacta aggccccaag taggaatgcc ttagttttcg gggttaacaa tgattaacac       120
     tgagcctcac acccacgcga tgccctcagc tcctcgctca gcgctctcac caacagccgt       180
     agcccgcagc cccgctggac accggttctc catccccgca gcgtagcccg gaacatggta       240
     gctgccatct ttacctgcta cgccagcctt ctgtgcgcgc aactgtctgg tcccgccccg       300
     tcctgcgcga gctgctgccc aggcaggttc gccggtgcga gcgtaaaggg gcggagctag       360
     gactgccttg ggcggtacaa atagcaggga accgcgcggt cgctcagcag tgacgtgaca       420
     cgcagcccac ggtctgtact gacgcgccct cgcttcttcc tctttctcga ctccatcttc       480
     gcggtagctg ggaccgccgt tcaggtaaga atggggcctt ggctggatcc gaagggcttg       540
     tagcaggttg gctgcggggt cagaaggcgc ggggggaacc gaagaacggg gcctgctccg       600
     tggccctgct ccagtcccta tccgaactcc ttgggaggca ctggccttcc gcacgtgagc       660
     cgccgcgacc accatcccgt cgcgatcgtt tctggaccgc tttccactcc caaatctcct       720
     ttatcccaga gcatttcttg gcttctctta caagccgtct tttctttact cagtcgccaa       780
     tatgcagctc tttgtccgcg cccaggagct acacaccttc gaggtgaccg gccaggaaac       840
     ggtcgcccag atcaaggtaa ggctgcttgg tgcgccctgg gttccatttt cttgtgctct       900
     tcactctcgc ggcccgaggg aacgcttacg agccttatct ttccctgtag gctcatgtag       960
     cctcactgga gggcattgcc ccggaagatc aagtcgtgct cctggcaggc gcgcccctgg      1020
     aggatgaggc cactctgggc cagtgcgggg tggaggccct gactaccctg gaagtagcag      1080
     gccgcatgct tggaggtgag tgagagagga atgttctttg aagtaccggt aagcgtctag      1140
     tgagtgtggg gtgcatagtc ctgacagctg agtgtcacac ctatggtaat agagtacttc      1200
     tcactgtctt cagttcagag tgattcttcc tgtttacatc cctcatgttg aacacagacg      1260
     tccatgggag actgagccag agtgtagttg tatttcagtc acatcacgag atcctagtct      1320
     ggttatcagc ttccacacta aaaattaggt cagaccaggc cccaaagtgc tctataaatt      1380
     agaagctgga agatcctgaa atgaaactta agatttcaag gtcaaatatc tgcaactttg      1440
     ttctcattac ctattgggcg cagcttctct ttaaaggctt gaattgagaa aagaggggtt      1500
     ctgctgggtg gcaccttctt gctcttacct gctggtgcct tcctttccca ctacaggtaa      1560
     agtccatggt tccctggccc gtgctggaaa agtgagaggt cagactccta aggtgagtga      1620
     gagtattagt ggtcatggtg ttaggacttt ttttcctttc acagctaaac caagtccctg      1680
     ggctcttact cggtttgcct tctccctccc tggagatgag cctgagggaa gggatgctag      1740
     gtgtggaaga caggaaccag ggcctgatta accttccctt ctccaggtgg ccaaacagga      1800
     gaagaagaag aagaagacag gtcgggctaa gcggcggatg cagtacaacc ggcgctttgt      1860
     caacgttgtg cccacctttg gcaagaagaa gggccccaat gccaactctt aagtcttttg      1920
     taattctggc tttctctaat aaaaaagcca cttagttcag tcatcgcatt gtttcatctt      1980
     tacttgcaag gcctcaggga gaggtgtgct tctcgg                                2016
//
</pre>
</td></tr></table><p>

<H2>
    Output file format
</H2>


The output is a text representation of the feature table.

<p>


<a name="output.1"></a>
<h3>Output files for usage example </h3>
<p><h3>File: x65921.showfeat </h3>
<table width="90%"><tr><td bgcolor="#CCFFCC">
<pre>
X65921
H.sapiens fau 1 gene
|==========================================================| 2016
|----------------------------------------------------------&gt; source
           |--&gt;                                              exon
           |--&gt;                                              mRNA
              |-------&gt;                                      intron
                      |-&gt;                                    exon
                      |-&gt;                                    mRNA
                      |-&gt;                                    CDS
                        |--&gt;                                 intron
                           |---&gt;                             CDS
                           |---&gt;                             exon
                           |---&gt;                             mRNA
                               |-------------&gt;               intron
                                             |&gt;              CDS
                                             |&gt;              exon
                                             |&gt;              mRNA
                                               |----&gt;        intron
                                                    |--&gt;     CDS
                                                    |--&gt;     exon
                                                    |--&gt;     mRNA
                                                        &gt;    polyA_signal

</pre>
</td></tr></table><p>

<a name="output.2"></a>
<h3>Output files for usage example 2</h3>
<p><h3>File: x65921.showfeat </h3>
<table width="90%"><tr><td bgcolor="#CCFFCC">
<pre>
X65921
H.sapiens fau 1 gene
|==========================================================| 2016
|----------------------------------------------------------&gt; 1-2016 source
           |--&gt;       |-&gt;  |---&gt;             |&gt;     |--&gt;     408-504,774-856,951-1095,1557-1612,1787-1912 mRNA
           |--&gt;                                              408-504 exon
              |-------&gt;                                      505-773 intron
                      |-&gt;                                    774-856 exon
                      |-&gt;  |---&gt;             |&gt;     |--&gt;     782-856,951-1095,1557-1612,1787-1912 CDS
                        |--&gt;                                 857-950 intron
                           |---&gt;                             951-1095 exon
                               |-------------&gt;               1096-1556 intron
                                             |&gt;              1557-1612 exon
                                               |----&gt;        1613-1786 intron
                                                    |--&gt;     1787-1912 exon
                                                        &gt;    1938-1943 polyA_signal

</pre>
</td></tr></table><p>

<a name="output.3"></a>
<h3>Output files for usage example 3</h3>
<p><h3>File: x65921.showfeat </h3>
<table width="90%"><tr><td bgcolor="#CCFFCC">
<pre>
782-856 CDS
951-1095 CDS
1557-1612 CDS
1787-1912 CDS

</pre>
</td></tr></table><p>


<H2>
    Data files
</H2>

None.

<H2>
    Notes
</H2>

<p>Sequence regions to annotate may be specified as a set or ranges. Optionally, forward and reverse sense features, and features of unspecified sense (for example, CG-rich regions), are displayed.  Optionally, features from the same source and of the same type and sense may all be printed on the same line, although this may make it hard to distinguish overlapping features.</p>

<p>By default, any features in the feature table are displayed however there are options to <b>restrict</b> this by feature source, type, feature tag and feature tag value.</p>

<p>Features may be sorted by Type, Start or Source, left unsorted (in which case the same order as the input is used), or features for the coding regions presented all together.</p>

<p>Various options control what else is displayed, including the sequence ID, description, scale line, width of display, strand of the features, feature source, start and end positions, feature type, and feature tags and tag values.  By default if any tag/value pair in a feature matches the specified tag and value, then all the tags/value pairs of that feature will be displayed. If this is set to be true, then only those tag/value pairs in a feature that match the specified tag and value will be displayed.</p>

<p>Optionally, output can be in HTML.</p>


<H2>
    References
</H2>

None.


<H2>
    Warnings
</H2>

None.


<H2>
    Diagnostic Error Messages
</H2>

None.


<H2>
    Exit status
</H2>

It always exits with a status of 0.

<H2>
    Known bugs
</H2>

None.


<h2><a name="See also">See also</a></h2>
<table border cellpadding=4 bgcolor="#FFFFF0">
<tr><th>Program name</th>
<th>Description</th></tr>
<tr>
<td><a href="abiview.html">abiview</a></td>
<td>Display the trace in an ABI sequencer file</td>
</tr>

<tr>
<td><a href="cirdna.html">cirdna</a></td>
<td>Draws circular maps of DNA constructs</td>
</tr>

<tr>
<td><a href="coderet.html">coderet</a></td>
<td>Extract CDS, mRNA and translations from feature tables</td>
</tr>

<tr>
<td><a href="extractfeat.html">extractfeat</a></td>
<td>Extract features from sequence(s)</td>
</tr>

<tr>
<td><a href="lindna.html">lindna</a></td>
<td>Draws linear maps of DNA constructs</td>
</tr>

<tr>
<td><a href="maskfeat.html">maskfeat</a></td>
<td>Write a sequence with masked features</td>
</tr>

<tr>
<td><a href="pepnet.html">pepnet</a></td>
<td>Draw a helical net for a protein sequence</td>
</tr>

<tr>
<td><a href="pepwheel.html">pepwheel</a></td>
<td>Draw a helical wheel diagram for a protein sequence</td>
</tr>

<tr>
<td><a href="prettyplot.html">prettyplot</a></td>
<td>Draw a sequence alignment with pretty formatting</td>
</tr>

<tr>
<td><a href="prettyseq.html">prettyseq</a></td>
<td>Write a nucleotide sequence and its translation to file</td>
</tr>

<tr>
<td><a href="remap.html">remap</a></td>
<td>Display restriction enzyme binding sites in a nucleotide sequence</td>
</tr>

<tr>
<td><a href="seealso.html">seealso</a></td>
<td>Finds programs with similar function to a specified program</td>
</tr>

<tr>
<td><a href="showalign.html">showalign</a></td>
<td>Display a multiple sequence alignment in pretty format</td>
</tr>

<tr>
<td><a href="showdb.html">showdb</a></td>
<td>Displays information on configured databases</td>
</tr>

<tr>
<td><a href="showseq.html">showseq</a></td>
<td>Displays sequences with features in pretty format</td>
</tr>

<tr>
<td><a href="sixpack.html">sixpack</a></td>
<td>Display a DNA sequence with 6-frame translation and ORFs</td>
</tr>

<tr>
<td><a href="textsearch.html">textsearch</a></td>
<td>Search the textual description of sequence(s)</td>
</tr>

<tr>
<td><a href="twofeat.html">twofeat</a></td>
<td>Finds neighbouring pairs of features in sequence(s)</td>
</tr>

</table>

<H2>
    Author(s)
</H2>
Gary Williams (gwilliam&nbsp;&copy;&nbsp;rfcgr.mrc.ac.uk)
<br>
MRC Rosalind Franklin Centre for Genomics Research
Wellcome Trust Genome Campus, Hinxton, Cambridge, CB10 1SB, UK




<H2>
    History
</H2>

Written 1999 - Gary Williams
<p>

Dec 2001 - added -sort nosort option to get the features in the input order

<H2>
    Target users
</H2>
This program is intended to be used by everyone and everything, from naive users to embedded scripts.



<H2>
    Comments
</H2>
None

</BODY>
</HTML>

