<HTML>

<HEAD>
  <TITLE>
  EMBOSS: showseq
  </TITLE>
</HEAD>
<BODY BGCOLOR="#FFFFFF" text="#000000">

<table align=center border=0 cellspacing=0 cellpadding=0>
<tr><td valign=top>
<A HREF="/" ONMOUSEOVER="self.status='Go to the EMBOSS home page';return true"><img border=0 src="emboss_icon.jpg" alt="" width=150 height=48></a>
</td>
<td align=left valign=middle>
<b><font size="+6">
showseq
</font></b>
</td></tr>
</table>
<br>&nbsp;
<p>


<H2>
Wiki
</H2>

The master copies of EMBOSS documentation are available
at <a href="http://emboss.open-bio.org/wiki/Appdocs">
http://emboss.open-bio.org/wiki/Appdocs</a>
on the EMBOSS Wiki.

<p>
Please help by correcting and extending the Wiki pages.

<H2>
    Function
</H2>
Displays sequences with features in pretty format
<H2>
    Description
</H2>

<p><b>showseq</b> displays one or more nucleic acid sequences, with
features, in a style suitable for publication. The output is sent to
screen by default but can be written to file. You may pick a format
from a list, alternatively, use the many options to control what is
output and in what format. Optionally, the sequence feature table can
be displayed. The sequence can be translated, using the specified
genetic code tables. Also recognition sites and/or cut sites of
restriction enzymes from the REBASE database may be displayed.  There
are various other options for controlling how the sequence is
displayed and numbered and the output can be formatted for HTML.</p>


<H2>
    Usage
</H2>
<b>Here is a sample session with showseq</b>
<p>

<p>
<table width="90%"><tr><td bgcolor="#CCFFFF"><pre>

% <b>showseq tembl:x13776 -sbeg 1 -send 100 </b>
Displays sequences with features in pretty format
Things to display
         0 : Enter your own list of things to display
         1 : Sequence only
         2 : Default sequence with features
         3 : Pretty sequence
         4 : One frame translation
         5 : Three frame translations
         6 : Six frame translations
         7 : Restriction enzyme map
         8 : Baroque
Display format [2]: <b></b>
Output file [x13776.showseq]: <b></b>

</pre></td></tr></table><p>
<p>
<a href="#input.1">Go to the input files for this example</a><br><a href="#output.1">Go to the output files for this example</a><p><p>
<p>
<b>Example 2</b>
<p>
The standard list of output formats are only a small selection of the possible ways in which a sequence might be displayed.  Precise control over the output format is acheived by selecting the qualifier '-format 0'  (Option 0 in the list of things to display).  For example, by choosing format '0' and then specifying that we want to display the things: 'b,s,t,c', we will output the sequence in the following way: 
<p>

<p>
<table width="90%"><tr><td bgcolor="#CCFFFF"><pre>

% <b>showseq tembl:x13776 -sbeg 1 -send 120 </b>
Displays sequences with features in pretty format
Things to display
         0 : Enter your own list of things to display
         1 : Sequence only
         2 : Default sequence with features
         3 : Pretty sequence
         4 : One frame translation
         5 : Three frame translations
         6 : Six frame translations
         7 : Restriction enzyme map
         8 : Baroque
Display format [2]: <b>0</b>
Specify your own things to display
         S : Sequence
         B : Blank line
         1 : Frame1 translation
         2 : Frame2 translation
         3 : Frame3 translation
        -1 : CompFrame1 translation
        -2 : CompFrame2 translation
        -3 : CompFrame3 translation
         T : Ticks line
         N : Number ticks line
         C : Complement sequence
         F : Features
         R : Restriction enzyme cut sites in forward sense
        -R : Restriction enzyme cut sites in reverse sense
         A : Annotation
Enter a list of things to display [B,N,T,S,A,F]: <b>b,s,t,c</b>
Output file [x13776.showseq]: <b></b>

</pre></td></tr></table><p>
<p>
<a href="#output.2">Go to the output files for this example</a><p><p>
<p>
<b>Example 3</b>
<p>
Display only the sequence: 
<p>

<p>
<table width="90%"><tr><td bgcolor="#CCFFFF"><pre>

% <b>showseq tembl:x13776 -sbeg 1 -send 100 -noname -nodesc -format 0 -thing S </b>
Displays sequences with features in pretty format
Output file [x13776.showseq]: <b></b>

</pre></td></tr></table><p>
<p>
<a href="#output.3">Go to the output files for this example</a><p><p>
<p>
<b>Example 4</b>
<p>
Number the sequence lines in the margin:  
<p>

<p>
<table width="90%"><tr><td bgcolor="#CCFFFF"><pre>

% <b>showseq tembl:x13776 -sbeg 1 -send 100 -format 1 -number </b>
Displays sequences with features in pretty format
Output file [x13776.showseq]: <b></b>

</pre></td></tr></table><p>
<p>
<a href="#output.4">Go to the output files for this example</a><p><p>
<p>
<b>Example 5</b>
<p>
Start the numbering at a specified value ('123' in this case): 
<p>

<p>
<table width="90%"><tr><td bgcolor="#CCFFFF"><pre>

% <b>showseq tembl:x13776 -sbeg 1 -send 100 -format 1 -number -offset 123 </b>
Displays sequences with features in pretty format
Output file [x13776.showseq]: <b></b>

</pre></td></tr></table><p>
<p>
<a href="#output.5">Go to the output files for this example</a><p><p>
<p>
<b>Example 6</b>
<p>
Make selected regions uppercase. (Use '-slower' to force the rest of the sequence to be lowercase).  
<p>

<p>
<table width="90%"><tr><td bgcolor="#CCFFFF"><pre>

% <b>showseq tembl:x13776 -sbeg 1 -send 100 -format 1 -slower -upper '8-24,65-81' </b>
Displays sequences with features in pretty format
Output file [x13776.showseq]: <b></b>

</pre></td></tr></table><p>
<p>
<a href="#output.6">Go to the output files for this example</a><p><p>
<p>
<b>Example 7</b>
<p>
Translate selected regions:  
<p>

<p>
<table width="90%"><tr><td bgcolor="#CCFFFF"><pre>

% <b>showseq tembl:x13776 -sbeg 1 -send 120 -format 5 -trans 25-49,66-76 </b>
Displays sequences with features in pretty format
Output file [x13776.showseq]: <b></b>

</pre></td></tr></table><p>
<p>
<a href="#output.7">Go to the output files for this example</a><p><p>
<p>
<b>Example 8</b>
<p>
Add your own annotation to the display: 
<p>

<p>
<table width="90%"><tr><td bgcolor="#CCFFFF"><pre>

% <b>showseq tembl:x13776 -sbeg 1 -send 100 -format 2 -send 120 -annotation '13-26 binding site 15-15 SNP' </b>
Displays sequences with features in pretty format
Output file [x13776.showseq]: <b></b>

</pre></td></tr></table><p>
<p>
<a href="#output.8">Go to the output files for this example</a><p><p>

<H2>
    Command line arguments
</H2>
<table CELLSPACING=0 CELLPADDING=3 BGCOLOR="#f5f5ff" ><tr><td>
<pre>
Displays sequences with features in pretty format
Version: EMBOSS:6.2.0

   Standard (Mandatory) qualifiers (* if not always prompted):
  [-sequence]          seqall     (Gapped) nucleotide sequence(s) filename and
                                  optional format, or reference (input USA)
   -format             menu       [2] Display format (Values: 0 (Enter your
                                  own list of things to display); 1 (Sequence
                                  only); 2 (Default sequence with features); 3
                                  (Pretty sequence); 4 (One frame
                                  translation); 5 (Three frame translations);
                                  6 (Six frame translations); 7 (Restriction
                                  enzyme map); 8 (Baroque))
*  -things             menu       [B,N,T,S,A,F] Specify a list of one or more
                                  code characters in the order in which you
                                  wish things to be displayed one above the
                                  other down the page. For example if you wish
                                  to see things displayed in the order:
                                  sequence, complement sequence, ticks line,
                                  frame 1 translation, blank line; then you
                                  should enter 'S,C,T,1,B'. (Values: S
                                  (Sequence); B (Blank line); 1 (Frame1
                                  translation); 2 (Frame2 translation); 3
                                  (Frame3 translation); -1 (CompFrame1
                                  translation); -2 (CompFrame2 translation);
                                  -3 (CompFrame3 translation); T (Ticks line);
                                  N (Number ticks line); C (Complement
                                  sequence); F (Features); R (Restriction
                                  enzyme cut sites in forward sense); -R
                                  (Restriction enzyme cut sites in reverse
                                  sense); A (Annotation))
  [-outfile]           outfile    [*.showseq] Output file name

   Additional (Optional) qualifiers:
   -translate          range      [If this is left blank the complete sequence
                                  is translated.] Regions to translate (if
                                  translating).
                                  If this is left blank the complete sequence
                                  is translated.
                                  A set of regions is specified by a set of
                                  pairs of positions.
                                  The positions are integers.
                                  They are separated by any non-digit,
                                  non-alpha character.
                                  Examples of region specifications are:
                                  24-45, 56-78
                                  1:45, 67=99;765..888
   -revtranslate       range      [If this is left blank the complete reverse
                                  sequence is translated.] Regions to
                                  translate (if translating).
                                  If this is left blank the complete sequence
                                  is translated.
                                  A set of regions is specified by a set of
                                  pairs of positions.
                                  The positions are integers.
                                  They are separated by any non-digit,
                                  non-alpha character.
                                  Examples of region specifications are:
                                  78-56, 45-24,
                                  888..765, 99=67; 45:1
   -uppercase          range      [If this is left blank, then the sequence
                                  case is left alone.] Regions to put in
                                  uppercase.
                                  If this is left blank, then the sequence
                                  case is left alone.
                                  A set of regions is specified by a set of
                                  pairs of positions.
                                  The positions are integers.
                                  They are separated by any non-digit,
                                  non-alpha character.
                                  Examples of region specifications are:
                                  24-45, 56-78
                                  1:45, 67=99;765..888
                                  1,5,8,10,23,45,57,99
   -highlight          range      [(full sequence)] Regions to colour if
                                  formatting for HTML.
                                  If this is left blank, then the sequence is
                                  left alone.
                                  A set of regions is specified by a set of
                                  pairs of positions.
                                  The positions are integers.
                                  They are followed by any valid HTML font
                                  colour.
                                  Examples of region specifications are:
                                  24-45 blue 56-78 orange
                                  1-100 green 120-156 red
                                  A file of ranges to colour (one range per
                                  line) can be specified as '@filename'.
   -annotation         range      [If this is left blank, then no annotation
                                  is added.] Regions to annotate by marking.
                                  If this is left blank, then no annotation is
                                  added.
                                  A set of regions is specified by a set of
                                  pairs of positions followed by optional
                                  text.
                                  The positions are integers.
                                  They are followed by any text (but not
                                  digits when on the command-line).
                                  Examples of region specifications are:
                                  24-45 new domain 56-78 match to Mouse
                                  1-100 First part 120-156 oligo
                                  A file of ranges to annotate (one range per
                                  line) can be specified as '@filename'.
   -enzymes            string     [all] The name 'all' reads in all enzyme
                                  names from the REBASE database. You can
                                  specify enzymes by giving their names with
                                  commas between then, such as:
                                  'HincII,hinfI,ppiI,hindiii'.
                                  The case of the names is not important. You
                                  can specify a file of enzyme names to read
                                  in by giving the name of the file holding
                                  the enzyme names with a '@' character in
                                  front of it, for example, '@enz.list'.
                                  Blank lines and lines starting with a hash
                                  character or '!' are ignored and all other
                                  lines are concatenated together with a comma
                                  character ',' and then treated as the list
                                  of enzymes to search for.
                                  An example of a file of enzyme names is:
                                  ! my enzymes
                                  HincII, ppiII
                                  ! other enzymes
                                  hindiii
                                  HinfI
                                  PpiI (Any string)
   -table              menu       [0] Genetic code to use (Values: 0
                                  (Standard); 1 (Standard (with alternative
                                  initiation codons)); 2 (Vertebrate
                                  Mitochondrial); 3 (Yeast Mitochondrial); 4
                                  (Mold, Protozoan, Coelenterate Mitochondrial
                                  and Mycoplasma/Spiroplasma); 5
                                  (Invertebrate Mitochondrial); 6 (Ciliate
                                  Macronuclear and Dasycladacean); 9
                                  (Echinoderm Mitochondrial); 10 (Euplotid
                                  Nuclear); 11 (Bacterial); 12 (Alternative
                                  Yeast Nuclear); 13 (Ascidian Mitochondrial);
                                  14 (Flatworm Mitochondrial); 15
                                  (Blepharisma Macronuclear); 16
                                  (Chlorophycean Mitochondrial); 21 (Trematode
                                  Mitochondrial); 22 (Scenedesmus obliquus);
                                  23 (Thraustochytrium Mitochondrial))
   -sourcematch        string     [*] By default any feature source in the
                                  feature table is shown. You can set this to
                                  match any feature source you wish to show.
                                  The source name is usually either the name
                                  of the program that detected the feature or
                                  it is the feature table (eg: EMBL) that the
                                  feature came from.
                                  The source may be wildcarded by using '*'.
                                  If you wish to show more than one source,
                                  separate their names with the character '|',
                                  eg:
                                  gene* | embl (Any string)
   -typematch          string     [*] By default any feature type in the
                                  feature table is shown. You can set this to
                                  match any feature type you wish to show.
                                  See http://www3.ebi.ac.uk/Services/WebFeat/
                                  for a list of the EMBL feature types and see
                                  Appendix A of the Swissprot user manual in
                                  http://www.expasy.ch/txt/userman.txt for a
                                  list of the Swissprot feature types.
                                  The type may be wildcarded by using '*'.
                                  If you wish to show more than one type,
                                  separate their names with the character '|',
                                  eg:
                                  *UTR | intron (Any string)
   -sensematch         integer    [0 - any sense, 1 - forward sense, -1 -
                                  reverse sense] By default any feature type
                                  in the feature table is shown. You can set
                                  this to match any feature sense you wish to
                                  show. 0 - any sense, 1 - forward sense, -1 -
                                  reverse sense (Any integer value)
   -minscore           float      [0.0] Minimum score of feature to display
                                  (see also maxscore) (Any numeric value)
   -maxscore           float      [0.0] Maximum score of feature to display.
                                  If both minscore and maxscore are zero (the
                                  default), then any score is ignored (Any
                                  numeric value)
   -tagmatch           string     [*] Tags are the types of extra values that
                                  a feature may have. For example in the EMBL
                                  feature table, a 'CDS' type of feature may
                                  have the tags '/codon', '/codon_start',
                                  '/db_xref', '/EC_number', '/evidence',
                                  '/exception', '/function', '/gene',
                                  '/label', '/map', '/note', '/number',
                                  '/partial', '/product', '/protein_id',
                                  '/pseudo', '/standard_name', '/translation',
                                  '/transl_except', '/transl_table', or
                                  '/usedin'. Some of these tags also have
                                  values, for example '/gene' can have the
                                  value of the gene name.
                                  By default any feature tag in the feature
                                  table is shown. You can set this to match
                                  any feature tag you wish to show.
                                  The tag may be wildcarded by using '*'.
                                  If you wish to show more than one tag,
                                  separate their names with the character '|',
                                  eg:
                                  gene | label (Any string)
   -valuematch         string     [*] Tag values are the values associated
                                  with a feature tag. Tags are the types of
                                  extra values that a feature may have. For
                                  example in the EMBL feature table, a 'CDS'
                                  type of feature may have the tags '/codon',
                                  '/codon_start', '/db_xref', '/EC_number',
                                  '/evidence', '/exception', '/function',
                                  '/gene', '/label', '/map', '/note',
                                  '/number', '/partial', '/product',
                                  '/protein_id', '/pseudo', '/standard_name',
                                  '/translation', '/transl_except',
                                  '/transl_table', or '/usedin'. Only some of
                                  these tags can have values, for example
                                  '/gene' can have the value of the gene name.
                                  By default any feature tag value in the
                                  feature table is shown. You can set this to
                                  match any feature tag valueyou wish to show.
                                  The tag value may be wildcarded by using
                                  '*'.
                                  If you wish to show more than one tag value,
                                  separate their names with the character
                                  '|', eg:
                                  pax* | 10 (Any string)
   -stricttags         boolean    [N] By default if any tag/value pair in a
                                  feature matches the specified tag and value,
                                  then all the tags/value pairs of that
                                  feature will be displayed. If this is set to
                                  be true, then only those tag/value pairs in
                                  a feature that match the specified tag and
                                  value will be displayed.

   Advanced (Unprompted) qualifiers:
   -mfile              datafile   [Emethylsites.dat] Restriction enzyme
                                  methylation data file (optional)
   -flatreformat       boolean    [N] This changes the output format to one
                                  where the recognition site is indicated by a
                                  row of '===' characters and the cut site is
                                  pointed to by a '>' character in the
                                  forward sense, or a '<' in the reverse sense
                                  strand.
   -mincuts            integer    [1] This sets the minimum number of cuts for
                                  any restriction enzyme that will be
                                  considered. Any enzymes that cut fewer times
                                  than this will be ignored. (Integer from 1
                                  to 1000)
   -maxcuts            integer    [2000000000] This sets the maximum number of
                                  cuts for any restriction enzyme that will
                                  be considered. Any enzymes that cut more
                                  times than this will be ignored. (Integer up
                                  to 2000000000)
   -sitelen            integer    [4] This sets the minimum length of the
                                  restriction enzyme recognition site. Any
                                  enzymes with sites shorter than this will be
                                  ignored. (Integer from 2 to 20)
   -single             boolean    [N] If this is set then this forces the
                                  values of the mincuts and maxcuts qualifiers
                                  to both be 1. Any other value you may have
                                  set them to will be ignored.
   -[no]blunt          boolean    [Y] This allows those enzymes which cut at
                                  the same position on the forward and reverse
                                  strands to be considered.
   -[no]sticky         boolean    [Y] This allows those enzymes which cut at
                                  different positions on the forward and
                                  reverse strands, leaving an overhang, to be
                                  considered.
   -[no]ambiguity      boolean    [Y] This allows those enzymes which have one
                                  or more 'N' ambiguity codes in their
                                  pattern to be considered
   -plasmid            boolean    [N] If this is set then this allows searches
                                  for restriction enzyme recognition site and
                                  cut postions that span the end of the
                                  sequence to be considered.
   -methylation        boolean    [N] If this is set then RE recognition sites
                                  will not match methylated bases.
   -[no]commercial     boolean    [Y] If this is set, then only those enzymes
                                  with a commercial supplier will be searched
                                  for. This qualifier is ignored if you have
                                  specified an explicit list of enzymes to
                                  search for, rather than searching through
                                  'all' the enzymes in the REBASE database. It
                                  is assumed that, if you are asking for an
                                  explicit enzyme, then you probably know
                                  where to get it from and so all enzymes
                                  names that you have asked to be searched
                                  for, and which cut, will be reported whether
                                  or not they have a commercial supplier.
   -[no]limit          boolean    [Y] This limits the reporting of enzymes to
                                  just one enzyme from each group of
                                  isoschizomers. The enzyme chosen to
                                  represent an isoschizomer group is the
                                  prototype indicated in the data file
                                  'embossre.equ', which is created by the
                                  program 'rebaseextract'. If you prefer
                                  different prototypes to be used, make a copy
                                  of embossre.equ in your home directory and
                                  edit it. If this value is set to be false
                                  then all of the input enzymes will be
                                  reported. You might like to set this to
                                  false if you are supplying an explicit set
                                  of enzymes rather than searching 'all' of
                                  them.
   -orfminsize         integer    [0] This sets the minimum size of Open
                                  Reading Frames (ORFs) to display in the
                                  translations. All other translation regions
                                  are masked by changing the amino acids to
                                  '-' characters. (Integer 0 or more)
   -threeletter        boolean    [N] Display protein sequences in
                                  three-letter code
   -number             boolean    [N] Number the sequences
   -width              integer    [60] Width of sequence to display (Integer 1
                                  or more)
   -length             integer    [0] Line length of page (0 for indefinite)
                                  (Integer 0 or more)
   -margin             integer    [10] Margin around sequence for numbering
                                  (Integer 0 or more)
   -[no]name           boolean    [Y] Set this to be false if you do not wish
                                  to display the ID name of the sequence
   -[no]description    boolean    [Y] Set this to be false if you do not wish
                                  to display the description of the sequence
   -offset             integer    [1] Offset to start numbering the sequence
                                  from (Any integer value)
   -html               boolean    [N] Use HTML formatting

   Associated qualifiers:

   "-sequence" associated qualifiers
   -sbegin1            integer    Start of each sequence to be used
   -send1              integer    End of each sequence to be used
   -sreverse1          boolean    Reverse (if DNA)
   -sask1              boolean    Ask for begin/end/reverse
   -snucleotide1       boolean    Sequence is nucleotide
   -sprotein1          boolean    Sequence is protein
   -slower1            boolean    Make lower case
   -supper1            boolean    Make upper case
   -sformat1           string     Input sequence format
   -sdbname1           string     Database name
   -sid1               string     Entryname
   -ufo1               string     UFO features
   -fformat1           string     Features format
   -fopenfile1         string     Features file name

   "-outfile" associated qualifiers
   -odirectory2        string     Output directory

   General qualifiers:
   -auto               boolean    Turn off prompts
   -stdout             boolean    Write first file to standard output
   -filter             boolean    Read first file from standard input, write
                                  first file to standard output
   -options            boolean    Prompt for standard and additional values
   -debug              boolean    Write debug output to program.dbg
   -verbose            boolean    Report some/full command line options
   -help               boolean    Report command line options and exit. More
                                  information on associated and general
                                  qualifiers can be found with -help -verbose
   -warning            boolean    Report warnings
   -error              boolean    Report errors
   -fatal              boolean    Report fatal errors
   -die                boolean    Report dying program messages
   -version            boolean    Report version number and exit

</pre>
</td></tr></table>
<P>

<table border cellspacing=0 cellpadding=3 bgcolor="#ccccff">
<tr bgcolor="#FFFFCC">
<th align="left">Qualifier</th>
<th align="left">Type</th>
<th align="left">Description</th>
<th align="left">Allowed values</th>
<th align="left">Default</th>
</tr>

<tr bgcolor="#FFFFCC">
<th align="left" colspan=5>Standard (Mandatory) qualifiers</th>
</tr>

<tr bgcolor="#FFFFCC">
<td>[-sequence]<br>(Parameter 1)</td>
<td>seqall</td>
<td>(Gapped) nucleotide sequence(s) filename and optional format, or reference (input USA)</td>
<td>Readable sequence(s)</td>
<td><b>Required</b></td>
</tr>

<tr bgcolor="#FFFFCC">
<td>-format</td>
<td>list</td>
<td>Display format</td>
<td><table><tr><td>0</td> <td><i>(Enter your own list of things to display)</i></td></tr><tr><td>1</td> <td><i>(Sequence only)</i></td></tr><tr><td>2</td> <td><i>(Default sequence with features)</i></td></tr><tr><td>3</td> <td><i>(Pretty sequence)</i></td></tr><tr><td>4</td> <td><i>(One frame translation)</i></td></tr><tr><td>5</td> <td><i>(Three frame translations)</i></td></tr><tr><td>6</td> <td><i>(Six frame translations)</i></td></tr><tr><td>7</td> <td><i>(Restriction enzyme map)</i></td></tr><tr><td>8</td> <td><i>(Baroque)</i></td></tr></table></td>
<td>2</td>
</tr>

<tr bgcolor="#FFFFCC">
<td>-things</td>
<td>list</td>
<td>Specify a list of one or more code characters in the order in which you wish things to be displayed one above the other down the page. For example if you wish to see things displayed in the order: sequence, complement sequence, ticks line, frame 1 translation, blank line; then you should enter 'S,C,T,1,B'.</td>
<td><table><tr><td>S</td> <td><i>(Sequence)</i></td></tr><tr><td>B</td> <td><i>(Blank line)</i></td></tr><tr><td>1</td> <td><i>(Frame1 translation)</i></td></tr><tr><td>2</td> <td><i>(Frame2 translation)</i></td></tr><tr><td>3</td> <td><i>(Frame3 translation)</i></td></tr><tr><td>-1</td> <td><i>(CompFrame1 translation)</i></td></tr><tr><td>-2</td> <td><i>(CompFrame2 translation)</i></td></tr><tr><td>-3</td> <td><i>(CompFrame3 translation)</i></td></tr><tr><td>T</td> <td><i>(Ticks line)</i></td></tr><tr><td>N</td> <td><i>(Number ticks line)</i></td></tr><tr><td>C</td> <td><i>(Complement sequence)</i></td></tr><tr><td>F</td> <td><i>(Features)</i></td></tr><tr><td>R</td> <td><i>(Restriction enzyme cut sites in forward sense)</i></td></tr><tr><td>-R</td> <td><i>(Restriction enzyme cut sites in reverse sense)</i></td></tr><tr><td>A</td> <td><i>(Annotation)</i></td></tr></table></td>
<td>B,N,T,S,A,F</td>
</tr>

<tr bgcolor="#FFFFCC">
<td>[-outfile]<br>(Parameter 2)</td>
<td>outfile</td>
<td>Output file name</td>
<td>Output file</td>
<td><i>&lt;*&gt;</i>.showseq</td>
</tr>

<tr bgcolor="#FFFFCC">
<th align="left" colspan=5>Additional (Optional) qualifiers</th>
</tr>

<tr bgcolor="#FFFFCC">
<td>-translate</td>
<td>range</td>
<td>Regions to translate (if translating).
If this is left blank the complete sequence is translated.
A set of regions is specified by a set of pairs of positions.
The positions are integers.
They are separated by any non-digit, non-alpha character.
Examples of region specifications are:
24-45, 56-78
1:45, 67=99;765..888</td>
<td>Sequence range</td>
<td>If this is left blank the complete sequence is translated.</td>
</tr>

<tr bgcolor="#FFFFCC">
<td>-revtranslate</td>
<td>range</td>
<td>Regions to translate (if translating).
If this is left blank the complete sequence is translated.
A set of regions is specified by a set of pairs of positions.
The positions are integers.
They are separated by any non-digit, non-alpha character.
Examples of region specifications are:
78-56, 45-24,
888..765, 99=67; 45:1</td>
<td>Sequence range</td>
<td>If this is left blank the complete reverse sequence is translated.</td>
</tr>

<tr bgcolor="#FFFFCC">
<td>-uppercase</td>
<td>range</td>
<td>Regions to put in uppercase.
If this is left blank, then the sequence case is left alone.
A set of regions is specified by a set of pairs of positions.
The positions are integers.
They are separated by any non-digit, non-alpha character.
Examples of region specifications are:
24-45, 56-78
1:45, 67=99;765..888
1,5,8,10,23,45,57,99</td>
<td>Sequence range</td>
<td>If this is left blank, then the sequence case is left alone.</td>
</tr>

<tr bgcolor="#FFFFCC">
<td>-highlight</td>
<td>range</td>
<td>Regions to colour if formatting for HTML.
If this is left blank, then the sequence is left alone.
A set of regions is specified by a set of pairs of positions.
The positions are integers.
They are followed by any valid HTML font colour.
Examples of region specifications are:
24-45 blue 56-78 orange
1-100 green 120-156 red
A file of ranges to colour (one range per line) can be specified as '@filename'.</td>
<td>Sequence range</td>
<td><i>full sequence</i></td>
</tr>

<tr bgcolor="#FFFFCC">
<td>-annotation</td>
<td>range</td>
<td>Regions to annotate by marking.
If this is left blank, then no annotation is added.
A set of regions is specified by a set of pairs of positions followed by optional text.
The positions are integers.
They are followed by any text (but not digits when on the command-line).
Examples of region specifications are:
24-45 new domain 56-78 match to Mouse
1-100 First part 120-156 oligo
A file of ranges to annotate (one range per line) can be specified as '@filename'.</td>
<td>Sequence range</td>
<td>If this is left blank, then no annotation is added.</td>
</tr>

<tr bgcolor="#FFFFCC">
<td>-enzymes</td>
<td>string</td>
<td>The name 'all' reads in all enzyme names from the REBASE database. You can specify enzymes by giving their names with commas between then, such as: 'HincII,hinfI,ppiI,hindiii'.
The case of the names is not important. You can specify a file of enzyme names to read in by giving the name of the file holding the enzyme names with a '@' character in front of it, for example, '@enz.list'.
Blank lines and lines starting with a hash character or '!' are ignored and all other lines are concatenated together with a comma character ',' and then treated as the list of enzymes to search for.
An example of a file of enzyme names is:
! my enzymes
HincII, ppiII
! other enzymes
hindiii
HinfI
PpiI</td>
<td>Any string</td>
<td>all</td>
</tr>

<tr bgcolor="#FFFFCC">
<td>-table</td>
<td>list</td>
<td>Genetic code to use</td>
<td><table><tr><td>0</td> <td><i>(Standard)</i></td></tr><tr><td>1</td> <td><i>(Standard (with alternative initiation codons))</i></td></tr><tr><td>2</td> <td><i>(Vertebrate Mitochondrial)</i></td></tr><tr><td>3</td> <td><i>(Yeast Mitochondrial)</i></td></tr><tr><td>4</td> <td><i>(Mold, Protozoan, Coelenterate Mitochondrial and Mycoplasma/Spiroplasma)</i></td></tr><tr><td>5</td> <td><i>(Invertebrate Mitochondrial)</i></td></tr><tr><td>6</td> <td><i>(Ciliate Macronuclear and Dasycladacean)</i></td></tr><tr><td>9</td> <td><i>(Echinoderm Mitochondrial)</i></td></tr><tr><td>10</td> <td><i>(Euplotid Nuclear)</i></td></tr><tr><td>11</td> <td><i>(Bacterial)</i></td></tr><tr><td>12</td> <td><i>(Alternative Yeast Nuclear)</i></td></tr><tr><td>13</td> <td><i>(Ascidian Mitochondrial)</i></td></tr><tr><td>14</td> <td><i>(Flatworm Mitochondrial)</i></td></tr><tr><td>15</td> <td><i>(Blepharisma Macronuclear)</i></td></tr><tr><td>16</td> <td><i>(Chlorophycean Mitochondrial)</i></td></tr><tr><td>21</td> <td><i>(Trematode Mitochondrial)</i></td></tr><tr><td>22</td> <td><i>(Scenedesmus obliquus)</i></td></tr><tr><td>23</td> <td><i>(Thraustochytrium Mitochondrial)</i></td></tr></table></td>
<td>0</td>
</tr>

<tr bgcolor="#FFFFCC">
<td>-sourcematch</td>
<td>string</td>
<td>By default any feature source in the feature table is shown. You can set this to match any feature source you wish to show.
The source name is usually either the name of the program that detected the feature or it is the feature table (eg: EMBL) that the feature came from.
The source may be wildcarded by using '*'.
If you wish to show more than one source, separate their names with the character '|', eg:
gene* | embl</td>
<td>Any string</td>
<td>*</td>
</tr>

<tr bgcolor="#FFFFCC">
<td>-typematch</td>
<td>string</td>
<td>By default any feature type in the feature table is shown. You can set this to match any feature type you wish to show.
See http://www3.ebi.ac.uk/Services/WebFeat/ for a list of the EMBL feature types and see Appendix A of the Swissprot user manual in http://www.expasy.ch/txt/userman.txt for a list of the Swissprot feature types.
The type may be wildcarded by using '*'.
If you wish to show more than one type, separate their names with the character '|', eg:
*UTR | intron</td>
<td>Any string</td>
<td>*</td>
</tr>

<tr bgcolor="#FFFFCC">
<td>-sensematch</td>
<td>integer</td>
<td>By default any feature type in the feature table is shown. You can set this to match any feature sense you wish to show. 0 - any sense, 1 - forward sense, -1 - reverse sense</td>
<td>Any integer value</td>
<td>0 - any sense, 1 - forward sense, -1 - reverse sense</td>
</tr>

<tr bgcolor="#FFFFCC">
<td>-minscore</td>
<td>float</td>
<td>Minimum score of feature to display (see also maxscore)</td>
<td>Any numeric value</td>
<td>0.0</td>
</tr>

<tr bgcolor="#FFFFCC">
<td>-maxscore</td>
<td>float</td>
<td>Maximum score of feature to display.
If both minscore and maxscore are zero (the default), then any score is ignored</td>
<td>Any numeric value</td>
<td>0.0</td>
</tr>

<tr bgcolor="#FFFFCC">
<td>-tagmatch</td>
<td>string</td>
<td>Tags are the types of extra values that a feature may have. For example in the EMBL feature table, a 'CDS' type of feature may have the tags '/codon', '/codon_start', '/db_xref', '/EC_number', '/evidence', '/exception', '/function', '/gene', '/label', '/map', '/note', '/number', '/partial', '/product', '/protein_id', '/pseudo', '/standard_name', '/translation', '/transl_except', '/transl_table', or '/usedin'. Some of these tags also have values, for example '/gene' can have the value of the gene name.
By default any feature tag in the feature table is shown. You can set this to match any feature tag you wish to show.
The tag may be wildcarded by using '*'.
If you wish to show more than one tag, separate their names with the character '|', eg:
gene | label</td>
<td>Any string</td>
<td>*</td>
</tr>

<tr bgcolor="#FFFFCC">
<td>-valuematch</td>
<td>string</td>
<td>Tag values are the values associated with a feature tag. Tags are the types of extra values that a feature may have. For example in the EMBL feature table, a 'CDS' type of feature may have the tags '/codon', '/codon_start', '/db_xref', '/EC_number', '/evidence', '/exception', '/function', '/gene', '/label', '/map', '/note', '/number', '/partial', '/product', '/protein_id', '/pseudo', '/standard_name', '/translation', '/transl_except', '/transl_table', or '/usedin'. Only some of these tags can have values, for example '/gene' can have the value of the gene name. By default any feature tag value in the feature table is shown. You can set this to match any feature tag valueyou wish to show.
The tag value may be wildcarded by using '*'.
If you wish to show more than one tag value, separate their names with the character '|', eg:
pax* | 10</td>
<td>Any string</td>
<td>*</td>
</tr>

<tr bgcolor="#FFFFCC">
<td>-stricttags</td>
<td>boolean</td>
<td>By default if any tag/value pair in a feature matches the specified tag and value, then all the tags/value pairs of that feature will be displayed. If this is set to be true, then only those tag/value pairs in a feature that match the specified tag and value will be displayed.</td>
<td>Boolean value Yes/No</td>
<td>No</td>
</tr>

<tr bgcolor="#FFFFCC">
<th align="left" colspan=5>Advanced (Unprompted) qualifiers</th>
</tr>

<tr bgcolor="#FFFFCC">
<td>-mfile</td>
<td>datafile</td>
<td>Restriction enzyme methylation data file (optional)</td>
<td>Data file</td>
<td>Emethylsites.dat</td>
</tr>

<tr bgcolor="#FFFFCC">
<td>-flatreformat</td>
<td>boolean</td>
<td>This changes the output format to one where the recognition site is indicated by a row of '===' characters and the cut site is pointed to by a '&gt;' character in the forward sense, or a '&lt;' in the reverse sense strand.</td>
<td>Boolean value Yes/No</td>
<td>No</td>
</tr>

<tr bgcolor="#FFFFCC">
<td>-mincuts</td>
<td>integer</td>
<td>This sets the minimum number of cuts for any restriction enzyme that will be considered. Any enzymes that cut fewer times than this will be ignored.</td>
<td>Integer from 1 to 1000</td>
<td>1</td>
</tr>

<tr bgcolor="#FFFFCC">
<td>-maxcuts</td>
<td>integer</td>
<td>This sets the maximum number of cuts for any restriction enzyme that will be considered. Any enzymes that cut more times than this will be ignored.</td>
<td>Integer up to 2000000000</td>
<td>2000000000</td>
</tr>

<tr bgcolor="#FFFFCC">
<td>-sitelen</td>
<td>integer</td>
<td>This sets the minimum length of the restriction enzyme recognition site. Any enzymes with sites shorter than this will be ignored.</td>
<td>Integer from 2 to 20</td>
<td>4</td>
</tr>

<tr bgcolor="#FFFFCC">
<td>-single</td>
<td>boolean</td>
<td>If this is set then this forces the values of the mincuts and maxcuts qualifiers to both be 1. Any other value you may have set them to will be ignored.</td>
<td>Boolean value Yes/No</td>
<td>No</td>
</tr>

<tr bgcolor="#FFFFCC">
<td>-[no]blunt</td>
<td>boolean</td>
<td>This allows those enzymes which cut at the same position on the forward and reverse strands to be considered.</td>
<td>Boolean value Yes/No</td>
<td>Yes</td>
</tr>

<tr bgcolor="#FFFFCC">
<td>-[no]sticky</td>
<td>boolean</td>
<td>This allows those enzymes which cut at different positions on the forward and reverse strands, leaving an overhang, to be considered.</td>
<td>Boolean value Yes/No</td>
<td>Yes</td>
</tr>

<tr bgcolor="#FFFFCC">
<td>-[no]ambiguity</td>
<td>boolean</td>
<td>This allows those enzymes which have one or more 'N' ambiguity codes in their pattern to be considered</td>
<td>Boolean value Yes/No</td>
<td>Yes</td>
</tr>

<tr bgcolor="#FFFFCC">
<td>-plasmid</td>
<td>boolean</td>
<td>If this is set then this allows searches for restriction enzyme recognition site and cut postions that span the end of the sequence to be considered.</td>
<td>Boolean value Yes/No</td>
<td>No</td>
</tr>

<tr bgcolor="#FFFFCC">
<td>-methylation</td>
<td>boolean</td>
<td>If this is set then RE recognition sites will not match methylated bases.</td>
<td>Boolean value Yes/No</td>
<td>No</td>
</tr>

<tr bgcolor="#FFFFCC">
<td>-[no]commercial</td>
<td>boolean</td>
<td>If this is set, then only those enzymes with a commercial supplier will be searched for. This qualifier is ignored if you have specified an explicit list of enzymes to search for, rather than searching through 'all' the enzymes in the REBASE database. It is assumed that, if you are asking for an explicit enzyme, then you probably know where to get it from and so all enzymes names that you have asked to be searched for, and which cut, will be reported whether or not they have a commercial supplier.</td>
<td>Boolean value Yes/No</td>
<td>Yes</td>
</tr>

<tr bgcolor="#FFFFCC">
<td>-[no]limit</td>
<td>boolean</td>
<td>This limits the reporting of enzymes to just one enzyme from each group of isoschizomers. The enzyme chosen to represent an isoschizomer group is the prototype indicated in the data file 'embossre.equ', which is created by the program 'rebaseextract'. If you prefer different prototypes to be used, make a copy of embossre.equ in your home directory and edit it. If this value is set to be false then all of the input enzymes will be reported. You might like to set this to false if you are supplying an explicit set of enzymes rather than searching 'all' of them.</td>
<td>Boolean value Yes/No</td>
<td>Yes</td>
</tr>

<tr bgcolor="#FFFFCC">
<td>-orfminsize</td>
<td>integer</td>
<td>This sets the minimum size of Open Reading Frames (ORFs) to display in the translations. All other translation regions are masked by changing the amino acids to '-' characters.</td>
<td>Integer 0 or more</td>
<td>0</td>
</tr>

<tr bgcolor="#FFFFCC">
<td>-threeletter</td>
<td>boolean</td>
<td>Display protein sequences in three-letter code</td>
<td>Boolean value Yes/No</td>
<td>No</td>
</tr>

<tr bgcolor="#FFFFCC">
<td>-number</td>
<td>boolean</td>
<td>Number the sequences</td>
<td>Boolean value Yes/No</td>
<td>No</td>
</tr>

<tr bgcolor="#FFFFCC">
<td>-width</td>
<td>integer</td>
<td>Width of sequence to display</td>
<td>Integer 1 or more</td>
<td>60</td>
</tr>

<tr bgcolor="#FFFFCC">
<td>-length</td>
<td>integer</td>
<td>Line length of page (0 for indefinite)</td>
<td>Integer 0 or more</td>
<td>0</td>
</tr>

<tr bgcolor="#FFFFCC">
<td>-margin</td>
<td>integer</td>
<td>Margin around sequence for numbering</td>
<td>Integer 0 or more</td>
<td>10</td>
</tr>

<tr bgcolor="#FFFFCC">
<td>-[no]name</td>
<td>boolean</td>
<td>Set this to be false if you do not wish to display the ID name of the sequence</td>
<td>Boolean value Yes/No</td>
<td>Yes</td>
</tr>

<tr bgcolor="#FFFFCC">
<td>-[no]description</td>
<td>boolean</td>
<td>Set this to be false if you do not wish to display the description of the sequence</td>
<td>Boolean value Yes/No</td>
<td>Yes</td>
</tr>

<tr bgcolor="#FFFFCC">
<td>-offset</td>
<td>integer</td>
<td>Offset to start numbering the sequence from</td>
<td>Any integer value</td>
<td>1</td>
</tr>

<tr bgcolor="#FFFFCC">
<td>-html</td>
<td>boolean</td>
<td>Use HTML formatting</td>
<td>Boolean value Yes/No</td>
<td>No</td>
</tr>

<tr bgcolor="#FFFFCC">
<th align="left" colspan=5>Associated qualifiers</th>
</tr>

<tr bgcolor="#FFFFCC">
<th align="left" colspan=5>"-sequence" associated seqall qualifiers
</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -sbegin1<br>-sbegin_sequence</td>
<td>integer</td>
<td>Start of each sequence to be used</td>
<td>Any integer value</td>
<td>0</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -send1<br>-send_sequence</td>
<td>integer</td>
<td>End of each sequence to be used</td>
<td>Any integer value</td>
<td>0</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -sreverse1<br>-sreverse_sequence</td>
<td>boolean</td>
<td>Reverse (if DNA)</td>
<td>Boolean value Yes/No</td>
<td>N</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -sask1<br>-sask_sequence</td>
<td>boolean</td>
<td>Ask for begin/end/reverse</td>
<td>Boolean value Yes/No</td>
<td>N</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -snucleotide1<br>-snucleotide_sequence</td>
<td>boolean</td>
<td>Sequence is nucleotide</td>
<td>Boolean value Yes/No</td>
<td>N</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -sprotein1<br>-sprotein_sequence</td>
<td>boolean</td>
<td>Sequence is protein</td>
<td>Boolean value Yes/No</td>
<td>N</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -slower1<br>-slower_sequence</td>
<td>boolean</td>
<td>Make lower case</td>
<td>Boolean value Yes/No</td>
<td>N</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -supper1<br>-supper_sequence</td>
<td>boolean</td>
<td>Make upper case</td>
<td>Boolean value Yes/No</td>
<td>N</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -sformat1<br>-sformat_sequence</td>
<td>string</td>
<td>Input sequence format</td>
<td>Any string</td>
<td>&nbsp;</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -sdbname1<br>-sdbname_sequence</td>
<td>string</td>
<td>Database name</td>
<td>Any string</td>
<td>&nbsp;</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -sid1<br>-sid_sequence</td>
<td>string</td>
<td>Entryname</td>
<td>Any string</td>
<td>&nbsp;</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -ufo1<br>-ufo_sequence</td>
<td>string</td>
<td>UFO features</td>
<td>Any string</td>
<td>&nbsp;</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -fformat1<br>-fformat_sequence</td>
<td>string</td>
<td>Features format</td>
<td>Any string</td>
<td>&nbsp;</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -fopenfile1<br>-fopenfile_sequence</td>
<td>string</td>
<td>Features file name</td>
<td>Any string</td>
<td>&nbsp;</td>
</tr>

<tr bgcolor="#FFFFCC">
<th align="left" colspan=5>"-outfile" associated outfile qualifiers
</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -odirectory2<br>-odirectory_outfile</td>
<td>string</td>
<td>Output directory</td>
<td>Any string</td>
<td>&nbsp;</td>
</tr>

<tr bgcolor="#FFFFCC">
<th align="left" colspan=5>General qualifiers</th>
</tr>

<tr bgcolor="#FFFFCC">
<td> -auto</td>
<td>boolean</td>
<td>Turn off prompts</td>
<td>Boolean value Yes/No</td>
<td>N</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -stdout</td>
<td>boolean</td>
<td>Write first file to standard output</td>
<td>Boolean value Yes/No</td>
<td>N</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -filter</td>
<td>boolean</td>
<td>Read first file from standard input, write first file to standard output</td>
<td>Boolean value Yes/No</td>
<td>N</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -options</td>
<td>boolean</td>
<td>Prompt for standard and additional values</td>
<td>Boolean value Yes/No</td>
<td>N</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -debug</td>
<td>boolean</td>
<td>Write debug output to program.dbg</td>
<td>Boolean value Yes/No</td>
<td>N</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -verbose</td>
<td>boolean</td>
<td>Report some/full command line options</td>
<td>Boolean value Yes/No</td>
<td>Y</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -help</td>
<td>boolean</td>
<td>Report command line options and exit. More information on associated and general qualifiers can be found with -help -verbose</td>
<td>Boolean value Yes/No</td>
<td>N</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -warning</td>
<td>boolean</td>
<td>Report warnings</td>
<td>Boolean value Yes/No</td>
<td>Y</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -error</td>
<td>boolean</td>
<td>Report errors</td>
<td>Boolean value Yes/No</td>
<td>Y</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -fatal</td>
<td>boolean</td>
<td>Report fatal errors</td>
<td>Boolean value Yes/No</td>
<td>Y</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -die</td>
<td>boolean</td>
<td>Report dying program messages</td>
<td>Boolean value Yes/No</td>
<td>Y</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -version</td>
<td>boolean</td>
<td>Report version number and exit</td>
<td>Boolean value Yes/No</td>
<td>N</td>
</tr>

</table>


<H2>
    Input file format
</H2>

<b>showseq</b> reads normal sequence USAs.

<p>


<a name="input.1"></a>
<h3>Input files for usage example </h3>

'tembl:x13776' is a sequence entry in the example nucleic acid database 'tembl'
<p>
<p><h3>Database entry: tembl:x13776</h3>
<table width="90%"><tr><td bgcolor="#FFCCFF">
<pre>
ID   X13776; SV 1; linear; genomic DNA; STD; PRO; 2167 BP.
XX
AC   X13776; M43175;
XX
DT   19-APR-1989 (Rel. 19, Created)
DT   14-NOV-2006 (Rel. 89, Last updated, Version 24)
XX
DE   Pseudomonas aeruginosa amiC and amiR gene for aliphatic amidase regulation
XX
KW   aliphatic amidase regulator; amiC gene; amiR gene.
XX
OS   Pseudomonas aeruginosa
OC   Bacteria; Proteobacteria; Gammaproteobacteria; Pseudomonadales;
OC   Pseudomonadaceae; Pseudomonas.
XX
RN   [1]
RP   1167-2167
RA   Rice P.M.;
RT   ;
RL   Submitted (16-DEC-1988) to the EMBL/GenBank/DDBJ databases.
RL   Rice P.M., EMBL, Postfach 10-2209, Meyerhofstrasse 1, 6900 Heidelberg, FRG.
XX
RN   [2]
RP   1167-2167
RX   DOI; 10.1016/0014-5793(89)80249-2.
RX   PUBMED; 2495988.
RA   Lowe N., Rice P.M., Drew R.E.;
RT   "Nucleotide sequence of the aliphatic amidase regulator gene (amiR) of
RT   Pseudomonas aeruginosa";
RL   FEBS Lett. 246(1-2):39-43(1989).
XX
RN   [3]
RP   1-1292
RX   PUBMED; 1907262.
RA   Wilson S., Drew R.;
RT   "Cloning and DNA sequence of amiC, a new gene regulating expression of the
RT   Pseudomonas aeruginosa aliphatic amidase, and purification of the amiC
RT   product";
RL   J. Bacteriol. 173(16):4914-4921(1991).
XX
RN   [4]
RP   1-2167
RA   Rice P.M.;
RT   ;
RL   Submitted (04-SEP-1991) to the EMBL/GenBank/DDBJ databases.
RL   Rice P.M., EMBL, Postfach 10-2209, Meyerhofstrasse 1, 6900 Heidelberg, FRG.
XX
DR   GOA; Q51417.
DR   InterPro; IPR003211; AmiSUreI_transpt.
DR   UniProtKB/Swiss-Prot; Q51417; AMIS_PSEAE.


<font color=red>  [Part of this file has been deleted for brevity]</font>

FT                   /replace=""
FT                   /note="ClaI fragment deleted in pSW36,  constitutive
FT                   phenotype"
FT   misc_feature    1
FT                   /note="last base of an XhoI site"
FT   misc_feature    648..653
FT                   /note="end of 658bp XhoI fragment, deletion in  pSW3 causes
FT                   constitutive expression of amiE"
FT   conflict        1281
FT                   /replace="g"
FT                   /citation=[3]
XX
SQ   Sequence 2167 BP; 363 A; 712 C; 730 G; 362 T; 0 other;
     ggtaccgctg gccgagcatc tgctcgatca ccaccagccg ggcgacggga actgcacgat        60
     ctacctggcg agcctggagc acgagcgggt tcgcttcgta cggcgctgag cgacagtcac       120
     aggagaggaa acggatggga tcgcaccagg agcggccgct gatcggcctg ctgttctccg       180
     aaaccggcgt caccgccgat atcgagcgct cgcacgcgta tggcgcattg ctcgcggtcg       240
     agcaactgaa ccgcgagggc ggcgtcggcg gtcgcccgat cgaaacgctg tcccaggacc       300
     ccggcggcga cccggaccgc tatcggctgt gcgccgagga cttcattcgc aaccgggggg       360
     tacggttcct cgtgggctgc tacatgtcgc acacgcgcaa ggcggtgatg ccggtggtcg       420
     agcgcgccga cgcgctgctc tgctacccga ccccctacga gggcttcgag tattcgccga       480
     acatcgtcta cggcggtccg gcgccgaacc agaacagtgc gccgctggcg gcgtacctga       540
     ttcgccacta cggcgagcgg gtggtgttca tcggctcgga ctacatctat ccgcgggaaa       600
     gcaaccatgt gatgcgccac ctgtatcgcc agcacggcgg cacggtgctc gaggaaatct       660
     acattccgct gtatccctcc gacgacgact tgcagcgcgc cgtcgagcgc atctaccagg       720
     cgcgcgccga cgtggtcttc tccaccgtgg tgggcaccgg caccgccgag ctgtatcgcg       780
     ccatcgcccg tcgctacggc gacggcaggc ggccgccgat cgccagcctg accaccagcg       840
     aggcggaggt ggcgaagatg gagagtgacg tggcagaggg gcaggtggtg gtcgcgcctt       900
     acttctccag catcgatacg cccgccagcc gggccttcgt ccaggcctgc catggtttct       960
     tcccggagaa cgcgaccatc accgcctggg ccgaggcggc ctactggcag accttgttgc      1020
     tcggccgcgc cgcgcaggcc gcaggcaact ggcgggtgga agacgtgcag cggcacctgt      1080
     acgacatcga catcgacgcg ccacaggggc cggtccgggt ggagcgccag aacaaccaca      1140
     gccgcctgtc ttcgcgcatc gcggaaatcg atgcgcgcgg cgtgttccag gtccgctggc      1200
     agtcgcccga accgattcgc cccgaccctt atgtcgtcgt gcataacctc gacgactggt      1260
     ccgccagcat gggcggggga ccgctcccat gagcgccaac tcgctgctcg gcagcctgcg      1320
     cgagttgcag gtgctggtcc tcaacccgcc gggggaggtc agcgacgccc tggtcttgca      1380
     gctgatccgc atcggttgtt cggtgcgcca gtgctggccg ccgccggaag ccttcgacgt      1440
     gccggtggac gtggtcttca ccagcatttt ccagaatggc caccacgacg agatcgctgc      1500
     gctgctcgcc gccgggactc cgcgcactac cctggtggcg ctggtggagt acgaaagccc      1560
     cgcggtgctc tcgcagatca tcgagctgga gtgccacggc gtgatcaccc agccgctcga      1620
     tgcccaccgg gtgctgcctg tgctggtatc ggcgcggcgc atcagcgagg aaatggcgaa      1680
     gctgaagcag aagaccgagc agctccagga ccgcatcgcc ggccaggccc ggatcaacca      1740
     ggccaaggtg ttgctgatgc agcgccatgg ctgggacgag cgcgaggcgc accagcacct      1800
     gtcgcgggaa gcgatgaagc ggcgcgagcc gatcctgaag atcgctcagg agttgctggg      1860
     aaacgagccg tccgcctgag cgatccgggc cgaccagaac aataacaaga ggggtatcgt      1920
     catcatgctg ggactggttc tgctgtacgt tggcgcggtg ctgtttctca atgccgtctg      1980
     gttgctgggc aagatcagcg gtcgggaggt ggcggtgatc aacttcctgg tcggcgtgct      2040
     gagcgcctgc gtcgcgttct acctgatctt ttccgcagca gccgggcagg gctcgctgaa      2100
     ggccggagcg ctgaccctgc tattcgcttt tacctatctg tgggtggccg ccaaccagtt      2160
     cctcgag                                                                2167
//
</pre>
</td></tr></table><p>

<p>

You can specify a file of ranges to display in uppercase by giving
the '-uppercase' qualifier the value '@' followed by the name of the
file containing the ranges.  (eg: '-upper @myfile').

<p>
          
The format of the range file is:

<p>

<ul>
      <li>Comment lines start with '#' in the first column.
      <li>Comment lines and blank lines are ignored.
      <li>The line may start with white-space.
      <li>There are two positive (integer) numbers per line separated by one
                or more space or
      <li>TAB characters.
      <li>The second number must be greater or equal to the first number.  
      <li>There can be optional text after the two numbers to annotate the line.
      <li>White-space before or after the text is removed.
</ul>

<p>

An example range file is:

<p>

<pre>          
# this is my set of ranges
12   23                           
 4   5       this is like 12-23, but smaller
67   10348   interesting region
</pre>

<p>

You can specify a file of ranges to highlight in a different colour
when outputting in HTML format (using the '-html' qualifier) by giving
the '-highlight' qualifier the value '@' followed by the name of the
file containing the ranges.  (eg: '-highlight @myfile').

<p>

The format of this file is very similar to the format of the above
uppercase range file, except that the text after the start and end
positions is used as the HTML colour name.  This colour name is used 'as
is' when specifying the colour in HTML in a '<FONT COLOR=xxx>'
construct, (where 'xxx' is the name of the colour).

<p>

The standard names of HTML font colours are given in:

<br>

<A HREF="http://http://www.w3.org/TR/REC-html40/types.html">
http://http://www.w3.org/TR/REC-html40/types.html</A>
and
<A HREF="http://www.ausmall.com.au/freegraf/ncolour2.htm">
http://www.ausmall.com.au/freegraf/ncolour2.htm</A>
and
<A HREF="http://mindprod.com/htmlcolours.html">
http://mindprod.com/htmlcolours.html</A>
(amongst other places).


<p>

An example highlight range file is:

<p>

<pre>          
# this is my set of ranges
12   23         red
 4   5          darkturquoise
67   10348      #FFE4E1
</pre>

<p>

You can specify a file of ranges to annotate
by giving
the '-annotate' qualifier the value '@' followed by the name of the
file containing the ranges.  (eg: '-annotate @myfile').

<p>
     
The format of this file is very similar to the format of the above
highlight range file, except that the text after the start and end
positions is used as the displayed text of the annotated region.

<p>

An example annotation range file is:

<p>
   
<pre>
# this is my set of ranges
12   23         exon 1
 4   5          CAP site
67   10348      exon 2
</pre>

<p>


You can specify a file of enzyme names to read in by giving the
'-enzymes' qualifier the name of the file holding the enzyme names with
a '@' character in front of it, for example, '@enz.list'.

<p>

Blank lines and lines starting with a '#' or '!' character are ignored
and all other lines are concatenated together with a comma character ','
and then treated as the list of enzymes to search for.

<p>
         
An example of a file of enzyme names is:

<p>

<pre>      
# my enzymes
HincII, ppiI
# other enzymes
hindiii
HinfI
</pre>

<p>            


<H2>
    Output file format
</H2>


<a name="output.1"></a>
<h3>Output files for usage example </h3>
<p><h3>File: x13776.showseq </h3>
<table width="90%"><tr><td bgcolor="#CCFFCC">
<pre>
X13776
Pseudomonas aeruginosa amiC and amiR gene for aliphatic amidase
regulation

                   10        20        30        40        50        60        
          ----:----|----:----|----:----|----:----|----:----|----:----|
          ggtaccgctggccgagcatctgctcgatcaccaccagccgggcgacgggaactgcacgat
                 |===============|
                 promoter note="proposed rpoN-dependent promoter"
          |
          misc_feature note="last base of an XhoI site"

                   70        80        90        100       
          ----:----|----:----|----:----|----:----|
          ctacctggcgagcctggagcacgagcgggttcgcttcgta
              |===============|
          promoter note="proposed rpoN-dependent promoter"

</pre>
</td></tr></table><p>

<a name="output.2"></a>
<h3>Output files for usage example 2</h3>
<p><h3>File: x13776.showseq </h3>
<table width="90%"><tr><td bgcolor="#CCFFCC">
<pre>
X13776
Pseudomonas aeruginosa amiC and amiR gene for aliphatic amidase
regulation

          ggtaccgctggccgagcatctgctcgatcaccaccagccgggcgacgggaactgcacgat
          ----:----|----:----|----:----|----:----|----:----|----:----|
          ccatggcgaccggctcgtagacgagctagtggtggtcggcccgctgcccttgacgtgcta

          ctacctggcgagcctggagcacgagcgggttcgcttcgtacggcgctgagcgacagtcac
          ----:----|----:----|----:----|----:----|----:----|----:----|
          gatggaccgctcggacctcgtgctcgcccaagcgaagcatgccgcgactcgctgtcagtg

</pre>
</td></tr></table><p>

<a name="output.3"></a>
<h3>Output files for usage example 3</h3>
<p><h3>File: x13776.showseq </h3>
<table width="90%"><tr><td bgcolor="#CCFFCC">
<pre>
          ggtaccgctggccgagcatctgctcgatcaccaccagccgggcgacgggaactgcacgat
          ctacctggcgagcctggagcacgagcgggttcgcttcgta

</pre>
</td></tr></table><p>

<a name="output.4"></a>
<h3>Output files for usage example 4</h3>
<p><h3>File: x13776.showseq </h3>
<table width="90%"><tr><td bgcolor="#CCFFCC">
<pre>
X13776
Pseudomonas aeruginosa amiC and amiR gene for aliphatic amidase
regulation
        1 ggtaccgctggccgagcatctgctcgatcaccaccagccgggcgacgggaactgcacgat 60
       61 ctacctggcgagcctggagcacgagcgggttcgcttcgta 100

</pre>
</td></tr></table><p>

<a name="output.5"></a>
<h3>Output files for usage example 5</h3>
<p><h3>File: x13776.showseq </h3>
<table width="90%"><tr><td bgcolor="#CCFFCC">
<pre>
X13776
Pseudomonas aeruginosa amiC and amiR gene for aliphatic amidase
regulation
      123 ggtaccgctggccgagcatctgctcgatcaccaccagccgggcgacgggaactgcacgat 182
      183 ctacctggcgagcctggagcacgagcgggttcgcttcgta 222

</pre>
</td></tr></table><p>

<a name="output.6"></a>
<h3>Output files for usage example 6</h3>
<p><h3>File: x13776.showseq </h3>
<table width="90%"><tr><td bgcolor="#CCFFCC">
<pre>
X13776
Pseudomonas aeruginosa amiC and amiR gene for aliphatic amidase
regulation
          ggtaccgCTGGCCGAGCATCTGCTcgatcaccaccagccgggcgacgggaactgcacgat
          ctacCTGGCGAGCCTGGAGCAcgagcgggttcgcttcgta

</pre>
</td></tr></table><p>

<a name="output.7"></a>
<h3>Output files for usage example 7</h3>
<p><h3>File: x13776.showseq </h3>
<table width="90%"><tr><td bgcolor="#CCFFCC">
<pre>
X13776
Pseudomonas aeruginosa amiC and amiR gene for aliphatic amidase
regulation

                   10        20        30        40        50        60        
          ----:----|----:----|----:----|----:----|----:----|----:----|
          ggtaccgctggccgagcatctgctcgatcaccaccagccgggcgacgggaactgcacgat

                                  R  S  P  P  A  G  R  R  V           
                                                                      
                                                                      
                 |===============|
                 promoter note="proposed rpoN-dependent promoter"
          |
          misc_feature note="last base of an XhoI site"

                   70        80        90        100       110       120       
          ----:----|----:----|----:----|----:----|----:----|----:----|
          ctacctggcgagcctggagcacgagcgggttcgcttcgtacggcgctgagcgacagtcac

                                                                      
                 A  S  L                                              
                                                                      
              |===============|
              promoter note="proposed rpoN-dependent promoter"

</pre>
</td></tr></table><p>

<a name="output.8"></a>
<h3>Output files for usage example 8</h3>
<p><h3>File: x13776.showseq </h3>
<table width="90%"><tr><td bgcolor="#CCFFCC">
<pre>
X13776
Pseudomonas aeruginosa amiC and amiR gene for aliphatic amidase
regulation

                   10        20        30        40        50        60        
          ----:----|----:----|----:----|----:----|----:----|----:----|
          ggtaccgctggccgagcatctgctcgatcaccaccagccgggcgacgggaactgcacgat
                      |------------|
                      binding site
                        |
                        SNP
                 |===============|
                 promoter note="proposed rpoN-dependent promoter"
          |
          misc_feature note="last base of an XhoI site"

                   70        80        90        100       110       120       
          ----:----|----:----|----:----|----:----|----:----|----:----|
          ctacctggcgagcctggagcacgagcgggttcgcttcgtacggcgctgagcgacagtcac
              |===============|
              promoter note="proposed rpoN-dependent promoter"

</pre>
</td></tr></table><p>

<p>

Most of the variants of the output format have already been described in
the 'Description' and 'Usage' sections, but here is some more just to
fill out this section ;-)

<p>

The output format is extremely variable and under the control of the 
qualifiers used.

<p>
          
The sequence can be formatted for HTML display by using the '-html'
qualifier.  The top and tail html tags &lt;HEAD&gt;, &lt;BODY&gt; etc.  are not
included as it is expected that the output of this program will be
included in a more extensive HTML page and so these parts are left to
the user to provide.

<p>

The name of the sequence is displayed, followed by the description of
the sequence.  These can be turned off with the '-noname' and
'-nodescription' qualifiers.

<p>
      
Then the sequence is output, one line at a time.  Any associated
information to be displayed is also output above and below the sequence
line, as specified by the '-format' and or '-things' qualifiers.  (See
the 'Description' section for detals).

<p>
          
The margins around the sequence are specified by the use of the
'-margin' qaulifier and any numbering of the sequence and its
translations are placed in the margin.

<p>

A display of the restriction enzyme cut sites can be selected via
'-format 6' option or the '-format 0 -thing b,r,s,-r' style of options.

<p>

The option '-format 7' will produce a formatted display of cut sites on
the sequence, with the six-frame translation below it.  The cut sites
are indicated by a slash character '\' that points to the poition
between the nucleotides where the cuts occur.  Cuts by many enzymes at
the same position are indicated by stacking the enzyme names on top of
each other.

<p>

At the end the section header 'Enzymes that cut' is displayed followed
by a list of the enzymes that cut the specified sequence and the number
of times that they cut.

<p>

The '-flatreformat' qualifier changes the display to emphasise the
recognition site of the restriction enzyme, which is indicated by a row
of '=' characters.  The cut site if pointed to by a '>' or '<' character
and if the cut site is not within or imemdiately adjacent to the
recognition site, they are linked by a row or '.' characters.

<p>

The name of the enzyme is displayed above (or below when the reverse
sense site if displayed) the recognition site.  The name of the enzyme
is also displayed above the cut site if this occurs on a different
display line to the recognition site (i.e.  if it wraps onto the next
line of sequence).




<H2>
    Data files
</H2>

<p>
EMBOSS data files are distributed with the application and stored
in the standard EMBOSS data directory, which is defined
by the EMBOSS environment variable EMBOSS_DATA.

<p>

To see the available EMBOSS data files, run:
<p>
<pre>
% embossdata -showall
</pre>
<p>
To fetch one of the data files (for example 'Exxx.dat') into your
current directory for you to inspect or modify, run:

<pre>

% embossdata -fetch -file Exxx.dat

</pre>
<p>

Users can provide their own data files in their own directories.
Project specific files can be put in the current directory, or for
tidier directory listings in a subdirectory called
".embossdata". Files for all EMBOSS runs can be put in the user's home
directory, or again in a subdirectory called ".embossdata".

<p>
The directories are searched in the following order:

<ul>
   <li> . (your current directory)
   <li> .embossdata (under your current directory)
   <li> ~/ (your home directory)
   <li> ~/.embossdata
</ul>
<p>


<p>
The Genetic Code data files are based on the NCBI genetic code tables.
Their names and descriptions are:

<dl>
<dt>EGC.0 </dt><dd>
      Standard (Differs from GC.1 in that it only has initiation site 'AUG')
<dt>EGC.1 </dt><dd>
      Standard
<dt>EGC.2 </dt><dd>
      Vertebrate Mitochodrial
<dt>EGC.3 </dt><dd>
      Yeast Mitochondrial
<dt>EGC.4 </dt><dd>
      Mold, Protozoan, Coelenterate Mitochondrial and Mycoplasma/Spiroplasma
<dt>EGC.5 </dt><dd>
      Invertebrate Mitochondrial
<dt>EGC.6 </dt><dd>
      Ciliate Macronuclear and Dasycladacean
<dt>EGC.9 </dt><dd>
      Echinoderm Mitochondrial
<dt>EGC.10 </dt><dd>
      Euplotid Nuclear
<dt>EGC.11 </dt><dd>
      Bacterial
<dt>EGC.12 </dt><dd>
      Alternative Yeast Nuclear
<dt>EGC.13 </dt><dd>
      Ascidian Mitochondrial
<dt>EGC.14 </dt><dd>
      Flatworm Mitochondrial
<dt>EGC.15</dt><dd>
      Blepharisma Macronuclear
<dt>EGC.16</dt><dd>
      Chlorophycean Mitochondrial
<dt>EGC.21</dt><dd>
      Trematode Mitochondrial
<dt>EGC.22</dt><dd>
      Scenedesmus obliquus
<dt>EGC.23</dt><dd>
      Thraustochytrium Mitochondrial
</dl>

<p>

The format of these files is very simple.

<p>   

It consists of several lines of optional comments, each starting with
a '#' character.

<p>

These are followed the line: 'Genetic Code [n]', where 'n' is the
number of the genetic code file.

<p> 

This is followed by the description of the code and then by four lines
giving the IUPAC one-letter code of the translated amino acid, the
start codons (indicdated by an 'M') and the three bases of the codon,
lined up one on top of the other.
   
<p>

For example:

<pre>
   
------------------------------------------------------------------------------
# Genetic Code Table
#
# Obtained from: http://www.ncbi.nlm.nih.gov/collab/FT/genetic_codes.html
# and: http://www3.ncbi.nlm.nih.gov/htbin-post/Taxonomy/wprintgc?mode=c
#
# Differs from Genetic Code [1] only in that the initiation sites have been
# changed to only 'AUG'

Genetic Code [0]
Standard
   
AAs  =   FFLLSSSSYY**CC*WLLLLPPPPHHQQRRRRIIIMTTTTNNKKSSRRVVVVAAAADDEEGGGG
Starts = -----------------------------------M----------------------------
Base1  = TTTTTTTTTTTTTTTTCCCCCCCCCCCCCCCCAAAAAAAAAAAAAAAAGGGGGGGGGGGGGGGG
Base2  = TTTTCCCCAAAAGGGGTTTTCCCCAAAAGGGGTTTTCCCCAAAAGGGGTTTTCCCCAAAAGGGG
Base3  = TCAGTCAGTCAGTCAGTCAGTCAGTCAGTCAGTCAGTCAGTCAGTCAGTCAGTCAGTCAGTCAG
------------------------------------------------------------------------------

</pre>

<p>      
The EMBOSS REBASE restriction enzyme data files are stored in
directory 'data/REBASE/*' under the EMBOSS installation directory.

<p>

These files must first be set up using the program <a
href="rebaseextract.html">'<b>rebaseextract</b>'</a>.  Running
'rebaseextract' may be the job of your system manager.

<p>
      
The data files are stored in the REBASE directory of the standard EMBOSS
data directory. The names are:

<ul>
<li> embossre.enz     Cleavage information
<li> embossre.ref     Reference/methylation information
<li> embossre.sup     Supplier information
</ul>
	 
The column information is described at the top of the data files
	 
<p>
	 
The reported enzyme from any one group of isoschizomers (the prototype)
is specified in the REBASE database and the information is held in the
data file 'embossre.equ'.  You may edit this file to set your own
preferred prototype, if you wish. 
	 
<p>
	 
The format of the file "embossre.equ" is
<br>
Enzyme-name Prototype-name
	 
<p>
	 
i.e.  two columns of enzyme names separated by a space.  The first name
of the pair of enzymes is the name that is not preferred and the second
is the preferred (prototype) name. 
	 
	          

<H2>
    Notes
</H2>

<p>One or more things may be selected for display from a menu
(<tt>-things</tt> option).  The order of specified characters (upper
or lower case) determines the order in the output:</p>
<pre>
s	Sequence
b	Blank line
1	Frame 1 translation
2	Frame 2 translation
3	Frame 3 translation
-1	Frame -1 translation
-2	Frame -2 translation
-2	Frame -3 translation
t	Ticks line
n	Number ticks line
c	Complement sequence
f	Features (from the feature table or from a command line -ufo file)
r	Restriction enzyme cut sites in the forward sense
-r	Restriction enzyme cut sites in the reverse sense
a	User Annotation
</pre>

<p>Alternatively, there is a selection of pre-defined formats to
choose from. The codes from above used in the list of standard formats
are:</p>
<pre>
Sequence only:                  S A
Default sequence:               B N T S A F
Pretty sequence:                B N T S A
One frame translation:          B N T S B 1 A F
Three frame translations:       B N T S B 1 2 3 A F
Six frame translations:         B N T S B 1 2 3 T -3 -2 -1 A F
Restriction enzyme map:         B R S N T C -R B 1 2 3 T -3 -2 -1 A
Baroque:                        B 1 2 3 N T R S T C -R T -3 -2 -1 A F</pre>



<p>The default standard format displays the following: for every new
line that the sequence starts to write, the output display will
contain first a blank line (<tt>b</tt>), then the position numbers of
the ticks (<tt>n</tt>) then the ticks every 10 characters (<tt>t</tt>)
then the sequence itself (<tt>s</tt>) then any user-supplied
annotation (<tt>a</tt>) then the features from the feature table
(<tt>f</tt>). Subsequent lines of the sequence output will repeat this
format.</p>

<p>The sequence can be translated, using the specified genetic code
tables. The translation can be done in one, three or six frames. The
translation can be displayed in one-letter or three-letter amino acid
codes. The translation can optionally be displayed only when it is in
open reading frames (ORFs) of a specified minimum size. One or more
specified regions of the sequence can be individually translated.</p>

<p>The output can be formatted for HTML. If the output is being
formatted for HTML, then specified regions of the sequence can be
displayed in any valid HTML colours.</p>

<p>This program can use REBASE data to find the recognition sites
and/or cut sites of restriction enzymes in a nucleic acid sequence.
This program can display the cut sites on both strands.
The <tt>-flatreformat</tt> option displays not only the cut sites
which many other restriction cut-site programs will show, but also
shows the recognition site.</p>

<p>The Restriction Enzyme database (REBASE) is a collection of
information about restriction enzymes and related proteins. It
contains published and unpublished references, recognition and
cleavage sites, isoschizomers, commercial availability, methylation
sensitivity, crystal and sequence data. DNA methyltransferases, homing
endonucleases, nicking enzymes, specificity subunits and control
proteins are also included. Most recently, putative DNA
methyltransferases and restriction enzymes, as predicted from analysis
of genomic sequences, are also listed. The home page of REBASE
is: <a href="http://rebase.neb.com/">http://rebase.neb.com/</a></p>

<p>If the sequence is in EMBL, Genbank or SwissProt format, the
feature table of the sequence can be displayed with the sequence. GFF
file features can also be displayed if they are included on the
command line using <tt>-ufo=file</tt>.</p>

<p>
Other display options include: The displayed sequence can be numbered
either by numbering the start and ending positions, or by placing a
ruler with ticks above or below the sequence. An initial position to
start the numbering from can be set.  The width of a line, and width
of a margin around the sequence reserved for numbering can be set.
Specified regions of the sequence can be displayed in uppercase to
highlight them.</p>


<H2>
    References
</H2>

None.

<H2>
    Warnings
</H2>

None.

<H2>
    Diagnostic Error Messages
</H2>

None.

<H2>
    Exit status
</H2>

It always exits with status 0.

<H2>
    Known bugs
</H2>

None known.

<h2><a name="See also">See also</a></h2>
<table border cellpadding=4 bgcolor="#FFFFF0">
<tr><th>Program name</th>
<th>Description</th></tr>
<tr>
<td><a href="abiview.html">abiview</a></td>
<td>Display the trace in an ABI sequencer file</td>
</tr>

<tr>
<td><a href="backtranambig.html">backtranambig</a></td>
<td>Back-translate a protein sequence to ambiguous nucleotide sequence</td>
</tr>

<tr>
<td><a href="backtranseq.html">backtranseq</a></td>
<td>Back-translate a protein sequence to a nucleotide sequence</td>
</tr>

<tr>
<td><a href="cirdna.html">cirdna</a></td>
<td>Draws circular maps of DNA constructs</td>
</tr>

<tr>
<td><a href="coderet.html">coderet</a></td>
<td>Extract CDS, mRNA and translations from feature tables</td>
</tr>

<tr>
<td><a href="lindna.html">lindna</a></td>
<td>Draws linear maps of DNA constructs</td>
</tr>

<tr>
<td><a href="pepnet.html">pepnet</a></td>
<td>Draw a helical net for a protein sequence</td>
</tr>

<tr>
<td><a href="pepwheel.html">pepwheel</a></td>
<td>Draw a helical wheel diagram for a protein sequence</td>
</tr>

<tr>
<td><a href="plotorf.html">plotorf</a></td>
<td>Plot potential open reading frames in a nucleotide sequence</td>
</tr>

<tr>
<td><a href="prettyplot.html">prettyplot</a></td>
<td>Draw a sequence alignment with pretty formatting</td>
</tr>

<tr>
<td><a href="prettyseq.html">prettyseq</a></td>
<td>Write a nucleotide sequence and its translation to file</td>
</tr>

<tr>
<td><a href="recoder.html">recoder</a></td>
<td>Find restriction sites to remove (mutate) with no translation change</td>
</tr>

<tr>
<td><a href="redata.html">redata</a></td>
<td>Retrieve information from REBASE restriction enzyme database</td>
</tr>

<tr>
<td><a href="remap.html">remap</a></td>
<td>Display restriction enzyme binding sites in a nucleotide sequence</td>
</tr>

<tr>
<td><a href="restover.html">restover</a></td>
<td>Find restriction enzymes producing a specific overhang</td>
</tr>

<tr>
<td><a href="restrict.html">restrict</a></td>
<td>Report restriction enzyme cleavage sites in a nucleotide sequence</td>
</tr>

<tr>
<td><a href="seealso.html">seealso</a></td>
<td>Finds programs with similar function to a specified program</td>
</tr>

<tr>
<td><a href="showalign.html">showalign</a></td>
<td>Display a multiple sequence alignment in pretty format</td>
</tr>

<tr>
<td><a href="showdb.html">showdb</a></td>
<td>Displays information on configured databases</td>
</tr>

<tr>
<td><a href="showfeat.html">showfeat</a></td>
<td>Display features of a sequence in pretty format</td>
</tr>

<tr>
<td><a href="showorf.html">showorf</a></td>
<td>Display a nucleotide sequence and translation in pretty format</td>
</tr>

<tr>
<td><a href="showpep.html">showpep</a></td>
<td>Displays protein sequences with features in pretty format</td>
</tr>

<tr>
<td><a href="silent.html">silent</a></td>
<td>Find restriction sites to insert (mutate) with no translation change</td>
</tr>

<tr>
<td><a href="sixpack.html">sixpack</a></td>
<td>Display a DNA sequence with 6-frame translation and ORFs</td>
</tr>

<tr>
<td><a href="textsearch.html">textsearch</a></td>
<td>Search the textual description of sequence(s)</td>
</tr>

<tr>
<td><a href="transeq.html">transeq</a></td>
<td>Translate nucleic acid sequences</td>
</tr>

</table>

<H2>
    Author(s)
</H2>
Gary Williams (gwilliam&nbsp;&copy;&nbsp;rfcgr.mrc.ac.uk)
<br>
MRC Rosalind Franklin Centre for Genomics Research
Wellcome Trust Genome Campus, Hinxton, Cambridge, CB10 1SB, UK



<H2>
    History
</H2>

Written 1999 - GWW
<br>
23 Aug 2000 - features display added - GWW
<br>
20 Nov 2001 - feature matches and annotation display added - GWW
<br>
16 Dec 2008 - limited to nucleotide only. Use showpep for proteins - PMR

<H2>
    Target users
</H2>
This program is intended to be used by everyone and everything, from naive users to embedded scripts.

<H2>
    Comments
</H2>
None

</BODY>
</HTML>

