<HTML>
<HEAD>
  <TITLE>
  EMBOSS: sirna
  </TITLE>
</HEAD>
<BODY BGCOLOR="#FFFFFF" text="#000000">

<table align=center border=0 cellspacing=0 cellpadding=0>
<tr><td valign=top>
<A HREF="/" ONMOUSEOVER="self.status='Go to the EMBOSS home page';return true"><img border=0 src="/images/emboss_icon.jpg" alt="" width=150 height=48></a>
</td>
<td align=left valign=middle>
<b><font size="+6">
sirna
</font></b>
</td></tr>
</table>
<br>&nbsp;
<p>


<H2>
Wiki
</H2>

The master copies of EMBOSS documentation are available
at <a href="http://emboss.open-bio.org/wiki/Appdocs">
http://emboss.open-bio.org/wiki/Appdocs</a>
on the EMBOSS Wiki.

<p>
Please help by correcting and extending the Wiki pages.

<H2>
    Function
</H2>
Find siRNA duplexes in mRNA

<H2>
    Description
</H2>
<p>Finds siRNA duplexes in mRNA.  The output is a standard EMBOSS report file. The siRNAs are reported in order of best score first. <b>sirna</b> reports both the sense and antisense siRNAs as 5' to 3'. </p>


<H2>
Algorithm
</H2>

<pre>
for each input sequence:

    find the start position of the CDS in the feature table
    if there is no such CDS, take the -sbegin position as the CDS start

    for each 23 base window along the sequence:

        set the score for this window = 0
        if base 2 of the window is not 'a': ignore this window
        if the window is within 50 bases of the CDS start: ignore this window
	if the window is within 100 bases of the CDS: score = -2
	measure the %GC of the 20 bases from position 2 to 21 of the window
	for the following %GC values change the score:
		%GC <= 25% (<= 5 bases): ignore this window
		%GC 30% (6 bases): score + 0
		%GC 35% (7 bases): score + 2
		%GC 40% (8 bases): score + 4
		%GC 45% (9 bases): score + 5
		%GC 50% (10 bases): score + 6
		%GC 55% (11 bases): score + 5
		%GC 60% (12 bases): score + 4
		%GC 65% (13 bases): score + 2
		%GC 70% (14 bases): score + 0
		%GC >= 75% (>= 15 bases): ignore this window
	if the window starts with a 'AA': score + 3
	if the window does not start 'AA' and it is required: ignore this window
	if the window ends with a 'TT': score + 1
	if the window does not end 'TT' and it is required: ignore this window
	if 4 G's in a row are found: ignore this window
	if any 4 bases in a row are present and not required: ignore this window
	if PolIII probes are required and the window is not NARN(17)YNN: ignore this window
        if the score is > 0: store this window for output
	
    sort the windows found by their score
    output the 23-base windows to the sequence file
    if the 'context' qualifier is specified, output window bases 1 and 2 in brackets to the report file
    take the window bases 3 to 21, add 'dTdT' output to the report file
    take the window bases 3 to 21, reverse complement, add 'dTdT' output to the report file
</pre>



<H2>
    Usage
</H2>
Here is a sample session with <b>sirna</b>
<p>

<p>
<table width="90%"><tr><td bgcolor="#CCFFFF"><pre>

% <b>sirna </b>
Find siRNA duplexes in mRNA
Input nucleotide sequence(s): <b>tembl:x65923</b>
Output report [x65923.sirna]: <b></b>
output sequence(s) [x65923.fasta]: <b></b>

</pre></td></tr></table><p>
<p>
<a href="#input.1">Go to the input files for this example</a><br><a href="#output.1">Go to the output files for this example</a><p><p>
<p>
<b>Example 2</b>
<p>
Show the first two bases of the 23 base target region in brackets. These do not form part of the sequence to be ordered, but it is useful to see if the 23 base region starts with an 'AA'. 
<p>

<p>
<table width="90%"><tr><td bgcolor="#CCFFFF"><pre>

% <b>sirna -context </b>
Find siRNA duplexes in mRNA
Input nucleotide sequence(s): <b>tembl:x65923</b>
Output report [x65923.sirna]: <b></b>
output sequence(s) [x65923.fasta]: <b></b>

</pre></td></tr></table><p>
<p>
<a href="#output.2">Go to the output files for this example</a><p><p>



<H2>
    Command line arguments
</H2>
<table CELLSPACING=0 CELLPADDING=3 BGCOLOR="#f5f5ff" ><tr><td>
<pre>
Find siRNA duplexes in mRNA
Version: EMBOSS:6.5.6.0

   Standard (Mandatory) qualifiers:
  [-sequence]          seqall     Nucleotide sequence(s) filename and optional
                                  format, or reference (input USA)
  [-outfile]           report     [*.sirna] The output is a table of the
                                  forward and reverse parts of the 21 base
                                  siRNA duplex. Both the forward and reverse
                                  sequences are written 5' to 3', ready to be
                                  ordered. The last two bases have been
                                  replaced by 'dTdT'. The starting position of
                                  the 23 base region and the %GC content is
                                  also given. If you wish to see the complete
                                  23 base sequence, then either look at the
                                  sequence in the other output file, or use
                                  the qualifier '-context' which will display
                                  the 23 bases of the forward sequence in this
                                  report with the first two bases in
                                  brackets. These first two bases do not form
                                  part of the siRNA probe to be ordered.
                                  (default -rformat table)
  [-outseq]            seqoutall  [<sequence>.<format>] This is a file of the
                                  sequences of the 23 base regions that the
                                  siRNAs are selected from. You may use it to
                                  do searches of mRNA databases (e.g. REFSEQ)
                                  to confirm that the probes are unique to the
                                  gene you wish to use it on.

   Additional (Optional) qualifiers:
   -poliii             boolean    [N] This option allows you to select only
                                  the 21 base probes that start with a purine
                                  and so can be expressed from Pol III
                                  expression vectors. This is the NARN(17)YNN
                                  pattern that has been suggested by Tuschl et
                                  al.
   -aa                 boolean    [N] This option allows you to select only
                                  those 23 base regions that start with AA. If
                                  this option is not selected then regions
                                  that start with AA will be favoured by
                                  giving them a higher score, but regions that
                                  do not start with AA will also be reported.
   -tt                 boolean    [N] This option allows you to select only
                                  those 23 base regions that end with TT. If
                                  this option is not selected then regions
                                  that end with TT will be favoured by giving
                                  them a higher score, but regions that do not
                                  end with TT will also be reported.
   -[no]polybase       boolean    [Y] If this option is FALSE then only those
                                  23 base regions that have no repeat of 4 or
                                  more of any bases in a row will be reported.
                                  No regions will ever be reported that have
                                  4 or more G's in a row.
   -context            boolean    [N] The output report file gives the
                                  sequences of the 21 base siRNA regions ready
                                  to be ordered. This does not give you an
                                  indication of the 2 bases before the 21
                                  bases. It is often interesting to see which
                                  of the suggested possible probe regions have
                                  an 'AA' in front of them (i.e. it is useful
                                  to see which of the 23 base regions start
                                  with an 'AA'). This option displays the
                                  whole 23 bases of the region with the first
                                  two bases in brackets, e.g. '(AA)' to give
                                  you some context for the probe region. YOU
                                  SHOULD NOT INCLUDE THE TWO BASES IN BRACKETS
                                  WHEN YOU PLACE AN ORDER FOR THE PROBES.

   Advanced (Unprompted) qualifiers: (none)
   Associated qualifiers:

   "-sequence" associated qualifiers
   -sbegin1            integer    Start of each sequence to be used
   -send1              integer    End of each sequence to be used
   -sreverse1          boolean    Reverse (if DNA)
   -sask1              boolean    Ask for begin/end/reverse
   -snucleotide1       boolean    Sequence is nucleotide
   -sprotein1          boolean    Sequence is protein
   -slower1            boolean    Make lower case
   -supper1            boolean    Make upper case
   -scircular1         boolean    Sequence is circular
   -sformat1           string     Input sequence format
   -iquery1            string     Input query fields or ID list
   -ioffset1           integer    Input start position offset
   -sdbname1           string     Database name
   -sid1               string     Entryname
   -ufo1               string     UFO features
   -fformat1           string     Features format
   -fopenfile1         string     Features file name

   "-outfile" associated qualifiers
   -rformat2           string     Report format
   -rname2             string     Base file name
   -rextension2        string     File name extension
   -rdirectory2        string     Output directory
   -raccshow2          boolean    Show accession number in the report
   -rdesshow2          boolean    Show description in the report
   -rscoreshow2        boolean    Show the score in the report
   -rstrandshow2       boolean    Show the nucleotide strand in the report
   -rusashow2          boolean    Show the full USA in the report
   -rmaxall2           integer    Maximum total hits to report
   -rmaxseq2           integer    Maximum hits to report for one sequence

   "-outseq" associated qualifiers
   -osformat3          string     Output seq format
   -osextension3       string     File name extension
   -osname3            string     Base file name
   -osdirectory3       string     Output directory
   -osdbname3          string     Database name to add
   -ossingle3          boolean    Separate file for each entry
   -oufo3              string     UFO features
   -offormat3          string     Features format
   -ofname3            string     Features file name
   -ofdirectory3       string     Output directory

   General qualifiers:
   -auto               boolean    Turn off prompts
   -stdout             boolean    Write first file to standard output
   -filter             boolean    Read first file from standard input, write
                                  first file to standard output
   -options            boolean    Prompt for standard and additional values
   -debug              boolean    Write debug output to program.dbg
   -verbose            boolean    Report some/full command line options
   -help               boolean    Report command line options and exit. More
                                  information on associated and general
                                  qualifiers can be found with -help -verbose
   -warning            boolean    Report warnings
   -error              boolean    Report errors
   -fatal              boolean    Report fatal errors
   -die                boolean    Report dying program messages
   -version            boolean    Report version number and exit

</pre>
</td></tr></table>
<P>
<table border cellspacing=0 cellpadding=3 bgcolor="#ccccff">
<tr bgcolor="#FFFFCC">
<th align="left">Qualifier</th>
<th align="left">Type</th>
<th align="left">Description</th>
<th align="left">Allowed values</th>
<th align="left">Default</th>
</tr>

<tr bgcolor="#FFFFCC">
<th align="left" colspan=5>Standard (Mandatory) qualifiers</th>
</tr>

<tr bgcolor="#FFFFCC">
<td>[-sequence]<br>(Parameter 1)</td>
<td>seqall</td>
<td>Nucleotide sequence(s) filename and optional format, or reference (input USA)</td>
<td>Readable sequence(s)</td>
<td><b>Required</b></td>
</tr>

<tr bgcolor="#FFFFCC">
<td>[-outfile]<br>(Parameter 2)</td>
<td>report</td>
<td>The output is a table of the forward and reverse parts of the 21 base siRNA duplex. Both the forward and reverse sequences are written 5' to 3', ready to be ordered. The last two bases have been replaced by 'dTdT'. The starting position of the 23 base region and the %GC content is also given. If you wish to see the complete 23 base sequence, then either look at the sequence in the other output file, or use the qualifier '-context' which will display the 23 bases of the forward sequence in this report with the first two bases in brackets. These first two bases do not form part of the siRNA probe to be ordered.</td>
<td>(default -rformat table)</td>
<td><i>&lt;*&gt;</i>.sirna</td>
</tr>

<tr bgcolor="#FFFFCC">
<td>[-outseq]<br>(Parameter 3)</td>
<td>seqoutall</td>
<td>This is a file of the sequences of the 23 base regions that the siRNAs are selected from. You may use it to do searches of mRNA databases (e.g. REFSEQ) to confirm that the probes are unique to the gene you wish to use it on.</td>
<td>Writeable sequence(s)</td>
<td><i>&lt;*&gt;</i>.<i>format</i></td>
</tr>

<tr bgcolor="#FFFFCC">
<th align="left" colspan=5>Additional (Optional) qualifiers</th>
</tr>

<tr bgcolor="#FFFFCC">
<td>-poliii</td>
<td>boolean</td>
<td>This option allows you to select only the 21 base probes that start with a purine and so can be expressed from Pol III expression vectors. This is the NARN(17)YNN pattern that has been suggested by Tuschl et al.</td>
<td>Boolean value Yes/No</td>
<td>No</td>
</tr>

<tr bgcolor="#FFFFCC">
<td>-aa</td>
<td>boolean</td>
<td>This option allows you to select only those 23 base regions that start with AA. If this option is not selected then regions that start with AA will be favoured by giving them a higher score, but regions that do not start with AA will also be reported.</td>
<td>Boolean value Yes/No</td>
<td>No</td>
</tr>

<tr bgcolor="#FFFFCC">
<td>-tt</td>
<td>boolean</td>
<td>This option allows you to select only those 23 base regions that end with TT. If this option is not selected then regions that end with TT will be favoured by giving them a higher score, but regions that do not end with TT will also be reported.</td>
<td>Boolean value Yes/No</td>
<td>No</td>
</tr>

<tr bgcolor="#FFFFCC">
<td>-[no]polybase</td>
<td>boolean</td>
<td>If this option is FALSE then only those 23 base regions that have no repeat of 4 or more of any bases in a row will be reported. No regions will ever be reported that have 4 or more G's in a row.</td>
<td>Boolean value Yes/No</td>
<td>Yes</td>
</tr>

<tr bgcolor="#FFFFCC">
<td>-context</td>
<td>boolean</td>
<td>The output report file gives the sequences of the 21 base siRNA regions ready to be ordered. This does not give you an indication of the 2 bases before the 21 bases. It is often interesting to see which of the suggested possible probe regions have an 'AA' in front of them (i.e. it is useful to see which of the 23 base regions start with an 'AA'). This option displays the whole 23 bases of the region with the first two bases in brackets, e.g. '(AA)' to give you some context for the probe region. YOU SHOULD NOT INCLUDE THE TWO BASES IN BRACKETS WHEN YOU PLACE AN ORDER FOR THE PROBES.</td>
<td>Boolean value Yes/No</td>
<td>No</td>
</tr>

<tr bgcolor="#FFFFCC">
<th align="left" colspan=5>Advanced (Unprompted) qualifiers</th>
</tr>

<tr>
<td colspan=5>(none)</td>
</tr>

<tr bgcolor="#FFFFCC">
<th align="left" colspan=5>Associated qualifiers</th>
</tr>

<tr bgcolor="#FFFFCC">
<td align="left" colspan=5>"-sequence" associated seqall qualifiers
</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -sbegin1<br>-sbegin_sequence</td>
<td>integer</td>
<td>Start of each sequence to be used</td>
<td>Any integer value</td>
<td>0</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -send1<br>-send_sequence</td>
<td>integer</td>
<td>End of each sequence to be used</td>
<td>Any integer value</td>
<td>0</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -sreverse1<br>-sreverse_sequence</td>
<td>boolean</td>
<td>Reverse (if DNA)</td>
<td>Boolean value Yes/No</td>
<td>N</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -sask1<br>-sask_sequence</td>
<td>boolean</td>
<td>Ask for begin/end/reverse</td>
<td>Boolean value Yes/No</td>
<td>N</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -snucleotide1<br>-snucleotide_sequence</td>
<td>boolean</td>
<td>Sequence is nucleotide</td>
<td>Boolean value Yes/No</td>
<td>N</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -sprotein1<br>-sprotein_sequence</td>
<td>boolean</td>
<td>Sequence is protein</td>
<td>Boolean value Yes/No</td>
<td>N</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -slower1<br>-slower_sequence</td>
<td>boolean</td>
<td>Make lower case</td>
<td>Boolean value Yes/No</td>
<td>N</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -supper1<br>-supper_sequence</td>
<td>boolean</td>
<td>Make upper case</td>
<td>Boolean value Yes/No</td>
<td>N</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -scircular1<br>-scircular_sequence</td>
<td>boolean</td>
<td>Sequence is circular</td>
<td>Boolean value Yes/No</td>
<td>N</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -sformat1<br>-sformat_sequence</td>
<td>string</td>
<td>Input sequence format</td>
<td>Any string</td>
<td>&nbsp;</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -iquery1<br>-iquery_sequence</td>
<td>string</td>
<td>Input query fields or ID list</td>
<td>Any string</td>
<td>&nbsp;</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -ioffset1<br>-ioffset_sequence</td>
<td>integer</td>
<td>Input start position offset</td>
<td>Any integer value</td>
<td>0</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -sdbname1<br>-sdbname_sequence</td>
<td>string</td>
<td>Database name</td>
<td>Any string</td>
<td>&nbsp;</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -sid1<br>-sid_sequence</td>
<td>string</td>
<td>Entryname</td>
<td>Any string</td>
<td>&nbsp;</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -ufo1<br>-ufo_sequence</td>
<td>string</td>
<td>UFO features</td>
<td>Any string</td>
<td>&nbsp;</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -fformat1<br>-fformat_sequence</td>
<td>string</td>
<td>Features format</td>
<td>Any string</td>
<td>&nbsp;</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -fopenfile1<br>-fopenfile_sequence</td>
<td>string</td>
<td>Features file name</td>
<td>Any string</td>
<td>&nbsp;</td>
</tr>

<tr bgcolor="#FFFFCC">
<td align="left" colspan=5>"-outfile" associated report qualifiers
</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -rformat2<br>-rformat_outfile</td>
<td>string</td>
<td>Report format</td>
<td>Any string</td>
<td>table</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -rname2<br>-rname_outfile</td>
<td>string</td>
<td>Base file name</td>
<td>Any string</td>
<td>&nbsp;</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -rextension2<br>-rextension_outfile</td>
<td>string</td>
<td>File name extension</td>
<td>Any string</td>
<td>&nbsp;</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -rdirectory2<br>-rdirectory_outfile</td>
<td>string</td>
<td>Output directory</td>
<td>Any string</td>
<td>&nbsp;</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -raccshow2<br>-raccshow_outfile</td>
<td>boolean</td>
<td>Show accession number in the report</td>
<td>Boolean value Yes/No</td>
<td>N</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -rdesshow2<br>-rdesshow_outfile</td>
<td>boolean</td>
<td>Show description in the report</td>
<td>Boolean value Yes/No</td>
<td>N</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -rscoreshow2<br>-rscoreshow_outfile</td>
<td>boolean</td>
<td>Show the score in the report</td>
<td>Boolean value Yes/No</td>
<td>Y</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -rstrandshow2<br>-rstrandshow_outfile</td>
<td>boolean</td>
<td>Show the nucleotide strand in the report</td>
<td>Boolean value Yes/No</td>
<td>Y</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -rusashow2<br>-rusashow_outfile</td>
<td>boolean</td>
<td>Show the full USA in the report</td>
<td>Boolean value Yes/No</td>
<td>N</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -rmaxall2<br>-rmaxall_outfile</td>
<td>integer</td>
<td>Maximum total hits to report</td>
<td>Any integer value</td>
<td>0</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -rmaxseq2<br>-rmaxseq_outfile</td>
<td>integer</td>
<td>Maximum hits to report for one sequence</td>
<td>Any integer value</td>
<td>0</td>
</tr>

<tr bgcolor="#FFFFCC">
<td align="left" colspan=5>"-outseq" associated seqoutall qualifiers
</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -osformat3<br>-osformat_outseq</td>
<td>string</td>
<td>Output seq format</td>
<td>Any string</td>
<td>&nbsp;</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -osextension3<br>-osextension_outseq</td>
<td>string</td>
<td>File name extension</td>
<td>Any string</td>
<td>&nbsp;</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -osname3<br>-osname_outseq</td>
<td>string</td>
<td>Base file name</td>
<td>Any string</td>
<td>&nbsp;</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -osdirectory3<br>-osdirectory_outseq</td>
<td>string</td>
<td>Output directory</td>
<td>Any string</td>
<td>&nbsp;</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -osdbname3<br>-osdbname_outseq</td>
<td>string</td>
<td>Database name to add</td>
<td>Any string</td>
<td>&nbsp;</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -ossingle3<br>-ossingle_outseq</td>
<td>boolean</td>
<td>Separate file for each entry</td>
<td>Boolean value Yes/No</td>
<td>N</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -oufo3<br>-oufo_outseq</td>
<td>string</td>
<td>UFO features</td>
<td>Any string</td>
<td>&nbsp;</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -offormat3<br>-offormat_outseq</td>
<td>string</td>
<td>Features format</td>
<td>Any string</td>
<td>&nbsp;</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -ofname3<br>-ofname_outseq</td>
<td>string</td>
<td>Features file name</td>
<td>Any string</td>
<td>&nbsp;</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -ofdirectory3<br>-ofdirectory_outseq</td>
<td>string</td>
<td>Output directory</td>
<td>Any string</td>
<td>&nbsp;</td>
</tr>

<tr bgcolor="#FFFFCC">
<th align="left" colspan=5>General qualifiers</th>
</tr>

<tr bgcolor="#FFFFCC">
<td> -auto</td>
<td>boolean</td>
<td>Turn off prompts</td>
<td>Boolean value Yes/No</td>
<td>N</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -stdout</td>
<td>boolean</td>
<td>Write first file to standard output</td>
<td>Boolean value Yes/No</td>
<td>N</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -filter</td>
<td>boolean</td>
<td>Read first file from standard input, write first file to standard output</td>
<td>Boolean value Yes/No</td>
<td>N</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -options</td>
<td>boolean</td>
<td>Prompt for standard and additional values</td>
<td>Boolean value Yes/No</td>
<td>N</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -debug</td>
<td>boolean</td>
<td>Write debug output to program.dbg</td>
<td>Boolean value Yes/No</td>
<td>N</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -verbose</td>
<td>boolean</td>
<td>Report some/full command line options</td>
<td>Boolean value Yes/No</td>
<td>Y</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -help</td>
<td>boolean</td>
<td>Report command line options and exit. More information on associated and general qualifiers can be found with -help -verbose</td>
<td>Boolean value Yes/No</td>
<td>N</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -warning</td>
<td>boolean</td>
<td>Report warnings</td>
<td>Boolean value Yes/No</td>
<td>Y</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -error</td>
<td>boolean</td>
<td>Report errors</td>
<td>Boolean value Yes/No</td>
<td>Y</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -fatal</td>
<td>boolean</td>
<td>Report fatal errors</td>
<td>Boolean value Yes/No</td>
<td>Y</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -die</td>
<td>boolean</td>
<td>Report dying program messages</td>
<td>Boolean value Yes/No</td>
<td>Y</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -version</td>
<td>boolean</td>
<td>Report version number and exit</td>
<td>Boolean value Yes/No</td>
<td>N</td>
</tr>

</table>

<H2>
    Input file format
</H2>

<p>

The input is a standard EMBOSS sequence query (also known as a 'USA').

<p>
Major sequence database sources defined as standard in EMBOSS
installations include srs:embl, srs:uniprot and ensembl

<p>

Data can also be read from sequence output in any supported format
written by an EMBOSS or third-party application.

<p>
The input format can be specified by using the
command-line qualifier <tt>-sformat xxx</tt>, where 'xxx' is replaced
by the name of the required format.  The available format names are:
gff (gff3), gff2, embl (em), genbank (gb, refseq), ddbj, refseqp, pir
(nbrf), swissprot (swiss, sw), dasgff and debug.

<p>

See:
<A href="http://emboss.sf.net/docs/themes/SequenceFormats.html">
http://emboss.sf.net/docs/themes/SequenceFormats.html</A>
for further information on sequence formats.


<p>


<a name="input.1"></a>
<h3>Input files for usage example </h3>

'tembl:x65923' is a sequence entry in the example nucleic acid database 'tembl'
<p>
<p><h3>Database entry: tembl:x65923</h3>
<table width="90%"><tr><td bgcolor="#FFCCFF">
<pre>
ID   X65923; SV 1; linear; mRNA; STD; HUM; 518 BP.
XX
AC   X65923;
XX
DT   13-MAY-1992 (Rel. 31, Created)
DT   18-APR-2005 (Rel. 83, Last updated, Version 11)
XX
DE   H.sapiens fau mRNA
XX
KW   fau gene.
XX
OS   Homo sapiens (human)
OC   Eukaryota; Metazoa; Chordata; Craniata; Vertebrata; Euteleostomi; Mammalia;
OC   Eutheria; Euarchontoglires; Primates; Haplorrhini; Catarrhini; Hominidae;
OC   Homo.
XX
RN   [1]
RP   1-518
RA   Michiels L.M.R.;
RT   ;
RL   Submitted (29-APR-1992) to the INSDC.
RL   L.M.R. Michiels, University of Antwerp, Dept of Biochemistry,
RL   Universiteisplein 1, 2610 Wilrijk, BELGIUM
XX
RN   [2]
RP   1-518
RX   PUBMED; 8395683.
RA   Michiels L., Van der Rauwelaert E., Van Hasselt F., Kas K., Merregaert J.;
RT   "fau cDNA encodes a ubiquitin-like-S30 fusion protein and is expressed as
RT   an antisense sequence in the Finkel-Biskis-Reilly murine sarcoma virus";
RL   Oncogene 8(9):2537-2546(1993).
XX
DR   Ensembl-Gn; ENSG00000149806; Homo_sapiens.
DR   Ensembl-Tr; ENST00000279259; Homo_sapiens.
DR   Ensembl-Tr; ENST00000434372; Homo_sapiens.
DR   Ensembl-Tr; ENST00000525297; Homo_sapiens.
DR   Ensembl-Tr; ENST00000526555; Homo_sapiens.
DR   Ensembl-Tr; ENST00000527548; Homo_sapiens.
DR   Ensembl-Tr; ENST00000529259; Homo_sapiens.
DR   Ensembl-Tr; ENST00000529639; Homo_sapiens.
DR   Ensembl-Tr; ENST00000531743; Homo_sapiens.
XX
FH   Key             Location/Qualifiers
FH
FT   source          1..518
FT                   /organism="Homo sapiens"
FT                   /chromosome="11q"
FT                   /map="13"
FT                   /mol_type="mRNA"
FT                   /clone_lib="cDNA"
FT                   /clone="pUIA 631"
FT                   /tissue_type="placenta"
FT                   /db_xref="taxon:9606"
FT   misc_feature    57..278
FT                   /note="ubiquitin like part"
FT   CDS             57..458
FT                   /gene="fau"
FT                   /db_xref="GDB:135476"
FT                   /db_xref="GOA:P35544"
FT                   /db_xref="GOA:P62861"
FT                   /db_xref="H-InvDB:HIT000322806.14"
FT                   /db_xref="HGNC:3597"
FT                   /db_xref="InterPro:IPR000626"
FT                   /db_xref="InterPro:IPR006846"
FT                   /db_xref="InterPro:IPR019954"
FT                   /db_xref="InterPro:IPR019955"
FT                   /db_xref="InterPro:IPR019956"
FT                   /db_xref="PDB:2L7R"
FT                   /db_xref="UniProtKB/Swiss-Prot:P35544"
FT                   /db_xref="UniProtKB/Swiss-Prot:P62861"
FT                   /protein_id="CAA46716.1"
FT                   /translation="MQLFVRAQELHTFEVTGQETVAQIKAHVASLEGIAPEDQVVLLAG
FT                   APLEDEATLGQCGVEALTTLEVAGRMLGGKVHGSLARAGKVRGQTPKVAKQEKKKKKTG
FT                   RAKRRMQYNRRFVNVVPTFGKKKGPNANS"
FT   misc_feature    98..102
FT                   /note="nucleolar localization signal"
FT   misc_feature    279..458
FT                   /note="S30 part"
FT   polyA_signal    484..489
FT   polyA_site      509
XX
SQ   Sequence 518 BP; 125 A; 139 C; 148 G; 106 T; 0 other;
     ttcctctttc tcgactccat cttcgcggta gctgggaccg ccgttcagtc gccaatatgc        60
     agctctttgt ccgcgcccag gagctacaca ccttcgaggt gaccggccag gaaacggtcg       120
     cccagatcaa ggctcatgta gcctcactgg agggcattgc cccggaagat caagtcgtgc       180
     tcctggcagg cgcgcccctg gaggatgagg ccactctggg ccagtgcggg gtggaggccc       240
     tgactaccct ggaagtagca ggccgcatgc ttggaggtaa agttcatggt tccctggccc       300
     gtgctggaaa agtgagaggt cagactccta aggtggccaa acaggagaag aagaagaaga       360
     agacaggtcg ggctaagcgg cggatgcagt acaaccggcg ctttgtcaac gttgtgccca       420
     cctttggcaa gaagaagggc cccaatgcca actcttaagt cttttgtaat tctggctttc       480
     tctaataaaa aagccactta gttcagtcaa aaaaaaaa                               518
//
</pre>
</td></tr></table><p>


<H2>
    Output file format
</H2>

<p>

The output is a standard EMBOSS report file. 

<p>

The results can be output in one of several styles by using the
command-line qualifier <tt>-rformat xxx</tt>, where 'xxx' is replaced
by the name of the required format.  The available format names are:
embl, genbank, gff, pir, swiss, dasgff, debug, listfile, dbmotif,
diffseq, draw, restrict, excel, feattable, motif, nametable, regions,
seqtable, simple, srs, table, tagseq.

<p>

See:
<A href="http://emboss.sf.net/docs/themes/ReportFormats.html">
http://emboss.sf.net/docs/themes/ReportFormats.html</A>
for further information on report formats.

<p>
<p>

<b>sirna</b> 
outputs a report format file. The default format is 'table'.

<p>


<a name="output.1"></a>
<h3>Output files for usage example </h3>
<p><h3>File: x65923.sirna</h3>
<table width="90%"><tr><td bgcolor="#CCFFCC">
<pre>
########################################
# Program: sirna
# Rundate: Sun 15 Jul 2012 12:00:00
# Commandline: sirna
#    -sequence tembl:x65923
# Report_format: table
# Report_file: x65923.sirna
########################################

#=======================================
#
# Sequence: X65923     from: 1   to: 518
# HitCount: 85
#
# CDS region found in feature table starting at 57
#
#=======================================

  Start     End  Strand   Score    GC%             Sense_siRNA         Antisense_siRNA
    308     330       +   9.000   50.0 AAGUGAGAGGUCAGACUCCdTdT GGAGUCUGACCUCUCACUUdTdT
    309     331       +   9.000   50.0 AGUGAGAGGUCAGACUCCUdTdT AGGAGUCUGACCUCUCACUdTdT
    310     332       +   9.000   50.0 GUGAGAGGUCAGACUCCUAdTdT UAGGAGUCUGACCUCUCACdTdT
    351     373       +   9.000   50.0 GAAGAAGAAGACAGGUCGGdTdT CCGACCUGUCUUCUUCUUCdTdT
    166     188       +   8.000   55.0 GAUCAAGUCGUGCUCCUGGdTdT CCAGGAGCACGACUUGAUCdTdT
    279     301       +   8.000   55.0 AGUUCAUGGUUCCCUGGCCdTdT GGCCAGGGAACCAUGAACUdTdT
    330     352       +   8.000   55.0 GGUGGCCAAACAGGAGAAGdTdT CUUCUCCUGUUUGGCCACCdTdT
    354     376       +   8.000   55.0 GAAGAAGACAGGUCGGGCUdTdT AGCCCGACCUGUCUUCUUCdTdT
    357     379       +   8.000   55.0 GAAGACAGGUCGGGCUAAGdTdT CUUAGCCCGACCUGUCUUCdTdT
    393     415       +   8.000   55.0 CCGGCGCUUUGUCAACGUUdTdT AACGUUGACAAAGCGCCGGdTdT
    253     275       +   7.000   60.0 GUAGCAGGCCGCAUGCUUGdTdT CAAGCAUGCGGCCUGCUACdTdT
    280     302       +   7.000   60.0 GUUCAUGGUUCCCUGGCCCdTdT GGGCCAGGGAACCAUGAACdTdT
    339     361       +   7.000   40.0 ACAGGAGAAGAAGAAGAAGdTdT CUUCUUCUUCUUCUCCUGUdTdT
    340     362       +   7.000   40.0 CAGGAGAAGAAGAAGAAGAdTdT UCUUCUUCUUCUUCUCCUGdTdT
    348     370       +   7.000   40.0 GAAGAAGAAGAAGACAGGUdTdT ACCUGUCUUCUUCUUCUUCdTdT
    375     397       +   7.000   60.0 GCGGCGGAUGCAGUACAACdTdT GUUGUACUGCAUCCGCCGCdTdT
    408     430       +   7.000   60.0 CGUUGUGCCCACCUUUGGCdTdT GCCAAAGGUGGGCACAACGdTdT
    429     451       +   7.000   60.0 GAAGAAGGGCCCCAAUGCCdTdT GGCAUUGGGGCCCUUCUUCdTdT
    432     454       +   7.000   60.0 GAAGGGCCCCAAUGCCAACdTdT GUUGGCAUUGGGGCCCUUCdTdT
    435     457       +   7.000   60.0 GGGCCCCAAUGCCAACUCUdTdT AGAGUUGGCAUUGGGGCCCdTdT
    488     510       +   7.000   40.0 AAAGCCACUUAGUUCAGUCdTdT GACUGAACUAAGUGGCUUUdTdT
    489     511       +   7.000   40.0 AAGCCACUUAGUUCAGUCAdTdT UGACUGAACUAAGUGGCUUdTdT
    490     512       +   7.000   40.0 AGCCACUUAGUUCAGUCAAdTdT UUGACUGAACUAAGUGGCUdTdT
    491     513       +   7.000   40.0 GCCACUUAGUUCAGUCAAAdTdT UUUGACUGAACUAAGUGGCdTdT
    129     151       +   6.000   55.0 GGCUCAUGUAGCCUCACUGdTdT CAGUGAGGCUACAUGAGCCdTdT
    165     187       +   6.000   50.0 AGAUCAAGUCGUGCUCCUGdTdT CAGGAGCACGACUUGAUCUdTdT
    278     300       +   6.000   50.0 AAGUUCAUGGUUCCCUGGCdTdT GCCAGGGAACCAUGAACUUdTdT
    314     336       +   6.000   50.0 GAGGUCAGACUCCUAAGGUdTdT ACCUUAGGAGUCUGACCUCdTdT
    321     343       +   6.000   50.0 GACUCCUAAGGUGGCCAAAdTdT UUUGGCCACCUUAGGAGUCdTdT
    323     345       +   6.000   50.0 CUCCUAAGGUGGCCAAACAdTdT UGUUUGGCCACCUUAGGAGdTdT
    329     351       +   6.000   50.0 AGGUGGCCAAACAGGAGAAdTdT UUCUCCUGUUUGGCCACCUdTdT


<font color=red>  [Part of this file has been deleted for brevity]</font>

    374     396       +   5.000   55.0 AGCGGCGGAUGCAGUACAAdTdT UUGUACUGCAUCCGCCGCUdTdT
    383     405       +   5.000   55.0 UGCAGUACAACCGGCGCUUdTdT AAGCGCCGGUUGUACUGCAdTdT
    387     409       +   5.000   55.0 GUACAACCGGCGCUUUGUCdTdT GACAAAGCGCCGGUUGUACdTdT
    390     412       +   5.000   55.0 CAACCGGCGCUUUGUCAACdTdT GUUGACAAAGCGCCGGUUGdTdT
    392     414       +   5.000   55.0 ACCGGCGCUUUGUCAACGUdTdT ACGUUGACAAAGCGCCGGUdTdT
    407     429       +   5.000   55.0 ACGUUGUGCCCACCUUUGGdTdT CCAAAGGUGGGCACAACGUdTdT
    428     450       +   5.000   55.0 AGAAGAAGGGCCCCAAUGCdTdT GCAUUGGGGCCCUUCUUCUdTdT
    431     453       +   5.000   55.0 AGAAGGGCCCCAAUGCCAAdTdT UUGGCAUUGGGGCCCUUCUdTdT
    434     456       +   5.000   60.0 AGGGCCCCAAUGCCAACUCdTdT GAGUUGGCAUUGGGGCCCUdTdT
    444     466       +   5.000   35.0 UGCCAACUCUUAAGUCUUUdTdT AAAGACUUAAGAGUUGGCAdTdT
    487     509       +   5.000   35.0 AAAAGCCACUUAGUUCAGUdTdT ACUGAACUAAGUGGCUUUUdTdT
    123     145       +   4.000   50.0 GAUCAAGGCUCAUGUAGCCdTdT GGCUACAUGAGCCUUGAUCdTdT
    125     147       +   4.000   50.0 UCAAGGCUCAUGUAGCCUCdTdT GAGGCUACAUGAGCCUUGAdTdT
    128     150       +   4.000   50.0 AGGCUCAUGUAGCCUCACUdTdT AGUGAGGCUACAUGAGCCUdTdT
    155     177       +   4.000   50.0 UUGCCCCGGAAGAUCAAGUdTdT ACUUGAUCUUCCGGGGCAAdTdT
    234     256       +   4.000   60.0 GGCCCUGACUACCCUGGAAdTdT UUCCAGGGUAGUCAGGGCCdTdT
    259     281       +   4.000   60.0 GGCCGCAUGCUUGGAGGUAdTdT UACCUCCAAGCAUGCGGCCdTdT
    266     288       +   4.000   40.0 UGCUUGGAGGUAAAGUUCAdTdT UGAACUUUACCUCCAAGCAdTdT
    342     364       +   4.000   40.0 GGAGAAGAAGAAGAAGAAGdTdT CUUCUUCUUCUUCUUCUCCdTdT
    347     369       +   4.000   40.0 AGAAGAAGAAGAAGACAGGdTdT CCUGUCUUCUUCUUCUUCUdTdT
    359     381       +   4.000   60.0 AGACAGGUCGGGCUAAGCGdTdT CGCUUAGCCCGACCUGUCUdTdT
    111     133       +   3.000   55.0 AACGGUCGCCCAGAUCAAGdTdT CUUGAUCUGGGCGACCGUUdTdT
    113     135       +   3.000   65.0 CGGUCGCCCAGAUCAAGGCdTdT GCCUUGAUCUGGGCGACCGdTdT
    172     194       +   3.000   70.0 GUCGUGCUCCUGGCAGGCGdTdT CGCCUGCCAGGAGCACGACdTdT
    443     465       +   3.000   35.0 AUGCCAACUCUUAAGUCUUdTdT AAGACUUAAGAGUUGGCAUdTdT
    456     478       +   3.000   35.0 AGUCUUUUGUAAUUCUGGCdTdT GCCAGAAUUACAAAAGACUdTdT
    468     490       +   3.000   30.0 UUCUGGCUUUCUCUAAUAAdTdT UUAUUAGAGAAAGCCAGAAdTdT
    484     506       +   3.000   30.0 UAAAAAAGCCACUUAGUUCdTdT GAACUAAGUGGCUUUUUUAdTdT
    108     130       +   2.000   60.0 GGAAACGGUCGCCCAGAUCdTdT GAUCUGGGCGACCGUUUCCdTdT
    135     157       +   2.000   60.0 UGUAGCCUCACUGGAGGGCdTdT GCCCUCCAGUGAGGCUACAdTdT
    139     161       +   2.000   60.0 GCCUCACUGGAGGGCAUUGdTdT CAAUGCCCUCCAGUGAGGCdTdT
    150     172       +   2.000   60.0 GGGCAUUGCCCCGGAAGAUdTdT AUCUUCCGGGGCAAUGCCCdTdT
    171     193       +   2.000   65.0 AGUCGUGCUCCUGGCAGGCdTdT GCCUGCCAGGAGCACGACUdTdT
    201     223       +   2.000   65.0 GGAUGAGGCCACUCUGGGCdTdT GCCCAGAGUGGCCUCAUCCdTdT
    204     226       +   2.000   65.0 UGAGGCCACUCUGGGCCAGdTdT CUGGCCCAGAGUGGCCUCAdTdT
    245     267       +   2.000   65.0 CCCUGGAAGUAGCAGGCCGdTdT CGGCCUGCUACUUCCAGGGdTdT
    256     278       +   2.000   65.0 GCAGGCCGCAUGCUUGGAGdTdT CUCCAAGCAUGCGGCCUGCdTdT
    285     307       +   2.000   65.0 UGGUUCCCUGGCCCGUGCUdTdT AGCACGGGCCAGGGAACCAdTdT
    338     360       +   2.000   35.0 AACAGGAGAAGAAGAAGAAdTdT UUCUUCUUCUUCUCCUGUUdTdT
    345     367       +   2.000   35.0 GAAGAAGAAGAAGAAGACAdTdT UGUCUUCUUCUUCUUCUUCdTdT
    486     508       +   2.000   35.0 AAAAAGCCACUUAGUUCAGdTdT CUGAACUAAGUGGCUUUUUdTdT

#---------------------------------------
#---------------------------------------

#---------------------------------------
# Total_sequences: 1
# Total_length: 518
# Reported_sequences: 1
# Reported_hitcount: 85
#---------------------------------------
</pre>
</td></tr></table><p>
<p><h3>File: x65923.fasta</h3>
<table width="90%"><tr><td bgcolor="#CCFFCC">
<pre>
&gt;X65923_308 %GC 50.0 Score 9 H.sapiens fau mRNA
aaaagtgagaggtcagactccta
&gt;X65923_309 %GC 50.0 Score 9 H.sapiens fau mRNA
aaagtgagaggtcagactcctaa
&gt;X65923_310 %GC 50.0 Score 9 H.sapiens fau mRNA
aagtgagaggtcagactcctaag
&gt;X65923_351 %GC 50.0 Score 9 H.sapiens fau mRNA
aagaagaagaagacaggtcgggc
&gt;X65923_166 %GC 55.0 Score 8 H.sapiens fau mRNA
aagatcaagtcgtgctcctggca
&gt;X65923_279 %GC 55.0 Score 8 H.sapiens fau mRNA
aaagttcatggttccctggcccg
&gt;X65923_330 %GC 55.0 Score 8 H.sapiens fau mRNA
aaggtggccaaacaggagaagaa
&gt;X65923_354 %GC 55.0 Score 8 H.sapiens fau mRNA
aagaagaagacaggtcgggctaa
&gt;X65923_357 %GC 55.0 Score 8 H.sapiens fau mRNA
aagaagacaggtcgggctaagcg
&gt;X65923_393 %GC 55.0 Score 8 H.sapiens fau mRNA
aaccggcgctttgtcaacgttgt
&gt;X65923_253 %GC 60.0 Score 7 H.sapiens fau mRNA
aagtagcaggccgcatgcttgga
&gt;X65923_280 %GC 60.0 Score 7 H.sapiens fau mRNA
aagttcatggttccctggcccgt
&gt;X65923_339 %GC 40.0 Score 7 H.sapiens fau mRNA
aaacaggagaagaagaagaagaa
&gt;X65923_340 %GC 40.0 Score 7 H.sapiens fau mRNA
aacaggagaagaagaagaagaag
&gt;X65923_348 %GC 40.0 Score 7 H.sapiens fau mRNA
aagaagaagaagaagacaggtcg
&gt;X65923_375 %GC 60.0 Score 7 H.sapiens fau mRNA
aagcggcggatgcagtacaaccg
&gt;X65923_408 %GC 60.0 Score 7 H.sapiens fau mRNA
aacgttgtgcccacctttggcaa
&gt;X65923_429 %GC 60.0 Score 7 H.sapiens fau mRNA
aagaagaagggccccaatgccaa
&gt;X65923_432 %GC 60.0 Score 7 H.sapiens fau mRNA
aagaagggccccaatgccaactc
&gt;X65923_435 %GC 60.0 Score 7 H.sapiens fau mRNA
aagggccccaatgccaactctta
&gt;X65923_488 %GC 40.0 Score 7 H.sapiens fau mRNA
aaaaagccacttagttcagtcaa
&gt;X65923_489 %GC 40.0 Score 7 H.sapiens fau mRNA
aaaagccacttagttcagtcaaa
&gt;X65923_490 %GC 40.0 Score 7 H.sapiens fau mRNA
aaagccacttagttcagtcaaaa
&gt;X65923_491 %GC 40.0 Score 7 H.sapiens fau mRNA
aagccacttagttcagtcaaaaa
&gt;X65923_129 %GC 55.0 Score 6 H.sapiens fau mRNA
aaggctcatgtagcctcactgga


<font color=red>  [Part of this file has been deleted for brevity]</font>

gaggccctgactaccctggaagt
&gt;X65923_259 %GC 60.0 Score 4 H.sapiens fau mRNA
caggccgcatgcttggaggtaaa
&gt;X65923_266 %GC 40.0 Score 4 H.sapiens fau mRNA
catgcttggaggtaaagttcatg
&gt;X65923_342 %GC 40.0 Score 4 H.sapiens fau mRNA
caggagaagaagaagaagaagac
&gt;X65923_347 %GC 40.0 Score 4 H.sapiens fau mRNA
gaagaagaagaagaagacaggtc
&gt;X65923_359 %GC 60.0 Score 4 H.sapiens fau mRNA
gaagacaggtcgggctaagcggc
&gt;X65923_111 %GC 55.0 Score 3 H.sapiens fau mRNA
gaaacggtcgcccagatcaaggc
&gt;X65923_113 %GC 65.0 Score 3 H.sapiens fau mRNA
aacggtcgcccagatcaaggctc
&gt;X65923_172 %GC 70.0 Score 3 H.sapiens fau mRNA
aagtcgtgctcctggcaggcgcg
&gt;X65923_443 %GC 35.0 Score 3 H.sapiens fau mRNA
caatgccaactcttaagtctttt
&gt;X65923_456 %GC 35.0 Score 3 H.sapiens fau mRNA
taagtcttttgtaattctggctt
&gt;X65923_468 %GC 30.0 Score 3 H.sapiens fau mRNA
aattctggctttctctaataaaa
&gt;X65923_484 %GC 30.0 Score 3 H.sapiens fau mRNA
aataaaaaagccacttagttcag
&gt;X65923_108 %GC 60.0 Score 2 H.sapiens fau mRNA
caggaaacggtcgcccagatcaa
&gt;X65923_135 %GC 60.0 Score 2 H.sapiens fau mRNA
catgtagcctcactggagggcat
&gt;X65923_139 %GC 60.0 Score 2 H.sapiens fau mRNA
tagcctcactggagggcattgcc
&gt;X65923_150 %GC 60.0 Score 2 H.sapiens fau mRNA
gagggcattgccccggaagatca
&gt;X65923_171 %GC 65.0 Score 2 H.sapiens fau mRNA
caagtcgtgctcctggcaggcgc
&gt;X65923_201 %GC 65.0 Score 2 H.sapiens fau mRNA
gaggatgaggccactctgggcca
&gt;X65923_204 %GC 65.0 Score 2 H.sapiens fau mRNA
gatgaggccactctgggccagtg
&gt;X65923_245 %GC 65.0 Score 2 H.sapiens fau mRNA
taccctggaagtagcaggccgca
&gt;X65923_256 %GC 65.0 Score 2 H.sapiens fau mRNA
tagcaggccgcatgcttggaggt
&gt;X65923_285 %GC 65.0 Score 2 H.sapiens fau mRNA
catggttccctggcccgtgctgg
&gt;X65923_338 %GC 35.0 Score 2 H.sapiens fau mRNA
caaacaggagaagaagaagaaga
&gt;X65923_345 %GC 35.0 Score 2 H.sapiens fau mRNA
gagaagaagaagaagaagacagg
&gt;X65923_486 %GC 35.0 Score 2 H.sapiens fau mRNA
taaaaaagccacttagttcagtc
</pre>
</td></tr></table><p>

<a name="output.2"></a>
<h3>Output files for usage example 2</h3>
<p><h3>File: x65923.sirna</h3>
<table width="90%"><tr><td bgcolor="#CCFFCC">
<pre>
########################################
# Program: sirna
# Rundate: Sun 15 Jul 2012 12:00:00
# Commandline: sirna
#    -context
#    -sequence tembl:x65923
# Report_format: table
# Report_file: x65923.sirna
########################################

#=======================================
#
# Sequence: X65923     from: 1   to: 518
# HitCount: 85
#
# The forward sense sequence shows the first 2 bases of
# the 23 base region in brackets, this should be ignored
# when ordering siRNA probes.
# CDS region found in feature table starting at 57
#
#=======================================

  Start     End  Strand   Score    GC%                 Sense_siRNA         Antisense_siRNA
    308     330       +   9.000   50.0 (AA)AAGUGAGAGGUCAGACUCCdTdT GGAGUCUGACCUCUCACUUdTdT
    309     331       +   9.000   50.0 (AA)AGUGAGAGGUCAGACUCCUdTdT AGGAGUCUGACCUCUCACUdTdT
    310     332       +   9.000   50.0 (AA)GUGAGAGGUCAGACUCCUAdTdT UAGGAGUCUGACCUCUCACdTdT
    351     373       +   9.000   50.0 (AA)GAAGAAGAAGACAGGUCGGdTdT CCGACCUGUCUUCUUCUUCdTdT
    166     188       +   8.000   55.0 (AA)GAUCAAGUCGUGCUCCUGGdTdT CCAGGAGCACGACUUGAUCdTdT
    279     301       +   8.000   55.0 (AA)AGUUCAUGGUUCCCUGGCCdTdT GGCCAGGGAACCAUGAACUdTdT
    330     352       +   8.000   55.0 (AA)GGUGGCCAAACAGGAGAAGdTdT CUUCUCCUGUUUGGCCACCdTdT
    354     376       +   8.000   55.0 (AA)GAAGAAGACAGGUCGGGCUdTdT AGCCCGACCUGUCUUCUUCdTdT
    357     379       +   8.000   55.0 (AA)GAAGACAGGUCGGGCUAAGdTdT CUUAGCCCGACCUGUCUUCdTdT
    393     415       +   8.000   55.0 (AA)CCGGCGCUUUGUCAACGUUdTdT AACGUUGACAAAGCGCCGGdTdT
    253     275       +   7.000   60.0 (AA)GUAGCAGGCCGCAUGCUUGdTdT CAAGCAUGCGGCCUGCUACdTdT
    280     302       +   7.000   60.0 (AA)GUUCAUGGUUCCCUGGCCCdTdT GGGCCAGGGAACCAUGAACdTdT
    339     361       +   7.000   40.0 (AA)ACAGGAGAAGAAGAAGAAGdTdT CUUCUUCUUCUUCUCCUGUdTdT
    340     362       +   7.000   40.0 (AA)CAGGAGAAGAAGAAGAAGAdTdT UCUUCUUCUUCUUCUCCUGdTdT
    348     370       +   7.000   40.0 (AA)GAAGAAGAAGAAGACAGGUdTdT ACCUGUCUUCUUCUUCUUCdTdT
    375     397       +   7.000   60.0 (AA)GCGGCGGAUGCAGUACAACdTdT GUUGUACUGCAUCCGCCGCdTdT
    408     430       +   7.000   60.0 (AA)CGUUGUGCCCACCUUUGGCdTdT GCCAAAGGUGGGCACAACGdTdT
    429     451       +   7.000   60.0 (AA)GAAGAAGGGCCCCAAUGCCdTdT GGCAUUGGGGCCCUUCUUCdTdT
    432     454       +   7.000   60.0 (AA)GAAGGGCCCCAAUGCCAACdTdT GUUGGCAUUGGGGCCCUUCdTdT
    435     457       +   7.000   60.0 (AA)GGGCCCCAAUGCCAACUCUdTdT AGAGUUGGCAUUGGGGCCCdTdT
    488     510       +   7.000   40.0 (AA)AAAGCCACUUAGUUCAGUCdTdT GACUGAACUAAGUGGCUUUdTdT
    489     511       +   7.000   40.0 (AA)AAGCCACUUAGUUCAGUCAdTdT UGACUGAACUAAGUGGCUUdTdT
    490     512       +   7.000   40.0 (AA)AGCCACUUAGUUCAGUCAAdTdT UUGACUGAACUAAGUGGCUdTdT
    491     513       +   7.000   40.0 (AA)GCCACUUAGUUCAGUCAAAdTdT UUUGACUGAACUAAGUGGCdTdT
    129     151       +   6.000   55.0 (AA)GGCUCAUGUAGCCUCACUGdTdT CAGUGAGGCUACAUGAGCCdTdT
    165     187       +   6.000   50.0 (GA)AGAUCAAGUCGUGCUCCUGdTdT CAGGAGCACGACUUGAUCUdTdT
    278     300       +   6.000   50.0 (UA)AAGUUCAUGGUUCCCUGGCdTdT GCCAGGGAACCAUGAACUUdTdT


<font color=red>  [Part of this file has been deleted for brevity]</font>

    374     396       +   5.000   55.0 (UA)AGCGGCGGAUGCAGUACAAdTdT UUGUACUGCAUCCGCCGCUdTdT
    383     405       +   5.000   55.0 (GA)UGCAGUACAACCGGCGCUUdTdT AAGCGCCGGUUGUACUGCAdTdT
    387     409       +   5.000   55.0 (CA)GUACAACCGGCGCUUUGUCdTdT GACAAAGCGCCGGUUGUACdTdT
    390     412       +   5.000   55.0 (UA)CAACCGGCGCUUUGUCAACdTdT GUUGACAAAGCGCCGGUUGdTdT
    392     414       +   5.000   55.0 (CA)ACCGGCGCUUUGUCAACGUdTdT ACGUUGACAAAGCGCCGGUdTdT
    407     429       +   5.000   55.0 (CA)ACGUUGUGCCCACCUUUGGdTdT CCAAAGGUGGGCACAACGUdTdT
    428     450       +   5.000   55.0 (CA)AGAAGAAGGGCCCCAAUGCdTdT GCAUUGGGGCCCUUCUUCUdTdT
    431     453       +   5.000   55.0 (GA)AGAAGGGCCCCAAUGCCAAdTdT UUGGCAUUGGGGCCCUUCUdTdT
    434     456       +   5.000   60.0 (GA)AGGGCCCCAAUGCCAACUCdTdT GAGUUGGCAUUGGGGCCCUdTdT
    444     466       +   5.000   35.0 (AA)UGCCAACUCUUAAGUCUUUdTdT AAAGACUUAAGAGUUGGCAdTdT
    487     509       +   5.000   35.0 (AA)AAAAGCCACUUAGUUCAGUdTdT ACUGAACUAAGUGGCUUUUdTdT
    123     145       +   4.000   50.0 (CA)GAUCAAGGCUCAUGUAGCCdTdT GGCUACAUGAGCCUUGAUCdTdT
    125     147       +   4.000   50.0 (GA)UCAAGGCUCAUGUAGCCUCdTdT GAGGCUACAUGAGCCUUGAdTdT
    128     150       +   4.000   50.0 (CA)AGGCUCAUGUAGCCUCACUdTdT AGUGAGGCUACAUGAGCCUdTdT
    155     177       +   4.000   50.0 (CA)UUGCCCCGGAAGAUCAAGUdTdT ACUUGAUCUUCCGGGGCAAdTdT
    234     256       +   4.000   60.0 (GA)GGCCCUGACUACCCUGGAAdTdT UUCCAGGGUAGUCAGGGCCdTdT
    259     281       +   4.000   60.0 (CA)GGCCGCAUGCUUGGAGGUAdTdT UACCUCCAAGCAUGCGGCCdTdT
    266     288       +   4.000   40.0 (CA)UGCUUGGAGGUAAAGUUCAdTdT UGAACUUUACCUCCAAGCAdTdT
    342     364       +   4.000   40.0 (CA)GGAGAAGAAGAAGAAGAAGdTdT CUUCUUCUUCUUCUUCUCCdTdT
    347     369       +   4.000   40.0 (GA)AGAAGAAGAAGAAGACAGGdTdT CCUGUCUUCUUCUUCUUCUdTdT
    359     381       +   4.000   60.0 (GA)AGACAGGUCGGGCUAAGCGdTdT CGCUUAGCCCGACCUGUCUdTdT
    111     133       +   3.000   55.0 (GA)AACGGUCGCCCAGAUCAAGdTdT CUUGAUCUGGGCGACCGUUdTdT
    113     135       +   3.000   65.0 (AA)CGGUCGCCCAGAUCAAGGCdTdT GCCUUGAUCUGGGCGACCGdTdT
    172     194       +   3.000   70.0 (AA)GUCGUGCUCCUGGCAGGCGdTdT CGCCUGCCAGGAGCACGACdTdT
    443     465       +   3.000   35.0 (CA)AUGCCAACUCUUAAGUCUUdTdT AAGACUUAAGAGUUGGCAUdTdT
    456     478       +   3.000   35.0 (UA)AGUCUUUUGUAAUUCUGGCdTdT GCCAGAAUUACAAAAGACUdTdT
    468     490       +   3.000   30.0 (AA)UUCUGGCUUUCUCUAAUAAdTdT UUAUUAGAGAAAGCCAGAAdTdT
    484     506       +   3.000   30.0 (AA)UAAAAAAGCCACUUAGUUCdTdT GAACUAAGUGGCUUUUUUAdTdT
    108     130       +   2.000   60.0 (CA)GGAAACGGUCGCCCAGAUCdTdT GAUCUGGGCGACCGUUUCCdTdT
    135     157       +   2.000   60.0 (CA)UGUAGCCUCACUGGAGGGCdTdT GCCCUCCAGUGAGGCUACAdTdT
    139     161       +   2.000   60.0 (UA)GCCUCACUGGAGGGCAUUGdTdT CAAUGCCCUCCAGUGAGGCdTdT
    150     172       +   2.000   60.0 (GA)GGGCAUUGCCCCGGAAGAUdTdT AUCUUCCGGGGCAAUGCCCdTdT
    171     193       +   2.000   65.0 (CA)AGUCGUGCUCCUGGCAGGCdTdT GCCUGCCAGGAGCACGACUdTdT
    201     223       +   2.000   65.0 (GA)GGAUGAGGCCACUCUGGGCdTdT GCCCAGAGUGGCCUCAUCCdTdT
    204     226       +   2.000   65.0 (GA)UGAGGCCACUCUGGGCCAGdTdT CUGGCCCAGAGUGGCCUCAdTdT
    245     267       +   2.000   65.0 (UA)CCCUGGAAGUAGCAGGCCGdTdT CGGCCUGCUACUUCCAGGGdTdT
    256     278       +   2.000   65.0 (UA)GCAGGCCGCAUGCUUGGAGdTdT CUCCAAGCAUGCGGCCUGCdTdT
    285     307       +   2.000   65.0 (CA)UGGUUCCCUGGCCCGUGCUdTdT AGCACGGGCCAGGGAACCAdTdT
    338     360       +   2.000   35.0 (CA)AACAGGAGAAGAAGAAGAAdTdT UUCUUCUUCUUCUCCUGUUdTdT
    345     367       +   2.000   35.0 (GA)GAAGAAGAAGAAGAAGACAdTdT UGUCUUCUUCUUCUUCUUCdTdT
    486     508       +   2.000   35.0 (UA)AAAAAGCCACUUAGUUCAGdTdT CUGAACUAAGUGGCUUUUUdTdT

#---------------------------------------
#---------------------------------------

#---------------------------------------
# Total_sequences: 1
# Total_length: 518
# Reported_sequences: 1
# Reported_hitcount: 85
#---------------------------------------
</pre>
</td></tr></table><p>

<p>

The siRNAs are reported in order of best score first.

<p>

<b>sirna</b> reports both the sense and antisense siRNAs as 5' to 3'. 

<H2>
    Data files
</H2>

None.

<H2>
    Notes
</H2>

<p>RNA interference (RNAi) is a phenomenon whereby small interfering RNA strands (siRNA) inhibit gene expression at the level of transcription or translation of specific genes.  RNAi is a defence mechanism against viruses and is important in regulating development and genome maintenance. siRNA are double stranded RNA molecules where one or the other strand is strongly complementary to a target RNA strand. Once they bind to a target, a nuclease protein guided by the siRNA cleaves the target and renders it untranslateable.</p>


<p>Gene silencing using RNAi has been used to determine the function of many genes in Drosophilia, C. elegans, and many plant species. The duration of knockdown by siRNA can typically last for 7-10 days, and has been shown to transfer to daughter cells. Of further note, siRNAs are effective at quantities much lower than alternative gene silencing methodologies, including antisense and ribozyme based strategies.</p>

<p>Due to various mechanisms of antiviral response to long dsRNA, RNAi at first proved more difficult to establish in mammalian species. Then, Tuschl, Elbashir, and others discovered that RNAi can be elicited very effectively by well-defined 21-base duplex RNAs. When these small interfering RNA, or siRNA, are added in duplex form with a transfection agent to mammalian cell cultures, the 21-base-pair RNA acts in concert with cellular components to silence the gene with sequence homology to one of the siRNA sequences. Strategies for the design of effective siRNA sequences have been recently documented, most notably by Sayda Elbashir, Thomas Tuschl, et al.</p>

<p>Their studies of mammalian RNAi suggest that the most efficient gene-silencing effect is achieved using double-stranded siRNA having a 19-nucleotide complementary region and a 2-nucleotide 3' overhang at each end. Current models of the RNAi mechanism suggest that the antisense siRNA strand recognizes the specific gene target.</p>

<p>In gene-specific RNAi, the coding region (CDS) of the mRNA is usually targeted. The search for an appropriate target sequence should begin 50-100 nucleotides downstream of the start codon. UTR-binding proteins and/or translation initiation complexes may interfere with the binding of the siRNP endonuclease complex. Tuschl, Elbashir et al. say that they have successfully used siRNAs targetting the 3' UTR. To avoid interference from mRNA regulatory proteins, sequences in the 5' untranslated region or near the start codon should not be targeted.</p>

<p>A set of rules for the design of siRNA has been suggested <a href="http://www.mpibpc.gwdg.de/abteilungen/100/105/sirna.html">http://www.mpibpc.gwdg.de/abteilungen/100/105/sirna.html</a> based on the work of Tuschl, Elbashir et al. They suggest searching for 23-nt sequence motif AA(N19)TT (N, any nucleotide) and select hits with approx. 50% G/C-content (30% to 70% has also worked in for them). If no suitable sequences are found, the search is extended using the motif NA(N21). The sequence of the sense siRNA corresponds to (N19)TT or N21 (position 3 to 23 of the 23-nt motif), respectively. In the latter case, they convert the 3' end of the sense siRNA to TT.</p>

<p>The rationale for this sequence conversion is to generate a symmetric duplex with respect to the sequence composition of the sense and antisense 3' overhangs. The antisense siRNA is synthesized as the complement to position 1 to 21 of the 23-nt motif. Because position 1 of the 23-nt motif is not recognized sequence-specifically by the antisense siRNA, the 3'-most nucleotide residue of the antisense siRNA, can be chosen deliberately. However, the penultimate nucleotide of the antisense siRNA (complementary to position 2 of the 23-nt motif) should always be complementary to the targeted sequence. For simplifying chemical synthesis, they always use TT.</p>

<p>More recently, they preferentially select siRNAs corresponding to the target motif NAR(N17)YNN, where R is purine (A, G) and Y is pyrimidine (C, U). The respective 21-nt sense and antisense siRNAs therefore begin with a purine nucleotide and can also be expressed from pol III expression vectors without a change in targeting site; expression of RNAs from pol III promoters is only efficient when the first transcribed nucleotide is a purine.</p>

<p>They always design siRNAs with symmetric 3' TT overhangs, believing that symmetric 3' overhangs help to ensure that the siRNPs are formed with approximately equal ratios of sense and antisense target RNA-cleaving siRNPs Please note that the modification of the overhang of the sense sequence of the siRNA duplex is not expected to affect targeted mRNA recognition, as the antisense siRNA strand guides target recognition. In summary, no matter what you do to your overhangs, siRNAs should still function to a reasonable extent. However, using TT in the 3' overhang will always help your RNA synthesis company to let you know when you accidentally order a siRNA sequences 3' to 5' rather than in the recommended format of 5' to 3'. <b>sirna</b> reports both the sense and antisense siRNAs as 5' to 3'.</p>

<p>Xeragon.com also suggest that choosing a region of the mRNA with a GC content as close as possible to 50% is a more important consideration than choosing a target sequence that begins with AA. They also suggest that a key consideration in target selection is to avoid having more than three guanosines in a row, since poly G sequences can hyperstack and form agglomerates that potentially interfere with the siRNA silencing mechanism.</p>

<p>siRNAs appear to effectively silence genes in more than 80% of cases. Current data indicate that there are regions of some mRNAs where gene silencing does not work. To help ensure that a given target gene is silenced, it is advised that at least two target sequences as far apart on the gene as possible be chosen.</p>




<h3>Coding region specification</h3>
It's possible (although the evidence is unclear) that regulatory protein binding to regions in and near the untranslated 5' region might interfere with the RNAi process. Therefore, this program avoids choosing siRNA probes from the 5' UTR and from the first 50 bases of the coding region. The second 50 bases of the coding region has a penalty associated with it to reduce the reporting of possible siRNA probes in this region.</p>

If the input sequence has a feature table specifying a coding region, then this will be used, else you can specify the start of the coding region, where this is known by the <tt>-sbegin</tt> command-line qualifier (which is normally used to specify the start of the region of a sequence that should be analysed in all EMBOSS programs).

<b>sirna</b> looks at the feature table of the input mRNA sequence to find the coding regions (CDS). It will ignore the 5' UTR and the first 50 bases of the CDS. It will assign a penalty of 2 points to any siRNA in positions 51 to 100 in the CDS. If there is no CDS in the feature table, you can specify the CDS by using the command-line qualifier <tt>-sbegin</tt> to indicate where the CDS should start. If there is no CDS in the feature table and you do not use the command-line qualifier <tt>-sbegin</tt>, then <b>sirna</b> will assume that the CDS region is not known and will look for siRNAs in the whole of the sequence with no penaties associated with the location within the sequence.
All these confusing regions
There are a lot of references to 23 base regions, 21 base regions, 19 base regions, etc. in any description of siRNA.

Perhaps an example with a sequence would be clearer?

The 23 base region, in this case starting with an <tt>AA</tt>, might typically look like:

<pre>5' AAGUGAGAGGUCAGACUCCUATC</pre>

The sense siRNA is made from the 19 bases of positions 3 to 21 of the 23 base target region, so:

<pre>5'   GUGAGAGGUCAGACUCCUA</pre>

and then typically d(TT) is added, so:

<pre>5'   GUGAGAGGUCAGACUCCUAdTdT</pre>

The antisense siRNA sequence is made from bases 3 to 21 of the target region, so:

<pre>5'   GUGAGAGGUCAGACUCCUA sense
3'   CACUCUCCAGUCUGAGGAU antisense 3' -> 5'</pre>

so the antisense sequence that should be ordered with d(TT) added is:

<pre>5'   UAGGAGUCUGACCUCUCACdTdT antisense 5' -> 3'</pre>

<H2>
    References
</H2>

<ol>

<li>Elbashir, S. M., et al. (2001a). Duplexes of 21-nucleotide RNAs
    mediate RNA interference in mammalian cell culture. Nature 411:
    494-498.

<li>
    Elbashir, S. M., W. Lendeckel and T. Tuschl (2001b). RNA
    interference is mediated by 21 and 22 nt RNAs. Genes & Dev. 15:
    188-200.

</ol>



<H2>
    Warnings
</H2>

It is assumed that the input sequence is mRNA.

<H2>
    Diagnostic Error Messages
</H2>

None.

<H2>
    Exit status
</H2>

It always exits with status 0.

<H2>
    Known bugs
</H2>

None.

<h2><a name="See also">See also</a></h2>
<table border cellpadding=4 bgcolor="#FFFFF0">
<tr><th>Program name</th>
<th>Description</th></tr>
<tr>
<td><a href="banana.html">banana</a></td>
<td>Plot bending and curvature data for B-DNA</td>
</tr>

<tr>
<td><a href="btwisted.html">btwisted</a></td>
<td>Calculate the twisting in a B-DNA sequence</td>
</tr>

<tr>
<td><a href="einverted.html">einverted</a></td>
<td>Find inverted repeats in nucleotide sequences</td>
</tr>

<tr>
<td><a href="marscan.html">marscan</a></td>
<td>Find matrix/scaffold recognition (MRS) signatures in DNA sequences</td>
</tr>

<tr>
<td><a href="trimest.html">trimest</a></td>
<td>Remove poly-A tails from nucleotide sequences</td>
</tr>

</table>


<H2>
    Author(s)
</H2>

Gary Williams formerly at:
<br>
MRC Rosalind Franklin Centre for Genomics Research
Wellcome Trust Genome Campus, Hinxton, Cambridge, CB10 1SB, UK

<p>
Please report all bugs to the EMBOSS bug team (emboss-bug&nbsp;&copy;&nbsp;emboss.open-bio.org) not to the original author.

<H2>
    History
</H2>

Written (November 2002) - Gary Williams.

<H2>
    Target users
</H2>
This program is intended to be used by everyone and everything, from naive users to embedded scripts.

<H2>
    Comments
</H2>
None


</BODY>
</HTML>

