<HTML>
<HEAD>
  <TITLE>
  EMBOSS: sixpack
  </TITLE>
</HEAD>
<BODY BGCOLOR="#FFFFFF" text="#000000">



<table align=center border=0 cellspacing=0 cellpadding=0>
<tr><td valign=top>
<A HREF="/" ONMOUSEOVER="self.status='Go to the EMBOSS home page';return true"><img border=0 src="emboss_icon.jpg" alt="" width=150 height=48></a>
</td>
<td align=left valign=middle>
<b><font size="+6">
sixpack
</font></b>
</td></tr>
</table>
<br>&nbsp;
<p>


<H2>
Wiki
</H2>

The master copies of EMBOSS documentation are available
at <a href="http://emboss.open-bio.org/wiki/Appdocs">
http://emboss.open-bio.org/wiki/Appdocs</a>
on the EMBOSS Wiki.

<p>
Please help by correcting and extending the Wiki pages.

<H2>
    Function
</H2>
Display a DNA sequence with 6-frame translation and ORFs
<!--
DON'T WRITE ANYTHING HERE.
IT IS DONE FOR YOU.
-->




<H2>
    Description
</H2>

<p><b>sixpack</b> reads a DNA sequence and writes an output file
giving out the forward and reverse sense sequences with the three
forward and (optionally) three reverse translations in a pretty
display format.  A genetic code may be specified for the translation.
There are various options to control the appearance of the output
file.  It also writes a file of protein sequences corresponding to any
open reading frames that are larger than the specified minimum size:
the default of 1 base shows all possible open reading frames.</p>

<H2>
    Algorithm
</H2>

<p>The program takes the following steps:</p>
<pre>
 The nucleic acid sequence is read in.
 The required genetic code is read in from the EGC* data files.
 The three forward and three reverse translations are created.
 The name and description are written to the ouput display file.
 Any required regions to be changed to upper case are changed.
 Any required regions to be highlighted in HTML colour tags are changed.
 The reverse sense sequence is placed below the forward sequence.
 The forward translations are placed above the sequences.
 The reverse translation are placed below the sequences.
 The display is written out, split at the ends of lines.
 Any ORFs that are longer than the specified minimum size are written to the output sequence file.
</pre>

<H2>
    Usage
</H2>
<b>Here is a sample session with sixpack</b>
<p>

<p>
<table width="90%"><tr><td bgcolor="#CCFFFF"><pre>

% <b>sixpack </b>
Display a DNA sequence with 6-frame translation and ORFs
Input nucleotide sequence: <b>tembl:x13776</b>
Output file [x13776.sixpack]: <b></b>
protein output sequence(s) [x13776.fasta]: <b></b>

</pre></td></tr></table><p>
<p>
<a href="#input.1">Go to the input files for this example</a><br><a href="#output.1">Go to the output files for this example</a><p><p>



<H2>
    Command line arguments
</H2>

<table CELLSPACING=0 CELLPADDING=3 BGCOLOR="#f5f5ff" ><tr><td>
<pre>
Display a DNA sequence with 6-frame translation and ORFs
Version: EMBOSS:6.3.0

   Standard (Mandatory) qualifiers:
  [-sequence]          sequence   Nucleotide sequence filename and optional
                                  format, or reference (input USA)
  [-outfile]           outfile    [*.sixpack] Output file name
   -outseq             seqoutall  [<sequence>.<format>] ORF sequence output

   Additional (Optional) qualifiers:
   -table              menu       [0] Genetics code used for the translation
                                  (Values: 0 (Standard); 1 (Standard (with
                                  alternative initiation codons)); 2
                                  (Vertebrate Mitochondrial); 3 (Yeast
                                  Mitochondrial); 4 (Mold, Protozoan,
                                  Coelenterate Mitochondrial and
                                  Mycoplasma/Spiroplasma); 5 (Invertebrate
                                  Mitochondrial); 6 (Ciliate Macronuclear and
                                  Dasycladacean); 9 (Echinoderm
                                  Mitochondrial); 10 (Euplotid Nuclear); 11
                                  (Bacterial); 12 (Alternative Yeast Nuclear);
                                  13 (Ascidian Mitochondrial); 14 (Flatworm
                                  Mitochondrial); 15 (Blepharisma
                                  Macronuclear); 16 (Chlorophycean
                                  Mitochondrial); 21 (Trematode
                                  Mitochondrial); 22 (Scenedesmus obliquus);
                                  23 (Thraustochytrium Mitochondrial))
   -[no]firstorf       boolean    [Y] Count the beginning of a sequence as a
                                  possible ORF, even if it's inferior to the
                                  minimal ORF size.
   -[no]lastorf        boolean    [Y] Count the end of a sequence as a
                                  possible ORF, even if it's not finishing
                                  with a STOP, or inferior to the minimal ORF
                                  size.
   -mstart             boolean    [N] Displays only ORFs starting with an M.

   Advanced (Unprompted) qualifiers:
   -[no]reverse        boolean    [Y] Display also the translation of the DNA
                                  sequence in the 3 reverse frames
   -orfminsize         integer    [1] Minimum size of Open Reading Frames
                                  (ORFs) to display in the translations.
                                  (Integer 1 or more)
   -uppercase          range      [If this is left blank, then the sequence
                                  case is left alone.] Regions to put in
                                  uppercase.
                                  If this is left blank, then the sequence
                                  case is left alone.
                                  A set of regions is specified by a set of
                                  pairs of positions.
                                  The positions are integers.
                                  They are separated by any non-digit,
                                  non-alpha character.
                                  Examples of region specifications are:
                                  24-45, 56-78
                                  1:45, 67=99;765..888
                                  1,5,8,10,23,45,57,99
   -highlight          range      [(full sequence)] Regions to colour if
                                  formatting for HTML.
                                  If this is left blank, then the sequence is
                                  left alone.
                                  A set of regions is specified by a set of
                                  pairs of positions.
                                  The positions are integers.
                                  They are followed by any valid HTML font
                                  colour.
                                  Examples of region specifications are:
                                  24-45 blue 56-78 orange
                                  1-100 green 120-156 red
                                  A file of ranges to colour (one range per
                                  line) can be specified as '@filename'.
   -[no]number         boolean    [Y] Number the sequence at the beginning and
                                  the end of each line.
   -width              integer    [60] Number of nucleotides displayed on each
                                  line (Integer 1 or more)
   -length             integer    [0] Line length of page (0 for indefinite)
                                  (Integer 0 or more)
   -margin             integer    [10] Margin around sequence for numbering.
                                  (Integer 0 or more)
   -[no]name           boolean    [Y] Set this to be false if you do not wish
                                  to display the ID name of the sequence.
   -[no]description    boolean    [Y] Set this to be false if you do not wish
                                  to display the description of the sequence.
   -offset             integer    [1] Number from which you want the DNA
                                  sequence to be numbered. (Any integer value)
   -html               boolean    [N] Use HTML formatting

   Associated qualifiers:

   "-sequence" associated qualifiers
   -sbegin1            integer    Start of the sequence to be used
   -send1              integer    End of the sequence to be used
   -sreverse1          boolean    Reverse (if DNA)
   -sask1              boolean    Ask for begin/end/reverse
   -snucleotide1       boolean    Sequence is nucleotide
   -sprotein1          boolean    Sequence is protein
   -slower1            boolean    Make lower case
   -supper1            boolean    Make upper case
   -sformat1           string     Input sequence format
   -sdbname1           string     Database name
   -sid1               string     Entryname
   -ufo1               string     UFO features
   -fformat1           string     Features format
   -fopenfile1         string     Features file name

   "-outfile" associated qualifiers
   -odirectory2        string     Output directory

   "-outseq" associated qualifiers
   -osformat           string     Output seq format
   -osextension        string     File name extension
   -osname             string     Base file name
   -osdirectory        string     Output directory
   -osdbname           string     Database name to add
   -ossingle           boolean    Separate file for each entry
   -oufo               string     UFO features
   -offormat           string     Features format
   -ofname             string     Features file name
   -ofdirectory        string     Output directory

   General qualifiers:
   -auto               boolean    Turn off prompts
   -stdout             boolean    Write first file to standard output
   -filter             boolean    Read first file from standard input, write
                                  first file to standard output
   -options            boolean    Prompt for standard and additional values
   -debug              boolean    Write debug output to program.dbg
   -verbose            boolean    Report some/full command line options
   -help               boolean    Report command line options and exit. More
                                  information on associated and general
                                  qualifiers can be found with -help -verbose
   -warning            boolean    Report warnings
   -error              boolean    Report errors
   -fatal              boolean    Report fatal errors
   -die                boolean    Report dying program messages
   -version            boolean    Report version number and exit

</pre>
</td></tr></table>
<P>
<table border cellspacing=0 cellpadding=3 bgcolor="#ccccff">
<tr bgcolor="#FFFFCC">
<th align="left">Qualifier</th>
<th align="left">Type</th>
<th align="left">Description</th>
<th align="left">Allowed values</th>
<th align="left">Default</th>
</tr>

<tr bgcolor="#FFFFCC">
<th align="left" colspan=5>Standard (Mandatory) qualifiers</th>
</tr>

<tr bgcolor="#FFFFCC">
<td>[-sequence]<br>(Parameter 1)</td>
<td>sequence</td>
<td>Nucleotide sequence filename and optional format, or reference (input USA)</td>
<td>Readable sequence</td>
<td><b>Required</b></td>
</tr>

<tr bgcolor="#FFFFCC">
<td>[-outfile]<br>(Parameter 2)</td>
<td>outfile</td>
<td>Output file name</td>
<td>Output file</td>
<td><i>&lt;*&gt;</i>.sixpack</td>
</tr>

<tr bgcolor="#FFFFCC">
<td>-outseq</td>
<td>seqoutall</td>
<td>ORF sequence output</td>
<td>Writeable sequence(s)</td>
<td><i>&lt;*&gt;</i>.<i>format</i></td>
</tr>

<tr bgcolor="#FFFFCC">
<th align="left" colspan=5>Additional (Optional) qualifiers</th>
</tr>

<tr bgcolor="#FFFFCC">
<td>-table</td>
<td>list</td>
<td>Genetics code used for the translation</td>
<td><table><tr><td>0</td> <td><i>(Standard)</i></td></tr><tr><td>1</td> <td><i>(Standard (with alternative initiation codons))</i></td></tr><tr><td>2</td> <td><i>(Vertebrate Mitochondrial)</i></td></tr><tr><td>3</td> <td><i>(Yeast Mitochondrial)</i></td></tr><tr><td>4</td> <td><i>(Mold, Protozoan, Coelenterate Mitochondrial and Mycoplasma/Spiroplasma)</i></td></tr><tr><td>5</td> <td><i>(Invertebrate Mitochondrial)</i></td></tr><tr><td>6</td> <td><i>(Ciliate Macronuclear and Dasycladacean)</i></td></tr><tr><td>9</td> <td><i>(Echinoderm Mitochondrial)</i></td></tr><tr><td>10</td> <td><i>(Euplotid Nuclear)</i></td></tr><tr><td>11</td> <td><i>(Bacterial)</i></td></tr><tr><td>12</td> <td><i>(Alternative Yeast Nuclear)</i></td></tr><tr><td>13</td> <td><i>(Ascidian Mitochondrial)</i></td></tr><tr><td>14</td> <td><i>(Flatworm Mitochondrial)</i></td></tr><tr><td>15</td> <td><i>(Blepharisma Macronuclear)</i></td></tr><tr><td>16</td> <td><i>(Chlorophycean Mitochondrial)</i></td></tr><tr><td>21</td> <td><i>(Trematode Mitochondrial)</i></td></tr><tr><td>22</td> <td><i>(Scenedesmus obliquus)</i></td></tr><tr><td>23</td> <td><i>(Thraustochytrium Mitochondrial)</i></td></tr></table></td>
<td>0</td>
</tr>

<tr bgcolor="#FFFFCC">
<td>-[no]firstorf</td>
<td>boolean</td>
<td>Count the beginning of a sequence as a possible ORF, even if it's inferior to the minimal ORF size.</td>
<td>Boolean value Yes/No</td>
<td>Yes</td>
</tr>

<tr bgcolor="#FFFFCC">
<td>-[no]lastorf</td>
<td>boolean</td>
<td>Count the end of a sequence as a possible ORF, even if it's not finishing with a STOP, or inferior to the minimal ORF size.</td>
<td>Boolean value Yes/No</td>
<td>Yes</td>
</tr>

<tr bgcolor="#FFFFCC">
<td>-mstart</td>
<td>boolean</td>
<td>Displays only ORFs starting with an M.</td>
<td>Boolean value Yes/No</td>
<td>No</td>
</tr>

<tr bgcolor="#FFFFCC">
<th align="left" colspan=5>Advanced (Unprompted) qualifiers</th>
</tr>

<tr bgcolor="#FFFFCC">
<td>-[no]reverse</td>
<td>boolean</td>
<td>Display also the translation of the DNA sequence in the 3 reverse frames</td>
<td>Boolean value Yes/No</td>
<td>Yes</td>
</tr>

<tr bgcolor="#FFFFCC">
<td>-orfminsize</td>
<td>integer</td>
<td>Minimum size of Open Reading Frames (ORFs) to display in the translations.</td>
<td>Integer 1 or more</td>
<td>1</td>
</tr>

<tr bgcolor="#FFFFCC">
<td>-uppercase</td>
<td>range</td>
<td>Regions to put in uppercase.
If this is left blank, then the sequence case is left alone.
A set of regions is specified by a set of pairs of positions.
The positions are integers.
They are separated by any non-digit, non-alpha character.
Examples of region specifications are:
24-45, 56-78
1:45, 67=99;765..888
1,5,8,10,23,45,57,99</td>
<td>Sequence range</td>
<td>If this is left blank, then the sequence case is left alone.</td>
</tr>

<tr bgcolor="#FFFFCC">
<td>-highlight</td>
<td>range</td>
<td>Regions to colour if formatting for HTML.
If this is left blank, then the sequence is left alone.
A set of regions is specified by a set of pairs of positions.
The positions are integers.
They are followed by any valid HTML font colour.
Examples of region specifications are:
24-45 blue 56-78 orange
1-100 green 120-156 red
A file of ranges to colour (one range per line) can be specified as '@filename'.</td>
<td>Sequence range</td>
<td><i>full sequence</i></td>
</tr>

<tr bgcolor="#FFFFCC">
<td>-[no]number</td>
<td>boolean</td>
<td>Number the sequence at the beginning and the end of each line.</td>
<td>Boolean value Yes/No</td>
<td>Yes</td>
</tr>

<tr bgcolor="#FFFFCC">
<td>-width</td>
<td>integer</td>
<td>Number of nucleotides displayed on each line</td>
<td>Integer 1 or more</td>
<td>60</td>
</tr>

<tr bgcolor="#FFFFCC">
<td>-length</td>
<td>integer</td>
<td>Line length of page (0 for indefinite)</td>
<td>Integer 0 or more</td>
<td>0</td>
</tr>

<tr bgcolor="#FFFFCC">
<td>-margin</td>
<td>integer</td>
<td>Margin around sequence for numbering.</td>
<td>Integer 0 or more</td>
<td>10</td>
</tr>

<tr bgcolor="#FFFFCC">
<td>-[no]name</td>
<td>boolean</td>
<td>Set this to be false if you do not wish to display the ID name of the sequence.</td>
<td>Boolean value Yes/No</td>
<td>Yes</td>
</tr>

<tr bgcolor="#FFFFCC">
<td>-[no]description</td>
<td>boolean</td>
<td>Set this to be false if you do not wish to display the description of the sequence.</td>
<td>Boolean value Yes/No</td>
<td>Yes</td>
</tr>

<tr bgcolor="#FFFFCC">
<td>-offset</td>
<td>integer</td>
<td>Number from which you want the DNA sequence to be numbered.</td>
<td>Any integer value</td>
<td>1</td>
</tr>

<tr bgcolor="#FFFFCC">
<td>-html</td>
<td>boolean</td>
<td>Use HTML formatting</td>
<td>Boolean value Yes/No</td>
<td>No</td>
</tr>

<tr bgcolor="#FFFFCC">
<th align="left" colspan=5>Associated qualifiers</th>
</tr>

<tr bgcolor="#FFFFCC">
<th align="left" colspan=5>"-sequence" associated sequence qualifiers
</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -sbegin1<br>-sbegin_sequence</td>
<td>integer</td>
<td>Start of the sequence to be used</td>
<td>Any integer value</td>
<td>0</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -send1<br>-send_sequence</td>
<td>integer</td>
<td>End of the sequence to be used</td>
<td>Any integer value</td>
<td>0</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -sreverse1<br>-sreverse_sequence</td>
<td>boolean</td>
<td>Reverse (if DNA)</td>
<td>Boolean value Yes/No</td>
<td>N</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -sask1<br>-sask_sequence</td>
<td>boolean</td>
<td>Ask for begin/end/reverse</td>
<td>Boolean value Yes/No</td>
<td>N</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -snucleotide1<br>-snucleotide_sequence</td>
<td>boolean</td>
<td>Sequence is nucleotide</td>
<td>Boolean value Yes/No</td>
<td>N</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -sprotein1<br>-sprotein_sequence</td>
<td>boolean</td>
<td>Sequence is protein</td>
<td>Boolean value Yes/No</td>
<td>N</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -slower1<br>-slower_sequence</td>
<td>boolean</td>
<td>Make lower case</td>
<td>Boolean value Yes/No</td>
<td>N</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -supper1<br>-supper_sequence</td>
<td>boolean</td>
<td>Make upper case</td>
<td>Boolean value Yes/No</td>
<td>N</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -sformat1<br>-sformat_sequence</td>
<td>string</td>
<td>Input sequence format</td>
<td>Any string</td>
<td>&nbsp;</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -sdbname1<br>-sdbname_sequence</td>
<td>string</td>
<td>Database name</td>
<td>Any string</td>
<td>&nbsp;</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -sid1<br>-sid_sequence</td>
<td>string</td>
<td>Entryname</td>
<td>Any string</td>
<td>&nbsp;</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -ufo1<br>-ufo_sequence</td>
<td>string</td>
<td>UFO features</td>
<td>Any string</td>
<td>&nbsp;</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -fformat1<br>-fformat_sequence</td>
<td>string</td>
<td>Features format</td>
<td>Any string</td>
<td>&nbsp;</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -fopenfile1<br>-fopenfile_sequence</td>
<td>string</td>
<td>Features file name</td>
<td>Any string</td>
<td>&nbsp;</td>
</tr>

<tr bgcolor="#FFFFCC">
<th align="left" colspan=5>"-outfile" associated outfile qualifiers
</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -odirectory2<br>-odirectory_outfile</td>
<td>string</td>
<td>Output directory</td>
<td>Any string</td>
<td>&nbsp;</td>
</tr>

<tr bgcolor="#FFFFCC">
<th align="left" colspan=5>"-outseq" associated seqoutall qualifiers
</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -osformat</td>
<td>string</td>
<td>Output seq format</td>
<td>Any string</td>
<td>&nbsp;</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -osextension</td>
<td>string</td>
<td>File name extension</td>
<td>Any string</td>
<td>&nbsp;</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -osname</td>
<td>string</td>
<td>Base file name</td>
<td>Any string</td>
<td>&nbsp;</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -osdirectory</td>
<td>string</td>
<td>Output directory</td>
<td>Any string</td>
<td>&nbsp;</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -osdbname</td>
<td>string</td>
<td>Database name to add</td>
<td>Any string</td>
<td>&nbsp;</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -ossingle</td>
<td>boolean</td>
<td>Separate file for each entry</td>
<td>Boolean value Yes/No</td>
<td>N</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -oufo</td>
<td>string</td>
<td>UFO features</td>
<td>Any string</td>
<td>&nbsp;</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -offormat</td>
<td>string</td>
<td>Features format</td>
<td>Any string</td>
<td>&nbsp;</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -ofname</td>
<td>string</td>
<td>Features file name</td>
<td>Any string</td>
<td>&nbsp;</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -ofdirectory</td>
<td>string</td>
<td>Output directory</td>
<td>Any string</td>
<td>&nbsp;</td>
</tr>

<tr bgcolor="#FFFFCC">
<th align="left" colspan=5>General qualifiers</th>
</tr>

<tr bgcolor="#FFFFCC">
<td> -auto</td>
<td>boolean</td>
<td>Turn off prompts</td>
<td>Boolean value Yes/No</td>
<td>N</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -stdout</td>
<td>boolean</td>
<td>Write first file to standard output</td>
<td>Boolean value Yes/No</td>
<td>N</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -filter</td>
<td>boolean</td>
<td>Read first file from standard input, write first file to standard output</td>
<td>Boolean value Yes/No</td>
<td>N</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -options</td>
<td>boolean</td>
<td>Prompt for standard and additional values</td>
<td>Boolean value Yes/No</td>
<td>N</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -debug</td>
<td>boolean</td>
<td>Write debug output to program.dbg</td>
<td>Boolean value Yes/No</td>
<td>N</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -verbose</td>
<td>boolean</td>
<td>Report some/full command line options</td>
<td>Boolean value Yes/No</td>
<td>Y</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -help</td>
<td>boolean</td>
<td>Report command line options and exit. More information on associated and general qualifiers can be found with -help -verbose</td>
<td>Boolean value Yes/No</td>
<td>N</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -warning</td>
<td>boolean</td>
<td>Report warnings</td>
<td>Boolean value Yes/No</td>
<td>Y</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -error</td>
<td>boolean</td>
<td>Report errors</td>
<td>Boolean value Yes/No</td>
<td>Y</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -fatal</td>
<td>boolean</td>
<td>Report fatal errors</td>
<td>Boolean value Yes/No</td>
<td>Y</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -die</td>
<td>boolean</td>
<td>Report dying program messages</td>
<td>Boolean value Yes/No</td>
<td>Y</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -version</td>
<td>boolean</td>
<td>Report version number and exit</td>
<td>Boolean value Yes/No</td>
<td>N</td>
</tr>

</table>

<H2>
    Input file format
</H2>

<b>sixpack</b> reads any normal sequence USAs.

<p>


<a name="input.1"></a>
<h3>Input files for usage example </h3>

'tembl:x13776' is a sequence entry in the example nucleic acid database 'tembl'
<p>
<p><h3>Database entry: tembl:x13776</h3>
<table width="90%"><tr><td bgcolor="#FFCCFF">
<pre>
ID   X13776; SV 1; linear; genomic DNA; STD; PRO; 2167 BP.
XX
AC   X13776; M43175;
XX
DT   19-APR-1989 (Rel. 19, Created)
DT   14-NOV-2006 (Rel. 89, Last updated, Version 24)
XX
DE   Pseudomonas aeruginosa amiC and amiR gene for aliphatic amidase regulation
XX
KW   aliphatic amidase regulator; amiC gene; amiR gene.
XX
OS   Pseudomonas aeruginosa
OC   Bacteria; Proteobacteria; Gammaproteobacteria; Pseudomonadales;
OC   Pseudomonadaceae; Pseudomonas.
XX
RN   [1]
RP   1167-2167
RA   Rice P.M.;
RT   ;
RL   Submitted (16-DEC-1988) to the EMBL/GenBank/DDBJ databases.
RL   Rice P.M., EMBL, Postfach 10-2209, Meyerhofstrasse 1, 6900 Heidelberg, FRG.
XX
RN   [2]
RP   1167-2167
RX   DOI; 10.1016/0014-5793(89)80249-2.
RX   PUBMED; 2495988.
RA   Lowe N., Rice P.M., Drew R.E.;
RT   "Nucleotide sequence of the aliphatic amidase regulator gene (amiR) of
RT   Pseudomonas aeruginosa";
RL   FEBS Lett. 246(1-2):39-43(1989).
XX
RN   [3]
RP   1-1292
RX   PUBMED; 1907262.
RA   Wilson S., Drew R.;
RT   "Cloning and DNA sequence of amiC, a new gene regulating expression of the
RT   Pseudomonas aeruginosa aliphatic amidase, and purification of the amiC
RT   product";
RL   J. Bacteriol. 173(16):4914-4921(1991).
XX
RN   [4]
RP   1-2167
RA   Rice P.M.;
RT   ;
RL   Submitted (04-SEP-1991) to the EMBL/GenBank/DDBJ databases.
RL   Rice P.M., EMBL, Postfach 10-2209, Meyerhofstrasse 1, 6900 Heidelberg, FRG.
XX
DR   GOA; Q51417.
DR   InterPro; IPR003211; AmiSUreI_transpt.
DR   UniProtKB/Swiss-Prot; Q51417; AMIS_PSEAE.


<font color=red>  [Part of this file has been deleted for brevity]</font>

FT                   /replace=""
FT                   /note="ClaI fragment deleted in pSW36,  constitutive
FT                   phenotype"
FT   misc_feature    1
FT                   /note="last base of an XhoI site"
FT   misc_feature    648..653
FT                   /note="end of 658bp XhoI fragment, deletion in  pSW3 causes
FT                   constitutive expression of amiE"
FT   conflict        1281
FT                   /replace="g"
FT                   /citation=[3]
XX
SQ   Sequence 2167 BP; 363 A; 712 C; 730 G; 362 T; 0 other;
     ggtaccgctg gccgagcatc tgctcgatca ccaccagccg ggcgacggga actgcacgat        60
     ctacctggcg agcctggagc acgagcgggt tcgcttcgta cggcgctgag cgacagtcac       120
     aggagaggaa acggatggga tcgcaccagg agcggccgct gatcggcctg ctgttctccg       180
     aaaccggcgt caccgccgat atcgagcgct cgcacgcgta tggcgcattg ctcgcggtcg       240
     agcaactgaa ccgcgagggc ggcgtcggcg gtcgcccgat cgaaacgctg tcccaggacc       300
     ccggcggcga cccggaccgc tatcggctgt gcgccgagga cttcattcgc aaccgggggg       360
     tacggttcct cgtgggctgc tacatgtcgc acacgcgcaa ggcggtgatg ccggtggtcg       420
     agcgcgccga cgcgctgctc tgctacccga ccccctacga gggcttcgag tattcgccga       480
     acatcgtcta cggcggtccg gcgccgaacc agaacagtgc gccgctggcg gcgtacctga       540
     ttcgccacta cggcgagcgg gtggtgttca tcggctcgga ctacatctat ccgcgggaaa       600
     gcaaccatgt gatgcgccac ctgtatcgcc agcacggcgg cacggtgctc gaggaaatct       660
     acattccgct gtatccctcc gacgacgact tgcagcgcgc cgtcgagcgc atctaccagg       720
     cgcgcgccga cgtggtcttc tccaccgtgg tgggcaccgg caccgccgag ctgtatcgcg       780
     ccatcgcccg tcgctacggc gacggcaggc ggccgccgat cgccagcctg accaccagcg       840
     aggcggaggt ggcgaagatg gagagtgacg tggcagaggg gcaggtggtg gtcgcgcctt       900
     acttctccag catcgatacg cccgccagcc gggccttcgt ccaggcctgc catggtttct       960
     tcccggagaa cgcgaccatc accgcctggg ccgaggcggc ctactggcag accttgttgc      1020
     tcggccgcgc cgcgcaggcc gcaggcaact ggcgggtgga agacgtgcag cggcacctgt      1080
     acgacatcga catcgacgcg ccacaggggc cggtccgggt ggagcgccag aacaaccaca      1140
     gccgcctgtc ttcgcgcatc gcggaaatcg atgcgcgcgg cgtgttccag gtccgctggc      1200
     agtcgcccga accgattcgc cccgaccctt atgtcgtcgt gcataacctc gacgactggt      1260
     ccgccagcat gggcggggga ccgctcccat gagcgccaac tcgctgctcg gcagcctgcg      1320
     cgagttgcag gtgctggtcc tcaacccgcc gggggaggtc agcgacgccc tggtcttgca      1380
     gctgatccgc atcggttgtt cggtgcgcca gtgctggccg ccgccggaag ccttcgacgt      1440
     gccggtggac gtggtcttca ccagcatttt ccagaatggc caccacgacg agatcgctgc      1500
     gctgctcgcc gccgggactc cgcgcactac cctggtggcg ctggtggagt acgaaagccc      1560
     cgcggtgctc tcgcagatca tcgagctgga gtgccacggc gtgatcaccc agccgctcga      1620
     tgcccaccgg gtgctgcctg tgctggtatc ggcgcggcgc atcagcgagg aaatggcgaa      1680
     gctgaagcag aagaccgagc agctccagga ccgcatcgcc ggccaggccc ggatcaacca      1740
     ggccaaggtg ttgctgatgc agcgccatgg ctgggacgag cgcgaggcgc accagcacct      1800
     gtcgcgggaa gcgatgaagc ggcgcgagcc gatcctgaag atcgctcagg agttgctggg      1860
     aaacgagccg tccgcctgag cgatccgggc cgaccagaac aataacaaga ggggtatcgt      1920
     catcatgctg ggactggttc tgctgtacgt tggcgcggtg ctgtttctca atgccgtctg      1980
     gttgctgggc aagatcagcg gtcgggaggt ggcggtgatc aacttcctgg tcggcgtgct      2040
     gagcgcctgc gtcgcgttct acctgatctt ttccgcagca gccgggcagg gctcgctgaa      2100
     ggccggagcg ctgaccctgc tattcgcttt tacctatctg tgggtggccg ccaaccagtt      2160
     cctcgag                                                                2167
//
</pre>
</td></tr></table><p>

<H2>
    Output file format
</H2>


<a name="output.1"></a>
<h3>Output files for usage example </h3>
<p><h3>File: x13776.sixpack</h3>
<table width="90%"><tr><td bgcolor="#CCFFCC">
<pre>
X13776
Pseudomonas aeruginosa amiC and amiR gene for aliphatic amidase
regulation


          G  T  A  G  R  A  S  A  R  S  P  P  A  G  R  R  E  L  H  D     F1
           V  P  L  A  E  H  L  L  D  H  H  Q  P  G  D  G  N  C  T  I    F2
            Y  R  W  P  S  I  C  S  I  T  T  S  R  A  T  G  T  A  R  S   F3
        1 ggtaccgctggccgagcatctgctcgatcaccaccagccgggcgacgggaactgcacgat 60
          ----:----|----:----|----:----|----:----|----:----|----:----|
        1 ccatggcgaccggctcgtagacgagctagtggtggtcggcccgctgcccttgacgtgcta 60
           P  V  A  P  R  A  D  A  R  D  G  G  A  P  R  R  S  S  C  S    F6
          X  Y  R  Q  G  L  M  Q  E  I  V  V  L  R  A  V  P  V  A  R     F5
            T  G  S  A  S  C  R  S  S  *  W  W  G  P  S  P  F  Q  V  I   F4


          L  P  G  E  P  G  A  R  A  G  S  L  R  T  A  L  S  D  S  H     F1
           Y  L  A  S  L  E  H  E  R  V  R  F  V  R  R  *  A  T  V  T    F2
            T  W  R  A  W  S  T  S  G  F  A  S  Y  G  A  E  R  Q  S  Q   F3
       61 ctacctggcgagcctggagcacgagcgggttcgcttcgtacggcgctgagcgacagtcac 120
          ----:----|----:----|----:----|----:----|----:----|----:----|
       61 gatggaccgctcggacctcgtgctcgcccaagcgaagcatgccgcgactcgctgtcagtg 120
           R  G  P  S  G  P  A  R  A  P  E  S  R  V  A  S  L  S  L  *    F6
          D  V  Q  R  A  Q  L  V  L  P  N  A  E  Y  P  A  S  R  C  D     F5
            *  R  A  L  R  S  C  S  R  T  R  K  T  R  R  Q  A  V  T  V   F4


          R  R  G  N  G  W  D  R  T  R  S  G  R  *  S  A  C  C  S  P     F1
           G  E  E  T  D  G  I  A  P  G  A  A  A  D  R  P  A  V  L  R    F2
            E  R  K  R  M  G  S  H  Q  E  R  P  L  I  G  L  L  F  S  E   F3
      121 aggagaggaaacggatgggatcgcaccaggagcggccgctgatcggcctgctgttctccg 180
          ----:----|----:----|----:----|----:----|----:----|----:----|
      121 tcctctcctttgcctaccctagcgtggtcctcgccggcgactagccggacgacaagaggc 180
           L  L  P  F  P  H  S  R  V  L  L  P  R  Q  D  A  Q  Q  E  G    F6
          C  S  L  F  R  I  P  D  C  W  S  R  G  S  I  P  R  S  N  E     F5
            P  S  S  V  S  P  I  A  G  P  A  A  A  S  R  G  A  T  R  R   F4


          K  P  A  S  P  P  I  S  S  A  R  T  R  M  A  H  C  S  R  S     F1
           N  R  R  H  R  R  Y  R  A  L  A  R  V  W  R  I  A  R  G  R    F2
            T  G  V  T  A  D  I  E  R  S  H  A  Y  G  A  L  L  A  V  E   F3
      181 aaaccggcgtcaccgccgatatcgagcgctcgcacgcgtatggcgcattgctcgcggtcg 240
          ----:----|----:----|----:----|----:----|----:----|----:----|
      181 tttggccgcagtggcggctatagctcgcgagcgtgcgcataccgcgtaacgagcgccagc 240
           F  G  A  D  G  G  I  D  L  A  R  V  R  I  A  C  Q  E  R  D    F6
          S  V  P  T  V  A  S  I  S  R  E  C  A  Y  P  A  N  S  A  T     F5
            F  R  R  *  R  R  Y  R  A  S  A  R  T  H  R  M  A  R  P  R   F4


          S  N  *  T  A  R  A  A  S  A  V  A  R  S  K  R  C  P  R  T     F1


<font color=red>  [Part of this file has been deleted for brevity]</font>

     1981 caacgacccgttctagtcgccagccctccaccgccactagttgaaggaccagccgcacga 2040
           T  A  P  C  S  *  R  D  P  P  P  P  S  *  S  G  P  R  R  A    F6
          P  Q  Q  A  L  D  A  T  P  L  H  R  H  D  V  E  Q  D  A  H     F5
            N  S  P  L  I  L  P  R  S  T  A  T  I  L  K  R  T  P  T  S   F4


          E  R  L  R  R  V  L  P  D  L  F  R  S  S  R  A  G  L  A  E     F1
           S  A  C  V  A  F  Y  L  I  F  S  A  A  A  G  Q  G  S  L  K    F2
            A  P  A  S  R  S  T  *  S  F  P  Q  Q  P  G  R  A  R  *  R   F3
     2041 gagcgcctgcgtcgcgttctacctgatcttttccgcagcagccgggcagggctcgctgaa 2100
          ----:----|----:----|----:----|----:----|----:----|----:----|
     2041 ctcgcggacgcagcgcaagatggactagaaaaggcgtcgtcggcccgtcccgagcgactt 2100
           S  R  R  R  R  T  R  G  S  R  K  R  L  L  R  A  P  S  A  S    F6
          Q  A  G  A  D  R  E  V  Q  D  K  G  C  C  G  P  L  A  R  Q     F5
            L  A  Q  T  A  N  *  R  I  K  E  A  A  A  P  C  P  E  S  F   F4


          G  R  S  A  D  P  A  I  R  F  Y  L  S  V  G  G  R  Q  P  V     F1
           A  G  A  L  T  L  L  F  A  F  T  Y  L  W  V  A  A  N  Q  F    F2
            P  E  R  *  P  C  Y  S  L  L  P  I  C  G  W  P  P  T  S  S   F3
     2101 ggccggagcgctgaccctgctattcgcttttacctatctgtgggtggccgccaaccagtt 2160
          ----:----|----:----|----:----|----:----|----:----|----:----|
     2101 ccggcctcgcgactgggacgataagcgaaaatggatagacacccaccggcggttggtcaa 2160
           P  R  L  A  S  G  A  I  R  K  *  R  D  T  P  P  R  W  G  T    F6
          L  G  S  R  Q  G  Q  *  E  S  K  G  I  Q  P  H  G  G  V  L     F5
            A  P  A  S  V  R  S  N  A  K  V  *  R  H  T  A  A  L  W  N   F4


          P  R  X                                                        F1
           L  E                                                          F2
            S  X                                                         F3
     2161 cctcgag 2167
          ----:--
     2161 ggagctc 2220
           G  R                                                          F6
          E  E  L                                                        F5
            R  S                                                         F4

##############################
Minimum size of ORFs : 1

Total ORFs in frame 1 :     8
Total ORFs in frame 2 :     5
Total ORFs in frame 3 :    13
Total ORFs in frame 4 :    10
Total ORFs in frame 5 :    16
Total ORFs in frame 6 :    15

Total ORFs :    67
##############################

</pre>
</td></tr></table><p>
<p><h3>File: x13776.fasta</h3>
<table width="90%"><tr><td bgcolor="#CCFFCC">
<pre>
&gt;X13776_1_ORF1  Translation of X13776 in frame 1, ORF 1, threshold 1, 53aa
GTAGRASARSPPAGRRELHDLPGEPGARAGSLRTALSDSHRRGNGWDRTRSGR
&gt;X13776_1_ORF2  Translation of X13776 in frame 1, ORF 2, threshold 1, 28aa
SACCSPKPASPPISSARTRMAHCSRSSN
&gt;X13776_1_ORF3  Translation of X13776 in frame 1, ORF 3, threshold 1, 52aa
TARAASAVARSKRCPRTPAATRTAIGCAPRTSFATGGYGSSWAATCRTRARR
&gt;X13776_1_ORF4  Translation of X13776 in frame 1, ORF 4, threshold 1, 43aa
CRWSSAPTRCSATRPPTRASSIRRTSSTAVRRRTRTVRRWRRT
&gt;X13776_1_ORF5  Translation of X13776 in frame 1, ORF 5, threshold 1, 23aa
FATTASGWCSSARTTSIRGKATM
&gt;X13776_1_ORF6  Translation of X13776 in frame 1, ORF 6, threshold 1, 72aa
CATCIASTAARCSRKSTFRCIPPTTTCSAPSSASTRRAPTWSSPPWWAPAPPSCIAPSPV
ATATAGGRRSPA
&gt;X13776_1_ORF7  Translation of X13776 in frame 1, ORF 7, threshold 1, 357aa
PPARRRWRRWRVTWQRGRWWSRLTSPASIRPPAGPSSRPAMVSSRRTRPSPPGPRRPTGR
PCCSAAPRRPQATGGWKTCSGTCTTSTSTRHRGRSGWSARTTTAACLRASRKSMRAACSR
SAGSRPNRFAPTLMSSCITSTTGPPAWAGDRSHERQLAARQPARVAGAGPQPAGGGQRRP
GLAADPHRLFGAPVLAAAGSLRRAGGRGLHQHFPEWPPRRDRCAARRRDSAHYPGGAGGV
RKPRGALADHRAGVPRRDHPAARCPPGAACAGIGAAHQRGNGEAEAEDRAAPGPHRRPGP
DQPGQGVADAAPWLGRARGAPAPVAGSDEAARADPEDRSGVAGKRAVRLSDPGRPEQ
&gt;X13776_1_ORF8  Translation of X13776 in frame 1, ORF 8, threshold 1, 88aa
QEGYRHHAGTGSAVRWRGAVSQCRLVAGQDQRSGGGGDQLPGRRAERLRRVLPDLFRSSR
AGLAEGRSADPAIRFYLSVGGRQPVPRX
&gt;X13776_2_ORF1  Translation of X13776 in frame 2, ORF 1, threshold 1, 35aa
VPLAEHLLDHHQPGDGNCTIYLASLEHERVRFVRR
&gt;X13776_2_ORF2  Translation of X13776 in frame 2, ORF 2, threshold 1, 252aa
ATVTGEETDGIAPGAAADRPAVLRNRRHRRYRALARVWRIARGRATEPRGRRRRSPDRNA
VPGPRRRPGPLSAVRRGLHSQPGGTVPRGLLHVAHAQGGDAGGRARRRAALLPDPLRGLR
VFAEHRLRRSGAEPEQCAAGGVPDSPLRRAGGVHRLGLHLSAGKQPCDAPPVSPARRHGA
RGNLHSAVSLRRRLAARRRAHLPGARRRGLLHRGGHRHRRAVSRHRPSLRRRQAAADRQP
DHQRGGGGEDGE
&gt;X13776_2_ORF3  Translation of X13776 in frame 2, ORF 3, threshold 1, 125aa
RGRGAGGGRALLLQHRYARQPGLRPGLPWFLPGERDHHRLGRGGLLADLVARPRRAGRRQ
LAGGRRAAAPVRHRHRRATGAGPGGAPEQPQPPVFAHRGNRCARRVPGPLAVARTDSPRP
LCRRA
&gt;X13776_2_ORF4  Translation of X13776 in frame 2, ORF 4, threshold 1, 210aa
PRRLVRQHGRGTAPMSANSLLGSLRELQVLVLNPPGEVSDALVLQLIRIGCSVRQCWPPP
EAFDVPVDVVFTSIFQNGHHDEIAALLAAGTPRTTLVALVEYESPAVLSQIIELECHGVI
TQPLDAHRVLPVLVSARRISEEMAKLKQKTEQLQDRIAGQARINQAKVLLMQRHGWDERE
AHQHLSREAMKRREPILKIAQELLGNEPSA
&gt;X13776_2_ORF5  Translation of X13776 in frame 2, ORF 5, threshold 1, 96aa
AIRADQNNNKRGIVIMLGLVLLYVGAVLFLNAVWLLGKISGREVAVINFLVGVLSACVAF
YLIFSAAAGQGSLKAGALTLLFAFTYLWVAANQFLE
&gt;X13776_3_ORF1  Translation of X13776 in frame 3, ORF 1, threshold 1, 429aa
YRWPSICSITTSRATGTARSTWRAWSTSGFASYGAERQSQERKRMGSHQERPLIGLLFSE
TGVTADIERSHAYGALLAVEQLNREGGVGGRPIETLSQDPGGDPDRYRLCAEDFIRNRGV
RFLVGCYMSHTRKAVMPVVERADALLCYPTPYEGFEYSPNIVYGGPAPNQNSAPLAAYLI
RHYGERVVFIGSDYIYPRESNHVMRHLYRQHGGTVLEEIYIPLYPSDDDLQRAVERIYQA
RADVVFSTVVGTGTAELYRAIARRYGDGRRPPIASLTTSEAEVAKMESDVAEGQVVVAPY
FSSIDTPASRAFVQACHGFFPENATITAWAEAAYWQTLLLGRAAQAAGNWRVEDVQRHLY


<font color=red>  [Part of this file has been deleted for brevity]</font>

SEPMNTTRSP
&gt;X13776_5_ORF11  Translation of X13776 in frame 5, ORF 11, threshold 1, 19aa
WRIRYAASGALFWFGAGPP
&gt;X13776_5_ORF12  Translation of X13776 in frame 5, ORF 12, threshold 1, 10aa
TMFGEYSKPS
&gt;X13776_5_ORF13  Translation of X13776 in frame 5, ORF 13, threshold 1, 3aa
GVG
&gt;X13776_5_ORF14  Translation of X13776 in frame 5, ORF 14, threshold 1, 20aa
QSSASARSTTGITALRVCDM
&gt;X13776_5_ORF15  Translation of X13776 in frame 5, ORF 15, threshold 1, 19aa
QPTRNRTPRLRMKSSAHSR
&gt;X13776_5_ORF16  Translation of X13776 in frame 5, ORF 16, threshold 1, 107aa
RSGSPPGSWDSVSIGRPPTPPSRFSCSTASNAPYACERSISAVTPVSENSRPISGRSWCD
PIRFLSCDCRSAPYEANPLVLQARQVDRAVPVARLVVIEQMLGQRYX
&gt;X13776_6_ORF1  Translation of X13776 in frame 6, ORF 1, threshold 1, 11aa
RGTGWRPPTDR
&gt;X13776_6_ORF2  Translation of X13776 in frame 6, ORF 2, threshold 1, 36aa
KRIAGSALRPSASPARLLRKRSGRTRRRRSARRPGS
&gt;X13776_6_ORF3  Translation of X13776 in frame 6, ORF 3, threshold 1, 7aa
SPPPPDR
&gt;X13776_6_ORF4  Translation of X13776 in frame 6, ORF 4, threshold 1, 8aa
SCPATRRH
&gt;X13776_6_ORF5  Translation of X13776 in frame 6, ORF 5, threshold 1, 14aa
ETAPRQRTAEPVPA
&gt;X13776_6_ORF6  Translation of X13776 in frame 6, ORF 6, threshold 1, 61aa
RYPSCYCSGRPGSLRRTARFPATPERSSGSARAASSLPATGAGAPRARPSHGAASATPWP
G
&gt;X13776_6_ORF7  Translation of X13776 in frame 6, ORF 7, threshold 1, 23aa
SGPGRRCGPGAARSSASASPFPR
&gt;X13776_6_ORF8  Translation of X13776 in frame 6, ORF 8, threshold 1, 18aa
CAAPIPAQAAPGGHRAAG
&gt;X13776_6_ORF9  Translation of X13776 in frame 6, ORF 9, threshold 1, 8aa
SRRGTPAR
&gt;X13776_6_ORF10  Translation of X13776 in frame 6, ORF 10, threshold 1, 16aa
SARAPRGFRTPPAPPG
&gt;X13776_6_ORF11  Translation of X13776 in frame 6, ORF 11, threshold 1, 22aa
CAESRRRAAQRSRRGGHSGKCW
&gt;X13776_6_ORF12  Translation of X13776 in frame 6, ORF 12, threshold 1, 32aa
RPRPPARRRLPAAASTGAPNNRCGSAARPGRR
&gt;X13776_6_ORF13  Translation of X13776 in frame 6, ORF 13, threshold 1, 5aa
PPPAG
&gt;X13776_6_ORF14  Translation of X13776 in frame 6, ORF 14, threshold 1, 407aa
GPAPATRAGCRAASWRSWERSPAHAGGPVVEVMHDDIRVGANRFGRLPADLEHAARIDFR
DARRQAAVVVLALHPDRPLWRVDVDVVQVPLHVFHPPVACGLRGAAEQQGLPVGRLGPGG
DGRVLREETMAGLDEGPAGGRIDAGEVRRDHHLPLCHVTLHLRHLRLAGGQAGDRRPPAV
AVATGDGAIQLGGAGAHHGGEDHVGARLVDALDGALQVVVGGIQRNVDFLEHRAAVLAIQ
VAHHMVAFPRIDVVRADEHHPLAVVANQVRRQRRTVLVRRRTAVDDVRRILEALVGGRVA
EQRVGALDHRHHRLARVRHVAAHEEPYPPVANEVLGAQPIAVRVAAGVLGQRFDRATADA
ALAVQLLDREQCAIRVRALDIGGDAGFGEQQADQRPLLVRSHPFPLL
&gt;X13776_6_ORF15  Translation of X13776 in frame 6, ORF 15, threshold 1, 39aa
LSLSAVRSEPARAPGSPGRSCSSRRPAGGDRADARPAVP
</pre>
</td></tr></table><p>

<H2>
    Data files
</H2>

<p>
EMBOSS data files are distributed with the application and stored
in the standard EMBOSS data directory, which is defined
by the EMBOSS environment variable EMBOSS_DATA.

<p>

To see the available EMBOSS data files, run:
<p>
<pre>
% embossdata -showall
</pre>
<p>
To fetch one of the data files (for example 'Exxx.dat') into your
current directory for you to inspect or modify, run:

<pre>

% embossdata -fetch -file Exxx.dat

</pre>
<p>

Users can provide their own data files in their own directories.
Project specific files can be put in the current directory, or for
tidier directory listings in a subdirectory called
".embossdata". Files for all EMBOSS runs can be put in the user's home
directory, or again in a subdirectory called ".embossdata".

<p>
The directories are searched in the following order:

<ul>
   <li> . (your current directory)
   <li> .embossdata (under your current directory)
   <li> ~/ (your home directory)
   <li> ~/.embossdata
</ul>
<p>

<p>
The Genetic Code data files are based on the NCBI genetic code tables.
Their names and descriptions are:

<dl>
<dt>EGC.0 </dt><dd>
      Standard (Differs from GC.1 in that it only has initiation site 'AUG')
<dt>EGC.1 </dt><dd>
      Standard
<dt>EGC.2 </dt><dd>
      Vertebrate Mitochodrial
<dt>EGC.3 </dt><dd>
      Yeast Mitochondrial
<dt>EGC.4 </dt><dd>
      Mold, Protozoan, Coelenterate Mitochondrial and Mycoplasma/Spiroplasma
<dt>EGC.5 </dt><dd>
      Invertebrate Mitochondrial
<dt>EGC.6 </dt><dd>
      Ciliate Macronuclear and Dasycladacean
<dt>EGC.9 </dt><dd>
      Echinoderm Mitochondrial
<dt>EGC.10 </dt><dd>
      Euplotid Nuclear
<dt>EGC.11 </dt><dd>
      Bacterial
<dt>EGC.12 </dt><dd>
      Alternative Yeast Nuclear
<dt>EGC.13 </dt><dd>
      Ascidian Mitochondrial
<dt>EGC.14 </dt><dd>
      Flatworm Mitochondrial
<dt>EGC.15</dt><dd>
      Blepharisma Macronuclear
<dt>EGC.16</dt><dd>
      Chlorophycean Mitochondrial
<dt>EGC.21</dt><dd>
      Trematode Mitochondrial
<dt>EGC.22</dt><dd>
      Scenedesmus obliquus
<dt>EGC.23</dt><dd>
      Thraustochytrium Mitochondrial
</dl>

<p>

The format of these files is very simple.

<p>   

It consists of several lines of optional comments, each starting with
a '#' character.

<p>

These are followed the line: 'Genetic Code [n]', where 'n' is the
number of the genetic code file.

<p> 

This is followed by the description of the code and then by four lines
giving the IUPAC one-letter code of the translated amino acid, the
start codons (indicdated by an 'M') and the three bases of the codon,
lined up one on top of the other.
   
<p>

For example:

<pre>
   
------------------------------------------------------------------------------
# Genetic Code Table
#
# Obtained from: http://www.ncbi.nlm.nih.gov/collab/FT/genetic_codes.html
# and: http://www3.ncbi.nlm.nih.gov/htbin-post/Taxonomy/wprintgc?mode=c
#
# Differs from Genetic Code [1] only in that the initiation sites have been
# changed to only 'AUG'

Genetic Code [0]
Standard
   
AAs  =   FFLLSSSSYY**CC*WLLLLPPPPHHQQRRRRIIIMTTTTNNKKSSRRVVVVAAAADDEEGGGG
Starts = -----------------------------------M----------------------------
Base1  = TTTTTTTTTTTTTTTTCCCCCCCCCCCCCCCCAAAAAAAAAAAAAAAAGGGGGGGGGGGGGGGG
Base2  = TTTTCCCCAAAAGGGGTTTTCCCCAAAAGGGGTTTTCCCCAAAAGGGGTTTTCCCCAAAAGGGG
Base3  = TCAGTCAGTCAGTCAGTCAGTCAGTCAGTCAGTCAGTCAGTCAGTCAGTCAGTCAGTCAGTCAG
------------------------------------------------------------------------------

</pre>


<H2>
    Notes
</H2>

<p>An open reading frame is defined in this program as any possible
translation between two STOP codons. Optionally, the beginning or end
of a sequence may be counted as an ORF even if it's less than the
minimal ORF size or (end only) lacking a STOP codon.  See the
-firstorf and -lastorf options.</p>

<H2>
    References
</H2>

None.

<H2>
    Warnings
</H2>

None.

<H2>
    Diagnostic Error Messages
</H2>

None.

<H2>
    Exit status
</H2>

It always exits with status 0.

<H2>
    Known bugs
</H2>

None.

<h2><a name="See also">See also</a></h2>
<table border cellpadding=4 bgcolor="#FFFFF0">
<tr><th>Program name</th>
<th>Description</th></tr>
<tr>
<td><a href="abiview.html">abiview</a></td>
<td>Display the trace in an ABI sequencer file</td>
</tr>

<tr>
<td><a href="backtranambig.html">backtranambig</a></td>
<td>Back-translate a protein sequence to ambiguous nucleotide sequence</td>
</tr>

<tr>
<td><a href="backtranseq.html">backtranseq</a></td>
<td>Back-translate a protein sequence to a nucleotide sequence</td>
</tr>

<tr>
<td><a href="cirdna.html">cirdna</a></td>
<td>Draws circular maps of DNA constructs</td>
</tr>

<tr>
<td><a href="coderet.html">coderet</a></td>
<td>Extract CDS, mRNA and translations from feature tables</td>
</tr>

<tr>
<td><a href="getorf.html">getorf</a></td>
<td>Finds and extracts open reading frames (ORFs)</td>
</tr>

<tr>
<td><a href="lindna.html">lindna</a></td>
<td>Draws linear maps of DNA constructs</td>
</tr>

<tr>
<td><a href="marscan.html">marscan</a></td>
<td>Finds matrix/scaffold recognition (MRS) signatures in DNA sequences</td>
</tr>

<tr>
<td><a href="pepnet.html">pepnet</a></td>
<td>Draw a helical net for a protein sequence</td>
</tr>

<tr>
<td><a href="pepwheel.html">pepwheel</a></td>
<td>Draw a helical wheel diagram for a protein sequence</td>
</tr>

<tr>
<td><a href="plotorf.html">plotorf</a></td>
<td>Plot potential open reading frames in a nucleotide sequence</td>
</tr>

<tr>
<td><a href="prettyplot.html">prettyplot</a></td>
<td>Draw a sequence alignment with pretty formatting</td>
</tr>

<tr>
<td><a href="prettyseq.html">prettyseq</a></td>
<td>Write a nucleotide sequence and its translation to file</td>
</tr>

<tr>
<td><a href="remap.html">remap</a></td>
<td>Display restriction enzyme binding sites in a nucleotide sequence</td>
</tr>

<tr>
<td><a href="seealso.html">seealso</a></td>
<td>Finds programs with similar function to a specified program</td>
</tr>

<tr>
<td><a href="showalign.html">showalign</a></td>
<td>Display a multiple sequence alignment in pretty format</td>
</tr>

<tr>
<td><a href="showdb.html">showdb</a></td>
<td>Displays information on configured databases</td>
</tr>

<tr>
<td><a href="showfeat.html">showfeat</a></td>
<td>Display features of a sequence in pretty format</td>
</tr>

<tr>
<td><a href="showorf.html">showorf</a></td>
<td>Display a nucleotide sequence and translation in pretty format</td>
</tr>

<tr>
<td><a href="showpep.html">showpep</a></td>
<td>Displays protein sequences with features in pretty format</td>
</tr>

<tr>
<td><a href="showseq.html">showseq</a></td>
<td>Displays sequences with features in pretty format</td>
</tr>

<tr>
<td><a href="syco.html">syco</a></td>
<td>Draw synonymous codon usage statistic plot for a nucleotide sequence</td>
</tr>

<tr>
<td><a href="tcode.html">tcode</a></td>
<td>Identify protein-coding regions using Fickett TESTCODE statistic</td>
</tr>

<tr>
<td><a href="textsearch.html">textsearch</a></td>
<td>Search the textual description of sequence(s)</td>
</tr>

<tr>
<td><a href="transeq.html">transeq</a></td>
<td>Translate nucleic acid sequences</td>
</tr>

<tr>
<td><a href="wobble.html">wobble</a></td>
<td>Plot third base position variability in a nucleotide sequence</td>
</tr>

</table>

<H2>
    Author(s)
</H2>

Thomas Laurent formerly at:
<br>
Lion Bioscience Ltd,
     Compass House,
     80-82 Newmarket Road,
     Cambridge,
     CB5 8DZ,
     UK

<p>
Please report all bugs to the EMBOSS bug team (emboss-bug&nbsp;&copy;&nbsp;emboss.open-bio.org) not to the original author.
<H2>
    History
</H2>


Written (November 2002) - Thomas Laurent


<H2>
    Target users
</H2>
This program is intended to be used by everyone and everything, from naive users to embedded scripts.


<H2>
    Comments
</H2>
None

</BODY>
</HTML>

