<HTML>

<HEAD>
  <TITLE>
  EMBOSS: tmap
  </TITLE>
</HEAD>
<BODY BGCOLOR="#FFFFFF" text="#000000">

<table align=center border=0 cellspacing=0 cellpadding=0>
<tr><td valign=top>
<A HREF="/" ONMOUSEOVER="self.status='Go to the EMBOSS home page';return true"><img border=0 src="emboss_icon.jpg" alt="" width=150 height=48></a>
</td>
<td align=left valign=middle>
<b><font size="+6">
tmap
</font></b>
</td></tr>
</table>
<br>&nbsp;
<p>


<H2>
    Function
</H2>
Displays membrane spanning regions

<H2>
    Description
</H2>

This program predicts transmembrane segments in proteins,    
utilising the algorithm described in:                        
"Persson, B. & Argos, P. (1994) Prediction of transmembrane  
segments in proteins utilising multiple sequence alignments   
J. Mol. Biol. 237, 182-192."                                 
<p>

<b>tmap</b> reads in one or more aligned protein sequences.
<p>

Two sets of propensity values are then used for the calculations: one
for the middle, hydrophobic portion and one for the terminal regions of
the transmembrane sequence spans.  Average propensity values are
calculated for each position along the alignment, with the contribution
from each sequence weighted according to its dissimilarity relative to
the other aligned sequences. 
<p>

Eight-residue segments are considered as potential cores of
transmembrane segments and elongated if thier middle propensity values
are above a threshold.  End propensity values are also considered as
stop signals.  Only helices with a length of 15 to 29 residues are
allowed and corrections for strictly conserved charged residues are
made. 

<p>

The method is more successful than predictions based upon single
sequences alone. 
<p>

The results are plotted on a graph and written to a text file.

<H2>
    Usage
</H2>
<b>Here is a sample session with tmap</b>
<p>

<p>
<table width="90%"><tr><td bgcolor="#CCFFFF"><pre>

% <b>tmap opsd.msf -out tmap.res -graph cps </b>
Displays membrane spanning regions

Created tmap.ps

</pre></td></tr></table><p>
<p>
<a href="#input.1">Go to the input files for this example</a><br><a href="#output.1">Go to the output files for this example</a><p><p>


<H2>
    Command line arguments
</H2>
<table CELLSPACING=0 CELLPADDING=3 BGCOLOR="#f5f5ff" ><tr><td>
<pre>
   Standard (Mandatory) qualifiers:
  [-sequences]         seqset     File containing a sequence alignment
   -graph              xygraph    [$EMBOSS_GRAPHICS value, or x11] Graph type
                                  (ps, hpgl, hp7470, hp7580, meta, cps, x11,
                                  tekt, tek, none, data, xterm, png, gif)
  [-outfile]           report     [*.tmap] Output report file name

   Additional (Optional) qualifiers: (none)
   Advanced (Unprompted) qualifiers: (none)
   Associated qualifiers:

   "-sequences" associated qualifiers
   -sbegin1            integer    Start of each sequence to be used
   -send1              integer    End of each sequence to be used
   -sreverse1          boolean    Reverse (if DNA)
   -sask1              boolean    Ask for begin/end/reverse
   -snucleotide1       boolean    Sequence is nucleotide
   -sprotein1          boolean    Sequence is protein
   -slower1            boolean    Make lower case
   -supper1            boolean    Make upper case
   -sformat1           string     Input sequence format
   -sdbname1           string     Database name
   -sid1               string     Entryname
   -ufo1               string     UFO features
   -fformat1           string     Features format
   -fopenfile1         string     Features file name

   "-graph" associated qualifiers
   -gprompt            boolean    Graph prompting
   -gdesc              string     Graph description
   -gtitle             string     Graph title
   -gsubtitle          string     Graph subtitle
   -gxtitle            string     Graph x axis title
   -gytitle            string     Graph y axis title
   -goutfile           string     Output file for non interactive displays
   -gdirectory         string     Output directory

   "-outfile" associated qualifiers
   -rformat2           string     Report format
   -rname2             string     Base file name
   -rextension2        string     File name extension
   -rdirectory2        string     Output directory
   -raccshow2          boolean    Show accession number in the report
   -rdesshow2          boolean    Show description in the report
   -rscoreshow2        boolean    Show the score in the report
   -rusashow2          boolean    Show the full USA in the report
   -rmaxall2           integer    Maximum total hits to report
   -rmaxseq2           integer    Maximum hits to report for one sequence

   General qualifiers:
   -auto               boolean    Turn off prompts
   -stdout             boolean    Write standard output
   -filter             boolean    Read standard input, write standard output
   -options            boolean    Prompt for standard and additional values
   -debug              boolean    Write debug output to program.dbg
   -verbose            boolean    Report some/full command line options
   -help               boolean    Report command line options. More
                                  information on associated and general
                                  qualifiers can be found with -help -verbose
   -warning            boolean    Report warnings
   -error              boolean    Report errors
   -fatal              boolean    Report fatal errors
   -die                boolean    Report dying program messages

</pre>
</td></tr></table>
<P>
<table border cellspacing=0 cellpadding=3 bgcolor="#ccccff">
<tr bgcolor="#FFFFCC">
<th align="left" colspan=2>Standard (Mandatory) qualifiers</th>
<th align="left">Allowed values</th>
<th align="left">Default</th>
</tr>

<tr>
<td>[-sequences]<br>(Parameter 1)</td>
<td>File containing a sequence alignment</td>
<td>Readable set of sequences</td>
<td><b>Required</b></td>
</tr>

<tr>
<td>-graph</td>
<td>Graph type</td>
<td>EMBOSS has a list of known devices, including ps, hpgl, hp7470, hp7580, meta, cps, x11, tekt, tek, none, data, xterm, png, gif</td>
<td><i>EMBOSS_GRAPHICS</i> value, or x11</td>
</tr>

<tr>
<td>[-outfile]<br>(Parameter 2)</td>
<td>Output report file name</td>
<td>Report output file</td>
<td><i>&lt;*&gt;</i>.tmap</td>
</tr>

<tr bgcolor="#FFFFCC">
<th align="left" colspan=2>Additional (Optional) qualifiers</th>
<th align="left">Allowed values</th>
<th align="left">Default</th>
</tr>

<tr>
<td colspan=4>(none)</td>
</tr>

<tr bgcolor="#FFFFCC">
<th align="left" colspan=2>Advanced (Unprompted) qualifiers</th>
<th align="left">Allowed values</th>
<th align="left">Default</th>
</tr>

<tr>
<td colspan=4>(none)</td>
</tr>

</table>


<H2>
    Input file format
</H2>


<b>tmap</b> reads a protein sequence USA for one or more aligned sequences.

<p>


<a name="input.1"></a>
<h3>Input files for usage example </h3>
<p><h3>File: opsd.msf</h3>
<table width="90%"><tr><td bgcolor="#FFCCFF">
<pre>
!!AA_MULTIPLE_ALIGNMENT 1.0

  opsd.msf MSF:  354 Type: P 15/07/06 CompCheck: 5414 ..

  Name: OPSD_HUMAN Len: 354  Check: 2647 Weight: 50.00
  Name: OPSD_XENLA Len: 354  Check: 2767 Weight: 50.00

//

           1                                               50
OPSD_HUMAN MNGTEGPNFYVPFSNATGVVRSPFEYPQYYLAEPWQFSMLAAYMFLLIVL
OPSD_XENLA MNGTEGPNFYVPMSNKTGVVRSPFDYPQYYLAEPWQYSALAAYMFLLILL

           51                                             100
OPSD_HUMAN GFPINFLTLYVTVQHKKLRTPLNYILLNLAVADLFMVLGGFTSTLYTSLH
OPSD_XENLA GLPINFMTLFVTIQHKKLRTPLNYILLNLVFANHFMVLCGFTVTMYTSMH

           101                                            150
OPSD_HUMAN GYFVFGPTGCNLEGFFATLGGEIALWSLVVLAIERYVVVCKPMSNFRFGE
OPSD_XENLA GYFIFGPTGCYIEGFFATLGGEVALWSLVVLAVERYIVVCKPMANFRFGE

           151                                            200
OPSD_HUMAN NHAIMGVAFTWVMALACAAPPLAGWSRYIPEGLQCSCGIDYYTLKPEVNN
OPSD_XENLA NHAIMGVAFTWIMALSCAAPPLFGWSRYIPEGMQCSCGVDYYTLKPEVNN

           201                                            250
OPSD_HUMAN ESFVIYMFVVHFTIPMIIIFFCYGQLVFTVKEAAAQQQESATTQKAEKEV
OPSD_XENLA ESFVIYMFIVHFTIPLIVIFFCYGRLLCTVKEAAAQQQESLTTQKAEKEV

           251                                            300
OPSD_HUMAN TRMVIIMVIAFLICWVPYASVAFYIFTHQGSNFGPIFMTIPAFFAKSAAI
OPSD_XENLA TRMVVIMVVFFLICWVPYAYVAFYIFTHQGSNFGPVFMTVPAFFAKSSAI

           301                                            350
OPSD_HUMAN YNPVIYIMMNKQFRNCMLTTICCGKNPLGD.DEASATVSKTETSQVAPA~
OPSD_XENLA YNPVIYIVLNKQFRNCLITTLCCGKNPFGDEDGSSAATSKTEASSVSSSQ

           351
OPSD_HUMAN ~~~~
OPSD_XENLA VSPA

</pre>
</td></tr></table><p>


<H2>
    Output file format
</H2>


<a name="output.1"></a>
<h3>Output files for usage example </h3>
<p><h3>File: tmap.res</h3>
<table width="90%"><tr><td bgcolor="#CCFFCC">
<pre>
########################################
# Program: tmap
# Rundate: Sun 15 Jul 2007 12:00:00
# Commandline: tmap
#    [-sequences] ../../data/opsd.msf
#    -outfile tmap.res
#    -graph cps
# Report_format: seqtable
# Report_file: tmap.res
########################################

#=======================================
#
# Sequence: Consensus     from: 1   to: 354
# HitCount: 7
#=======================================

  Start     End TransMem Sequence
     43      70        1 YMFLLIvLGxPINFlTLyVTvQHKKLRT
     71      98        2 PLNYILLNLxxAdxFMVLxGFTxTlYTS
    112     140        3 lEGFFATLGGEiALWSLVVLAiERYvVVC
    148     176        4 FGENHAIMGVAFTWvMALaCAAPPLxGWS
    201     229        5 ESFVIYMFvVHFTIPmIiIFFCYGqLvxT
    256     276        6 IMVixFLICWVPYAxVAFYIF
    285     305        7 PiFMTiPAFFAKSaAIYNPVI

#---------------------------------------
#---------------------------------------
#=======================================
#
# Sequence: OPSD_HUMAN     from: 1   to: 354
# HitCount: 7
#=======================================

  Start     End TransMem Sequence
     43      70        1 YMFLLIVLGFPINFLTLYVTVQHKKLRT
     71      98        2 PLNYILLNLAVADLFMVLGGFTSTLYTS
    112     140        3 LEGFFATLGGEIALWSLVVLAIERYVVVC
    148     176        4 FGENHAIMGVAFTWVMALACAAPPLAGWS
    201     229        5 ESFVIYMFVVHFTIPMIIIFFCYGQLVFT
    256     276        6 IMVIAFLICWVPYASVAFYIF
    285     305        7 PIFMTIPAFFAKSAAIYNPVI

#---------------------------------------
#---------------------------------------
#=======================================
#
# Sequence: OPSD_XENLA     from: 1   to: 354
# HitCount: 7
#=======================================

  Start     End TransMem Sequence
     43      70        1 YMFLLILLGLPINFMTLFVTIQHKKLRT
     71      98        2 PLNYILLNLVFANHFMVLCGFTVTMYTS
    112     140        3 IEGFFATLGGEVALWSLVVLAVERYIVVC
    148     176        4 FGENHAIMGVAFTWIMALSCAAPPLFGWS
    201     229        5 ESFVIYMFIVHFTIPLIVIFFCYGRLLCT
    256     276        6 IMVVFFLICWVPYAYVAFYIF
    285     305        7 PVFMTVPAFFAKSSAIYNPVI

#---------------------------------------
#---------------------------------------

#---------------------------------------
# Total_sequences: 3
# Total_hitcount: 21
#---------------------------------------
</pre>
</td></tr></table><p>
<p><h3>Graphics File: tmap.ps</h3>
<p><img src="tmap.1.tmap.gif" alt="[tmap results]">

<p>

A plot of the propensities to form the middle (solid line) and the end
(dashed line) of transmembrane regions is output.

<p>

Bars are displayed in the plot above the regions predicted as being most
likely to form transmembrane regions. 

<p>

The text file (specified by the <b>-outfile</b> option) gives a summary
of these regions. 

<p>


The transmembrane regions for the complete alignment are given first,
followed by the predictions for each individual sequence in the
alignment.


<H2>
    Data files
</H2>

None.

<H2>
    Notes
</H2>

None.


<H2>
    References
</H2>

<ol>

<li>
 "Persson, B. & Argos, P. (1994) Prediction of transmembrane  
 segments in proteins utilsing multiple sequence alignments   
 J. Mol. Biol. 237, 182-192."                                 

</ol>

<H2>
    Warnings
</H2>

None.


<H2>
    Diagnostic Error Messages
</H2>

None.


<H2>
    Exit status
</H2>

0 if successful.

<H2>
    Known bugs
</H2>

None.


<h2><a name="See also">See also</a></h2>
<table border cellpadding=4 bgcolor="#FFFFF0">
<tr><th>Program name</th><th>Description</th></tr>
<tr>
<td><a href="garnier.html">garnier</a></td>
<td>Predicts protein secondary structure</td>
</tr>

<tr>
<td><a href="helixturnhelix.html">helixturnhelix</a></td>
<td>Report nucleic acid binding motifs</td>
</tr>

<tr>
<td><a href="hmoment.html">hmoment</a></td>
<td>Hydrophobic moment calculation</td>
</tr>

<tr>
<td><a href="pepcoil.html">pepcoil</a></td>
<td>Predicts coiled coil regions</td>
</tr>

<tr>
<td><a href="pepnet.html">pepnet</a></td>
<td>Displays proteins as a helical net</td>
</tr>

<tr>
<td><a href="pepwheel.html">pepwheel</a></td>
<td>Shows protein sequences as helices</td>
</tr>

</table>

<H2>
    Author(s)
</H2>


Original program by Bengt Persson and Patrick Argos.
<p>

This application was modified for inclusion in EMBOSS by 

Ian Longden (il&nbsp;&copy;&nbsp;sanger.ac.uk)
<br>
Sanger Institute, Wellcome Trust Genome Campus, Hinxton,
Cambridge, CB10 1SA, UK.                      

<H2>
    History
</H2>

Completed 17th June 1999.

<H2>
    Target users
</H2>
This program is intended to be used by everyone and everything, from naive users to embedded scripts.



<H2>
    Comments
</H2>
The graph y axis is not labelled. It is some kind of score, but we
aer not sure exactly how to describe it.

</BODY>
</HTML>

