<HTML>
<HEAD>
  <TITLE>
  EMBOSS: cai
  </TITLE>
</HEAD>
<BODY BGCOLOR="#FFFFFF" text="#000000">

<!--#include file="header1.inc" -->
cai
<!--#include file="header2.inc" -->

<H2>
    Function
</H2>
<!--#include file="inc/cai.ione" -->

<H2>
    Description
</H2>

<p><b>cai</b> calculates the Codon Adaptation Index for a given nucleotide sequence, given a reference codon usage table.  The CAI index is a simple, effective measure of synonymous codon usage bias. It index assesses the extent to which selection has been effective in moulding the pattern of codon usage. In that respect it is useful for predicting the level of expression of a gene, for assessing the adaptation of viral genes to their hosts, and for making comparisons of codon usage in different organisms. The index may also give an approximate indication of the likely success of heterologous gene expression. 
</p>


<H2>
    Algorithm
</H2>
<p>The CAI index uses a reference set of highly expressed genes from a species to assess the relative merits of each codon.  A score for a gene sequence is calculated from the frequency of use of all codons in that gene sequence.</p>


<H2>
    Usage
</H2>
<!--#include file="inc/cai.usage" -->

<H2>
    Command line arguments
</H2>

<table CELLSPACING=0 CELLPADDING=3 BGCOLOR="#f5f5ff" ><tr><td>
<pre>
<!--#include file="inc/cai.ihelp" -->
</pre>
</td></tr></table>
<P>
<!--#include file="inc/cai.itable" -->


<H2>
    Input file format
</H2>

<b>cai</b> reads a nucleic acid sequence of a gene.

<p>

<!--#include file="inc/cai.input" -->

<H2>
    Output file format
</H2>

<b>cai</b> writes the Codon Adaptation Index to the output file.

<p>
<!--#include file="inc/cai.output" -->

<H2>
    Data files
</H2>

<p><b>cai</b> requires a reference codon usage table prepared from a set of genes which are known to be highly expressed.  This is specified by the <tt>-cfile</tt> option and must exist in the EMBOSS data directory.  The default codon usage table <tt>Eyeastcai.cut</tt> is the standard set of Saccharomyces cerevisiae highly expressed gene codon frequiencies. Another table (<tt>Eschpo_cai.cut</tt>) was prepared from a set of Schizosaccharomyces pombe genes by Peter Rice for the S. pombe sequencing team at the Sanger Centre, and is available in the EMBOSS data directory.  You should prepare your own codon usage table for your organism of interest.</p>

<p>
<!--#include file="inc/localfiles.ihtml" -->

<H2>
    Notes
</H2>

<p>Codons are nucleotide triplet that encode an amino acid residue in a polypeptide chain. There are four possible nucleotides in DNA; adenine (<tt>A</tt>), guanine (<tt>G</tt>), cytosine (<tt>C</tt>) and thymine (<tt>T</tt>), therefore 64 possible triplets to encode the 20 amino acids plus the translation termination signal. The encoding is therefore redundant, with all but two amino acids coded for by more than one triplet. Organisms often have a particular preference for one of the possible codons for a given amino acid.</p>

<p>Codon preferences reflect a balance between mutational bias and selection for efficiency of translation. In fast-growing microorganisms there are optimal codons that reflect the composition of the genomic tRNA pool and probably help achieve faster translation rates and high accuracy.  Such selection is expected to be strong in highly expressed genes, as is the case for Escherichia coli or Saccharomyces cerevisiae.  In contrast, codon usage optimization is normally absent in organisms with slower growing rates such as Homo sapiens (human), where codon preferences are determined by mutational biases characteristic to a particular genome.</p>

<p>Various factors are thought to influence codon usage bias in baceteria, including gene expression level already mentioned, %G+C composition (reflecting horizontal gene transfer or mutational bias), GC skew (reflecting strand-specific mutational bias), amino acid conservation, protein hydropathy, transcriptional selection, RNA stability, and optimal growth temperature.</p>
<p>Various methods have been used to analyze codon usage bias. CAI and methods such as the 'frequency of optimal codons' (Fop) are commonly used to predict gene expression levels.  Others such as the 'effective number of codons' (Nc) and Shannon entropy are used to measure codon usage evenness, whereas multivariate statistical methods, iincluding correspondence analysis and principal component analysis, may be used to analyze variations in codon usage between genes.</p>


<H2>
    References
</H2>

<ol>

<li>
Sharp PM., Li W-H.  "The codon adaptation index - a measure of
directional synonymous codon usage bias, and its potential
applications." Nucleic Acids Research 1987 vol 15, pp 1281-1295.

<li>
Synonymous codon usage in bacteria.
Curr Issues Mol Biol. 2001 Oct;3(4):91-7.

</ol>

<H2>
    Warnings
</H2>

None.

<H2>
    Diagnostic Error Messages
</H2>

None.

<H2>
    Exit status
</H2>

It always exits with status 0.

<H2>
    Known bugs
</H2>

None.

<!--#include file="inc/cai.isee" -->


<H2>
    Author(s)
</H2>
<!--#include file="inc/ableasby.address" -->


<H2>
    History
</H2>
<!--#include file="inc/cai.history" -->



<H2>
    Target users
</H2>
<!--#include file="inc/target.itxt" -->

<H2>
    Comments
</H2>
<!--#include file="inc/cai.comment" -->

</BODY>
</HTML>

