<HTML>
<HEAD>
  <TITLE>
  EMBOSS: cons
  </TITLE>
</HEAD>
<BODY BGCOLOR="#FFFFFF" text="#000000">


<!--#include file="header1.inc" -->
cons
<!--#include file="header2.inc" -->

<H2>
    Function
</H2>
<!--#include file="inc/cons.ione" -->

<H2>
    Description
</H2>

<p><b>cons</b> calculates a consensus sequence from a multiple
sequence alignment. To obtain the consensus, the sequence weights and
a scoring matrix are used to calculate a score for each amino acid
residue or nucleotide at each position in the alignment.  The highest
scoring residue goes into the consensus sequence if the score is
higher than a user-specified "plurality" value, otherwise, there is no
consensus at that position.</p>


<H2>
    Algorithm
</H2>
<p>To obtain the consensus, the sequence weights and a scoring matrix
are used to calculate a score at each position in the alignment as
follows. The residue (or nucleotide) i in an alignment column, is
compared to all other residues (j) in the same column. The score for i
is the sum over all residues j (not i=j) of the score(ij)*weight(j),
where score(ij) is taken from a nucleotide or protein scoring matrix
(see -datafile qualifier) and the "weight(j)" is the weighting given
to the sequence j, which is given in the alignment file.</p>

q<p>The highest scoring type of residue is then found in the column. If
the number of "positive matches" (see below) for this residue is
greater than the "plurality value" (see below), then this residue is
the consensus residue.  Otherwise there is no consensus for that
position and an 'n' (nucleotide sequence alignment) or an 'x' (protein
sequence alignment) character is written to the consensus
sequence.</p>

<p>The positive matches for a residue i are calculated as being the
sum of the corresponding sequence weights for all the residues that
increase the score of residue i (i.e. that have a positive score). The
"plurality" qualifier sets the cut-off for the number of positive
matches (weighted) below which there is no consensus.</p>



<H2>
    Usage
</H2>
<!--#include file="inc/cons.usage" -->


<H2>
    Command line arguments
</H2>

<table CELLSPACING=0 CELLPADDING=3 BGCOLOR="#f5f5ff" ><tr><td>
<pre>
<!--#include file="inc/cons.ihelp" -->
</pre>
</td></tr></table>
<P>
<!--#include file="inc/cons.itable" -->


<H2>
    Input file format
</H2>

The USA of a set of aligned sequences.

<p>
<!--#include file="inc/cons.input" -->

<H2>
    Output file format
</H2>
The output consists of a sequence file holding the consensus sequence. 

<p>
<!--#include file="inc/cons.output" -->



<H2>
    Data files
</H2>

<p><b>cons</b> uses the standard set of scoring matrix data files in
the EMBOSS data directory.</p>


<P>
<!--#include file="inc/localfiles.ihtml" -->


<H2>
    Notes
</H2>

<p>The "identity" qualifier provides an additional constrain to
"plurality" when determining a consensus residue at an alignment
site. "identity" sets the required number of identities at a site for
it to be included in the consensus.  If for example this is set to the
number of sequences in the alignment, then only a site with the same
residue in all sequences would be included in the consensus.</p>

<p>The "setcase" qualifier sets the threshold for the positive matches
above which the consensus residue is given is upper-case and below
which it is in lower-case.</p>


<H2>
    References
</H2>

None.

<H2>
    Warnings
</H2>

None.

<H2>
    Diagnostic Error Messages
</H2>

None.

<H2>
    Exit status
</H2>

It always exits with status 0.

<H2>
    Known bugs
</H2>

None.


<!--#include file="inc/cons.isee" -->

<H2>
    Author(s)
</H2>
<!--#include file="inc/tcarver.address" -->

<H2>
    History
</H2>
<!--#include file="inc/cons.history" -->

<H2>
    Target users
</H2>
<!--#include file="inc/target.itxt" -->

<H2>
    Comments
</H2>
<!--#include file="inc/cons.comment" -->


</BODY>
</HTML>

