<HTML>

<HEAD>
  <TITLE>
  EMBOSS: cpgreport
  </TITLE>
</HEAD>
<BODY BGCOLOR="#FFFFFF" text="#000000">

<!--#include file="header1.inc" -->
cpgreport
<!--#include file="header2.inc" -->

<H2>
    Function
</H2>
<!--#include file="inc/cpgreport.ione" -->

<H2>
    Description
</H2>

<p><b>cpgreport</b> identifies in a nucleotide sequence regions with higher than expected frequency of the dinucleotide <tt>CG</tt>.</p>
Each position in the sequence is scored using a running sum calculated from all positions in the sequence.  This is a different method to that typically used for identifying CpG islands, for example by <b>newcpgreport</b> and <b>cpgplot</b>.  This method overpredicts islands but finds the smaller ones around primary exons. An output file is written with information on the CpG-rich regions that are found.  A feature table of sequence features in these regions is also written.</p>


<H2>
Algorithm
</H2>
<p><b>cpgreport</b> scores each position in the sequence using a running sum calculated from all positions in the sequence, starting with the first and ending in the last. If there is not a <tt>CG</tt> dinucleotide at a position, the score is decremented, if there is one, the score is incremented by a constant (user-defined) value. If the score for a region in the sequence is higher than a threshold (17 at the moment) then a putative island is declared.  Sequence regions scoring above the threshold are searched for recursively.</p>


<H2>
    Usage
</H2>
<!--#include file="inc/cpgreport.usage" -->

<H2>
    Command line arguments
</H2>

<table CELLSPACING=0 CELLPADDING=3 BGCOLOR="#f5f5ff" ><tr><td>
<pre>
<!--#include file="inc/cpgreport.ihelp" -->
</pre>
</td></tr></table>
<P>
<!--#include file="inc/cpgreport.itable" -->

<H2>
    Input file format
</H2>

<b>cpgreport</b> reads one or more nucleotide sequences.

<p>
<!--#include file="inc/seqinformats.ihtml" -->

<p>
<!--#include file="inc/cpgreport.input" -->

<H2>
    Output file format
</H2>

<!--#include file="inc/cpgreport.output" -->
<p>

The first non-blank line of the output file 'rnu68037.cpgreport' is the
title line giving the program name, the name of sequence being analysed
and the start and end positions of the sequence. 

<p>

The second non-blank line contains the headings of the columns.

<p>

Subsequent lines contain columns with the following information:

<p>

<ul>
<li>The name of the sequence.
<li>The begin position and the end position of the CpG-rich region.
<li>The score of the CpG-rich region.
<li>The number of CpG's in the CpG-rich region.
<li>The %(G+C) in the CpG-rich region.
<li>The ratio of CpG to GpC in the CpG-rich region.
</ul>

<p>

If the count of GpC in the region is zero, then the ratio of CG/GC is
reported as '-'. 

<H2>
    Data files
</H2>

None.

<H2>
    Notes
</H2>

<p>"CpG" refers to a <tt>C</tt> nucleotide immediately followed by a <tt>G</tt>. The 'p' in 'CpG' refers to the phosphate group linking the two bases.  Regions of genomic sequences rich in the CpG pattern or "CpG islands" are resistant to methylation and tend to be associated with genes which are frequently switched on. It's been estimated that about half of all mammalian genes, and, possibly all mammalian house-keeping genes, have a CpG-rich region around their 5' end.  Non-mammalian vertebrates have some CpG islands that are associated with genes, but the association gets equivocal in the farther taxonomic groups. The detection of CpG island upstream of predicted exons or genes is evidence in support of a highly expressed gene.</p>

<p>As there is no official definition of what is a CpG island is or how to identify where they begin and end, we work with two definitions and thus two methods. These are:</p>

<p>1. <b>cpgplot</b> and <b>newcpgreport</b> use a sliding window within which the Observed/Expected ratio of CpG is calculated. For a sequence region to reported as a CpG island, it must satisfy the following contraints:</p>
<pre>
   Observed/Expected ratio > 0.6
   % C + % G > 50%
   Sequence Length > 200
</pre>

<p>2. <b>newcpgseek</b> and <b>cpgreport</b> use a running sum calculated from all positions in a sequence rather than a window to produce a score.  If there is not a <tt>CG</tt> dinucleotide at a position, the score is decremented, if there is one, the score is incremented by a constant (user-defined) value. If the score for a region in the sequence is higher than a threshold (17 at the moment) then a putative island is declared.  Sequence regions scoring above the threshold are searched for recursively.</p> 

<p>This method overpredicts islands but finds the smaller ones around primary exons. <b>newcpgseek</b> uses the same method as <b>cpgreport</b> but the output is different and more readable. For most purposes you should probably use <b>newcpgreport</b> rather than <b>cpgreport</b>. It is used to produce the human cpgisland database you can find on the EBI's ftp server as well as on the EBI's SRS server.</p>
<p><b>newcpgseek</b> and <b>cpgreport</b> both now display the actual CpG count, the (%C + %G) and the Observed/Expected ratio in the region where the score is above the threshold.</p>
<p>The <b>geecee</b> program measures CG content in the entire input sequence and is not to be used to detect CpG islands. It can be useful for detecting sequences that MIGHT contain an island.</p>


<H2>
    References
</H2>

None.

<H2>
    Warnings
</H2>

None.

<H2>
    Diagnostic Error Messages
</H2>

None.

<H2>
    Exit status
</H2>

    0 if successful.

<H2>
    Known bugs
</H2>

None.

<!--#include file="inc/cpgreport.isee" -->

As there is no official definition of what is a cpg island is, and worst
where they begin and end, we have to live with 2 definitions and thus
two methods.  These are:

<p>

1.  <b>newcpgseek</b> and <b>cpgreport</b> - both declare a putative
island if the score is higher than a threshold (17 at the moment).  They
now also displaying the actual CpG count, the % CG and the
observed/expected ration in the region where the score is above the
threshold.  This scoring method based on sum/frequencies overpredicts
islands but finds the smaller ones around primary exons.
<b>newcpgseek</b> uses the same method as <b>cpgreport</b> but the
output is different and more readable.

<p>

2.  <b>newcpgreport</b> and <b>cpgplot</b> use a sliding window within
which the Obs/Exp ratio of CpG is calculated.  The important thing to
note in this method is that an island, in order to be reported, is
defined as a region that satisfies the following contraints:

<p>
<pre>
   Obs/Exp ratio > 0.6
   % C + % G > 50%
   Length > 200.
</pre>
<p>

<p>

For all practical purposes you should probably use newcpgreport.  It is
actually used to produce the human cpgisland database you can find on
the EBI's ftp server as well as on the EBI's SRS server.

<p>

<b>geecee</b> measures CG content in the entire input sequence and is
not to be used to detect CpG islands.  It can be usefull for detecting
sequences that MIGHT contain an island.
          

<H2>
    Author(s)
</H2>


This program was originally written by 

<!--#include file="inc/gmicklem.address" -->

<p>
The algorithm was modified for inclusion in EGCG under the name 'CPGSPANS' by
<!--#include file="inc/rlopez.address" -->


<p>

This application was modified for inclusion in EMBOSS by 
<!--#include file="inc/ableasby.address" -->


<H2>
    History
</H2>
<!--#include file="inc/cpgreport.history" -->

<H2>
    Target users
</H2>
<!--#include file="inc/target.itxt" -->

<H2>
    Comments
</H2>
<!--#include file="inc/cpgreport.comment" -->

</BODY>
</HTML>

