
<HTML>

<HEAD>
  <TITLE>
  EMBOSS
  </TITLE>
</HEAD>
<BODY BGCOLOR="#FFFFFF" text="#000000">


<!--#include file="header1.inc" -->
Application Groups
<!--#include file="header2.inc" -->

<!--END OF HEADER-->





<table border cellpadding=4 bgcolor="#FFFFF0">

<tr><th>Group</th><th>Description</th></tr>

<tr>
<td><A HREF="acd_group.html">Acd</A></td>
<td>Acd file utilities</td>
</tr>
<tr>
<td><A HREF="alignment_consensus_group.html">Alignment consensus</A></td>
<td>Merging sequences to make a consensus</td>
</tr>
<tr>
<td><A HREF="alignment_differences_group.html">Alignment differences</A></td>
<td>Finding differences between sequences</td>
</tr>
<tr>
<td><A HREF="alignment_dot_plots_group.html">Alignment dot plots</A></td>
<td>Dot plot sequence comparisons</td>
</tr>
<tr>
<td><A HREF="alignment_global_group.html">Alignment global</A></td>
<td>Global sequence alignment</td>
</tr>
<tr>
<td><A HREF="alignment_local_group.html">Alignment local</A></td>
<td>Local sequence alignment</td>
</tr>
<tr>
<td><A HREF="alignment_multiple_group.html">Alignment multiple</A></td>
<td>Multiple sequence alignment</td>
</tr>
<tr>
<td><A HREF="display_group.html">Display</A></td>
<td>Publication-quality display</td>
</tr>
<tr>
<td><A HREF="edit_group.html">Edit</A></td>
<td>Sequence editing</td>
</tr>
<tr>
<td><A HREF="enzyme_kinetics_group.html">Enzyme kinetics</A></td>
<td>Enzyme kinetics calculations</td>
</tr>
<tr>
<td><A HREF="feature_tables_group.html">Feature tables</A></td>
<td>Manipulation and display of sequence annotation</td>
</tr>
<tr>
<td><A HREF="hmm_group.html">HMM</A></td>
<td>Hidden markov model analysis</td>
</tr>
<tr>
<td><A HREF="information_group.html">Information</A></td>
<td>Information and general help for users</td>
</tr>
<tr>
<td><A HREF="menus_group.html">Menus</A></td>
<td>Menu interface(s)</td>
</tr>
<tr>
<td><A HREF="nucleic_2d_structure_group.html">Nucleic 2d structure</A></td>
<td>Nucleic acid secondary structure</td>
</tr>
<tr>
<td><A HREF="nucleic_codon_usage_group.html">Nucleic codon usage</A></td>
<td>Codon usage analysis</td>
</tr>
<tr>
<td><A HREF="nucleic_composition_group.html">Nucleic composition</A></td>
<td>Composition of nucleotide sequences</td>
</tr>
<tr>
<td><A HREF="nucleic_cpg_islands_group.html">Nucleic CpG islands</A></td>
<td>CpG island detection and analysis</td>
</tr>
<tr>
<td><A HREF="nucleic_gene_finding_group.html">Nucleic gene finding</A></td>
<td>Predictions of genes and other genomic features</td>
</tr>
<tr>
<td><A HREF="nucleic_motifs_group.html">Nucleic motifs</A></td>
<td>Nucleic acid motif searches</td>
</tr>
<tr>
<td><A HREF="nucleic_mutation_group.html">Nucleic mutation</A></td>
<td>Nucleic acid sequence mutation</td>
</tr>
<tr>
<td><A HREF="nucleic_primers_group.html">Nucleic primers</A></td>
<td>Primer prediction</td>
</tr>
<tr>
<td><A HREF="nucleic_profiles_group.html">Nucleic profiles</A></td>
<td>Nucleic acid profile generation and searching</td>
</tr>
<tr>
<td><A HREF="nucleic_repeats_group.html">Nucleic repeats</A></td>
<td>Nucleic acid repeat detection</td>
</tr>
<tr>
<td><A HREF="nucleic_restriction_group.html">Nucleic restriction</A></td>
<td>Restriction enzyme sites in nucleotide sequences</td>
</tr>
<tr>
<td><A HREF="nucleic_rna_folding_group.html">Nucleic RNA folding</A></td>
<td>RNA folding methods and analysis</td>
</tr>
<tr>
<td><A HREF="nucleic_transcription_group.html">Nucleic transcription</A></td>
<td>Transcription factors, promoters and terminator prediction</td>
</tr>
<tr>
<td><A HREF="nucleic_translation_group.html">Nucleic translation</A></td>
<td>Translation of nucleotide sequence to protein sequence</td>
</tr>
<tr>
<td><A HREF="phylogeny_consensus_group.html">Phylogeny consensus</A></td>
<td>Phylogenetic consensus methods</td>
</tr>
<tr>
<td><A HREF="phylogeny_continuous_characters_group.html">Phylogeny continuous characters</A></td>
<td>Phylogenetic continuous character methods</td>
</tr>
<tr>
<td><A HREF="phylogeny_discrete_characters_group.html">Phylogeny discrete characters</A></td>
<td>Phylogenetic discrete character methods</td>
</tr>
<tr>
<td><A HREF="phylogeny_distance_matrix_group.html">Phylogeny distance matrix</A></td>
<td>Phylogenetic distance matrix methods</td>
</tr>
<tr>
<td><A HREF="phylogeny_gene_frequencies_group.html">Phylogeny gene frequencies</A></td>
<td>Phylogenetic gene frequency methods</td>
</tr>
<tr>
<td><A HREF="phylogeny_molecular_sequence_group.html">Phylogeny molecular sequence</A></td>
<td>Phylogenetic tree drawing methods</td>
</tr>
<tr>
<td><A HREF="phylogeny_tree_drawing_group.html">Phylogeny tree drawing</A></td>
<td>Phylogenetic molecular sequence methods</td>
</tr>
<tr>
<td><A HREF="protein_2d_structure_group.html">Protein 2d structure</A></td>
<td>Protein secondary structure</td>
</tr>
<tr>
<td><A HREF="protein_3d_structure_group.html">Protein 3d structure</A></td>
<td>Protein tertiary structure</td>
</tr>
<tr>
<td><A HREF="protein_composition_group.html">Protein composition</A></td>
<td>Composition of protein sequences</td>
</tr>
<tr>
<td><A HREF="protein_motifs_group.html">Protein motifs</A></td>
<td>Protein motif searches</td>
</tr>
<tr>
<td><A HREF="protein_mutation_group.html">Protein mutation</A></td>
<td>Protein sequence mutation</td>
</tr>
<tr>
<td><A HREF="protein_profiles_group.html">Protein profiles</A></td>
<td>Protein profile generation and searching</td>
</tr>
<tr>
<td><A HREF="test_group.html">Test</A></td>
<td>Testing tools, not for general use.</td>
</tr>
<tr>
<td><A HREF="utils_database_creation_group.html">Utils database creation</A></td>
<td>Database installation</td>
</tr>
<tr>
<td><A HREF="utils_database_indexing_group.html">Utils database indexing</A></td>
<td>Database indexing</td>
</tr>
<tr>
<td><A HREF="utils_misc_group.html">Utils misc</A></td>
<td>Utility tools</td>
</tr>


</table>


</BODY>
</HTML>
