<HTML>
<HEAD>
  <TITLE>
  EMBOSS: jaspscan
  </TITLE>
</HEAD>
<BODY BGCOLOR="#FFFFFF" text="#000000">

<!--#include file="header1.inc" -->
jaspscan
<!--#include file="header2.inc" -->

<H2>
    Function
</H2>
<!--#include file="inc/jaspscan.ione" -->

<!--
DON'T WRITE ANYTHING HERE.
IT IS DONE FOR YOU.
-->




<H2>
    Description
</H2>

<b>jaspscan</b> scans one or more DNA sequences for transcription factor binding sites from the JASPAR database. Matches are searched for using fast sequence word-matching, optionally allowing mismatches.  Because the binding sites are so small, there will be many spurious (false positive) matches.</p>
<p>An output file is written with information on the matches, including sequence ID and accession number, the start and end positions of the match in an input sequence and the sequence of the region where a match has been found. Binding factor information, where available, is given at the end of the matches for each matching entry.</p>

















<H2>
    Usage
</H2>

<!--  
	Example usage, as run from the command-line.
        Many examples illustrating different behaviours is good.
 -->

<!--#include file="inc/jaspscan.usage" -->



<H2>
    Command line arguments
</H2>

<table CELLSPACING=0 CELLPADDING=3 BGCOLOR="#f5f5ff" ><tr><td>
<pre>
<!--#include file="inc/jaspscan.ihelp" -->
</pre>
</td></tr></table>
<P>
<!--#include file="inc/jaspscan.itable" -->

<!--
DON'T WRITE ANYTHING HERE.
IT IS DONE FOR YOU.
-->








<H2>
    Input file format
</H2>


<b>jaspscan</b> reads one or more nucleotide sequences.


<p>
<!--#include file="inc/seqinformats.ihtml" -->

<p>
<!--#include file="inc/jaspscan.input" -->





<H2>
    Output file format
</H2>


<!--#include file="inc/reportformats.ihtml" -->

<p>
By default the report is in 'seqtable' format.

<p>

<!--#include file="inc/jaspscan.output" -->





<H2>
    Data files
</H2>

<!--#include file="inc/localfiles.ihtml" -->
<p>
<!--#include file="inc/jaspfiles.ihtml" -->






<H2>
    Notes
</H2>

<!-- 
        Restrictions.
        Interesting behaviour.
        Useful things you can do with this program.
   -->

None.







<H2>
    References
</H2>

<OL>

<LI>DNA binding sites: representation and discovery Bioinformatics. 2000
      Jan;16(1):16-23

<LI>Applied bioinformatics for the identification of regulatory elements
      Nat Rev Genet. 2004 Apr;5(4):276-87 

</OL>



<H2>
    Warnings
</H2>

<!-- 
        Potentially stupid things the program will let you do.
   -->

None.







<H2>
    Diagnostic Error Messages
</H2>

<!-- 
        Error messages specific to this program, eg:
        "FATAL xxx" - means you have not set up the xxx data using program <b>prog</b>.<p>
   -->

None.







<H2>
    Exit status
</H2>

<!-- 
        Description of the exit status for various error conditions
   -->

It always exits with status 0.








<H2>
    Known bugs
</H2>


<!-- 
        Bugs noted but not yet fixed.
   -->

None.








<!--
<H2>
    See also
</H2>
-->
<!--#include file="inc/jaspscan.isee" -->
<!-- 
        Add any comments about other associated programs (to prepare
        data files?) that seealso doesn't find. 
   -->










<H2>
    Author(s)
</H2>

<!--#include file="inc/ableasby.address" -->

<H2>
    History
</H2>
<!--
        Date written and what changes have been made go in this file.
   -->
<!--#include file="inc/jaspscan.history" -->



<H2>
    Target users
</H2>
<!--
        For general users, use this text
   -->
<!--#include file="inc/target.itxt" -->

<H2>
    Comments
</H2>
<!--
        User/developer/other comments go in this file.
   -->
<!--#include file="inc/jaspscan.comment" -->


</BODY>
</HTML>

