<HTML>

<HEAD>
  <TITLE>
  EMBOSS: merger
  </TITLE>
</HEAD>
<BODY BGCOLOR="#FFFFFF" text="#000000">

<!--#include file="header1.inc" -->
merger
<!--#include file="header2.inc" -->

<H2>
    Function
</H2>
<!--#include file="inc/merger.ione" -->


<H2>
    Description
</H2>
<p><b>merger</b> reads two overlapping input sequences of the same
type (typically nucleotide) and uses a global alignment algorithm
(Needleman & Wunsch) to optimally align the sequences.  A merged
sequence is generated from the alignment and writen to the output
file. The sequence alignment is also written.</p>



<H2>
Algorithm
</H2>

<p>It uses a global alignment algorithm (Needleman & Wunsch) to
optimally align the sequences and then creates a merged sequence from
the alignment. When there is a mismatch in the alignment between the
two sequences, the base included in the merged sequence is the base
from the sequence which has the best local sequence quality score.</p>

<p>The following heuristic is used to find the sequence quality score.
If one of the bases is a 'N', then the other sequence's base is used.
Otherwise, a window size around the disputed base is used to find the
local quality score. This window size is increased from 5, to 10 to 20
bases or until there is a clear decision on the best choice. If there
is no best choice after using a window of 20, then the base in the
first sequence is used.</p>

<p>To calculate the quality of a window of a sequence around a base:</p>

<pre>
    * quality = sequence value/length under window either side of the base
    * sequence value = sum of points in that window
    * unambiguous bases (ACGTU) score 2 points
    * ambiguous bases (MRWSYKVHDB) score 1 point
    * Ns score 0 points
    * off end of the sequence scores 0 points 
</pre>
<p>This heavily discriminates against the iffy bits at the end of
sequence reads.</p>



<H2>
    Usage
</H2>
<!--#include file="inc/merger.usage" -->


<p>

Typically, one of the sequences will need to be reverse-complemented
to put it into the correct orientation to make it join. For example:

<p>

<pre>
% merger file1.seq file2.seq -sreverse2 -outseq merged.seq
</pre>

<H2>
    Command line arguments
</H2>
<table CELLSPACING=0 CELLPADDING=3 BGCOLOR="#f5f5ff" ><tr><td>
<pre>
<!--#include file="inc/merger.ihelp" -->
</pre>
</td></tr></table>
<P>

<!--#include file="inc/merger.itable" -->


<H2>
    Input file format
</H2>

<b>merger</b> reads two nucleotide or protein sequences.

<p>
<!--#include file="inc/seqinformats.ihtml" -->



<p>

<!--#include file="inc/merger.input" -->

<H2>
    Output file format
</H2>


<!--#include file="inc/alignformats.ihtml" -->
 
<p>
<!--#include file="inc/seqformats.ihtml" -->
 

<p>

<!--#include file="inc/merger.output" -->

<H2>
    Data files
</H2>

It reads the scoring matrix for the alignment from the standard EMBOSS
'data' directory.  By default it is the file 'EBLOSUM62' (for proteins)
or the file 'EDNAFULL' (for nucleic sequences). 

<H2>
    Notes
</H2>

<p>This program was originally written to aid in the reconstruction of
mRNA sequences which had been sequenced from both ends as a 5' and 3'
EST (cDNA). eg. joining two reads produced by primer walking
sequencing.</p>

<p>The gap open and gap extension penalties have been set at a higher
level than is usual (50 and 5). This was experimentally determined to
give the best results with a set of poor quality EST test
sequences.</p>

<H2>
    References
</H2>

None.

<H2>
    Warnings
</H2>

<p>Care should be taken to reverse one of the sequences (e.g. using the qualifier <tt>-sreverse2</tt>) if this is required to get them both in the correct orientation..</p>
<p><b>merger</b> uses the memory-hungry Needleman & Wunsch alignment.  The required memory may be greater than the available memory when attempting to merge large (cosmid-sized or greater) sequences.</p>


<H2>
    Diagnostic Error Messages
</H2>

None.

<H2>
    Exit status
</H2>

It exits with a status of 0

<H2>
    Known bugs
</H2>

None.

<!--#include file="inc/merger.isee" -->

<H2>
    Author(s)
</H2>


<!--#include file="inc/gwilliam.address" -->


<H2>
    History
</H2>
<!--#include file="inc/merger.history" -->

<H2>
    Target users
</H2>
<!--#include file="inc/target.itxt" -->


<H2>
    Comments
</H2>
<!--#include file="inc/merger.comment" -->

</BODY>
</HTML>

