<HTML>

<HEAD>
  <TITLE>
  EMBOSS: needle
  </TITLE>
</HEAD>
<BODY BGCOLOR="#FFFFFF" text="#000000">

<!--#include file="header1.inc" -->
needle
<!--#include file="header2.inc" -->
<H2>
    Function
</H2>
<!--#include file="inc/needle.ione" -->


<H2>
    Description
</H2>


This program uses the Needleman-Wunsch global alignment algorithm to
find the optimum alignment (including gaps) of two sequences when
considering their entire length. 

<p>

What is the optimal alignment? Dynamic programming methods ensure the
optimal global alignment by exploring all possible alignments and
choosing the best.  It does this by reading in a scoring matrix that
contains values for every possible residue or nucleotide match.  Needle
finds an alignment with the maximum possible score where the score of an
alignment is equal to the sum of the matches taken from the scoring
matrix. 


<H3>
    Algorithm
</H3>

The Needleman-Wunsch algorithm is a member of the class of algorithms
that can calculate the best score and alignment in the order of mn
steps, (where 'n' and 'm' are the lengths of the two sequences).  These
dynamic programming algorithms were first developed for protein sequence
comparison by Needleman and Wunsch, though similar methods were
independently devised during the late 1960's and early 1970's for use in
the fields of speech processing and computer science.

<p>

An important problem is the treatment of gaps, i.e., spaces inserted to
optimise the alignment score.  A penalty is subtracted from the score
for each gap opened (the 'gap open' penalty) and a penalty is subtracted
from the score for the total number of gap spaces multiplied by a cost
(the 'gap extension' penalty).

<p>

Typically, the cost of extending a gap is set to be 5-10 times lower
than the cost for opening a gap. 

<p>

There are two ways to compute a penalty for a gap of n positions :

<p>

<pre>
gap opening penalty + (n - 1) * gap extension penalty
gap penalty + n * gap length penalty
</pre>

<p>

The first way is used by EMBOSS and WU-BLAST
<br>
The second way is used by NCBI-BLAST, GCG, Staden and CLUSTAL.
Fasta used it for a long time the first way, but Prof. Pearson decided 
recently to shift to the second.

<p>

The two methods are basically equivalent.

<p>

In a Needleman-Wunsch global alignment, the entire length of each 
sequence is aligned. This can be thought of as an overlap between the 
two sequences (one can be completely within the other, or their ends can 
overlap).

<p>

This leaves no penalty for the hanging ends of the overlap. In 
bioinformatics, it is usually reasonable to assume that the sequences 
are incomplete and there should be no penalty for failing to align the 
missing bases.


<H2>
    Usage
</H2>
<!--#include file="inc/needle.usage" -->

<H2>
    Command line arguments
</H2>
<table CELLSPACING=0 CELLPADDING=3 BGCOLOR="#f5f5ff" ><tr><td>
<pre>
<!--#include file="inc/needle.ihelp" -->
</pre>
</td></tr></table>
<P>

<!--#include file="inc/needle.itable" -->


<H2>
    Input file format
</H2>

<b>needle</b> reads any 2 sequence USAs of the same type (DNA or protein).

<p>

<!--#include file="inc/needle.input" -->

<H2>
    Output file format
</H2>

<!--#include file="inc/alignformats.ihtml" -->

<p>


<!--#include file="inc/needle.output" -->

<p>

The <b>Identity:</b> is the percentage of identical matches between the two
sequences over the reported aligned region (including any gaps in the length).

<p>

The <b>Similarity:</b> is the percentage of matches between the two
sequences over the reported aligned region (including any gaps in the length).
     


<H2>
    Data files
</H2>


For protein sequences EBLOSUM62 is used for the substitution
matrix. For nucleotide sequence, EDNAFULL is used. Others can be
specified.

<p>
<!--#include file="inc/localfiles.ihtml" -->

<H2>
    Notes
</H2>

<b>needle</b> is a true implementation of the Needleman-Wunsch
algorithm and so produces a full path matrix. It therefore cannot be
used with genome sized sequences unless you've a lot of memory and a
lot of time.

<H2>
    References
</H2>


<ol>

<li>
Needleman, S.  B.  and Wunsch, C.  D.  (1970) J.  Mol.  Biol.  48,
443-453. 

<li>
Kruskal, J.  B.  (1983) An overview of squence comparison In D.  Sankoff
and J.  B.  Kruskal, (ed.), Time warps, string edits and macromolecules:
the theory and practice of sequence comparison, pp.  1-44 Addison
Wesley. 
</ol>

<H2>
    Warnings
</H2>


<b>needle</b> is for aligning two sequences over their entire length.  This
works best with closely related sequences.  If you use needle to align
very distantly-related sequences, it will produce a result but much of
the alignment may have little or no biological significance. 

<p>

A true Needleman Wunsch implementation like <b>needle</b> needs memory
proportional to the product of the sequence lengths.  For two sequences
of length 10,000,000 and 1,000 it therefore needs memory proportional to
10,000,000,000 characters.  Two arrays of this size are produced, one of
ints and one of floats so multiply that figure by 8 to get the memory
usage in bytes.  That doesn't include other overheads.  Therefore only
use water and needle for accurate alignment of reasonably short
sequences.

<p>

If you run out of memory, try using 
<b>stretcher</b> instead.

<H2>
    Diagnostic Error Messages
</H2>

<pre>
Uncaught exception
 Assertion failed
 raised at ajmem.c:xxx
</pre>

<p>

Probably means you have run out of memory. Try using 
<b>stretcher</b> if this happens. 

<H2>
    Exit status
</H2>

  0 upon successful completion.

<H2>
    Known bugs
</H2>

None.

<!--#include file="inc/needle.isee" -->

<P>
When you want an alignment that covers the whole length of both
sequences, use <a href="needle.html">needle. </A> 

<p>
When you are trying to find the best region of similarity between two
sequences, use <a href="water.html">water.</a>
<p>

<a href="stretcher.html">stretcher</a> is a more suitable program to use to
find global alignments of very long sequences.


<H2>
    Author(s)
</H2>
<!--#include file="inc/ableasby.address" -->


<H2>
    History
</H2>
<!--#include file="inc/needle.history" -->

<H2>
    Target users
</H2>
<!--#include file="inc/target.itxt" -->


<H2>
    Comments
</H2>
<!--#include file="inc/needle.comment" -->

</BODY>
</HTML>

