<HTML>

<HEAD>
  <TITLE>
  EMBOSS: newcpgreport
  </TITLE>
</HEAD>
<BODY BGCOLOR="#FFFFFF" text="#000000">

<!--#include file="header1.inc" -->
newcpgreport
<!--#include file="header2.inc" -->
<H2>
    Function
</H2>
<!--#include file="inc/newcpgreport.ione" -->


<H2>
    Description
</H2>

<p><b>newcpgreport</b> identifies CpG islands in one or more
nucleotide sequences. The ratio of observered to expected number of GC
dinucleotides patterns is calculated over a window (sequence region)
which is moved along the sequence. The calculated ratios are used to
identify regions which match the program's definition of a "CpG
island" (a CG dinucleotide rich area).  A report file is written
giving the input sequence name, CpG island parameters and data on any
CpG islands that are found.</p>


<H2>
Algorithm
</H2>

<p>The ratio of observered to expected number of GC dinucleotides
patterns is calculated over a window of user-specified size
(<tt>-window</tt> parameter).  The window is slid along the sequence
and the ratio recalculated until the end of the sequence is
reached.</p>

<p>By default, <b>cpgplot</b> defines a CpG island as a region where,
over an average of 10 windows and not less than 200 bases, the
calculated (%G + %C) content is over 50% and the calculated
Observed/Expected ratio is over 0.6. These conditions can be modified
by setting the values of the appropriate parameters.</p>

<p>The Observed number of CpG patterns in a window is simply the
number of times a <tt>C</tt> is found followed immediately by
a <tt>G</tt>.</p>

<p>The Expected number of CpG patterns is calculated for each window
as the number of CpG dinucleotides you would expect to see in that
window based on the frequency of <tt>C</tt>'s and <tt>G</tt>'s in that
window. Thus, the Expected frequency of CpG's in a window is
calculated as the number of <tt>C</tt>s in the window multiplied by
the number of <tt>G</tt>s in the window, divided by the window length.

<pre>
	Expected = (number of C's  * number of G's) / window length
</pre>
</p>



<H2>
    Usage
</H2>
<!--#include file="inc/newcpgreport.usage" -->


<H2>
    Command line arguments
</H2>
<table CELLSPACING=0 CELLPADDING=3 BGCOLOR="#f5f5ff" ><tr><td>
<pre>
<!--#include file="inc/newcpgreport.ihelp" -->
</pre>
</td></tr></table>
<P>

<!--#include file="inc/newcpgreport.itable" -->


<H2>
    Input file format
</H2>


<b>newcpgreport</b> reads one or more nucleic acid sequences.

<p>

<!--#include file="inc/newcpgreport.input" -->

<H2>
    Output file format
</H2>


<!--#include file="inc/newcpgreport.output" -->


<H2>
    Data files
</H2>

None.

<H2>
    Notes
</H2>

<p>"CpG" refers to a <tt>C</tt> nucleotide immediately followed by
a <tt>G</tt>. The 'p' in 'CpG' refers to the phosphate group linking
the two bases. Regions of genomic sequences rich in the CpG pattern or
"CpG islands" are resistant to methylation and tend to be associated
with genes which are frequently switched on. It's been estimated that
about half of all mammalian genes, and, possibly all mammalian
house-keeping genes, have a CpG-rich region around their 5' end.
Non-mammalian vertebrates have some CpG islands that are associated
with genes, but the association gets equivocal in the farther
taxonomic groups. The detection of CpG island upstream of predicted
exons or genes is evidence in support of a highly expressed gene.</p>

<p><b>newcpgreport</b> is used in the production of the CpG Island
database 'CPGISLE'. It produces a report for potential CpG islands in
CPGISLE database entry format. See the FTP
site: <a href="ftp://ftp.ebi.ac.uk/pub/databases/cpgisle/">
ftp://ftp.ebi.ac.uk/pub/databases/cpgisle/</a>
for the finished database.</p>



<p>As there is no official definition of what is a CpG island is or
how to identify where they begin and end, we work with two definitions
and thus two methods. These are:</p>

<p>1. <b>cpgplot</b> and <b>newcpgreport</b> use a sliding window
within which the Observed/Expected ratio of CpG is calculated. For a
sequence region to reported as a CpG island, it must satisfy the
following contraints:

<pre>  Observed/Expected ratio > 0.6
   % C + % G > 50%
   Sequence Length > 200</pre>
</p>

<p>2. <b>newcpgseek</b> and <b>cpgreport</b> use a running sum
calculated from all positions in a sequence rather than a window to
produce a score.  If there is not a <tt>CG</tt> dinucleotide at a
position, the score is decremented, if there is one, the score is
incremented by a constant (user-defined) value. If the score for a
region in the sequence is higher than a threshold (17 at the moment)
then a putative island is declared.  Sequence regions scoring above
the threshold are searched for recursively.</p>

<p>This method overpredicts islands but finds the smaller ones around
primary exons. <b>newcpgseek</b> uses the same method
as <b>cpgreport</b> but the output is different and more readable. For
most purposes you should probably use <b>newcpgreport</b> rather
than <b>cpgreport</b>. It is used to produce the human cpgisland
database you can find on the EBI's ftp server as well as on the EBI's
SRS server.</p>

<p><b>newcpgseek</b> and <b>cpgreport</b> both now display the actual
CpG count, the (%C + %G) and the Observed/Expected ratio in the region
where the score is above the threshold.</p>

<p>The <b>geecee</b> program measures CG content in the entire input
sequence and is not to be used to detect CpG islands. It can be useful
for detecting sequences that MIGHT contain an island.</p>





<H2>
    References
</H2>


<ol>

<li>
Larsen F., Gundersen, G., Lopez L., Prydz H.
"CpG island as Gene Markers in the Human Genome"
Genomics  13:1095-1107 (1992)

</ol>

<H2>
    Warnings
</H2>

None.

<H2>
    Diagnostic Error Messages
</H2>

None.

<H2>
    Exit status
</H2>

It always exits with a status of 0.

<H2>
    Known bugs
</H2>

None.

<!--#include file="inc/newcpgreport.isee" -->

<H2>
    Author(s)
</H2>


<!--#include file="inc/rlopez.address" -->


<H2>
    History
</H2>
<!--#include file="inc/newcpgreport.history" -->

<H2>
    Target users
</H2>
<!--#include file="inc/target.itxt" -->


<H2>
    Comments
</H2>
<!--#include file="inc/newcpgreport.comment" -->

</BODY>
</HTML>

