<HTML>

<HEAD>
  <TITLE>
  EMBOSS: stretcher
  </TITLE>
</HEAD>
<BODY BGCOLOR="#FFFFFF" text="#000000">

<!--#include file="header1.inc" -->
stretcher
<!--#include file="header2.inc" -->

<H2>
    Function
</H2>
<!--#include file="inc/stretcher.ione" -->


<H2>
    Description
</H2>


<p><b>stretcher</b> calculates an optimal global alignment of two sequences using a modification of the classic dynamic programming algorithm which uses linear space.  The output is a standard alignment file.  The substitution matrix, gap insertion penalty and gap extension penalties used to calculate the alignment may be specified.</p>

<H2>
Algorithm
</H2>
<p>The standard sequence global alignment program using the Needleman & Wunsch algorithm, as implemented in the program <b>needle</b>, requires O(MN) space and O(N) time. This is standard computer-science language for it needing an amount of computer memory that is proportional to the product of the two sequences being aligned and taking an amount of time that is proportional to the shorter of the two sequences. So if a 1 kb and a 10 kb sequence take 10 Mega-words of memory and 10 minutes to align, you should expect that in order to align a 10 kb sequence and a 1 Mb sequence you will need approximately 10 Giga-words of memory and 100 minutes. Computer memory will rapidly be exhausted as the size of the aligned sequences increases.</p>

<p>This program implements the Myers and Miller algorithm for finding an optimal global alignment in an amount of computer memory that is only proportional to the size of the smaller sequence - O(N).</p>

<p>In computing, a benefit is seldom gained without a cost elsewhere. The cost of gaining a memory-efficient alignment is that it takes about twice the amount of time to do the alignment as the Needleman & Wunsch algorithm. In computer-science language the time is approximately O(2N). </p>


<H2>
    Usage
</H2>
<!--#include file="inc/stretcher.usage" -->


<H2>
    Command line arguments
</H2>
<table CELLSPACING=0 CELLPADDING=3 BGCOLOR="#f5f5ff" ><tr><td>
<pre>
<!--#include file="inc/stretcher.ihelp" -->
</pre>
</td></tr></table>
<P>

<!--#include file="inc/stretcher.itable" -->


<H2>
    Input file format
</H2>


<b>stretcher</b> reads any 2 sequence USAs of the same type (DNA or protein).

<p>


<!--#include file="inc/stretcher.input" -->


<H2>
    Output file format
</H2>

<!--#include file="inc/alignformats.ihtml" -->

<p>

The default output format is 'markx0'.

<p>

<!--#include file="inc/stretcher.output" -->

<H2>
    Data files
</H2>


For protein sequences EBLOSUM62 is used for the substitution
matrix. For nucleotide sequence, EDNAMAT is used. Others can be
specified.

<p>
EMBOSS data files are distributed with the application and stored
in the standard EMBOSS data directory, which is defined
by EMBOSS environment variable EMBOSS_DATA.

<p>

Users can provide their own data files in their own directories.
Project specific files can be put in the current directory, or for
tidier directory listings in a subdirectory called
".embossdata". Files for all EMBOSS runs can be put in the user's home
directory, or again in a subdirectory called ".embossdata".

<p>
The directories are searched in the following order:

<ul>
   <li> . (your current directory)
   <li> .embossdata (under your current directory)
   <li> ~/ (your home directory)
   <li> ~/.embossdata
</ul>

<H2>
    Notes
</H2>

<p>A global pairwise alignment is one where it is assumed that the two sequences have diverged from a common ancestor and that the program should try to stretch the two sequences, introducing gaps where necessary, in order to show the alignment over the whole length of the two sequences that best illustrates their similarities. In contrast, a local alignment program like <b>matcher</b> simply finds local, small parts of the two sequences where there is some similarity and makes no assumption about the whole length of the sequence needing to be similar.</p>

<H2>
    References
</H2>

<ol>

<li>
<a href="http://www.cs.arizona.edu/people/gene/PAPERS/linear.ps">
E.  Myers and W.  Miller, "Optimal Alignments in Linear Space," CABIOS
4, 1 (1988), 11-17.</a>

</ol>

<H2>
    Warnings
</H2>

<p>Demonstration of similarity is not evidence of homology!  This program will produce a global alignment even if there is no biological justification for thinking that there might be a common ancestor.</p>



<H2>
    Diagnostic Error Messages
</H2>

None.

<H2>
    Exit status
</H2>

It exits with a status of 0.

<H2>
    Known bugs
</H2>

None.

<!--#include file="inc/stretcher.isee" -->

<H2>
    Author(s)
</H2>
The original program was written by Gene Myers and Webb Miller in 1989.

<p>

This application was modified for inclusion in EMBOSS by 
<!--#include file="inc/ilongden.address" -->


<H2>
    History
</H2>

<!--#include file="inc/stretcher.history" -->

<H2>
    Target users
</H2>
<!--#include file="inc/target.itxt" -->


<H2>
    Comments
</H2>
<!--#include file="inc/stretcher.comment" -->

</BODY>
</HTML>

