Description: Fix typo to build emboss with PostgreSQL support
Author: Felix Geyer <fgeyer@debian.org>
Date: Sun, 29 Jul 2012 17:51:04 UTC
Bugs-Debian: http://bugs.debian.org/683190

--- emboss-6.4.0.orig/m4/postgresql.m4
+++ emboss-6.4.0/m4/postgresql.m4
@@ -98,7 +98,7 @@
             EMBCPPFLAGS=$CPPFLAGS
 	    EMBLDFLAGS=$LDFLAGS
             
-            CPPFLAGS="$POSTGRESSQL_CPPFLAGS $EMBCPPFLAGS"
+            CPPFLAGS="$POSTGRESQL_CPPFLAGS $EMBCPPFLAGS"
 	    LDFLAGS="$POSTGRESQL_LDFLAGS $EMBLDFLAGS"
 
             AC_LINK_IFELSE([AC_LANG_PROGRAM([[#include <stdio.h>
