Description: Fix typo to build emboss with PostgreSQL support
Author: Felix Geyer <fgeyer@debian.org>
Date: Sun, 29 Jul 2012 17:51:04 UTC
Bugs-Debian: http://bugs.debian.org/683190

--- a/m4/postgresql.m4
+++ b/m4/postgresql.m4
@@ -98,7 +98,7 @@ dnl not be installed
             EMBCPPFLAGS=$CPPFLAGS
 	    EMBLIBS=$LIBS
             
-            CPPFLAGS="$POSTGRESSQL_CPPFLAGS $EMBCPPFLAGS"
+            CPPFLAGS="$POSTGRESQL_CPPFLAGS $EMBCPPFLAGS"
 	    LIBS="$POSTGRESQL_LDFLAGS $EMBLIBS"
 
             AC_LINK_IFELSE([AC_LANG_PROGRAM([[#include <stdio.h>
