Description: Correct the padding to make it consistent 
Author: Gert Wollny <gw.fossdev@gmail.com>
Bug-Debian: http://bugs.debian.org/748395
Forwarded: no
Last-Update: 2014-05-19

--- emboss-6.6.0.orig/plplot/plplotP.h
+++ emboss-6.6.0/plplot/plplotP.h
@@ -378,7 +378,7 @@ typedef struct {
 	unsigned int Hershey;
 	PLUNICODE Unicode;
 	char Font;
-	char padding[3];		/* pad to align */
+	char padding[7];		/* pad to align */
 } Hershey_to_Unicode_table;
 
 extern int number_of_entries_in_hershey_to_unicode_table;
